---
title: "Chapter 2: Pasture plants employ diverse interaction strategies with AMF "
author: "Natascha Lewe"
output:
  word_document2:
    number_sections: yes
    toc_depth: 3
    reference_docx: bib/Test.docx
bibliography: testbookdown/20220513bib.bibtex
csl: bib/apa.csl
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
knitr::opts_chunk$set(message = FALSE)
knitr::opts_chunk$set(warning  = FALSE)
knitr::opts_chunk$set(dpi  = 300)

#bibliography: 20210813bib_lib_CK.bibtex
#csl: apa.csl


library(tidyverse)
library(bookdown) #brauche ich für render(file, output_format="word_document2") #für die crossreferences
#library(kableExtra) #https://cran.r-project.org/web/packages/kableExtra/vignettes/awesome_table_in_html.html
library (flextable)  #https://ardata-fr.github.io/flextable-book/
#library(citr)#when .bib angegeben ist kann man bei addins referencen anwählen
library(knitr)
library(vegan)
library(picante)
library(RColorBrewer)
library(bipartite)
library(readxl)

library(phyloseq)
##Build phylogenetic TREE######
#https://rstudio-pubs-static.s3.#amazonaws.com/345955_fba1ccbdcd8f424aa5505c15bfd75bf7.html
#Lib
library(stats)
library(ade4)
library(ape)
library(adegenet)
library(phangorn)
library(seqinr)
library(msa)
library(microbiome)
library (rstatix)
library(ggpubr)
library(metagMisc)
library(phylosmith)
library(picante)
library (rstatix)
library (factoextra)
library(FactoMineR)
library (MicEco)
library(png)
library (grid)
library (gridExtra)
library(iNEXT)
library (ggrepel)
set.seed(23111406)

#Voraussetzungen für dieses doc sind 
#Marsden0_dada2.R (R in Results Ordner)
#Ergebnisse sind in tabellenformat gespeichert in 

#Wichtige internet infos:
#word und markdown hin under her, mit Kommentaren etc.
#https://github.com/noamross/redoc
#redoc()  # macht es zu word
#dedoc() edited word zu Rmd wieder - Achtung sachen, die aus code chuns sind sollten nicht geedited werden - -sieht man dann  nicht mehr, daher ist das alles automatisch markiert (kann man auch ausschalten )
#render ("Ch2.Rmd", output_format = "word_document2")

# 
# If certain code chunks in your R Markdown documents are time-consuming to run, you may cache them by adding the chunk option cache = TRUE in the chunk header, and you are recommended to label such code chunks as well, e.g.,{r important-computing, cache=TRUE}


```

# Chapter 2
## Introduction

The composition of AMF communities vary among biomes, ecosystems, and along environmental gradients [@Kivlin2011; @Davison2015; @Davison2021]. In temperate grasslands, AMF species are present at high richness and abundance, making these ecosystems especially interesting for the investigation of plant-AMF-interactions [@Oepik2006]. AMF richness is often reported to have positive effects on plant productivity in these ecosystems [@Heijden1998; @Marcel2006; @Vogelsang2006], but its net effect depends on multiple factors, including soil properties or the species of the plant and AMF [@Stover2018]. Strong correlations between plant- and AMF-richness and community compositions among successional stages have also been reported in grasslands [@Neuenkamp2018; @DeLong2019a]. Plants and AMF have reciprocal effects on the richness and composition of their mutualistic partner, but their relative contribution as drivers of community composition is still under discussion [@Smilauer2021]. An understanding of the mechanisms of assembly in plant-AMF communities has been hindered by the difficulty of directly comparing grassland studies on AMF-communities because the composition of AMF communities in plant roots vary according to the available AMF pool [@Smilauer2020], which in turn is influenced by soil properties [@Valyi2015; @Bouffaud2017; @Horn2017], and stochastic dispersal processes [@Davison2021; @Chaudhary2020].

### Plant hosts as drivers of AMF community assemblage

Although several factors affect the AMF community in plant roots, the most direct control lies with the plant host itself [@Mangan2010]. Plants influence key processes of the AMF community assembly like spore germination, or the initiation and degree of colonisation via chemical signalling in a context-dependent way [@Raudaskoski2015; @Feng2019; @Harrison2005; @Cohen2013]. For example, both the plant’s developmental stage [@Smilauer2021] and environment, including resource abundance [@Mueller2019] or stressors such as salinity or herbivory [@Aroca2013; @Sveen2021] can affect the selection of AMF by the host plant. While some studies report little phylogenetic signal in plant-AMF interactions in grassland systems [@Veresoglou2014], others report that closely related plants tend to interact with similar AMF species (Dong2021, Scheublin 2004). Nevertheless, the plant functional group (e.g., grass *versus* forbs) appears to be a strong determinant of the AMF community composition [@Smilauer2020; @Sepp2019]. Grasses and forbs interact differently with AMF in terms of the richness, colonisation rate and composition of AMF, with grasses often hosting more taxa than forbs [@Smilauer2020; @Hannula2019; @Bunn2015; @Sepp2019]. Furthermore, phylogenetic host preferences at the AMF family level and differential plant productivity depending on the AMF partner have also been reported, with some plant lineages of diverse phylogeny showing higher growth responses to specific AMF genera [@Peralta2016; @Hoeksema2018]. In addition, plants preferentially reward more collaborative AMF [@Kiers2011; @Argueello2016]. For example, those AMF with beneficial traits like superior phosphorus provisioning, may receive more photosynthate C from their host plant [@Pel2018]. While both host preference and environmental conditions influence AMF communities on plant hosts, most plants seem to harbour diverse AMF communities, consisting of several AMF families with a range of traits [@Geel2018; @Argueello2016; @Wagg2011].


###	Generalist and specialist plants in the plant-AMF interaction

Species in communities play unique functional roles that drive ecological processes such as pollination, predation, or mycorrhization, which are paramount for the assembly and maintenance of ecological communities. These functional roles reflect a species’ interaction niche which depicts the “resources” (i.e., species) it interacts with in this specific functional role, such as all prey species of a predator. Furthermore, the functional role of a species can be characterised by the degree to which it utilises the resource (i.e., niche width). 
For example, pollinators might vary in the kind of flowers they visit, or plants in the number or identity of AMF species they interact with. When characterising the interaction niche width, species are recognised as specialists or generalists in their functional role. 
The presence of both interaction strategies contributes to community stability [@Dehling2021]. However, the definition of a specialist and generalist is not always straightforward, and several approaches have been applied [@Ponisio2019; @Bluethgen2006; @Rohr2014]. In bipartite networks, the term specialist is commonly applied to members of one guild interacting with few or phylogenetic related interaction partners from the other guild [@Bascompte2009]. The term has also been applied to species that interact with a specific set of interaction partners that differs from the other species in their guild, or in other words, species with contrasting interaction niches [@Ponisio2019; @Bluethgen2006]. 
Vice versa, a generalist is commonly defined as a species with many interactions or one with diverse interactions. 
These differences in the definitions of specialism and generalism are problematic because they lead to the pooling of species interaction traits which may vary in their effects on the community. For example, specialist mutualists that have non-overlapping interaction niches stabilise their communities by complementary effects [@Dehling2021] but might be vulnerable to extinction events [@Dennis2011]. In contrast, a specialist interacting with few species in a nested network is not susceptible to extinction [@Tylianakis2018]. Distinguishing the interaction niche properties of species that contribute to their specialism and generalism will therefore improve understanding of the assembly and maintenance of ecological communities.

Plant-AMF interactions are usually depicted as bipartite interaction networks (Fig. \@ref(fig:fig-bipartite-architecture)). In most studies, the interaction niche width of a focal species (i.e., generalist or specialist), is based on the number of the species it interacts with (e.g., degree or interaction richness) [@Bascompte2003; @Bascompte2006; @Ollerton2006; @Vazquez2009], while species’ interaction preferences are seldom considered. Field studies of grassland communities have found plant-AMF networks to have network architectures that are nested and with low modularity [@Geel2018] indicating that both guilds include species that are either generalists or specialists, but that their interaction niches strongly overlap [@Geel2018; @Heijden2015; @Lekberg2016; @Valyi2015]. 
Improved understanding of the properties of the plant-AMF interaction niches could help elucidate how these inherent properties of species influence their consequent interaction network architectures. However, an understanding of the interaction niche width requires that additional aspects of specialism in the plant-AMF symbiosis are considered. Specifically, two independent traits contribute to specialism, causing a plant species to host few AMF: plants may either be indiscriminate in their interactions but host few taxa, or conversely, plants may be selective in their interactions showing specificity for individual AMF taxa or alternatively the phylogenetic relatedness of the interacting partners [@Wehner2014]. 
An over-dispersed phylogenetic signal of the AMF community, for example, might be advantageous for the plant under changing environmental conditions as less related AMF taxa comprise a set of differential traits [@Maherali2012]. On the other hand, a phylogenetically clustered AMF community indicates preferential interaction with specific AMF taxa, which may have been selected by the plant for the unique benefits they provide (e.g., nutrient acquisition) or may be the result of successful competition for root space among AMF [@Chagnon2013]. Both strategies of specialism come with different trade-offs, which directly impact the species fitness and subsequently the outcomes of AMF community assembly. Trade-offs that arise for a generalist plant are typically increased carbon costs versus beneficial complementary effects of multiple AMF. Generalists might not be able to differentiate between cooperative and non-cooperative AMF which could lead to increased carbon costs relative to phosphorus gain for the plant [@Kiers2008; @Bever2009]. On the other hand, a more diverse AMF community might provide the plant host with a range of functionally complementary species [@Jansa2008; @Koide2000], which could buffer changing environmental conditions. Clearly a characterisation of plant interaction niche width that simultaneously considers both the numeric richness of interactions and plant preference for AMF has the potential to enhance our understanding of the assembly of mycorrhizal networks.

Despite a recent surge in knowledge on AMF communities driven by affordable sequencing approaches, detailed descriptions of AMF communities are still sparse [@Chagnon2014; @Chagnon2015]. Previously, AMF species were identified by spore morphology, a complicated and time-consuming method which examines spores in soil, rather than in plant roots and therefore does not necessarily represent a confirmed plant-AMF interaction. In contrast, molecular identification of AMF taxa from clean root tissue is generally considered a reliable indication of mycorrhization [@Chagnon2014]. Descriptions of plant-AMF interactions rely strongly on metrics of α-diversity, especially species richness. Until recently, the determination of AMF richness from sequencing data was a statistical challenge, because measures of species richness are highly dependent on sampling efforts, i.e., sequencing depth [@Willis2019]. Richness, in turn, influences several other metrics of diversity, and network properties [@Mariani2019; @Ulrich2009]. Rarefaction to an equal sampling size has been commonly used for the adjustment of samples [@Gotelli2001], but has been criticised, *inter alia*, for large loss of data [@McMurdie2014], and its biased representation of species’ richnesses [@Chao2012]. To avoid this bias, coverage-based rarefaction is a new method that rarefies samples to equal sampling completeness (coverage), not equal sampling size. Standardisation based on the sample coverage retains the degree of dissimilarity of communities of different richnesses and is therefore suitable when comparing species diversity in different samples [@Chao2012].



### Aim of the study 

The aim of this study was to gain insight into interaction niche properties of the members of a plant-AMF community and the relevance of these properties for the architecture of the assembled interaction network. First, I sought to characterise properties of the interaction niches of grassland plant species for AMF. Twenty plant species from an established pasture grassland plant community were grown separately using a common soil substrate sourced from a pasture environment. This allowed plant species to select their interactions from the endemic pool of soil microbes, without the influence of neighbour effects. I characterised the root AMF community by next-generation sequencing and quantified two aspects of the interaction: firstly, the interaction niche width, defined as both numeric and phylogenetic metrics of richness of the AMF interaction partners per plant species. 
Second, I characterised “specificity” of host plants for their AMF, a measure how taxonomically specific the plants are in their AMF partner choice. In grasslands, grasses typically exhibit greater numeric richness of AMF interactions than forbs. Generalist species usually interact with an unspecific range of partners possibly because they are less likely to select for specific AMF partners of higher mutualistic value. 
I therefore tested the hypothesis (**Hyp 1**) that the high numeric richness of AMF in grasses would coincide with less filtering of the available AMF pool relative to forbs of the same community. 
Next, I focused on the mechanisms underpinning plant specialism for AMF interactions in this plant community. Numeric specialism should be most advantageous to plants under opposite conditions; either when the specialist plant selects for the most beneficial partners in a given environment, or when the specialist plant selects for AMF that differ from those of other hosts to increase complementarity of resource use at the community scale. I therefore determined plant species’ relative AMF selectivity and characterised the overlap of plant species’ interactions for AMF.
This enabled me to test the hypothesis (**Hyp 2**) that numeric specialist plants employ a variety of strategies to select for specific AMF communities. Characterising the functional interaction strategies of species that contribute to their specialism and generalism will improve understanding of the assembly and maintenance of ecological communities.

\newpage

## Methods

### Study site and glasshouse experiment

A glasshouse experiment was conducted to evaluate the propensity of different plant species to interact with AMF communities of different richness, taxonomic and phylogenetic diversities. Microcosms consisting of field-collected soil and one individual of each of 20 plant species, which co-occurred within the pasture plant community, were constructed in replicates of five. Hereafter, the term treatment is used to describe each microcosm consisting either of only soil or soil and one individual plant.

#### Description of the study site

Soil and seeds were collected at Wairio Wetland (41°14'42''S, 175°15'18''E) on the Western shore of Lake Wairarapa, New Zealand. The collection area is an abandoned field formerly used for pastoral agriculture that was fenced to exclude cattle in 2007 [@Grant2012]. It represents a stable community consisting of mostly naturalised forb and graminoid species, with scattered stands of native kahikatea trees (*Dacrycarpus dacrydioides*) [@Beadel2000; @Gillon2014]. The soils of the study site are sandy and recent [@Landcare2016] with a silty texture in the upper 6 to 10 cm and a deep sandy layer (\>1m) [@Gillon2014]. The area is relatively fertile [@Sparling2008], with a total carbon content of 8%, and 0.75% nitrogen [@Landcare2016]. The inorganic P ("Olsen P") can be described as in the low to moderate region with values between 10 and 25 µg mL^-1^ [@Landcare2016; @Gillon2014; @Sparling2008]. The soil pH is slightly acidic to circumneutral with pH 6.0 to 7.2, with moderate cation-exchange capacity (25-40 cmol~c~ kg^-1^) [@Landcare2016; @Gillon2014].

#### Soil and seed collection

Soil was collected at six randomly selected locations within the study site. This was to ensure that the AMF pool represents what the plants would encounter in the field. At each location, the vegetation was removed and two 20 L pails of the upper 20 cm of soil were collected. The field collected soil was air dried and sieved to 4 mm to remove plant roots, and large soil particles. Soil was homogenised by manual mixing on a tarpaulin in a 1:1 ratio with an autoclaved mix of potting soil, coarse sand and pumice (1:2:1, respectively).

Seeds of 30 forbs and graminoids were collected over the summer of 2016 (January and February) at the study site. The plant species were identified, and the viability of the seeds determined through germination experiments. Twenty plant species were selected for this study to represent a viable and balanced mix of grasses and forbs typical for the study site [@Beadel2000]. Additionally, to include plant species with a broad range of interaction traits, two native species that occur only rarely in the wetland, *P. cita* and *U. uncinata*, were included. Where necessary, additional seed was sourced from local agricultural suppliers (see Table \@ref(tab:pl-list)).

#### Glasshouse experiments and harvest

Seeds were surface-sterilised in a 1 % sodium hypochlorite solution for 10 min, following 10 min in 70 % ethanol. After thorough washing with distilled water, the seeds were placed on moistened filter paper in petri dishes and left in a temperature-controlled room at 18 °C with a 12 h day-and-night cycle. Seeds were monitored regularly, and the filter paper moistened if needed. Petri dishes showing signs of fungal contamination were discarded. Individual seedlings were transplanted to 2.7 L pots containing 2.5 L soil-mix when the first true roots appeared. Five replicate pots per plant species were established and grown in a glasshouse over the summer of 2016/2017. Plants were watered as needed and the position of the pots on the benches randomised biweekly. Seedlings that developed from the seed bank were promptly removed. All plants were harvested after 140 days of growth.

### Sequencing of the fungal community

#### Sample preparartion and DNA extraction

After harvest, the root system of each individual plant was thoroughly washed in distilled water to remove soil. Roots were cut with a scalpel into small segments of approximately 0.5 to 1 cm length to subsample randomly for DNA extraction.
About 50 mg of pooled root segments were weighed into a 1.5 mL Eppendorf tube containing steel beads (Green Eppendorf Lysis Kit, Next Advance, Inc., USA) and homogenised for 2 x 30 s at 8,000 rpm (~ 6,000 X g) (Precellys Evolution, Bertin instruments, FR). 
CTAB extraction buffer (600 µL, 2 % hexadecyl(trimethyl)ammonium bromide, 1 % polyvinylpyrrolidone (PVP40), 100 mM Tris(hydroxymethyl)aminomethane hydrochloride (tris-HCl), 1.4 M NaCl, 20 mM Ethylenediaminetetraacetic acid (EDTA)) was added and the tube incubated at 60 °C for 1 hr [@Doyle1991a]. The tube was shaken carefully every 20 min to ensure uniform treatment of the root homogenate. After incubation, 600 µL chloroform were added and the mixture was centrifuged at 10,000 rpm (~ 9,200 X g) for 2 s (Eppendorf centrifuge 5415R, Germany) and the aqueous layer transferred to a new 1.5 mL tube using a wide bore pipette tip.
The sample was gently mixed with 800 µL cold isopropanol and put on ice for 10 min to precipitate the DNA. By centrifugation at 10,000 rpm for 60 s the DNA was pelleted, and the supernatant could be carefully discarded. The DNA pellet was washed twice with 800 µL of 80 % ethanol by gently mixing, followed by centrifugation at 10,000 rpm (~ 9,200 X g) for 1 min.
The supernatant was discarded and the pellet air-dried upside down at room temperature. The DNA pellet was resuspended in 40 µL TE buffer (10 nM Tris-HCl, 1mM EDTA, adjusted to pH = 7.4) and the quality and concentration of the DNA extracts assessed by spectrophotometry (Nanodrop NP80, Implen, Germany) before the DNA extract was frozen at -80 °C until further processing.

#### PCR and sequencing

To identify the AMF community the internal transcribed spacer 2 (ITS2) region of the eukaryotic ribosomal DNA was amplified by polymerase chain reaction (PCR) with the modified versions of the primers ITS3 (5'-**CAHCGATGAAGAACGYRG-3'**) and ITS4 (5'-**TCCTSCGCTTATTGATATGC**-3') [@Tedersoo2014]. The 50 µL PCR reaction mix contained the extracted DNA at a concentration of 2 ng µL^-1^, 25 µL of 2 x MyTaq^TM^ Red Mix (Bioline, London, UK), 1 µL of each 10 µM primer and 8 µL 5 x TBT-Par as amplification enhancer [@Samarakoon2013]. Thermal cycling conditions were as follows: 5 min at 95 °C, then 35 cycles at 95 °C for 45 s, 50 °C for 45 s, and 70 °C for 60 s; followed by 15 min at 72 °C (GeneAmp PCR System 2700, Applied Biosciences, USA). The PCR product was cleaned with ExoSAP-IT^TM^ (Thermo Fisher Scientific, USA) before it was sent to sequencing on an Illumina MiSeq (AGRF, Melbourne, Australia) to generate paired-end reads of 300 bp. The image analysis for the generated data was performed in real time by the MiSeq Control Software (MCS) 2.6.2.1 and Real Time Analysis (RTA) 1.18.54. The Illumina bcl2fastq 2.20.0.422 pipeline was used to infer the sequence data.

### Bioinformatics

#### Processing of raw sequence reads

Paired-end reads were trimmed using *cutadapt* 3.2 (minimum primer overlap = 6; maximal error rate = 0.12) to remove the primer sequences and quality checked, merged and filtered using the DADA2 ITS pipeline 1.18 (https://benjjneb.github.io/dada2/ITS_workflow.html) into R 4.0.4 [@R2021]. Filtering was applied using the default parameters except for the maximum allowance of errors for the reverse reads which were set at 4 (allowed error for forward reads = 2) due to their lower quality. Adjusting the error allowance allows for appropriate filtering of paired-end reads, because the truncation of reads to remove low quality regions is unsuitable for ITS sequences as they have variable lengths [@Tedersoo2015]. Pseudo-pooling of the samples was included in the DADA2 method to reduce the loss of sequence variants that might be rare in some samples but not in others. 
Inferring ASVs in pseudo-pooling mode includes an additional step in which each sample is compared against the whole set of ASVs of the data to identify rare reads. Next, chimeric reads were discarded using default parameters and the fungal ITS classifiers naïve Bayes trained on the dynamic UNITE 8.2 (2020) reference database to assign the fungal taxonomy [@Nilsson2018a]. 
To examine if the detected pool of fungal ASVs represents the total possible pool for each treatment, inter- and extrapolated rarefaction and sampling completeness curves were calculated in R package *iNEXT* `r packageVersion ("iNEXT")` [@Hsieh2016; @Chao2014]. 
The extrapolation was based on estimated richnesses [@Colwell2012] and included confidence intervals based on the bootstrap method [@Chao2012]. Additionally, for each treatment a sampling coverage curve was constructed, i.e., showing the proportion of the total sequence reads  that belong to the AMF species identified in the treatment. 
The samples were then re-sampled to the smallest common sample coverage (coverage-based rarefaction). All sampling curves were calculated in R package *iNEXT* [@Hsieh2016; @Chao2014], and the standardisation to equal coverage was computed using the function "phyloseq_coverage_raref" with 99 iterations in the package *metagMisc* `r packageVersion ("metagMisc")` (https://github.com/vmikk/metagMisc/) on the Rāpoi High Performance Compute Cluster (Victoria University of Wellington, NZ).
The data was then filtered to contain only sequences assigned to the subphylum Glomeromycotina, which are considered to be AMF. Fungal metabarcoding often results in many sequences that cannot be identified  at a meaningful taxonomic level [@Nilsson2016], I therefore used a phylogenetic approach, analysing all ASVs that were identified as subphylum Glomeromycotina as a proxy for an AMF species. 


### Description of the plant-AMF interaction niche

To comprehensively describe the plant species' interaction niche properties with AMF and to compare the plant-AMF interaction niches among plant species several aspects of the interaction were analysed. 
First, to describe the properties of the interacting AMF community for each plant species (and soil) and the within-plant species variability of the AMF community, several numeric and phylogenetic metrics were calculated. These included (1) AMF richness, Chao1 diversity, and Shannon's diversity index to measure the numeric AMF richness, (2) Faith's phylogenetic diversity (PD) to describe the phylogenetic richness and (3) the mean phylogenetic distance (MPD) to depict the phylogenetic divergence of each replicate. To describe the fundamental AMF community per treatment, the (4) total unique AMF species and their taxonomic assignment were determined. The variability among the replicates of each treatment's AMF community was examined with the (5) true $\beta$-diversity and (6) the proportion of core AMF species per treatment. To examine, if plants hosted different AMF from each other, the interaction niche overlap of the plants was examined by comparing their AMF communities to each other, using both multivariate analysis of the AMF diversity and metrics derived from plant-AMF bipartite networks. For multivariate analysis, this included (7) non-metric multidimensional scaling (NMDS) based on the Raup-Crick distances among the AMF communities of each sample to visualise both the similarity among the plants' AMF communities and their variability. To examine phylogenetic similarities of the AMF communities within and between plant hosts (8) NMDS based on phylogenetic distances (UniFrac) was done. For each distance measure (i.e., Raup-Crick distances and UniFrac distances), (9) permutational analysis of variance (PERMANOVA) and (10) permutational analysis of multivariate dispersion (PERMDISP) were both applied for statistical testing of differences among the AMF communities. Although each plant was grown alone, metrics of ecological bipartite networks can be applied to understand the potential for interaction niche overlap among the plant species regarding their AMF communities. Therefore, a bipartite network with plants and AMF as nodes was constructed, and nestedness was calculated using (11) a nestedness metric based on overlap and decreasing fill (NODF) and (12) the Johnson nestedness, accounting for degree distribution. The nestedness metrics were compared against different null models to examine the influence of the numeric richness on the nestedness. The (13) modularity of the network was calculated to identify the presence of modular plant-AMF interaction indicating interaction complementarity. All calculations were done in R 4.0.4 (R Core Team 2020) if not stated otherwise.

#### Numeric metrics - measures of $\alpha$-diversity

To describe a plants' numeric interaction niche width for arbuscular mycorrhizal fungi, several measures of $\alpha$-diversity were calculated. The AMF richness S~Ob~, Chao1 diversity and Shannon's diversity index H' were calculated in *phyloseq* version `r packageVersion ("phyloseq")`, and for each metric the arithmetic means per treatment were employed as indicators of $\alpha$-diversity [@phyloseq2013]. While S~Ob~ describes the numeric richness, Shannon's diversity incorporates the evenness (relative abundances) of each species [@Hill1973]. Chao1 is an abundance-based estimator that adds a correction factor to the observed number of species to estimate the true richness and is especially suitable for microbial data that includes many low-abundance species [@Chao1984; @Chao1987]. 
Normality and homoscedasticity of all diversities were checked visually using a qq-plot and by calculating Bartlett's test. A non-parametric Wilcoxon rank sum test was applied to test for significant differences in S~Ob~, Chao1 and H'. A *post-hoc* multiple pairwise comparison test for unequal variances (Games-Howell test in package rstatix v0.7.0 [@rstatix2021]) was applied to identify significant differences between treatments, i.e., plant species and plant families. For each treatment, the total number of unique ASVs ($\gamma$-diversity) that occur over all replicates was determined.

#### Phylogenetic metrics - Faith's phylogenetic diversity (PD) and mean phylogenetic distance (MPD)

To examine the phylogenetic diversities of the AMF communities as a measure of phylogenetic interaction niche width, the sequences were aligned using *ClustalW* [@Thompson1994] in the R package *msa* `r packageVersion("msa")` [@Bodenhofer2015]. After sequence alignment, the pairwise distances between the DNA sequences were calculated to build a phylogenetic tree based on neighbor joining algorithm using the R packages *seqinr* `r packageVersion("seqinr")`[@seqinr2007], and *ape* `r packageVersion("ape")` [@ape2019]. Based on the phylogenetic tree, two metrics, Faith's phylogenetic diversity (PD) and mean pairwise distance (MPD), describing the phylogenetic diversity and dispersion per AMF community respectively, were computed [@Faith1992; @Webb2002]. 
Faith's PD measures the total branch length of the phylogenetic tree per sample. Because Faith's PD positively correlates with the ASV richness of the sample, the z-score, i.e., the standardised Faith's phylogenetic distance (z-score~PD~) was calculated in *picante* `r packageVersion("picante")` [@Webb2010]. 
A null model was estimated based on 999 iterations in which the composition per sample was randomised but the ASV richness per sample maintained to account for the potential correlation between PD and ASV richness (independent swap algorithm) [@Gotelli2000]. 
The observed PD (PD~Ob~) was compared against the distribution of the Faith's PD~NULL~ and a p-value assigned. An observed PD was judged as random for *p*-values between 0.05 and 0.95. Significant differences from the randomised mean phylogenetic richness are indicated for *p*\>0.95 and PD~Ob~\>PD~NULL~ or *p*\<0.05 and PD~Ob~\<PD~NULL~). The MPD describes the phylogenetic dispersion of a sample's AMF community by measuring the mean branch length [@Webb2000] between each pair of AMF ASVs from a distance-based phylogenetic tree containing all AMF species of the dataset. 
To evaluate if an AMF community is clustered, over-dispersed or a random set of the overall AMF phylogeny, the standardised MPD (z-score~MPD~) was computed based on randomised AMF communities. For that, a null model was generated based on 999 iterations using the randomisation algorithm as described above (independent swap algorithm) [@Gotelli2000]. The observed value for the MPD (MPD~Ob~) was compared to the distribution of the MPD of the null models (MPD~NULL~). A *p*-value was assigned and MPD~Ob~ judged as random for *p*-values between 0.05 and 0.95. Significant differences from the randomised mean indicate either phylogenetic over-dispersion (for *p*\>0.95 and MPD~Ob~\>MPD~NULL~) or a closer phylogenetic relationship, i.e., clustering (for *p*\<0.05 and MPD~Ob~\<MPD~NULL~). For both metrics, PD and MPD, the samples per treatment were merged into one sample to calculate the z-scores for PD and MPD per treatment as described above.

#### Compositional heterogeneity of the replicates - within treatment β-diversity

There are many different β-diversity measures that can describe the compositional heterogeneity of AMF sequence variants among treatment replicates [@Anderson2011; @Tuomisto2010]. Here, the numeric β-diversity was calculated yielding the number of compositional units (CU) per treatment (true β-diversity): CU = $\gamma$/ S̅~Ob~) [@Tuomisto2010; @Whittaker1960]. The resulting values for CU were adjusted to the number of replicates per treatment, leading to values between 1 for the smallest number of CUs, and 5 for the highest possible number of CUs. 
Taking the identity of the interacting AMF species into account, the variability among the replicates can be examined based on the proportional abundance of a treatments' core interacting species. Here, the sum of the core species per treatment that occur in more than 60 % of their replicates was divided by the unique number of AMF per treatment ($\gamma$), to yield β~core~.

#### Principal component analysis of $\alpha$- and $\beta$-diversities

Each of the numeric and phylogenetic metrics of $\alpha$-diversity (S~Ob~, Chao1, H', $\gamma$, PD, MPD) and β-diversity (CU, β~core~) can be interpreted as an axis of a multidimensional volume. 
Positioning the plant species according to their interaction niche properties inside this hypervolume results in an index of the species' interaction strategy. To identify treatment groups that have similar interaction strategies, principal component analysis (PCA) was applied. For that, the numeric and phylogenetic metrics of $\alpha$-diversity and β-diversity were calculated per treatment as described above and included as variables for PCA which was calculated and visualised using R package *FactoMineR* `r packageVersion("FactoMineR")` [@Husson2010] and *factoextra* `r packageVersion("factoextra")` [@factoextra2020] using variables that were scaled to unit variance. The percentage contribution of each variable to the principal components per dimension was calculated and visualised. Both, low number of replicates and low AMF richness can distort metrics like $\beta$-diversity or phylogenetic distance measures, and this was characteristic for most individuals of the Fabaceae, Asteraceae and Cyperaceae which had very small numbers of AMF ASVs, often also at low read counts, and several samples from these families were lost due to not including AMF. 
Therefore, a second PCA, including only soil, and the members of the Plantaginaceae and the Poaceae was calculated and visualised to allow a more detailed examination of these higher-diversity treatments. 

```{r metrics-table, tab.cap = "Overview of metrics used for the description of the plant-AMF interaction niche. "}

metrics_table <- read_xlsx("dataCh2/metric_table.xlsx")

metrics_table %>%  flextable () %>% 
  theme_vanilla() %>% 
  set_table_properties( layout = "autofit", width =1)
```



### Comparison of plant-AMF interaction niches and description of niche overlap

#### Venn diagram, dissimilarity- and distance-based measures

Niche overlap of the plant-AMF interaction niches of the functional type of the plants and plant families were visualised by Venn diagrams, using package *MicEco* `r packageVersion("MicEco")` [@MicEco2021]. To account for high variability of ASVs at the AMF species level, closely related tree tips were agglomerated based on cophenetic distances (at a height of h = 0.1 of the tree) using package *phyloseg*. For the plant species, interaction niche overlap was described by comparing the variations among their core AMF communities. Two measures of $\beta$-diversity were calculated, based either on a dissimilarity matrix or a distance matrix of the presence-absence data, and their results visualised by computing ordinations. First, the core AMF community per treatment was determined; only taxa that were present in 60 % of all replicates per treatment were included. 
Most metrics of $\beta$-diversity are not independent of the ASVs richness [@Modin2020], i.e., differences in ASV richness among samples increase their $\beta$-diversity, even if they select for the same AMF ASVs. To account for that, the Raup-Crick dissimilarity β~RC~ was used as a first measure, which computes the deviations from a probabilistic null model, while holding the values of the richness constant in the randomisations of the null models [@Chase2011]. The second measure is based on phylogenetic $\beta$-diversity by calculating UniFrac distances [@Lozupone2005] among the samples. The result for both metrics is a matrix each, which is then analysed by non-metric multidimensional scaling (NMDS). First, a NMDS scree plot that shows the final stress vs the number of dimensions (k) was computed. For the NMDS, the number of dimensions was chosen after which additional axes would not provide much additional reduction in stress. Next, the resulting stress of the NMDS at the chosen dimension was compared to the stress values obtained by computing NMDS on randomised data (1000 permutations with preserved AMF richness per sample). The solution obtained by NMDS was only selected when it was significantly better than the solutions from randomised data. The R packages *vegan* `r packageVersion("vegan")` [@oksanen2020], *phyloseq* `r packageVersion("phyloseq")` and *phylosmith* `r packageVersion("phylosmith")` [@Smith2022]) were used for computation. In addition to the analysis via NMDS, testing for differences between AMF community composition of plant species was done performing permutational multivariate analysis of variance (PERMANOVA) with the adonis function (R package *vegan*). A pairwise PERMANOVA was conducted to compare pairwise differences between each plant species using Benjamini & Hochberg (BH) correction for multiple testing [@Benjamini1995]. Because PERMANOVA tests the null hypothesis that centroids and variance of the defined groups are equivalent, an additional test for equality of variances was performed. For that, a permutational analysis of multivariate dispersion (PERMDISP) [@Anderson2006] was conducted. PERMANOVA, PERMDISP and pairwise PERMDISP were conducted in package *vegan*, and for pairwise PERMANOVA the R package *RVAideMemoire* `r packageVersion("RVAideMemoire")` [@RVA2021] used. All permutational tests were run with constrained permutations (n = 1000) to account for the difference in $\alpha$-diversity per plant species.

#### Nestedness and modularity

The overlap between interaction niches of species can be described using nestedness metrics for bipartite networks. First, an interaction matrix was constructed with columns of plant species and rows of AMF ASVs. All replicates per plant species were merged resulting in one sample per plant species with presence-absence information for each AMF. 
Presence-absence data was used because sequencing data is compositional and read abundances do not necessarily hold meaningful information [@Gloor2017]. Next, metrics to quantify nestedness were calculated based on the binary interaction matrix and on random null models. Lastly, the statistical significance of the nestedness of the observed matrix was assessed by comparison to that of the null models. The first metric is the nested overlap and decreasing fill (NODF) [@AlmeidaNeto2008] which can be used to calculate nestedness independently among rows and among columns of the interaction matrix, therefore allowing for evaluation of the contribution from both the AMF and the plants to the observed nestedness. To account for Type I and Type II errors, two different algorithms were chosen to generate null models: the FF (fixed-fixed) algorithm preserves both the degree of the rows and the columns to account for the inherent properties of the species (nodes) [@Gotelli2000], but is very conservative and prone to Type II errors [@Ulrich2009]. In contrasts, the SS (swappable-swappable) null model shuffles interactions randomly, conserving the matrix dimensions [@Beckett2014] and can suffer from Type I errors (tendency to falsely detect nestedness). Nestedness was considered significant if 95 % or more of the random matrices had smaller NODF values that the original interaction matrix. The second nestedness metric takes into account how the degree distribution of the nodes influences nestedness values, i.e., based on the higher probability of interaction for nodes with higher degrees [@Jonhson2013]. The JDMnestedness was calculated and compared against the FF null model as described by [@Jonhson2013]. 
Nestedness calculations that included null model generation were performed with the software package *FALCON* [@Beckett2014] in R, and additionally the NODF for AMF and plants separately was calculated using the R package *bipartite* `r packageVersion("bipartite")` [@bipartite2009; @bipartite2011]. Modularity was determined using Strona and Veech's [-@Strona2015] modularity index derived from node overlap and segregation (NOS) in *bipartite.* NOS is based on a probabilistic approach that determines whether the interactions shared between species of the same guild are different from random, i.e., as expected by chance, without the need to calculate random null models. The modularity index derived from NOS takes values between 0 (complete randomness) and 1 (maximum modularity).

\newpage

## Results

```{r ps-data-upload}

#Get ps (prepared, and saved from chunk phyloseq-object)
ps_ALL_0 <- read_rds("dataCh2/ps_ALL.rds")

#meta
 metaM0 <- read_xlsx("dataCh2/meta/M0_meta.xlsx")
## change meta table to short PlaSpe names
sample_data(ps_ALL_0) <- sample_data (metaM0 %>% data.frame (row.names = "sampleID"))

#Use metamMisq for
# ps_ALL_iter <-  phyloseq_coverage_raref(ps_ALL_0, iter =99) #produces 99 ps objects in list
# write_rds (ps_ALL_iter, "results/ps_ALL_iter99.rds")
# ps_ALL_iter <- read_rds("results/ps_ALL_iter99.rds")
# #get mean result of 99 iterations of resampling
# ALL_coverage_tables <-
#   map(ps_ALL_iter, otu_table) %>%
#   map (data.frame) %>%   map (t) %>%
#   map (function (df) as_tibble (df,rownames = "sampleID" )) %>%
#   map_dfr (bind_rows, .id = "TableNumber") %>%
#   pivot_longer(!c(TableNumber, sampleID), names_to =  "ASV_ID", values_to = "counts") %>%
#   group_by (sampleID, ASV_ID) %>% select (!TableNumber) %>% 
#   summarize (mean_counts = ceiling (mean(counts))) %>%
#   unique () %>%
#   pivot_wider(id_cols = , names_from = "sampleID", values_from = "mean_counts") %>%
#   data.frame()
# rownames (ALL_coverage_tables)<- ALL_coverage_tables$ASV_ID
# ALL_coverage_tables <- ALL_coverage_tables[,-1]
# 
# 
# # # get back into ps_ALL as otu_table
# ps_ALL <- ps_ALL_0  # to get a ps_ALL object
# otu_table (ps_ALL) <- otu_table(ALL_coverage_tables, taxa_are_rows = T)


#All fungi####
# ps_Fungi  <- subset_taxa (ps_ALL, Kingdom =="Fungi")  #all fungi selected
# ps_Fungi  <- prune_samples (sample_sums(ps_Fungi)>=1, ps_Fungi)  # samples that are empty after selection of fungi are removed

# get only Glo####
 ps_Glo <- read_rds("resultsCh2/ps_Glo.rds")

# change meta

sample_data (ps_Glo) <- sample_data (metaM0 %>% data.frame (row.names = "sampleID"))
#subset
# ps_Glo  <- subset_taxa(ps_ALL, Phylum == "Glomeromycota")
# ps_Glo  <- prune_taxa(taxa_sums(ps_Glo) >= 1, ps_Glo) # ASVs removed that are 0
# #because some Glo ASVs might be lost some samples might be empty, remove them
# ps_Glo  <- prune_samples(sample_sums(ps_Glo)>=1, ps_Glo)
# 
# # #GET data here instead of 
#  write_rds(ps_Glo, "results/ps_Glo.rds")

#Use presence-absence data only (binary)
ps_bin  <- metagMisc::phyloseq_standardize_otu_abundance (ps_Glo, method = "pa")

#agglomerate all replicates onto one sample, 
#otu_table counts will be reassigned as the mean of all the samples that are merged together. 
ps_bin_PlaSpe <- merge_samples(ps_bin, group = "PlantSpeciesfull") ### gives the sums
 
#total number of all ASVs from sample inference####
No_ASVs  <- ntaxa (ps_ALL_0)


# Glo Table ASVs####
 ASV_table_Glo  <- 
  otu_table (ps_Glo) %>% 
  data.frame() %>%  
   as_tibble (rownames = "ASV_ID") %>%   
   left_join ((tax_table (ps_Glo) %>% data.frame() %>%  as_tibble (rownames = "ASV_ID")), by = "ASV_ID")
  
 #number of ASvs per Glo Family####
 AMF_Fam_ASVs<- 
   ASV_table_Glo  %>%  
   group_by( Family)   %>%  
   tally()  %>%  
   arrange (desc(n))  %>%  
   as.data.frame()
 
 # All ASVs####
 AMF_all_ASVs<- ntaxa (ps_Glo)
 
 ## Percent of Glo Families in dataset (based on reads,####
 #counts per sample (um die % auszurechnen)
 total_counts_per_per_sample  <- 
   ASV_table_Glo  %>%  
   select (-c(Kingdom, Phylum, Class, Order, Genus, Species))  %>%  
   pivot_longer (!c(Family, ASV_ID), names_to = "sampleID", values_to="counts")  %>%  group_by(sampleID)  %>% 
   summarise(total_count=sum(counts)) 
 
 #  TOTAL read count for Glomeromycota ####
Glo_total_counts  <-ASV_table_Glo  %>%  
    select (-c(Kingdom, Phylum, Class, Order, Genus, Species))  %>%  
    pivot_longer (!c(Family, ASV_ID), names_to = "sampleID", values_to="counts")  %>%
   summarize(total=sum(across(counts)))

# relative number of reads per Glo family in whole dataset, in Percent ####
Glo_Fam_perc_reads  <- 
ASV_table_Glo  %>%  
    select (-c(Kingdom, Phylum, Class, Order, Genus, Species))  %>%  
    pivot_longer (!c(Family, ASV_ID), names_to = "sampleID", values_to="counts")  %>%
  group_by (Family)  %>% 
  summarize(Fam_sum=sum(counts)) %>% 
       add_column (Glo_total_counts)  %>% 
  mutate (Glo_Fam_perc = 100  * Fam_sum/total)  %>% 
  arrange(desc(Fam_sum))  %>% 
  as.data.frame()

 #No of removed samples by Glo and rarefying
 rem_samples <- nsamples(ps_ALL_0)-nsamples(ps_Glo)
 #
 richness_ps_ALL_0 <- estimate_richness(ps_ALL_0, measures = "Observed")

```

```{r dada2-results}
df_dada_1  <- read_csv("resultsCh2/dada2_pseudopooling.csv")#das ist der Vergleich der ASV Anzahl von indpendent dada2 und pseudopooling dada2
df_dada_reads <- read_csv("resultsCh2/track_reads_M0_PP.csv") #Tabelle aller Schritte von filtering, merging etc. für independent dada2 sample inference)
PPvsInd_ASVs  <-  df_dada_1    %>%  as_tibble()  %>% 
  group_by(mode) %>% 
  summarise(meanASV=mean(observed),sdASV = sd(observed)) # vergleich der beiden dada2 approaches - pseudopooling versus independent

#Anzahl readsmean
N_reads0<-df_dada_reads  %>% summarise(all_reads= sum(input))
N_reads<-df_dada_reads  %>% summarise(all_reads= sum(nonchim))
average_reads0  <- df_dada_reads  %>% summarise(ave_reads= mean(input))
average_reads  <- df_dada_reads  %>% summarise(ave_reads= mean(nonchim))

# ASV_table

```

```{r phyloseq-object}

# #get files#### ---- later, when rarefy needed, we have to use ALL sequences... and then select the Glomeromycota
# #tree####
# #Tree for Glo sequences only#### tree for full dataset is in dada2 script
# #Fasta file with all sequences
# ASV_fasta  <- seqinr::read.fasta(file = "results/ASVs.fa", seqtype = "DNA", as.string = TRUE)
# #Subset fasta file for Glo sequences
# ASV_fasta_Glo  <- ASV_fasta[names(ASV_fasta) %in% ASV_table_Glo$ASV_ID]
# seqinr::write.fasta(ASV_fasta_Glo, names = names(ASV_fasta_Glo), file.out= "results/ASV_fasta_Glo.fa")
# #Import fasta file as dna sequences for later alignment
# dna_Glo   <- readDNAStringSet("results/ASV_fasta_Glo.fa")
# #Alignment with ClustalW###
# ASVs_align_Glo  <-msa(dna_Glo, "ClustalW")
# #Convert for different package, here seqinr
# ASVs_align_Glo_forseqinr  <- msaConvert(ASVs_align_Glo, type= "seqinr::alignment")
# 
# #Build distance matrix as prep for tree###
# d_Glo <- seqinr::dist.alignment(ASVs_align_Glo_forseqinr, "identity")  #seqinr package
# #phylogenetic tree
# #neighbor joining
# ASV_tree_Glo  <- nj(d_Glo)  #ape package
# 
# #check if nJ was appropriate
# # x  <- as.vector(d_Glo)
# # y  <- as.vector(as.dist(cophenetic(ASV_tree_Glo)))
# # 
# # plot (x, y, xlab = "original distance", ylab = "distance in the tree",
# #       main = "Is NJ appropriate?", pch = 20, col = transp("black", 0.1), cex = 3)
# # abline(lm(y ~ x))
# # cor(x, y)^2
# ##when points on line, then fine. looked good enough
# 
# 
# 
# ##Matrix =otu-table herstellen für das Package####
# matrix  <-  ASV_table_Glo  %>%  
#   select(-c(Kingdom, Phylum, Class, Family, Order, Genus, Species))  %>% 
#   as.data.frame() 
# rownames(matrix)  <- matrix$ASV_ID
# matrix  <- matrix[,-1]
# 
# #OTU-table prep
# #here ASV abundances, ASVs in rows, samples in columns, nur GLO
# OTU  <- otu_table(as.matrix(matrix), taxa_are_rows = TRUE) # makes otu_table for phyloseq
# 
# #Sample data 
# # is my meta table 
# sampledata  <- sample_data (data.frame (metaM0, row.names = "sampleID")) ##sampledata for phyloseq
# 
# #taxa 
# # Add taxa data - maybe do as explained for their trait data#
# ##data frame with rows = ASVs, colums = traits (here: taxa names)
# taxa_df_Glo  <-  taxa_table %>%  
#   filter(Phylum=="Glomeromycota")  %>% 
#   select (!Kingdom) %>% 
#   as.data.frame()
# rownames(taxa_df_Glo)  <- taxa_df_Glo$ASV_ID
# TAX  <- tax_table(as.matrix(taxa_df_Glo[,-1])) #taxa table for phyloseq
# 
# #Tree
# #class(ASV_tree_Glo) #is it phylo? Yes, is okay already
# 
# ##Put together into phyloseq object
# ps_Glo  <- phyloseq(OTU, TAX, ASV_tree_Glo, sampledata)


# #Phyloseq_all####
# #OTUtable 
# OTU_ALL  <- otu_table ( ASV_counts, taxa_are_rows =  TRUE) # change to OTU_table object 
# 
# #taxa table
# taxa_test <- as.matrix(read.table("results/ASVs_taxonomy_simple.tsv", header=T,row.names=1, check.names=F, sep="\t")) # load
# TAX_ALL  <- tax_table(gsub(taxa_test[, colnames(taxa_test)],   pattern = "[a-z]__", replacement = ""))
#                 # remove name strings p__ etc with gsub. change to phyloseq taxa table with tax_table ()
# 
# #Sample data 
# # is my meta table 
# sampledata  <- sample_data (data.frame (metaM0, row.names = "sampleID")) ##sampledata for phyloseq

# ##Phyloseqtree for the full dataset here for reload if needed:####
# #
# dna<- readDNAStringSet ("results/ASVs.fa")
# #
# #
# ASVs_align  <-msa(dna, "ClustalW")
# ASVs_align_forseqinr  <- msaConvert(ASVs_align, type= "seqinr::alignment")
# #
### Citing this package
# If you use this package for research that is published later,
#you are kindly asked to cite it as follows:
#   U. Bodenhofer, E. Bonatesta, C. Horejš-Kainrath, and S. Hochreiter (2015).
#msa: an R package for multiple sequence alignment. Bioinformatics 31(24):3997-3999. DOI: 10.1093/bioinformatics/btv494.
 # Moreover, we insist that, any time you use/cite the package,
#you also cite the original paper in which the algorithm/method/package
#that you have been using has been introduced:
#   ClustalW:
#   J. D. Thompson, D. G. Higgins, and T. J. Gibson (1994).CLUSTAL W: improving the sensitivity of progressive multiple sequence alignment through sequence weighting, position-specific gap
#penalties and weight matrix choice. Nucleic Acids Res., 22(22):4673 4680. DOI: 10.1093/nar/22.22.4673.

# #Build distance matrix as prep for tree
# d <- dist.alignment(ASVs_align_forseqinr, "identity")  #seqinr package
# #tree
# #neighbor joining
# ASV_tree  <- nj(d)  #ape package
# ps_ALL  <- phyloseq(OTU_ALL, TAX_ALL, sampledata, ASV_tree)
# write_rds(ps_ALL, "data/ps_ALL.rds")


```

### Sequence data

The dataset included samples from 20 different plant species from five plant families, and soil samples (Table \@ref(tab:pl-list)). For each plant species and soil, five replicates were procured except for *A. capillaris* (four replicates), *E. repens* (three replicates), *P. cita* (three replicates) and *U. uncinata* (three replicates), for which the missing replicates did not reach harvesting age. Consequently, sequencing results for `r nsamples(ps_ALL_0)` samples were obtained with an average of `r round (average_reads0$ave_reads[1]) %>% prettyNum(big.mark = ",")` reads per sample and a total read number of `r round (N_reads0$all_reads[1]) %>%  prettyNum(big.mark = ",")`. After filtering, an average of `r round(average_reads$ave_reads[1]) %>% prettyNum(big.mark = ",")` or `r round(average_reads$ave_reads *100 /average_reads0$ave_reads)` % remained of the initial sequencing reads. The DADA2 pipeline resulted in a total number of `r ntaxa(ps_ALL_0) %>% prettyNum(big.mark = ",")` ASVs, and an average number of `r round(PPvsInd_ASVs$meanASV[2])` $\pm$ `r round(PPvsInd_ASVs$sdASV[2])` ASVs per sample, ranging from `r min (richness_ps_ALL_0)` to `r max (richness_ps_ALL_0)` ASVs across all samples were assigned. As samples were rarefied to their minimal coverage, which was at 99.9 % coverage, the average read count declined to `r round (mean (readcount(ps_ALL_0)))%>% prettyNum(big.mark = ",")` per sample. ASVs not belonging to the subphylum Glomeromycotina were removed from the data leaving `r AMF_all_ASVs` ASVs with a total count of `r Glo_total_counts$total[1]%>% prettyNum(big.mark = ",")` reads. This resulted in the removal of `r rem_samples` samples (Table \@ref(tab:pl-list)), in which ASVs belonging to the Glomeromycotina could not be detected. 

Overall, the frequencies of the ASVs and their distribution among the Glomeromycotinian families varied greatly: `r round (Glo_Fam_perc_reads$Glo_Fam_perc[1])` % of the reads were assigned to the family `r Glo_Fam_perc_reads$Family[1]`, followed by the `r Glo_Fam_perc_reads$Family[2]` (`r round (Glo_Fam_perc_reads$Glo_Fam_perc[2])` %), `r Glo_Fam_perc_reads$Family[3]` (`r round (Glo_Fam_perc_reads$Glo_Fam_perc[3])` %), `r  Glo_Fam_perc_reads$Family[4]` (`r round (Glo_Fam_perc_reads$Glo_Fam_perc[4])` %), `r Glo_Fam_perc_reads$Family[5]` (`r round (Glo_Fam_perc_reads$Glo_Fam_perc[5])` %), `r Glo_Fam_perc_reads$Family[6]` (`r round (Glo_Fam_perc_reads$Glo_Fam_perc[6])` %) and `r Glo_Fam_perc_reads$Family[8]` (`r round (Glo_Fam_perc_reads$Glo_Fam_perc[8], digits=2)` %). A total `r round (Glo_Fam_perc_reads$Glo_Fam_perc[7], digits=2)` % of the read counts were unidentified at the family level. If considering the diversity of ASVs affiliated with these families, a different picture emerges. Here, the `r AMF_Fam_ASVs$Family[1]` and  `r AMF_Fam_ASVs$Family[2]` dominate with a total of `r AMF_Fam_ASVs$n[1]` detected sequences each, followed by the  `r AMF_Fam_ASVs$Family[3]` (`r AMF_Fam_ASVs$n[3]` ASVs), `r AMF_Fam_ASVs$Family[4]` (`r AMF_Fam_ASVs$n[4]` ASVs), `r AMF_Fam_ASVs$Family[5]` (`r AMF_Fam_ASVs$n[5]` ASVs), `r AMF_Fam_ASVs$Family[6]` (`r AMF_Fam_ASVs$n[6]` ASVs), `r AMF_Fam_ASVs$Family[8]` (`r AMF_Fam_ASVs$n[8]` ASVs), and `r AMF_Fam_ASVs$n[7]` sequences were unidentified at the family level.

```{r pl-list, tab.cap = "Studied pasture plant species. Of the whole set of species, eleven were from the Poacaea, four species were Asteraceae, followed by two species each from the Fabaceae and Plantaginaceae, and one plant species originated from the Cyperaceae.", tab.id = "pl-list",  label = "pl-list" }

#calculate number of replicates for unique figure and or pl-list table####
#Problem : some plant species have only 2 or 3 replicates- the cumulative ASV number is not comparable, need to normalise it by including the number of the replicates 
#repl after selecting Glo ####
replicates_samples_after_Glo  <-
  ASV_table_Glo  %>% 
    select(-c(Kingdom, Phylum, Class, Family, Order, Genus, Species))  %>% 
    gather (!ASV_ID, key=sampleID, value = ASV_counts)  %>%
    left_join(metaM0, by="sampleID") %>% 
    filter (ASV_counts!=0) %>% 
    group_by (sampleID) %>%  
    tally(name="numberASVs") %>% 
    left_join(metaM0, by="sampleID") %>%  group_by(PlantSpeciesfull) %>%  tally() %>% 
 dplyr:: rename ("repl"= "n")

#repl before selecting Glo####
#use this because samples were only removed because they had 0 Glo, but they count as replicates, too!
replicates_samples_before  <-
    otu_table (ps_ALL_0) %>%  data.frame () %>%  as_tibble (rownames = "ASV_ID") %>% 
    gather (!ASV_ID, key=sampleID, value = ASV_counts)  %>%
    left_join(metaM0, by="sampleID") %>% 
    group_by (sampleID) %>%  
    tally(name="numberASVs") %>% 
    left_join(metaM0, by="sampleID") %>%  group_by(PlantSpeciesfull) %>%  tally() %>% 
    dplyr::rename ("repl"= "n")

## the table of plants ####
meta_plants_ch2 <- read_xlsx("dataCh2/meta/M0_meta.xlsx", sheet = "replicates" ) %>%  select ("Plant species", Abbreviation)


 metaM0 %>%  select (PlantSpeciesfull,  PlaSpe, PlantFamily) %>% 
  left_join (replicates_samples_before, by = "PlantSpeciesfull")  %>% 
  left_join (replicates_samples_after_Glo, by = "PlantSpeciesfull") %>% 
  unique () %>%  
  left_join (meta_plants_ch2, by = c("PlaSpe"= "Abbreviation")) %>% 
   select (-PlantSpeciesfull ) %>% 
   relocate ("Plant species") %>% 
  flextable ()  %>% 
  italic (j = "Plant species") %>% 
  set_header_labels(PlaSpe = "abbrevation" ,  PlantFamily = "Plant family" , 
                    repl.x = "# of replicates", repl.y = "# of replicates with AMF", Origin.of.seeds = "Origin of seeds") %>%   theme_vanilla() %>% 
  set_table_properties( width = 1, layout = "autofit") %>%  
  flextable::footnote (i = 17, j = 1, value = as_paragraph (c("Origin of seeds: nzseeds.co.nz") ))


```


```{r plot-rarefaction-ALL, fig.cap = "A) Rarefaction (solid line segments) and extrapolation (dotted line segments) sampling curves of fungal ASV richness for each plant species and soil. B) Sample completeness curves by sequence reads. c) Sample coverage curves. Shaded areas in each plot show the 95% confidence interval of the extrapolation. ", fig.height=8, fig.width=12, dpi = 300 }
#agglomerate to Plant Species
ps_ALL_perPlaSpe  <- merge_samples(ps_ALL_0, group = "PlantSpeciesfull")

#make df for Glo_PlaSpe data (empty samples samples pruned )
comm_df_ALL_PlaSpe  <- data.frame (otu_table (ps_ALL_perPlaSpe)) 

#rarefaction in iNEXT 
iNext_ALL_PlaSpe_richness <- iNEXT (t (comm_df_ALL_PlaSpe), 
                                q= 0, # richness
                                datatype =  "abundance", 
                                endpoint = NULL, 
                                knots = 400 )
# plot 
p1 <- ggiNEXT(iNext_ALL_PlaSpe_richness, type = 1) +
  theme_classic() + 
  scale_shape_manual(values = c(0,1, 2,3,4,5,6,7, 8,9,10, 12, 13,14,15, 16,17,22, 23,24, 25)) + 
  scale_y_continuous (name = "ASV richness") +
  scale_x_continuous(name = "Number of sequence reads")+
  theme (axis.text = element_text(size =14), 
         axis.title = element_text(size =18),
         legend.text = element_text(face = "italic"))+
 guides(     colour=guide_legend(title="Plant species"), 
             fill=guide_legend(title="Plant species"), 
             shape=guide_legend(title="Plant species"))


p2 <- ggiNEXT(iNext_ALL_PlaSpe_richness, type = 2) +
  theme_classic() + 
  scale_shape_manual(values = c(0,1, 2,3,4,5,6,7, 8,9,10, 12, 13,14,15, 16,17,22, 23,24, 25)) + 
  scale_x_continuous(name = "Number of sequence reads")+
  theme (axis.text = element_text(size =14), 
         axis.title = element_text(size =18), 
         legend.text = element_text(face = "italic"))+
 guides(     colour=guide_legend(title="Plant species"), 
             fill=guide_legend(title="Plant species"), 
             shape=guide_legend(title="Plant species"))

p3 <- ggiNEXT(iNext_ALL_PlaSpe_richness, type = 3) +
  theme_classic() + 
  scale_shape_manual(values = c(0,1, 2,3,4,5,6,7, 8,9,10, 12, 13,14,15, 16,17,22, 23,24, 25)) + 
  scale_y_continuous(name = "ASV richness")+
  scale_x_continuous (labels = scales::percent) + 
  theme (axis.text = element_text(size =14), 
         axis.title = element_text(size =18),  
         legend.text = element_text(face = "italic"))+
 guides(     colour=guide_legend(title="Plant species"), 
             fill=guide_legend(title="Plant species"), 
             shape=guide_legend(title="Plant species"))

ggarrange (p1,p2,p3, ncol = 3, common.legend = T, labels = "AUTO", legend = "bottom")


```


```{r plot-rarefaction-AMF, include =F,  fig.cap= "Rarefaction (solid line segments) and extraplotaion (dotted line segments) sampling curves of AMF ASV richness for each plant species and soil. Shaded areas show the 95% confidence interval of the extrapolation. A) Rarefaction and extrapolation based on sequencing reads. B) Coverage-based rarefaction and extrapolation sampling curves.", fig.height=8, fig.width=10, dpi = 300}


#agglomerate to Plant Species
ps_Glo_perPlaSpe  <- merge_samples(ps_Glo, group = "PlantSpeciesfull")

#make df for Glo_PlaSpe data (empty samples samples pruned )
comm_df_Glo_PlaSpe  <- data.frame (otu_table (ps_Glo_perPlaSpe)) # vegan expects samples as rows and 

#ASVs species as columns

#rarefaction instead 
iNext_PlaSpe_richness <- iNEXT (t (comm_df_Glo_PlaSpe), 
                       q= 0, # richness
                       datatype =  "abundance", 
                       endpoint = NULL, 
                       knots = 400 )
# plot 
p1 <- ggiNEXT(iNext_PlaSpe_richness, type = 1) +
  theme_classic() + 
  scale_shape_manual(values = c(0,1, 2,3,4,5,6,7, 8,9,10, 12, 13,14,15, 16,17,22, 23,24, 25)) + 
  scale_y_continuous (name = "ASV richness") +
  scale_x_continuous(name = "Number of sequence reads", limits = c(-100, 30010)) +
  theme (axis.text = element_text(size =16), 
         axis.title = element_text(size =20),  
         legend.text = element_text (face = "italic"))


p2 <- ggiNEXT(iNext_PlaSpe_richness, type = 3) +
  theme_classic() + 
  scale_shape_manual(values = c(0,1, 2,3,4,5,6,7, 8,9,10, 12, 13,14,15, 16,17,22, 23,24, 25)) + 
  scale_y_continuous (name = "ASV richness") +
  theme (axis.text = element_text(size =16), 
         axis.title = element_text(size =20),     
         legend.text = element_text (face = "italic"))


ggarrange (p1,p2, common.legend = T, labels = "AUTO", legend = "bottom")

```


### Plant-AMF interaction niche width

Several numeric and phylogenetic metrics were calculated to capture different aspects of the interaction niche width of the studied plant species regarding their mutualistic interaction with AMF.

#### The numeric interaction niche of the plant families

```{r species-degree-PlaFam-WelchANOVA}
#each replicate, then mean of Plant Family 
 meanASvsperFamily  <-
   ASV_table_Glo  %>% 
   select(-c(Kingdom, Phylum, Class, Family, Order, Genus, Species))  %>% 
   gather (!ASV_ID, key=sampleID, value = ASV_counts)  %>%
   left_join(metaM0, by="sampleID") %>% 
   filter (ASV_counts!=0) %>% 
   group_by (sampleID) %>%  
   tally(name="numberASVs") %>% 
   left_join(metaM0, by="sampleID") %>% 
 group_by(PlantFamily) %>% 
   summarize (meannumberASVsperFam =mean (numberASVs), sd=sd(numberASVs)) %>% 
   arrange (desc(meannumberASVsperFam))

#Welch  Test ####
Welch_PlFam  <- ASV_table_Glo  %>% 
  select(-c(Kingdom, Phylum, Class, Family, Order, Genus, Species))  %>% 
  gather (!ASV_ID, key=sampleID, value = ASV_counts) %>% 
  left_join(metaM0 ) %>% 
    left_join(replicates_samples_after_Glo, by = "PlantSpeciesfull") %>% 
  filter (ASV_counts!=0 ) %>% 
  filter (repl !=1) %>% 
  group_by (sampleID) %>%  
 tally(name="numberASVs") %>% 
  left_join(metaM0, by="sampleID") %>% 
  welch_anova_test(numberASVs ~ PlantFamily)


#Games Howell test ####
GamesHowell_PlFam  <- ASV_table_Glo  %>% 
  select(-c(Kingdom, Phylum, Class, Family, Order, Genus, Species))  %>% 
  gather (!ASV_ID, key=sampleID, value = ASV_counts)  %>%
  left_join(metaM0, by="sampleID") %>% 
  left_join(replicates_samples_after_Glo, by = "PlantSpeciesfull") %>% 
  filter (ASV_counts!=0 ) %>% 
  filter (repl !=1) %>% 
  group_by (sampleID) %>%  
 tally(name="numberASVs") %>% 
  left_join(metaM0, by="sampleID") %>% 
  games_howell_test(numberASVs ~ PlantFamily)

#Welch for Plant type - functional groups
Welch_PlType  <- ASV_table_Glo  %>% 
  select(-c(Kingdom, Phylum, Class, Family, Order, Genus, Species))  %>% 
  gather (!ASV_ID, key=sampleID, value = ASV_counts)  %>%
  left_join(metaM0, by="sampleID") %>% 
  filter (ASV_counts!=0, PlantType != "Soil") %>% 
  group_by (sampleID) %>%  
 tally(name="numberASVs") %>% 
  left_join(metaM0, by="sampleID") %>% 
  welch_anova_test(numberASVs ~ PlantType)

#Welch for Plant type - functional groups
Welch_PlFvGra  <- ASV_table_Glo  %>% 
  select(-c(Kingdom, Phylum, Class, Family, Order, Genus, Species))  %>% 
  gather (!ASV_ID, key=sampleID, value = ASV_counts)  %>%
  left_join(metaM0, by="sampleID") %>% 
  filter (ASV_counts!=0, PlantType != "Soil", PlantFamily != "Cyperaceae") %>% 
  group_by (sampleID) %>%  
 tally(name="numberASVs") %>% 
  left_join(metaM0, by="sampleID") %>% 
  welch_anova_test(numberASVs ~ PlantType)
```

```{r plot-ASV-speciesdegree-PlFamily, fig.cap="AMF richness per plant family and soil treatments. Glomeromicotinian ASVs were used as a proxy for AMF species. Horizontal lines show median values, boxes denote the interquartile range (IQR), while whiskers show 1.5$*$IQR ranges. Outliers are indicated by dots. No significances could be calculated between the Fabaceae (n=1 sample) and the other plant families. Significance of pairwise differences among families (Games-Howell test) are denoted: * p<0.05, ** p<0.005, *** p<0.0005, **** p<0.0001. ", fig.height=4, fig.width=3, dpi=300}
df <- GamesHowell_PlFam  %>%  filter (!p.adj.signif == "ns")  %>%   filter (!p.adj.signif == "")  %>%  
  dplyr::mutate(groups = purrr::pmap(.l = list(group1, group2), .f = c)) %>%
  dplyr::arrange(group1)
#Plot
p  <-   ASV_table_Glo  %>% 
  select(-c(Kingdom, Phylum, Class, Family, Order, Genus, Species))  %>% 
  gather (!ASV_ID, key=sampleID, value = ASV_counts)  %>%
  left_join(metaM0, by="sampleID") %>% 
  filter (ASV_counts!=0) %>% 
  group_by (sampleID) %>%  
 tally(name="numberASVs") %>% 
  left_join(metaM0, by="sampleID") %>% 
  group_by (PlantFamily) %>% 
  mutate(meannumberASVs=mean(numberASVs)) %>% 
  arrange(meannumberASVs)   %>% 
  ggplot(aes(x=PlantFamily, y=numberASVs, fill=PlantFamily))  +
  geom_boxplot()

p  +   stat_summary (fun=mean, geom="point", shape = 23)  +
  ylab(bquote("AMF richness" ~S[Ob]~"- number of ASVs")) +
  xlab ("Plant family and soil treatments") +
  theme_light () +
  theme(legend.position='none')+
  theme(axis.text.y= element_text(family= "sans", face= "italic", size = 11)) +
  theme (axis.text.x=element_text(family= "sans", size = 11, angle=45, hjust =1))  +
  theme (axis.title = element_text(family = "sans", size = 12 ),  
         legend.title=element_text(size=12), 
         legend.text=element_text(size=11)) +
  #scale_x_discrete(labels = c("Asteraceae", etc)) #ggf names mit trennstrich machen
  scale_fill_manual(values=c("Asteraceae"= "#edc948" ,"Cyperaceae" ="#5A6351" ,
                             "Fabaceae" = "#f28e2b" , 
                             "Plantaginaceae"= "#4e79a7","Soil"= "#9c755f",
                             "Poaceae" = "#7F9A65" )) +
  ggsignif::geom_signif (      comparisons = df$groups,
    annotation = c (df %>% pull (p.adj.signif)),
    step_increase = 0.06, vjust = 0.5)

#Plot functional groups forbs grasses
# p2  <-   ASV_table_Glo  %>% 
#   select(-c(Kingdom, Phylum, Class, Family, Order, Genus, Species))  %>% 
#   gather (!ASV_ID, key=sampleID, value = ASV_counts)  %>%
#   left_join(metaM0, by="sampleID") %>% 
#   filter (ASV_counts!=0) %>% 
#   group_by (sampleID) %>%  
#  tally(name="numberASVs") %>% 
#   left_join(metaM0, by="sampleID") %>% 
#   group_by (PlantType) %>% 
#   mutate(meannumberASVs=mean(numberASVs)) %>% 
#   arrange(meannumberASVs)   %>% 
#   ggplot(aes(x=PlantType, y=numberASVs, fill=PlantType))  +
#   geom_boxplot()


##Mean number AMF per Plant type -graminouids forbs
PlType  <-  ASV_table_Glo  %>% 
     select(-c(Kingdom, Phylum, Class, Family, Order, Genus, Species))  %>%
     gather (!ASV_ID, key=sampleID, value = ASV_counts)  %>%
     left_join(metaM0, by="sampleID") %>%
     filter (ASV_counts!=0) %>%
     group_by (sampleID, PlantType) %>%
    tally(name="numberASVs") %>% group_by (PlantType) %>% 
   summarize(meantype =mean (numberASVs), sd = sd (numberASVs))


PlPoa  <-  ASV_table_Glo  %>% 
     select(-c(Kingdom, Phylum, Class, Family, Order, Genus, Species))  %>%
     gather (!ASV_ID, key=sampleID, value = ASV_counts)  %>%
     left_join(metaM0, by="sampleID") %>%
     filter (ASV_counts!=0, PlantFamily != "Cyperaceae") %>%
     group_by (sampleID, PlantType) %>%
    tally(name="numberASVs") %>% group_by (PlantType) %>% 
   summarize(meantype =mean (numberASVs), sd = sd (numberASVs))


```

Plant host identity influenced the composition of the AMF community in plant roots both at the plant species and family level. The five different plant families showed significant differences in their numeric interactions with AMF (Figure \@ref(fig:plot-ASV-speciesdegree-PlFamily)). The mean AMF species richness, calculated from the total number of AMF species per plant individual, was larger in the Plantaginaceae (S̅~Ob~ = `r round(meanASvsperFamily$meannumberASVsperFam[1])` ± `r round(meanASvsperFamily$sd[1])`) and Poaceae (S̅~Ob~ = `r round(meanASvsperFamily$meannumberASVsperFam[2])` ± `r round(meanASvsperFamily$sd[2])`) than in the `r meanASvsperFamily$PlantFamily[4]` (S̅~Ob~= `r round(meanASvsperFamily$meannumberASVsperFam[4])` ± `r round(meanASvsperFamily$sd[4])`), `r meanASvsperFamily$PlantFamily[5]` (S̅~Ob~ = `r round(meanASvsperFamily$meannumberASVsperFam[5])` ± `r round(meanASvsperFamily$sd[5])`) or `r meanASvsperFamily$PlantFamily[6]` (S̅~Ob~ = `r round(meanASvsperFamily$meannumberASVsperFam[6])` ± `r round(meanASvsperFamily$sd[6])`) (Welch's ANOVA, F(`r Welch_PlFam$DFn`, `r round (Welch_PlFam$DFd,2)`) = `r Welch_PlFam$statistic`, *p* \< 0.001). A Games-Howell *post-hoc* test revealed that both the Plantaginaceae and Poaceae had a significantly higher AMF richness than the three other plant families, but the Plantaginaceae and Poaceae did not differ in their AMF richnesses. Likewise, the AMF richnesses of the Asteraceae, Fabaceae and Cyperaceae were not significantly different. If the functional groups forb versus graminoids are compared, the mean AMF richness of the forbs (`r round (PlType$meantype[1])` ± `r round (PlType$sd[1])`) was smaller than for the graminoids (`r round (PlType$meantype[2])` ± `r round (PlType$sd[2])`; Welch's ANOVA, F(`r Welch_PlType$DFn`, `r round (Welch_PlType$DFd,2)`) = `r Welch_PlType$statistic`, *p* \< 0.001), and also significantly smaller than that of the grasses (Poaceae) which had a mean AMF richness of `r round(PlPoa$meantype[2])` ± `r round(PlPoa$sd[2])` (Welch's ANOVA, F(`r Welch_PlFvGra$DFn`, `r round (Welch_PlFvGra$DFd,2)`) = `r Welch_PlFvGra$statistic`, *p* \< 0.001).

#### The numeric interaction niche width of the plant species ($\alpha$-diversity)

```{r ASVrichness-calc}

# Richness, a-diversity LINKS PER Species######
#ASV_richness# Table
#mean ASVs per sample####

meannumberASVspersample   <-
 ASV_table_Glo  %>% 
 select(-c(Kingdom, Phylum, Class, Family, Order, Genus, Species))  %>% 
 gather (!ASV_ID, key=sampleID, value = ASV_counts)  %>%
 left_join(metaM0, by="sampleID") %>% 
 filter (ASV_counts!=0) %>% 
 group_by (sampleID) %>%  
 tally(name="numberASVs") %>% 
 left_join(metaM0, by="sampleID") %>% 
 group_by (PlaSpe) %>% 
 mutate(meannumberASVs=mean(numberASVs)) %>% 
 arrange(meannumberASVs) 

#mean ASVs per plant species####
meannumberASVsper_species   <-
 ASV_table_Glo  %>% 
 select(-c(Kingdom, Phylum, Class, Family, Order, Genus, Species))  %>% 
 gather (!ASV_ID, key=sampleID, value = ASV_counts)  %>%
 left_join(metaM0, by="sampleID") %>% 
 filter (ASV_counts!=0) %>% 
 group_by (sampleID) %>%  
 tally(name="numberASVs") %>% 
 left_join(metaM0, by="sampleID") %>% 
 group_by (PlaSpe) %>% 
 mutate(meannumberASVs=mean(numberASVs)) %>% 
 arrange(meannumberASVs)   %>% 
 select (-c(sampleID, numberASVs)) %>% 
 group_by (PlantSpeciesfull, PlantFamily, PlantType)  %>% 
 summarize (mean_n_ASV_per_species  = mean(meannumberASVs))

# total ASVs in dataset####
totalASVs  <-ASV_table_Glo %>%  tally()

# unique ASVs per plant species ####
uniqueASVsperPlaSpe   <-  ASV_table_Glo  %>% 
 select(-c(Kingdom, Phylum, Class, Family, Order, Genus, Species))  %>% 
 gather (!ASV_ID, key=sampleID, value = ASV_counts)  %>%
 left_join(metaM0, by="sampleID") %>% 
 filter (ASV_counts!=0)  %>% 
 group_by(PlantSpeciesfull, ASV_ID)  %>% 
 tally ()  %>% 
 group_by(PlantSpeciesfull)  %>% 
 tally (name ="uniqueASVsperPlSpe")

## Use phyloseq ##

adiv_richness  <- estimate_richness(ps_Glo, measures = c("Observed", "Chao1", "Shannon")) %>% 
 as_tibble (rownames = "sampleID")

#Add Wilcoxon rank sum test 
adiv_kruskal_ob  <- adiv_richness %>%  
 left_join (metaM0, by = "sampleID")  %>%  
 rstatix::kruskal_test(Observed ~ PlaSpe)

adiv_kruskal_chao  <- adiv_richness %>%  
 left_join (metaM0, by = "sampleID")  %>%  
 rstatix::kruskal_test(Chao1 ~ PlaSpe)

adiv_kruskal_shannon  <- adiv_richness %>%  
 left_join (metaM0, by = "sampleID")  %>%  
rstatix::kruskal_test(Shannon ~ PlaSpe)

# get values for richnesses , especially min and max value
minmax_Sob  <- adiv_richness %>%  
 left_join(metaM0, by = "sampleID")  %>%  
 group_by (PlaSpe)   %>% 
 summarize (mean = mean (Observed))  %>%  arrange (mean)

minmax_chao1  <- adiv_richness %>%  
 left_join(metaM0, by = "sampleID")  %>%  
 group_by (PlaSpe)   %>% 
 summarize (mean = mean (Chao1))  %>%  arrange (mean)

minmax_Shannon  <- adiv_richness %>%  
 left_join(metaM0, by = "sampleID")  %>%  
 group_by (PlaSpe)   %>% 
 summarize (mean = mean (Shannon))  %>%  arrange (mean)
```



The AMF species richness S~Ob~, Chao1 diversity and Shannon's diversity index H' provide a proxy for the numeric interaction niche width of the mycorrhizal association of the respective plant species. Comparison of three diversity metrics S~Ob~, Chao1 and H' resulted in similar patterns of diversity, showing large differences among the AMF community $\alpha$-diversity of the plant species and soil (Figure \@ref(fig:plot-ASVrichness)). The mean S~Ob~ differed significantly among the plant species (Kruskal-Wallis $\chi$^2^ = `r round (adiv_kruskal_ob$statistic[1],2)`, *p* \< 0.001, df = `r adiv_kruskal_ob$df[1]`), ranging from `r round(meannumberASVspersample$meannumberASVs[1],1)` for *`r meannumberASVspersample$PlantSpeciesfull[1]`* to `r round(meannumberASVspersample$meannumberASVs[83],1)` for *`r meannumberASVspersample$PlantSpeciesfull[83]`*. The mean Chao1 diversity showed highly similar values to the observed AMF richness S~Ob~, also ranging from `r round (minmax_chao1$mean[1], 2)` to `r round (minmax_chao1$mean[21],2)` with significant differences in Chao1 diversity among the plant species (Kruskal-Wallis $\chi$^2^ = `r round (adiv_kruskal_chao$statistic[1],2)`, *p* \< 0.001, df = `r adiv_kruskal_chao$df[1]`). The Shannon diversity index H' showed a different pattern than S~Ob~ and Chao1 diversity with *P. cita*, for example, having a more even AMF community than other plants of similar or larger   S~Ob~, like *L. perenne*. Overall, the mean values for H' ranged from `r round(minmax_Shannon$mean[1], 2)` to `r round(minmax_Shannon$mean[21], 2)` and were significantly different among the treatments (Kruskal-Wallis $\chi$^2^ = `r round (adiv_kruskal_shannon$statistic[1],2)`, *p* \< 0.001, df = `r adiv_kruskal_shannon$df[1]`). The AMF community of the soil treatment was also included. Soil samples had an average of `r round(meannumberASVspersample$meannumberASVs[28],1)` ASVs, which was a greater number than for each plant species in the Asteraceae, Cyperaceae, and Fabaceae, indicating that their AMF community is not randomly assembled, but that these plants were either non-mycorrhizal, at least at the time of the sampling, or that selection processes were at play between the plant and the AMF that resulted in a low-mycorrhizal state.
On the other end of the spectrum were most species of the Poaceae and the Plantaginaceae, having large mean AMF richnesses with values around 50 ASVs per species (Figure \@ref(fig:plot-ASVrichness)). In contrast, *L. perenne* and the native grass *P. cita* hosted intermediate levels of AMF $\alpha$-diversity, similar to that found in soil samples. 


```{r plot-ASVrichness, fig.cap= "AMF $\\alpha$-diversity per plant species. Shown are Chao1 diversity, the AMF species richness S, and Shannon's diversity index H'. Horizontal lines show median values, boxes denote the interquartile range (IQR), while whiskers show 1.5*IQR ranges. Outliers are indicated by dots.",   fig.align="center", fig.width=9, fig.height=6}





#Plot richness
c
p1  <-   adiv_richness %>% 
 left_join(metaM0, by="sampleID") %>% 
 filter (Observed!=0) %>% 
 group_by (PlantSpeciesfull) %>% 
 mutate(meannumberASVs=mean(Observed)) %>% 
  select (sampleID, Observed, Chao1, Shannon, PlantSpeciesfull, meannumberASVs, PlantFamily)  %>% 
  pivot_longer(cols=c(Observed, Chao1, Shannon), names_to ="diversity_measure", values_to = "div", values_drop_na = TRUE) %>% 
 arrange(meannumberASVs)   %>% 
 mutate (diversity_measure2 = str_replace(diversity_measure, "Observed","Richness S")) %>% 
   mutate (diversity_measure3 = str_replace(diversity_measure2, "Shannon","Shannon's H'")) %>% 
   mutate (PlantFamily = str_replace(PlantFamily, "Soil","Soil control")) %>% 
 ggplot(aes(y=reorder(PlantSpeciesfull, -meannumberASVs), x=div, fill=PlantFamily))  +
 geom_boxplot()  +
  facet_wrap(vars(diversity_measure3), scales = "free_x")

p1 +  
stat_summary (fun=mean, geom="point", shape = 23)  +
 xlab("\u03b1-diversity") +
 ylab ("Plant species") +
 theme_bw () +
 theme(axis.text.y= element_text(family= "sans", face= "italic", size = 11)) +
 theme (axis.text.x=element_text(family= "sans", size = 11), legend.position = "bottom")  +
 theme (axis.title = element_text(family = "sans", size = 13 ),  
        legend.title=element_text(size=12), 
        legend.text=element_text(size=11)) +
 scale_fill_manual(values=c("Asteraceae"= "#edc948" ,"Cyperaceae" ="#5A6351" ,
                            "Fabaceae" = "#f28e2b" , 
                            "Plantaginaceae"= "#4e79a7","Soil control"= "#9c755f",
                            "Poaceae" = "#7F9A65" ), name = "Plant family") 
#Plot beta diversity among replicates
#β-diversity among AMF communities associated with individual plants.species 
#A) numeric β-diversity illustrating the relative compositional heterogeneity of the AMF communities associated with the plant species. 

comp_units   <- meannumberASVsper_species  %>% 
 left_join(uniqueASVsperPlaSpe)  %>% 
  left_join (replicates_samples_after_Glo) %>% 
 group_by (PlantSpeciesfull) %>% 
 mutate (unique = mean(uniqueASVsperPlSpe)) %>% 
 mutate (CUnits = 5* unique/(mean_n_ASV_per_species*repl)) %>% # repl adjusted for , times 5 to get back to number of 5 replicates 
  mutate ("1-CU" = 1 - CUnits) 

# #Plot boxplot species degree only ####
# p  <-   ASV_table_Glo  %>% 
#   select(-c(Kingdom, Phylum, Class, Family, Order, Genus, Species))  %>% 
#   gather (!ASV_ID, key=sampleID, value = ASV_counts)  %>%
#   left_join(metaM0, by="sampleID") %>% 
#   filter (ASV_counts!=0) %>% 
#   group_by (sampleID) %>%  
#  tally(name="numberASVs") %>% 
#   left_join(metaM0, by="sampleID") %>% 
#   group_by (PlaSpe) %>% 
#   mutate(meannumberASVs=mean(numberASVs)) %>% 
#   arrange(meannumberASVs)   %>% 
#   ggplot(aes(x=reorder(PlantSpeciesfull, -meannumberASVs), y=numberASVs, fill=PlantFamily))  +
#   geom_boxplot()
# 
# p +coord_flip () +
#   stat_summary (fun=mean, geom="point", shape = 23)  +
#   ylab("Species degree - number of ASVs") +
#   xlab ("Plant species") +
#   theme_light () +
#   theme(axis.text.y= element_text(family= "sans", face= "italic", size = 11)) +
#   theme (axis.text.x=element_text(family= "sans", size = 11))  +
#   theme (axis.title = element_text(family = "sans", size = 13 ),  
#          legend.title=element_text(size=12), 
#          legend.text=element_text(size=11, face = "italic")) +
#   scale_fill_manual(values=c("Asteraceae"= "#edc948" ,"Cyperaceae" ="#5A6351" ,
#                              "Fabaceae" = "#f28e2b" , 
#                             "Plantaginaceae"= "#4e79a7","Soil"= "#9c755f",
#                             "Poaceae" = "#7F9A65" )) 

```

#### The unique number of AMF species per plant species

The number of unique ASVs per plant species, i.e., their $\gamma$-diversity, is one of several proxies to describe the interaction niche width. Fig. \@ref(fig:plot-uniqueASVs-perPlant) shows the cumulative unique ASVs per plant species and soil, sorted by mean species richness. The composition of the AMF communities differed among treatments. Plant species with small cumulative AMF species values (species von Asteraceae, Cyperaceae and Fabaceae) showed highly variable AMF community compositions, while members of the Poaceae and Plantaginaceae had similar compositions of the AMF genera, with some exceptions. The AMF genus *Cetraspora*, for example, could only be detected in three plant species; *P. cita*, *S. arundinaceus* and *E. repens*, all of which hosted small numbers of unique ASVs in small relative abundances. In contrast, the AMF genus *Acaulospora* dominated the community, in terms of relative abundance in *L. perenne* and *P. pratense*, but could not be detected in *P. cita*, *S. arundinaceus*, *A. pratensis*, or *E. repens*. The genera *Archaespora* and *Claroideoglomus* were a proportionally dominant component of the AMF community in most members of the Poacaea and Plantaginaceae, and the highest number of unique ASVs were detected for the *Archaeospora* in these two plant families. Notable also was the even composition of the AMF genera in *P. cita* and in soil, which was also corroborated by their comparably high Shannon's diversity H' (Fig. \@ref(fig:plot-ASVrichness)). Furthermore, *P. cita* and *A. millefolium* hosted a significantly higher proportion of AMF from the genus *Glomus* than the other plant species.

```{r plot-uniqueASVs-perPlant, fig.cap = "Distribution of AMF genera per treatment.   A) Absolute number of unique ASVs per AMF genus and treatment ($\\gamma$-diversity). Note the different number of replicates for AchMil (4), AgrCap (4), CicInt (3), CirVul (4), ElyRep (3), LeoAut (2), MedSat (2), PoaCit (3), TriPra (1), and UncUnc (2). B) Relative read abundances per AMF genus and treatment.", fig.height= 6, fig.width=10}



#Plot unique ASVs per plant####
  # Verteilung der Glo genera per plant species 
p1  <-  ASV_table_Glo  %>%  
    select(-c(Kingdom, Phylum, Class, Order, Species))  %>% 
    gather (-c(ASV_ID, Genus, Family), key=sampleID, value = ASV_counts)  %>%
    left_join(metaM0, by="sampleID") %>% 
    filter (ASV_counts!=0)  %>% 
    group_by (ASV_ID, PlantSpeciesfull, Genus) %>% 
    tally () %>%
    group_by(Genus, PlantSpeciesfull )  %>% 
    tally () %>% 
  replace_na(list (Genus="unidentified"))   %>% 
    left_join (meannumberASVsper_species, by = "PlantSpeciesfull")  %>% 
    ggplot (aes(x=reorder (PlantSpeciesfull,-mean_n_ASV_per_species), y= n, fill=Genus))  +
    geom_bar(position="stack", stat ="identity") 

p1<-
  p1+ theme_light()+ 
  coord_flip () +
  ylab("Unique AMF ASVs per plant species") +
  xlab ("Plant species") +
  theme_light () +
  theme(axis.text.y= element_text(family= "sans", face= "italic", size = 12)) +
  theme (axis.text.x=element_text(family= "sans", size = 12))  +
  theme (axis.title = element_text(family = "sans", size = 14 ),  
         legend.title=element_text(size=13), 
         legend.text=element_text(size=11, face ="italic")) +
    scale_fill_manual(values = c( "unidentified" ="#C7EAE5", "Archaeospora"  = "#287D8EFF" , 
                                  "Acaulospora" =  "#C1A363" , 
                                  "Funneliformis" = "#F6E8C3", "Cetraspora"  = "#DFC27D", 
                                  "Claroideoglomus" =  "#20A386FF", 
                                  "Glomus" = "#80CDC1" , "Paraglomus" = "#35978F", 
                                  "Scutellospora"= "#01665E" ), name = "AMF genus"    )  
  p2   <- 
  ASV_table_Glo   %>%   
  select(-c(Kingdom, Phylum, Class, Order, Species, Family))  %>% 
  gather (-c(ASV_ID, Genus), key=sampleID, value = ASV_counts)  %>%
  group_by (sampleID, Genus)  %>% 
  summarise (ASV_sum_per_sample = sum(ASV_counts)) %>% 
replace_na(list (Genus="unidentified"))   %>% 
  left_join (metaM0, by = "sampleID")  %>% 
  group_by (PlantSpeciesfull, Genus)  %>% 
  summarise (mean_ASV_count_per_species = mean (ASV_sum_per_sample))  %>% 
  left_join (meannumberASVsper_species, by = "PlantSpeciesfull")  %>% 
  ggplot (aes(x=reorder (PlantSpeciesfull,-mean_n_ASV_per_species), 
              y= mean_ASV_count_per_species, fill=Genus))  +
  geom_bar(position="fill", stat ="identity") 

p2  <- 
  p2 + coord_flip () +
  ylab("Relative abundance of ASV reads (%)") +
  theme_light () +
  theme(axis.text.y= element_text(family= "sans", face= "italic", size = 12)) +
 theme (axis.text.x=element_text(family= "sans", size = 12))  +
  theme (axis.title = element_text(family = "sans", size = 14 ),  
         legend.title=element_text(size=13), 
         legend.text=element_text(size=11, face = "italic"),
         axis.title.y = element_blank()) +
  scale_fill_manual(values = c( "unidentified" ="#C7EAE5", "Archaeospora"  = "#287D8EFF" , 
                                "Acaulospora" =  "#C1A363" , 
                                "Funneliformis" = "#F6E8C3", "Cetraspora"  = "#DFC27D", 
                                "Claroideoglomus" =  "#20A386FF", 
                                "Glomus" = "#80CDC1" , "Paraglomus" = "#35978F", 
                                "Scutellospora"= "#01665E" ) , name = "AMF genus"   )  +
  scale_y_continuous(labels = scales::percent)+
  scale_x_discrete(labels = NULL)

ggarrange (p1,NULL, p2, labels = c("A","B", ""), nrow = 1, ncol =3, widths =  c(2,0.07,1.2), legend = "bottom", common.legend = T)


  
```

#### A phylogenetic view of the plants' interaction niche width

```{r PD-phylogenetic distances}


#Standardised PD####


#make df for Glo data (empty samples samples pruned )
comm_df_Glo  <- data.frame (t(otu_table (ps_Glo))) # vegan expects samples as rows and ASVs species as columns

#hier total randomised community data frame as null model
stand_pd_Glo_all  <- as_tibble (ses.pd(comm_df_Glo, phy_tree(ps_Glo), include.root = FALSE, null.model = "independentswap", runs=100, iterations=999), rownames="sampleID")

#In addition, the observed and standardised Faith's PD~PlaSpe~ were calculated for each plant species after agglomeration of the individual samples into one sample per plant species. Comparison of the agglomerated Faith's PD PlaSpe and Faith's PD I reveal ... yeah, the same as in ordination actually.

#agglomerate to Plant Species
ps_Glo_perPlaSpe  <- merge_samples(ps_Glo, group = "PlantSpeciesfull")

#make df for Glo_PlaSpe data (empty samples samples pruned )
comm_df_Glo_PlaSpe  <- data.frame (otu_table (ps_Glo_perPlaSpe)) # vegan expects samples as rows and ASVs species as columns

stand_pd_Glo_PlaSpe  <- as_tibble (ses.pd(comm_df_Glo_PlaSpe, phy_tree(ps_Glo), include.root = FALSE, null.model = "independentswap", runs=1000, iterations=999), rownames="sampleID")


```

Faith's phylogenetic diversity (PD) values varied from `r round (min(stand_pd_Glo_all$pd.obs, na.rm =T),2)` to `r round (max(stand_pd_Glo_all$pd.obs, na.rm =T),2)` (Fig. \@ref(fig:plot-PD)) and mean PD values were similar in pattern to the numeric diversity values (Fig. \@ref(fig:plot-ASVrichness)) with some exceptions. 
For example, soil had a higher PD than would be expected from its numeric richness S~Ob~. Overall, the Asteraceae, Cyperaceae and Fabaceae showed similarly small mean PD values in comparison to the Plantaginaceae and Poaceae, which were higher. 
The mean z-score of the PD differed notably from the pattern of the mean PD, and most samples' AMF communities were not significantly different from randomised AMF communities, indicating that most plant species hosted communities of intermediate phylogenetic richness (Table \@ref(tab:pd-table1)). 
However, after merging of the replicates per treatment, the resulting z-scores can be interpreted as indicator for the within-treatment phylogenetic similarity of the AMF communities (Table \@ref(tab:pd-table2)). 
This revealed that, contrary to Hypothesis 1, the grasses did not show less filtering of their AMF community than the forbs. Instead, several of the plant species studied from the Poaceae hosted AMF communities that were significant different from randomly assembled ones.
For example, individual *D. glomerata* plants seemed to host very similar AMF communities with a z-score~PD~ of `r round (stand_pd_Glo_PlaSpe$pd.obs.z[8], 2)` (*p* = `r round (stand_pd_Glo_PlaSpe$pd.obs.p[8], 2)`), as do *C. vulgare* (z-score~PD~ = `r round (stand_pd_Glo_PlaSpe$pd.obs.z[7], 2)`, *p* = `r round (stand_pd_Glo_PlaSpe$pd.obs.p[7], 2)`), *A. pratensis* (z-score~PD~ = `r round (stand_pd_Glo_PlaSpe$pd.obs.z[3], 2)`, *p* = `r round (stand_pd_Glo_PlaSpe$pd.obs.p[3], 2)`), *L. perenne* (z-score~PD~ = `r round (stand_pd_Glo_PlaSpe$pd.obs.z[12], 2)`, *p* = `r round (stand_pd_Glo_PlaSpe$pd.obs.p[12], 2)`), *H. lanatus* (z-score~PD~ = `r round (stand_pd_Glo_PlaSpe$pd.obs.z[10], 2)`, *p* = `r round (stand_pd_Glo_PlaSpe$pd.obs.p[10], 2)`), and *E. repens* (z-score~PD~ = `r round (stand_pd_Glo_PlaSpe$pd.obs.z[9], 2)`, *p* = `r round (stand_pd_Glo_PlaSpe$pd.obs.p[9], 2)`). 
In contrast to the plant species having hosted AMF communities of lower phylogenetic richness than expected from a random assembled community, the soil AMF community had a positive z-score~PD~ of `r round (stand_pd_Glo_PlaSpe$pd.obs.z[19], 2)`, indicating high phylogenetic richness of the AMF community (non-significant).

```{r plot-PD, fig.cap="Faith's phylogenetic diversity (PD). A) PD by plant species. B) PD effect size (z-score~PD~). Black squares denote PD of the treatment based on merged replicates, for which only significant results are shown based on a comparison of observed PD against PD from randomised communities. Horizontal lines show median values, empty squares show means, boxes denote the interquartile range (IQR), while whiskers show 1.5*IQR ranges. Outliers are indicated by dots.", fig.height= 6, fig.width=10}

p1  <-stand_pd_Glo_all  %>% 
  left_join (metaM0, by= "sampleID")  %>% 
  filter (!is.na(pd.obs)) %>% 
  group_by (PlaSpe) %>% 
  mutate(mean_pd.obs =mean(pd.obs)) %>% 
  arrange(mean_pd.obs)   %>% 
  left_join ((meannumberASVsper_species %>% ungroup () %>% select (-PlantFamily, PlantSpeciesfull, mean_n_ASV_per_species)), by = "PlantSpeciesfull")  %>% 
     mutate (PlantFamily = str_replace(PlantFamily, "Soil","Soil control")) %>% 
  ggplot (aes(x=reorder (PlantSpeciesfull,-mean_n_ASV_per_species),
             y=pd.obs, fill=PlantFamily))  +
  geom_boxplot() #  +   geom_point (data = (stand_pd_Glo_PlaSpe %>%  left_join(metaM0, by= c("sampleID"= "PlantSpeciesfull"))), aes (x= sampleID, y =pd.obs ), shape = 15)
#that are single points to see 
p1  <- p1 +coord_flip () +
  stat_summary (fun=mean, geom="point", shape = 23)  +
  ylab("Faith's phylogenetic diversity") +
  xlab ("Plant species") +
  theme_light () +
  theme(axis.text.y= element_text(family= "sans", face= "italic", size = 12)) +
  theme (axis.text.x=element_text(family= "sans", size = 12))  +
  theme (axis.title = element_text(family = "sans", size = 14 ),  
         legend.title=element_text(size=13), 
         legend.text=element_text(size=11)) +
  scale_fill_manual(values=c("Asteraceae"= "#edc948" ,"Cyperaceae" ="#5A6351" ,
                             "Fabaceae" = "#f28e2b" , 
                             "Plantaginaceae"= "#4e79a7","Soil control"= "#9c755f",
                             "Poaceae" = "#7F9A65" ), name = "Plant family") 
#Plot Faiths PD-mean effect size####
p2  <-  stand_pd_Glo_all %>% 
  left_join (metaM0, by= "sampleID")  %>% 
  filter (!is.na(pd.obs.z)) %>% 
  group_by (PlaSpe) %>% 
  mutate(mean_pd.obs.z =mean(pd.obs.z)) %>% 
  #arrange(mean_pd.obs.z)   %>% 
  left_join ((meannumberASVsper_species %>% ungroup () %>% 
                select (-PlantFamily, PlantSpeciesfull, mean_n_ASV_per_species)), by = "PlantSpeciesfull")  %>% #to sort like in richness plot
       mutate (PlantFamily = str_replace(PlantFamily, "Soil","Soil control")) %>% 
  ggplot (aes(x=reorder (PlantSpeciesfull,-mean_n_ASV_per_species), 
             y=pd.obs.z, fill=PlantFamily))  +  
  geom_boxplot() +
    geom_point (data = (stand_pd_Glo_PlaSpe %>% 
                          left_join (metaM0, by= c("sampleID" = "PlantSpeciesfull")) %>% 
                          filter (!between (pd.obs.p, 0.05, 0.95))%>% 
                          select (sampleID, PlantFamily, pd.obs.z)  %>% unique ()),
                aes(x=sampleID, y= pd.obs.z), shape = 15, size =3) 


p2 <- p2  +coord_flip () +
  stat_summary (fun=mean, geom="point", shape = 23)  +
  ylab("Faith's PD effect size (z-score)") +
 # xlab ("Plant Species") +
  theme_light () +
  theme(axis.text.y= element_text(family= "sans", face= "italic", size = 12)) +
  theme (axis.text.x=element_text(family= "sans", size = 12))  +
  theme (axis.title = element_text(family = "sans", size = 14 ),  
         legend.title=element_text(size=13), 
         legend.text=element_text(size=11), 
         axis.title.y = element_blank()) +
  scale_fill_manual(values=c("Asteraceae"= "#edc948" ,"Cyperaceae" ="#5A6351" ,
                             "Fabaceae" = "#f28e2b" , 
                             "Plantaginaceae"= "#4e79a7","Soil control"= "#9c755f",
                             "Poaceae" = "#7F9A65" ), name = "Plant family") +
  scale_x_discrete(labels = NULL)

ggarrange (NULL,p1,NULL,  p2, labels = c("","A","", "B"), hjust = 0.4, nrow = 1, ncol =4, widths =  c(0.05,1.8, 0.05,1), legend = "bottom", common.legend = T)


```


```{r MPD-sample-vs-PlaSpe}
dist_Glo  <- cophenetic(phy_tree(ps_Glo))
ses.MPD_Glo  <-  as_tibble (ses.mpd (comm_df_Glo, dist_Glo, null.model = "independentswap" ), rownames = "sampleID")

#agglomerate to Plant Species
ps_Glo_perPlaSpe  <- merge_samples(ps_Glo, group = "PlantSpeciesfull")

#make df for Glo_PlaSpe data (empty samples samples pruned )
comm_df_Glo_PlaSpe  <- data.frame (otu_table (ps_Glo_perPlaSpe)) # vegan expects samples as rows and ASVs species as columns

ses.MPD_Glo_PlaSpe  <-  as_tibble (ses.mpd (comm_df_Glo_PlaSpe, dist_Glo, null.model = "independentswap" ), rownames = "sampleID")# tree is the same as before

  
```


The mean phylogenetic distance (MPD) of the AMF communities from the Poaceae and Plantaginaceae were similar in their mean values, whereas the other plants had more variable values (Fig. \@ref(fig:plot-MPD)). 
Most of the treatments had AMF communities which were not significantly different from their randomised communities (Table \@ref(tab:mpd-table1)). 
Merging the replicates of each treatment into a single sample and calculating the z-score resulted in three species with significant negative z-score values, indicating that these plants hosted conserved AMF communities (Table \@ref(tab:mpd-table2)): *D. glomerata* (z-score~MPD~ = `r round (ses.MPD_Glo_PlaSpe$mpd.obs.z[8], 2)`, *p* = `r round (ses.MPD_Glo_PlaSpe$mpd.obs.p[8], 2)`), *A. millefolium* (z-score~MPD~ = `r round (ses.MPD_Glo_PlaSpe$mpd.obs.z[2], 2)`, *p* = `r round (ses.MPD_Glo_PlaSpe$mpd.obs.p[2], 2)`) and *H. lanatus* (z-score~MPD~ = `r round (ses.MPD_Glo_PlaSpe$mpd.obs.z[10], 2)`, *p* = `r round (ses.MPD_Glo_PlaSpe$mpd.obs.p[10], 2)`). In contrast, *P. pratense* and *B. willdenowii* had, with values of `r round (ses.MPD_Glo_PlaSpe$mpd.obs.z[17], 2)` (*p* = `r round (ses.MPD_Glo_PlaSpe$mpd.obs.p[17], 2)`) and `r round (ses.MPD_Glo_PlaSpe$mpd.obs.z[4], 2)` (*p* = `r round (ses.MPD_Glo_PlaSpe$mpd.obs.p[4], 2)`) respectively, over-dispersed AMF communities.


```{r plot-MPD, fig.cap= "Mean phylogenetic distances (MPD). A) MPD per treatment.  B) MPD effect size (z-score~MPD~). Black squares denote MPD of the treatment based on merged replicates, for which only significant results are shown based on a comparison of observed MPD against MPD from randomised communities. Test results can be found in the appendix. Horizontal lines show median values, empty squares show means, boxes denote the interquartile range (IQR), while whiskers show 1.5*IQR ranges. Outliers are indicated by dots. ", fig.height=6, fig.width=10}


#Plot MPD####
p1   <- 
  ses.MPD_Glo  %>% 
left_join (metaM0, by= "sampleID")  %>% 
  filter (!is.na(mpd.obs)) %>% 
  group_by (PlaSpe) %>% 
  mutate(mean_mpd.obs =mean(mpd.obs)) %>% 
  arrange(mean_mpd.obs)   %>% 
  left_join ((meannumberASVsper_species %>% ungroup () %>%
                select (-PlantFamily, PlantSpeciesfull, mean_n_ASV_per_species)), by = "PlantSpeciesfull")  %>% 
       mutate (PlantFamily = str_replace(PlantFamily, "Soil","Soil control")) %>% 
  ggplot (aes(x=reorder (PlantSpeciesfull,-mean_n_ASV_per_species),
             y=mpd.obs, fill=PlantFamily))  +
  geom_boxplot() 
#+geom_point (data = (ses.MPD_Glo_PlaSpe %>% left_join (metaM0, by= c("sampleID" = "PlantSpeciesfull"))), aes(x=sampleID, y= mpd.obs), shape = 15)
#small numbers are generalists here!

p1  <- p1 +coord_flip () +
  stat_summary (fun=mean, geom="point", shape = 23)  +
  ylab("Mean phylogenetic distance (MPD)") +
  xlab ("Plant species") +
  theme_light () +
 theme(axis.text.y= element_text(family= "sans", face= "italic", size = 12)) +
  theme (axis.text.x=element_text(family= "sans", size = 12))  +
  theme (axis.title = element_text(family = "sans", size = 14),  
         legend.title=element_text(size=13), 
         legend.text=element_text(size=11)) +
  scale_fill_manual(values=c("Asteraceae"= "#edc948" ,"Cyperaceae" ="#5A6351" ,
                             "Fabaceae" = "#f28e2b" , 
                             "Plantaginaceae"= "#4e79a7","Soil control"= "#9c755f",
                    "Poaceae" = "#7F9A65" ), name = "Plant family") 
  
   #Plot effect size    ####    
p2   <- 
  ses.MPD_Glo  %>% 
left_join (metaM0, by= "sampleID")  %>% 
  filter (!is.na(mpd.obs)) %>% 
  group_by (PlaSpe) %>% 
  mutate(mean_mpd.obs.z =mean(mpd.obs.z)) %>% 
  arrange(mean_mpd.obs.z)   %>% 
  left_join ((meannumberASVsper_species %>% ungroup () %>%
                select (-PlantFamily, PlantSpeciesfull, mean_n_ASV_per_species)), by = "PlantSpeciesfull")  %>% 
       mutate (PlantFamily = str_replace(PlantFamily, "Soil","Soil control")) %>% 
  ggplot (aes(x=reorder (PlantSpeciesfull,-mean_n_ASV_per_species),
             y=mpd.obs.z, fill=PlantFamily))  +
  geom_boxplot()  +
    geom_point (data = (ses.MPD_Glo_PlaSpe %>% 
                          left_join (metaM0, by= c("sampleID" = "PlantSpeciesfull")) %>% 
                          filter (! between(mpd.obs.p, 0.05, 0.95))), 
                aes(x=sampleID, y= mpd.obs.z), shape = 15, size =3)

#small numbers are generalists here!

p2  <- p2 +coord_flip () +
  stat_summary (fun=mean, geom="point", shape = 23)  +
  ylab("MPD effect size (z-score)") +
 # xlab ("Plant Species") +
  theme_light () +
 theme(axis.text.y= element_text(family= "sans", face= "italic", size = 12)) +
  theme (axis.text.x=element_text(family= "sans", size = 12))  +
  theme (axis.title = element_text(family = "sans", size = 14 ),  
         legend.title=element_text(size=13), 
         legend.text=element_text(size=11), 
         axis.title.y = element_blank()) +
  scale_fill_manual(values=c("Asteraceae"= "#edc948" ,"Cyperaceae" ="#5A6351" ,
                             "Fabaceae" = "#f28e2b" , 
                             "Plantaginaceae"= "#4e79a7","Soil control"= "#9c755f",
                             "Poaceae" = "#7F9A65" ),name = "Plant family") +
  scale_x_discrete(labels =NULL)




ggarrange (NULL,p1,NULL,  p2, labels = c("","A","", "B"), hjust = 0.4, nrow = 1, ncol =4, widths =  c(0.05,1.8, 0.05,1), legend = "bottom", common.legend = T)


# independentswap  - to compare if the PD of the sample is different from a random PD of the same richness - 
# Randomize community data matrix with the independent swap algorithm (Gotelli 2000) maintaining species occurrence frequency and sample species richness
#phylogeny.pool
#Randomize community data matrix by drawing species from pool of species occurring in the phylogeny (phylogeny pool) with equal probability

```

#### Interaction network and nestedness

```{r bipartite }
#the ps has the PlaSpe merged into one sample per Pla spe, binary data now
#ps_bin_Plaspe
ps_bin_PlaSpe_S  <- subset_samples(ps_bin_PlaSpe, !(PlantSpeciesfull %in% "Soil"))
#Plot

###Plot define colors generalist to specialist
colgrad2<-colorRampPalette(c( "#278d3e", "#eb6841")) #colour gradient green to orange
colplants  <- data.frame ("PlantFamily" = c("Poaceae", "Cyperaceae" ,"Asteraceae", 
                                            "Fabaceae","Plantaginaceae", "Soil" ),
                          "color" = c( "#7F9A65" ,"#5A6351" , "#edc948" ,"#f28e2b", "#4e79a7", "#9c755f" ))


#define colors 
Genus  <- c( "Archaeospora", "Acaulospora", "Funneliformis" , "Cetraspora" , 
   "Claroideoglomus",    "Glomus" , "Paraglomus" , 
   "Scutellospora", "unidentified" )  
colFams  <- c(   "#287D8EFF" ,   "#C1A363" , 
"#F6E8C3", "#DFC27D",  "#20A386FF", "#80CDC1" ,  "#35978F", 
 "#01665E", "#C7EAE5" ) 
dfcol  <- data.frame(tax_table(ps_bin_PlaSpe_S)) %>% 
  as_tibble  (rownames ="ASV_ID")  %>% 
  select (ASV_ID, Genus)  %>%   
  replace_na(list (Genus="unidentified")) %>% 
   left_join (as_tibble (data.frame (Genus, colFams))) 


#calculate nestedness, mod etc Strona 2015
 #sort from generalist to specialist
 sorted  <- sortweb (t(data.frame (otu_table(ps_bin_PlaSpe_S)))) %>% 
                       as_tibble(rownames ="ASV_ID")  %>% 
                    left_join  (dfcol, by ="ASV_ID") %>% data.frame(row.names = "ASV_ID")
 
modularity <- NOS(sorted)

```


```{r nestedness-metrics}
#calculation see nestedness.R due to long computing time
#NODF, sorted, nullmodel = SS swappable (Type   error)
matrix_nestedness  <- as.matrix(as.data.frame (otu_table(ps_bin_PlaSpe))) #pla as columns
matrix_nestedness_S  <- as.matrix(as.data.frame (otu_table(ps_bin_PlaSpe_S))) #pla as columns, without soil

NODF_SS  <-readRDS("resultsCh2/NODF_SS.rds")
NODF_FF  <-readRDS("resultsCh2/NODF_FF.rds")
Johnson_nested  <-readRDS("resultsCh2/Johnson_nested.rds")

#NODF for Pla and AMF respectively
NODF_both <- vegan::nestednodf(matrix_nestedness_S, order =T, weighted = F, wbinary= F)

##knitr::kable(NODF_both, caption = "Nestedness of plant-AMF bipartite network", "simple")
# prepare dataframe ausd en date 

```


The bipartite plant-AMF interaction network displayed a one-sided nested pattern (see Fig. \@ref(fig:plot-bipartite)). The overall NODF for the matrix was `r round(NODF_FF$Bin_t2$NODF$Measure,2)` (fill = `r round (NODF_both$fill * 100,1)`) with higher nestedness of the plant species (NODF~Pla~ = `r round (NODF_both$statistic[2], 0)`) than the AMF species (NODF~AMF~ = `r round (NODF_both$statistic[1], 0)`). 
Comparison of the nestedness in the empirical and randomised interaction networks differed depending on the null model applied. 
Whereas the use of the SS null model produced a significant different randomised NODF value (NODF~SS~ = `r round (NODF_SS$Bin_t1$NODF$Mean,1)`, *p* = `r NODF_SS$Bin_t1$NODF$pvalue`), indicating high nestedness of the observed interaction matrix, the randomisation based on the FF null model did not yield a significant difference of the observed NODF to NODF~FF~ (NODF~FF~ = `r round (NODF_FF$Bin_t2$NODF$Mean,2)`, p-value = `r round (NODF_FF$Bin_t2$NODF$pvalue,2)`). To further test if the detected degree of nestedness was a product of richness-dependant interaction probability, the JDM-nestedness was calculated, which did not indicate significant nestedness of the network (JDM = `r round (Johnson_nested$Bin_t4$JDMnestedness$Mean, 2)` , *p* = `r round (Johnson_nested$Bin_t4$JDMnestedness$pvalue,2 )`). Strona & Veechs's [@Strona2015] NOS metrics showed low modularity with m = `r round ( modularity$mod,2)` and high node overlap (`r round(modularity$Nbar,2)`) of the plant-AMF network.


#### $\beta$-diversities of the AMF communities

The percentage of the core AMF species, i.e., the species occurring in at least 60 % of the samples per treatment, indicates the variability of AMF among replicates of each treatment (Table \@ref(tab:specialisation)). Soil had the greatest $\beta$-diversity, only 33 % of its total AMF species were core species, whereas several plant species reached values up to 100 %, indicating very low variability among individuals (replicates) within a species (treatment). A notable exception among the Poaceae was *P. cita* with \~94 % core species in comparison to values located between \~42 % and \~65 % core species for other members of the Poaceae.
While the core species metric $\beta$~core~ was based on the identity of the species, the compositional units (CU) indicate numeric variability among the treatments. The values for CU reached from 2.2 to 5.0 and did not show a clear pattern for the plant species of each plant family. Several plant species (*L. autumnalis*, *M. sativa*, *T. pratense*) showed the highest possible numeric variability of 5, followed by *U. uncinata* and *C. vulgare* with 4.5 CUs. Notable was the high true $\beta$-diversity for soil and for *P. cita*. Interestingly, the numeric variability did not show the same pattern as the variability based on the AMF identity ($\beta$~core~). *E. repens* and *P. cita*, for example, had the same numeric variability, but were very different in their percentages of core species. *H. lanatus* had with a value of 2.2 CU the lowest true $\beta$-diversity.

```{r specialisation,  tab.cap = "$\\beta$-diversity of AMF communities per treatment. Percentage of AMF core species of total AMF species per treatment ($\\beta$~core~ in %), and compositional units (CU) per treatment, a measure of true $\\beta$-diversity, adjusted to the number of replicates per treatment. Values of CU can range between 1 and 5.", tab.id ="specialisation", label = "specialisation"}

#Specialisation

# taxa core
ps_Glo_core  <-phylosmith::taxa_core (ps_Glo, treatment = "PlaSpe", frequency =0.6)#see above
ps_core_PlaSpe  <- merge_samples(ps_Glo_core, group = "PlantSpeciesfull", fun = sum)

#calculate beta diversity as percentage 
cores  <- 
  as.data.frame (otu_table (ps_core_PlaSpe )) %>%   as_tibble(rownames = "PlantSpeciesfull") %>% 
  pivot_longer (!PlantSpeciesfull, names_to="ASV_ID", values_to  = "ASV_counts")  %>%  
  group_by (PlantSpeciesfull) %>%  
  dplyr::filter (ASV_counts != "0")  %>% 
  dplyr::tally(name="core") %>% 
  left_join(uniqueASVsperPlaSpe)  %>% 
  mutate (perc_core = 100* core/uniqueASVsperPlSpe) %>% 
  mutate (perc_core = round (perc_core,1) ) %>% 
  select (PlantSpeciesfull, perc_core)

cores %>% 
  left_join(comp_units %>% mutate (CU = round(CUnits,1)), by = "PlantSpeciesfull") %>%
  arrange (mean_n_ASV_per_species) %>% 
  select (PlantSpeciesfull, perc_core,CU) %>% 
  flextable () %>% 
 set_header_labels(PlantSpeciesfull = "Treatment", perc_core ="core AMF species in %") %>%  
  italic (j = "PlantSpeciesfull")  %>% autofit ()


# betas <-unlist (bluthgens_d$dprime) %>% 
#   as_tibble(rownames = "Treatment")  %>%  
#   mutate ("Blüthgens d'" = round (value, 2)) %>%  
#   select (!value) %>%  
#   left_join(cores, by = c("Treatment"= "PlantSpeciesfull")) %>% 
#   mutate (perc_core = round (perc_core,3) )

```

```{r AMFgeneralists, include = F, tab.cap =  "Distribution of AMF families in all 21 treatments.", tab.id = "AMFgeneralists", label =  "AMFgeneralists"}

#bipartite indeces####
##Matrix herstellen für das Package####
matrix  <-  ASV_table_Glo  %>%  
  select(-c(Kingdom, Phylum, Class, Family, Order, Genus, Species))  %>% 
  as.data.frame() 
rownames(matrix)  <- matrix$ASV_ID
matrix  <- matrix[,-1]

# Species degree per sample#####
species_degree  <-  specieslevel(matrix, index="degree")

#AMF ASVS generalists table#####
number_AMF_ASV_generalists  <-
  species_degree$`lower level`  %>% 
  arrange (desc(degree))  %>% 
  filter (degree>29) %>%
  tally()

Table_AMF_species_degree  <-
  species_degree$`lower level`  %>% 
  arrange (desc(degree))  %>% 
  filter (degree>29) %>% 
  as_tibble(rownames="ASV_ID") %>% 
  inner_join(tax_table (ps_Glo) %>% data.frame() %>% as_tibble (rownames= "ASV_ID"), by="ASV_ID")  %>% 
  select (c(ASV_ID, degree, Family, Genus, Species)) %>% 
  dplyr::rename("Species degree" = "degree", "ASV ID" = "ASV_ID")


#AMF Families verteilung in allen Plant species#####
Table_AMF_families_speciesDegree_per_PlantSpecies  <-
  ASV_table_Glo  %>%
select(-c(Kingdom, Phylum, Class, Order, Species))  %>%
gather (-c(ASV_ID, Genus, Family), key=sampleID, value = ASV_counts)  %>%
left_join(metaM0, by="sampleID")  %>% 
  filter (ASV_counts!=0)  %>% 
  group_by (Family, PlantSpeciesfull) %>% dplyr::count() %>% 
group_by(Family)  %>% dplyr::count()%>% 
   dplyr::rename("Number of treatments" = n) %>% filter (!is.na(Family)) %>% 
  flextable () 

Table_AMF_families_speciesDegree_per_PlantSpecies

```

#### Synopsis of measures of plant-AMF interaction niche width using principal component analysis (PCA)

```{r plot-PCA-all-metrics, fig.cap = "Principal component analysis (PCA) of numeric and phylogenetic $\\alpha$- and $\\beta$-diversities per treatment. A) PCA-biplot. Dots are treatments, arrows are metrics of diversity. contrib= contribution of variable to placement of treatment in biplot. B)-D) Contributions of metrics to the first three principal components (PC). PD = Faith's phylogenetic diversity, unique = number of unique ASVs per treatment ($\\gamma$-diversity), $\\beta$~core~ = percentage of AMF species occurring in 60% of replicates per treatment, MPD = mean phylogenetic distance, CU = compositional units (true $\\beta$-diversity), Chao1 = Chao1 diversity, Shannon = Shannon's diversity H', Richness = species richness S.", fig.height=8, fig.width=10}

#Get values for all metrics in one tibble#####
All_metrics  <-
  adiv_richness %>% 
  left_join(metaM0)  %>%  
  group_by(PlantSpeciesfull)  %>% 
  mutate (Chao1 = mean (Chao1)) %>% 
  mutate (Shannon = mean (Shannon)) %>%  
  select (sampleID, Chao1, Shannon) %>% 
  select (!sampleID) %>% 
  unique()  %>%  
  left_join (comp_units %>% select (!repl) %>% select (!"1-CU")) %>%   #includes CU and unique and richness
  left_join(cores) %>% 
  select (!c(PlantType,uniqueASVsperPlSpe, PlantFamily))  %>% 
  left_join (stand_pd_Glo_all %>% 
               left_join(ses.MPD_Glo, by ="sampleID")  %>%  
               left_join (metaM0)  %>% 
              drop_na ()  %>% 
                group_by (PlantSpeciesfull)  %>%   
               mutate (mean_pd = mean (pd.obs), mean_mpd = mean (mpd.obs)) %>%  
               select (PlantSpeciesfull,   mean_pd, mean_mpd,PlantFamily)   %>% 
               unique ()  )  %>% 
  dplyr::rename("Richness"= "mean_n_ASV_per_species", "CU" = "CUnits", "β(core)"  = "perc_core",
                 "PD" = "mean_pd", "MPD" = "mean_mpd" )  %>% 
  as.data.frame(row.names = NULL)

rownames(All_metrics)  <- All_metrics$PlantSpeciesfull
All_metrics  <- All_metrics[,-1]

#  Who are the generalists?
scaled_metrics  <-
  All_metrics %>%  
  as_tibble (rownames =  "tm") %>%  
  mutate_at (c(2:9), scale) %>% 
  mutate (b_core = `ß(core)`[,1] * (-1)) %>% 
  select (!(`ß(core)`)) %>% 
  pivot_longer(cols = c(2:8,10), values_to = "valueINP", names_to = "metric") %>% 
  ggplot(aes(x= tm, y=metric))  +
  geom_point (aes(size=valueINP, color = valueINP)) +
  coord_flip() +
  scale_color_gradient2(low = "#f90025", mid= "#d0bbb8", high = "#689178", na.value="#89e1ff") +
  ylab("Diversity metric") +
  xlab ("Plant Species") +
  theme_light () +
  theme(axis.text.y= element_text(family= "sans", face= "italic", size = 11)) +
  theme (axis.text.x=element_text(family= "sans", size = 11))  +
  theme (axis.title = element_text(family = "sans", size = 13 ),  
         legend.title=element_text(size=12), 
         legend.text=element_text(size=11))

#PCA
#PCA_all_metrics  <- vegan::rda (All_metrics, scale = T)
PCA_all_metrics <- PCA(All_metrics, quali.sup = 9,   scale.unit = T, graph = F)

PCA_arrows_metrics<-
  PCA_all_metrics$var$coord %>%  
  as_tibble (rownames = "metric") %>% 
  dplyr::rename("D1end" = "Dim.1", "D2end"= "Dim.2")

PCA_metric_data   <- PCA_all_metrics$ind$coord  %>%  as_tibble(rownames = "PlantSpeciesfull") %>% 
  left_join(metaM0 %>%  select (PlantSpeciesfull, PlantFamily) %>%  unique ())

PCA_eig_m_Dim1 <- round (PCA_all_metrics$eig[1,2],1)

PCA_eig_m_Dim2 <- round (PCA_all_metrics$eig[2,2],1)
#Plot 

plotPca  <- 
  PCA_metric_data %>%  
     mutate (PlantFamily = str_replace(PlantFamily, "Soil","Soil control")) %>% 
    ggplot (aes (x = Dim.1, y = Dim.2, color = PlantFamily)) + 
  geom_point (size =3) +
  geom_hline(yintercept = 0, lty = 2, color = "grey", alpha = 0.9) + 
  geom_vline(xintercept = 0, lty = 2, color = "grey", alpha = 0.9) + 
  #stat_ellipse(aes (x= Dim.1, y = Dim.2, color = group))  +   # thisis 95%confidence
  geom_segment(data = PCA_arrows_metrics, aes (x=0, xend= D1end*1, y = 0, yend = D2end*1), 
               arrow = arrow(length = unit(0.3, "picas")), color = "darkblue", inherit.aes = F)  +
  geom_text_repel ( data = PCA_arrows_metrics, aes (x  = D1end*1, y = D2end*1, label = metric), 
                   color = "darkblue", inherit.aes = F , force = 0.6) + 
  geom_text_repel(data = PCA_metric_data, aes (x = Dim.1, y = Dim.2, label = PlantSpeciesfull), fontface = "italic", inherit.aes = F) +
  theme_minimal() + 
  xlab(label = "PC1 (70.5 %)") +
  ylab ("PC2 (21.7 %)") + 
  guides (color= guide_legend( "Plant family")) +
  theme (legend.position = "bottom") +
    scale_color_manual(values=c("Asteraceae"= "#edc948" ,"Cyperaceae" ="#5A6351" ,
                             "Fabaceae" = "#f28e2b" , 
                             "Plantaginaceae"= "#4e79a7","Soil control"= "#9c755f",
                    "Poaceae" = "#7F9A65" )) 



# plotPca  <- fviz_pca_biplot(PCA_all_metrics,axes = c(1,2), 
#                             col.var = "contrib", 
#                             gradient.cols = c("#00AFBB", "#E7B800", "#FC4E07"), 
#                             repel =T)

plot1 <- fviz_contrib(PCA_all_metrics, choice = "var", axes = 1, 
                      font.main = c(size =12 ), title= "Contribution of variables to PC1")
plot2 <- fviz_contrib(PCA_all_metrics, choice = "var", axes = 2, 
                      font.main = c(size =12 ), title= "Contribution of variables to PC2")
plot3 <- fviz_contrib(PCA_all_metrics, choice = "var", axes = 3, 
                      font.main = c(size =12 ), title= "Contribution of variables to PC3")

ggarrange (plotPca,
           ggarrange (plot1, plot2, plot3, nrow = 3, ncol= 1, labels= c("B", "C", "D")), 
           nrow=1, ncol=2, widths =  c(2, 1), labels= "A") 


```

PCA of treatments along a selection of numeric and phylogenetic $\alpha$ and $\beta$-diversities shows a separation into groups along the principal component (PC) axes (Fig. \@ref(fig:plot-PCA-all-metrics)). The most notable separation along PC1 divides the treatments into such of low (left side) and high (right side) $\alpha$-diversities. Different placement along PC2 indicate differences in phylogenetic divergence, whereas the within-treatment $\beta$-diversity contributes most to the third dimension (PC3). Most of the grasses (Poaceae) are clustered together, being numeric generalists, with the exception of *P. cita* and *L. perenne*. Overall, the forbs are mostly spread along PC2, with *L. autumnalis* and *C. intybus* showing the highest difference to each other, indicating differences in phylogenetic divergence. 


```{r plot-PCA-all-metrics-Poaceae, fig.cap= "Principal component analysis (PCA) of numeric and phylogenetic $\\alpha$ and $\\beta$-diversities per Poacaean plant species. A) and B) PCA-biplots of A) Dimensions 1 and 2, and B) dimensions 2 and 3. Dots are plant species, arrows are metrics of diversity, i.e., properties of the interaction niche. contrib= contribution of variable to placement of treatment in biplot. C)-E) Contributions of metrics to the first three dimensions. PD = Faith's phylogenetic diversity, unique = number of unique ASVs per treatment ($\\gamma$-diversity), $\\beta$~core~ = percentage of AMF species occurring in 60% of replicates per plant species, MPD = mean phylogenetic distance, CU = compositional units (true $\\beta$-diversity), Chao1 = Chao1 diversity, Shannon = Shannon's diversity H', Richness = species richness S.", fig.height=8, fig.width=10}

#Get values for all metrics in one tibble#####
All_metrics_generalists  <-
  adiv_richness %>% 
  left_join(metaM0)  %>%  
  group_by(PlantSpeciesfull)  %>% 
  mutate (Chao1 = mean (Chao1)) %>% 
  mutate (Shannon = mean (Shannon)) %>%  
  select (sampleID, Chao1, Shannon) %>% 
  select (!sampleID) %>% 
  unique()  %>%  
  left_join (comp_units %>% select (!repl) %>% select (!"1-CU"))    %>%   #includes CU and unique and richness
  left_join(cores) %>% 
  select (!c(PlantType,uniqueASVsperPlSpe, PlantFamily))  %>% 
   left_join (stand_pd_Glo_all %>% 
               left_join(ses.MPD_Glo, by ="sampleID")  %>%  
               left_join (metaM0)  %>% 
              drop_na ()  %>% 
                group_by (PlantSpeciesfull)  %>%   
               mutate (mean_pd = mean (pd.obs.z), mean_mpd = mean (mpd.obs.z)) %>%  
               select (PlantSpeciesfull,   mean_pd, mean_mpd,PlantFamily)   %>% 
               unique ()  )  %>% 
  filter (PlantFamily == "Poaceae" | PlantFamily == "Plantaginaceae" | PlantFamily == "Soil") %>% 
  dplyr::rename("Richness"= "mean_n_ASV_per_species", "CU" = "CUnits", "β(core)"  = "perc_core",
                 "PD" = "mean_pd", "MPD" = "mean_mpd" )  %>% 
  as.data.frame(row.names = NULL)
rownames(All_metrics_generalists)  <- All_metrics_generalists$PlantSpeciesfull
All_metrics_generalists  <- All_metrics_generalists[,-1]

#PCA
#PCA_all_metrics  <- vegan::rda (All_metrics_generalists, scale = T)
PCA_all_metrics_generalists <- PCA(All_metrics_generalists, quali.sup = 9,   scale.unit = T, graph = F)


#Plot 

PCA_arrows_metrics_gen<-
  PCA_all_metrics_generalists$var$coord %>%  
  as_tibble (rownames = "metric") %>% 
  dplyr::rename("D1end" = "Dim.1", "D2end"= "Dim.2", "D3end"="Dim.3")

PCA_metric_data_gen   <- PCA_all_metrics_generalists$ind$coord  %>%  
  as_tibble(rownames = "PlantSpeciesfull") %>% 
  left_join(metaM0 %>%  select (PlantSpeciesfull, PlantFamily) %>%  unique ())

PCA_eig_m_Dim1_gen <- round (PCA_all_metrics_generalists$eig[1,2],1)
PCA_eig_m_Dim2_gen <- round (PCA_all_metrics_generalists$eig[2,2],1)
PCA_eig_m_Dim3_gen <- round (PCA_all_metrics_generalists$eig[3,2],1)
#Plot 

plotPca_gen1  <- 
  PCA_metric_data_gen %>% 
  mutate (PlantFamily = str_replace(PlantFamily, "Soil","Soil control")) %>% 
ggplot (aes (x = Dim.1, y = Dim.2, color = PlantFamily)) + 
  geom_point (size =3) +
  geom_hline(yintercept = 0, lty = 2, color = "grey", alpha = 0.9) + 
  geom_vline(xintercept = 0, lty = 2, color = "grey", alpha = 0.9) + 
  #stat_ellipse(aes (x= Dim.1, y = Dim.2, color = group))  +   # thisis 95%confidence
  geom_segment(data = PCA_arrows_metrics_gen, aes (x=0, xend= D1end*1, y = 0, yend = D2end*1), 
               arrow = arrow(length = unit(0.3, "picas")), color = "darkblue", inherit.aes = F)  +
  geom_text_repel ( data = PCA_arrows_metrics_gen, aes (x  = D1end*1, y = D2end*1, label = metric), 
                   color = "darkblue", inherit.aes = F , force = 0.6) + 
  geom_text_repel(data = PCA_metric_data_gen, aes (x = Dim.1, y = Dim.2, label = PlantSpeciesfull), fontface = "italic", inherit.aes = F) +
  theme_minimal() + 
  xlab(label = "PC1 (65.0 %)") +
  ylab ("PC2 (21.1 %)") + 
  guides (color= guide_legend( "Plant family")) +
  theme (legend.position = "bottom") +
    scale_color_manual(values=c(
      "Plantaginaceae"= "#4e79a7","Soil control"= "#9c755f",
                    "Poaceae" = "#7F9A65" )) 

plotPca_gen2  <- 
  PCA_metric_data_gen %>%
  mutate (PlantFamily = str_replace(PlantFamily, "Soil","Soil control")) %>% 
ggplot (aes (x = Dim.2, y = Dim.3, color = PlantFamily)) + 
  geom_point (size =3) +
  geom_hline(yintercept = 0, lty = 2, color = "grey", alpha = 0.9) + 
  geom_vline(xintercept = 0, lty = 2, color = "grey", alpha = 0.9) + 
  #stat_ellipse(aes (x= Dim.1, y = Dim.2, color = group))  +   # thisis 95%confidence
  geom_segment(data = PCA_arrows_metrics_gen, aes (x=0, xend= D2end*1, y = 0, yend = D3end*1), 
               arrow = arrow(length = unit(0.3, "picas")), color = "darkblue", inherit.aes = F)  +
  geom_text_repel ( data = PCA_arrows_metrics_gen, aes (x  = D2end*1, y = D3end*1, label = metric), 
                   color = "darkblue", inherit.aes = F , force = 0.6) + 
  geom_text_repel(data = PCA_metric_data_gen, aes (x = Dim.2, y = Dim.3, label = PlantSpeciesfull), fontface = "italic", inherit.aes = F) +
  theme_minimal() + 
  xlab(label = "PC1 (21.1 %)") +
  ylab ("PC2 (5.9 %)") + 
  guides (color= guide_legend( "Plant family")) +
  theme (legend.position = "bottom") +
    scale_color_manual(values=c(
      "Plantaginaceae"= "#4e79a7","Soil control"= "#9c755f",
                    "Poaceae" = "#7F9A65" )) 


# plotPca  <- fviz_pca_biplot(PCA_all_metrics,axes = c(1,2), col.var = "contrib", gradient.cols = c("#00AFBB", "#E7B800", "#FC4E07"), 
#                 repel =T)
# plotPca2  <- fviz_pca_biplot(PCA_all_metrics,axes = c(2,3), col.var = "contrib", gradient.cols = c("#00AFBB", "#E7B800", "#FC4E07"), 
#                 repel =T)

plot1 <- fviz_contrib(PCA_all_metrics_generalists, choice = "var", axes = 1,
                      font.main = c(size =12 ), title= "Contribution of variables to PC1")
plot2 <- fviz_contrib(PCA_all_metrics_generalists, choice = "var", axes = 2, 
                      font.main = c(size =12 ), title= "Contribution of variables to PC2")
plot3 <- fviz_contrib(PCA_all_metrics_generalists, choice = "var", axes = 3, 
                      font.main = c(size =12 ), title= "Contribution of variables to PC3")

ggarrange (ggarrange (plotPca_gen1,plotPca_gen2, ncol = 2, nrow =1, labels= c("A", "B"), common.legend = T, legend = "bottom"), 
           ggarrange (plot1, plot2, plot3, nrow = 1, ncol= 3,
                      widths = 2, labels= c( "C", "D", "E")), 
           nrow=2, ncol=1, heights= c(2,1) )


```

The examination of the PCA for Poaceae and Plantaginaceae supports the idea that the interaction niche properties differed among these species. Notable plants were *P. cita*, *L. perenne*, *D. glomerata*, *A. capillaris*  and  *P. pratense* with the first four plants showing a tendency to host similar AMF communities among their individuals, i.e., they had core AMF communities which were also of low phylogenetic divergence. Of these four plant species, two hosted AMF species of higher richness than the other two, indicating that numeric richness of the AMF community did not drive differences in $\beta$-diversity and MPD in my dataset. *P. pratense* showed the opposite tendency hosting divergent taxa, i.e., it had an over-dispersed AMF community, and highly dissimilar AMF communities among individuals.

### Interaction niche overlap

The functional plant groups as well as the individual plant families showed high overlap of their respective plant communities (Fig. \@ref(fig:venn)). Forbs and graminoids hosted mostly the same AMF species but also retained some AMF solely hosted by each group. After agglomerating the AMF ASVs at the tree tips, the percentage of shared AMF among functional groups increased notably, indicating that some of the differences between their AMF communities can be attributed to sequence variation. About half of the AMF species detected in the soil pool were not hosted by any of the 95 plant individuals. 
Comparing the overlap of the plant families shows that the AMF community of the Fabaceae and Cyperaceae were subsets of the Plantaginaceae and Poaceae, the Asteraceae shared most of their AMF with the Poaceae and Plantaginaceae, whereas the AMF community of the Plantaginaceae overlapped strongly with the one from the Poaceae. However, care must be taken in the interpretation of the overlap due to the unbalanced design of the study regarding the number of plant species per plant family and plant functional group.

```{r venn, fig.cap= "Venn diagrams of interaction niche overlap among plant functional groups and plant families. A) and C) Venn diagrams including all 309 AMF ASVs. B) and D) Venn diagrams of AMF species after cophenetic clustering of the tree tips at a height of h=0.1. A) and B) Overlap of AMF community between the functional group forbs and graminoids, and AMF detected in soil. C) and D) Overlap of AMF community among the studied plant families. Note that the number of species per family differed strongly (Poaceae = 11 species, Ateraceae = 5, Cyperaceae = 1, Plantaginaceae and Fabaceae = 2 species each). ", dpi=300, fig.height=6, fig.width= 8}

#Venn diagramm of functional groups and soil - full dataset
vennType1  <- ps_venn (ps_Glo, group = "PlantType")

vennType2  <- ps_venn (tip_glom (ps_Glo, h = 0.1), group = "PlantType")

#vennType3  <- ps_venn (tip_glom (ps_Glo, h = 0.2), group = "PlantType")


# and now the families
#subet ps
ps_Glo_plants  <- subset_samples(ps_Glo, PlantFamily != "Soil")

# venn diagram plant families
vennFam <- ps_venn (ps_Glo_plants, group = "PlantFamily")

vennFam2 <- ps_venn (tip_glom(ps_Glo_plants, h = 0.1), group = "PlantFamily")
#
ggarrange(NULL, NULL, vennType1, vennType2, NULL, NULL, vennFam, vennFam2, labels  = c ("", "","A", "B","", "", "C", "D"), 
          nrow = 4, ncol = 2, heights = c(0.1, 1, 0.1, 1.5))


```

```{r NMDS-raup-core, fig.cap= "Non-metric multidimentional scaling (NMDS) ordination based on Raup-Crick distances between core AMF communities. Shapes symbolise plant families, with a larger symbol denoting the centre for each treatment. The Raup-Crick distances account for the highly different AMF richness among the samples. Dots are colored by treatment." ,  fig.height=8, fig.width=10}

# core species #####
#ps_Glo_core <- taxa_core (ps_Glo,frequency = 0.6,  treatment = "PlaSpe") #see MPD
#ps_Glo_core  <-  prune_samples(sample_sums(ps_Glo_core)>0, ps_Glo_core)#remove empty samples

#as data.frame for vegdist function
df  <- as.data.frame (otu_table(ps_Glo_core))
df <- t(df)

#Calculate distance matrix##
dm  <-vegdist(df, method = "raup" ) #raupcrick distance for samples that have diff abundances and species richness see Beckett
#dm  <-vegdist(df, method = "bray", binary = T ) #sorensen


# Determine best number of dimensions###
# function NMDS.scree() 
# performs NMDS for 1-10 dimensions and plots the nr of dimensions vs the stress 
# - code changed after https://ourcodingclub.github.io/tutorials/ordination/

# NMDS.scree <- function(x) { #where x is the name of the data frame variable
#   plot(rep(1, 10), replicate(10, metaMDS(x, autotransform = F,trace =F, k = 1)$stress), xlim = c(1, 10),ylim = c(0, 0.30), xlab = "Number of Dimensions", ylab = "Stress", main = "NMDS scree plot")
#   for (i in 1:10) {
#     points(rep(i + 1,10),replicate(10, metaMDS(x, autotransform = F, k = i + 1)$stress))
#   }
# }
# 
# # Get scree plot to determine stress vs dimension in NMDS
# ScreeNMDS  <-NMDS.scree(dm)
# 
#Run NMDS
NMDS_dm  <- metaMDS (dm, distance = dm, trymax=1000, k =5, trace=F) #trace = F , then it stays quiet, k is number of dimension
#NMDS more robust than methods that include the magnitude of the distances, 
# get scores
scrs <- scores (NMDS_dm, display = "sites")
scrs <-as.data.frame(scrs) %>%  as_tibble(rownames = "sampleID") %>%  left_join (metaM0)
center <- aggregate(cbind (NMDS1, NMDS2) ~PlantSpeciesfull, data = scrs, FUN= mean) %>% 
  rename("NMDS1" = "cNMDS1", "NMDS2" = "cNMDS2") %>% 
  left_join (metaM0 %>%  select (PlantSpeciesfull, PlantFamily ) %>% unique ()) %>% 
       mutate (PlantFamily = str_replace(PlantFamily, "Soil","Soil control"))

segs <- left_join (scrs, center) %>% 
       mutate (PlantFamily = str_replace(PlantFamily, "Soil","Soil control"))


#Plot NMDS in ggplot using phyloseq
col  <- brewer.pal(8,  "Accent") # use all 8 colors from  palette
new_pal  <- colorRampPalette(col)  #then use with col = new_pal (21) for 21 different colors


p <- ggplot (segs , aes (x = NMDS1, y = NMDS2, color = PlantSpeciesfull, shape = PlantFamily))  + 
  geom_segment(segs, mapping = aes (xend = cNMDS1, yend = cNMDS2)) + 
  geom_point(data = segs, mapping =aes (x = NMDS1, y= NMDS2), size = 3) + 
  geom_point (data = center , aes (x= cNMDS1, y = cNMDS2), size = 5)



p  + 
  guides (color = guide_legend (order =2,  title = "Plant species", 
                                label.theme = element_text(face = "italic", family = "sans", size = 11),ncol = 1), 
          shape = guide_legend(order=1, title = "Plant family",legend.position = "bottom" ))  +
  scale_color_manual(values = new_pal(21))  + 
  theme_bw() + 
   # stat_ellipse(level=0.95) +
  scale_shape_manual (values= c(16,17, 18, 3,12,8 )) +
  annotate ("text", x= 0.2, y=0.4, label = "Stress = 0.11") #annotation is added by hand here, see NMDS_dm$stress
      # NMDS_dm$stress
#the stress value reflects how well the ordination summarizes the observed distances among the samples.
#stress value >0.2 poor and might not be interpretable, values <0.1 good, between 0.1 and 0.2 usable but see dexter et all - stress depends on samples etc, 
#some of the distances might be misleading

##Test ecological null model
# result<-oecosimu(comm = df, method = "quasiswap", 
#                  nestfun = metaMDS, autotransform = FALSE, k = 5, #add dimension
#                  distance = "raup", nsimul = 1000,  statistic = "stress", #add distance measure
#                  alternative = "less", trace = FALSE, maxit = 200, 
#                  trymax = 50, sratmax = 0.9999999)

# result$oecosimu$statistic # stress of my data NMDS
# result$oecosimu$means
# result$oecosimu$z
# result$oecosimu$pval #this is relevant for reporting

# Similar to a permutation test  - in oecosimu() the values of the statistic are computed using data generated from one of the Null models
# How MANY of the stress values generated using the Null model of the data are less than or equal to the observed stress value (OR: alternative = "two.sided" (=different from), alternative = "greater"(or greater than or equal to))   .
# Plot the stress values of simulated NMDS
#Plot quick and dirty to check
# hist(as.vector(result$oecosimu$simulated), xlim = c(0,max(result$oecosimu$simulated)+.05), xlab = "Stress value",ylab = "Frequency observed", main = "", breaks = 7)
# abline(v = result$oecosimu$statistic, col = "red", lty = 2) 


# #Pretty ggplot  stress values randomised vs stress observed
# simVector<-as.data.frame (as.vector(result$oecosimu$simulated))
# simVector<-cbind(simVector,seq(1:1000))
# 
# stresshisto <- ggplot(simVector,aes(x=as.vector(result$oecosimu$simulated))) +  
#   geom_histogram(col="black",size=.25) +
#   theme_minimal() + 
#   xlim(c(0.07, .2)) + ylim(c(0,600)) + #change limits as fitting to stress value
#   geom_segment(aes(x = result$oecosimu$statistic, y = 0, xend = result$oecosimu$statistic), yend = 600,linetype=2,color = "red",size=.5) + #change limits of plot as needed
#   labs(x="Stress value")  +
#   labs(y="Frequency observed")
# 

pair.perm_raup_core <- read_rds ("resultsCh2/pair_perm_raup_core.rds")
permdisp_pairs_raup_core <- read_rds ("resultsCh2/permdisp_pairs_raup_core.rds")
permanova_raup_core <- read_rds ("resultsCh2/permanova_raup_core.rds")
permdisp_raup_core <- read_rds ("resultsCh2/permdisp_raup_core.rds")

  

```

```{r NMDS-uf-core, fig.cap= "Non-metric multidimensional scaling (NMDS) ordination based on UniFrac distances between core AMF communities. Shapes symbolise plant families, with a larger symbol denoting the centre for each treatment. Unifrac distances measure the phylogenetic distance between the AMF taxa between two sample pairs. Dots are colored by treatment",  fig.height=8, fig.width=10 }

# core species #####
# ps_Glo_core <- taxa_core (ps_Glo,frequency = 1/2,  treatment = "PlaSpe")
# ps_Glo_core  <-  prune_samples(sample_sums(ps_Glo_core)>0, ps_Glo_core)#remove empty samples
# 
# #as data.frame for vegdist function
# df  <- as.data.frame (otu_table(ps_Glo_core))
# df <- t(df)

#Calculate distance matrix##
dm  <-UniFrac(ps_Glo_core, weighted = F ) #UniFrac

# tested hellinger transformation - NMDS are identical...
# Determine best number of dimensions###
# function NMDS.scree() 
# performs NMDS for 1-10 dimensions and plots the nr of dimensions vs the stress 
# - code changed after https://ourcodingclub.github.io/tutorials/ordination/

# NMDS.scree <- function(x) { #where x is the name of the data frame variable
#   plot(rep(1, 10), replicate(10, metaMDS(x, autotransform = F, k = 1)$stress), xlim = c(1, 10),ylim = c(0, 0.30), xlab = "Number of Dimensions", ylab = "Stress", main = "NMDS scree plot")
#   for (i in 1:10) {
#     points(rep(i + 1,10),replicate(10, metaMDS(x, autotransform = F, k = i + 1)$stress))
#   }
# }
# 
# # Get scree plot to determine stress vs dimension in NMDS
# ScreeNMDS  <-NMDS.scree(dm)

#Run NMDS
NMDS_dm  <- metaMDS (dm, distance = dm, trymax=1000, k =5, trace=F) #trace = F , then it stays quiet, k is number of dimension
#NMDS more robust than methods that include the magnitude of the distances, 

# get scores prepare tibble 
scrs <- scores (NMDS_dm, display = "sites")
scrs <-as.data.frame(scrs) %>%  as_tibble(rownames = "sampleID") %>%  left_join (metaM0)

# get centroids
center <- aggregate(cbind (NMDS1, NMDS2) ~PlantSpeciesfull, data = scrs, FUN= mean) %>%  
  rename("NMDS1" = "cNMDS1", "NMDS2" = "cNMDS2") %>% 
  left_join (metaM0 %>%  select (PlantSpeciesfull, PlantFamily ) %>% unique ()) %>%      mutate (PlantFamily = str_replace(PlantFamily, "Soil","Soil control")) 
segs <- left_join (scrs, center) %>%      mutate (PlantFamily = str_replace(PlantFamily, "Soil","Soil control")) 




#Plot NMDS in ggplot using phyloseq
col  <- brewer.pal(8,  "Accent") # use all 8 colors from  palette
new_pal  <- colorRampPalette(col)  #dann mit col = new_pal (21) verwenden


p <- ggplot (segs , aes (x = NMDS1, y = NMDS2, color = PlantSpeciesfull, shape = PlantFamily))  + 
  geom_segment(segs, mapping = aes (xend = cNMDS1, yend = cNMDS2)) + 
  geom_point(data = segs, mapping =aes (x = NMDS1, y= NMDS2), size = 3) + 
  geom_point (data = center , aes (x= cNMDS1, y = cNMDS2), size = 5)

p  +  guides (color = guide_legend (order =2,  title = "Plant species", 
                                    label.theme = element_text(face = "italic", family = "sans", size = 11),ncol = 1), 
              shape = guide_legend(order=1, title = "Plant family",legend.position = "bottom" ))  +
  scale_color_manual(values = new_pal(21))  + 
  theme_bw() + 
 # stat_ellipse(level=0.95) +
  scale_shape_manual (values= c(16,17, 15, 3,10,8 )) +
  annotate ("text", x= 0.2, y=0.5, label = "Stress = 0.08") 
# NMDS_dm$stress
#the stress value reflects how wel the ordination summarizes the observed distances among the samples.
#stress value >0.2 poor and might be uninterpretable, values <0.1 good, between 0.1 and 0.2 useable but see dexter et all - stress depends on samples etc, 
#some of the distances might be misleading

# ##Test ecological null model
# # Function that calculates NMDS on Unifrac distances of randomised community data
#                      
#  comdist.is  <- function (x) {
#   metaMDS( as.matrix (UniFrac(phyloseq_randomize(x, null_model = "phylogeny.pool"), weighted =F)), k=5)$stress  #change number of dimensions k #x is a phyloseq object
# }
 #get the stress for 1000 runs of NMDS 
 #null_stress <- replicate (1000, comdist.is(ps_Glo_core))
 
 #calculate how often stress of null model NMDS are smaller than observed stress of null model
 # p_value  <- 
 #  null_stress %>%  
 #  as_tibble() %>%  
 #  filter (value < NMDS_dm$stress) %>% 
 #  tally () %>% 
 #   mutate (p_value = n /1000)



# 
# 
# # Plot the stress values of simulated NMDS
# #Plot quick and dirty to check
# hist(null_stress, xlim = c(0.05,max(null_stress)+.05), xlab = "Stress value",ylab = "Frequency observed", main = "", breaks = 10)
#  abline(v = NMDS_dm$stress, col = "red", lty = 2) 

# #Replot in ggplot  stress values randomised vs 
# simVector<-as.data.frame (null_stress)
# simVector<-cbind(simVector,seq(1:1000))
# 
# stresshisto <- ggplot(simVector,aes(x=null_stress)) +  
#   geom_histogram(col="black",size=.25) +
#   theme_minimal() + 
#   xlim(c(0.05, .17)) + ylim(c(0,600)) + 
#   geom_segment(aes(x = NMDS_dm$stress, y = 0, xend = NMDS_dm$stress), yend = 600,linetype=2,color = "red",size=.5) + 
#   labs(x="Stress value")  +
#   labs(y="Frequency observed")

pair.perm_uf_core  <- read_rds ("resultsCh2/pair_perm_uf_core.rds")

permdisp_pairs_uf_core  <- read_rds ("resultsCh2/permdisp_pairs_uf_core.rds")

permanova_uf_core <- read_rds ("resultsCh2/permanova_uf_core.rds")
permdisp_uf_core <- read_rds ("resultsCh2/permdisp_uf_core.rds")

  

```

Non-metric multidimensional scaling (NMDS) ordination of the Raup-Crick distances of the AMF core communities of the treatments indicates that the AMF communities of the plants from the Poaceae and Plantaginaceae clustered somewhat together, whereas the plant species from the Asteraceae separated from each other and the Poaceae and Plantaginaceae (Fig. \@ref(fig:NMDS-raup-core)). However, testing of the similarity of the centroids by PERMANOVA shows that the AMF core communities did not vary significantly among plant species (PERMANOVA R^2^ = `r round (permanova_raup_core$aov.tab$R2[1],2)`, *p* = `r permanova_raup_core$aov.tab$"Pr(>F)"[1]`), therefore contradicting parts of hypothesis 2 that states that some specialists employ complementarity of resource use as strategy of specificity. Additional testing of the homogeneity of the multivariate dispersion (PERMDISP) disclosed the differences in dispersion among the treatments (PERMDISP F = `r round (permdisp_raup_core$tab$F[1], 2)`, *p* = `r permdisp_raup_core$tab$"Pr(>F)"[1]`). A pairwise PERMANOVA and PERMDISP test revealed that the differences among AMF core communities among plant families (e.g., Plantaginaceae versus Asteraceae) as indicated by the NMDS were a result of differences in dispersion (Tables \@ref(tab:PERMANOVA-raup-distance) and \@ref(tab:PERMDISP-raup)).

Analogously, NMDS ordination on the Unifrac distances of the AMF core communities suggests higher phylogenetic similarity among the AMF communities detected in the Poaceae, Plantaginaceae and soil samples, and some separation between the AMF communities determined in the other plant families (Fig. \@ref(fig:NMDS-uf-core)). PERMANOVA testing implies that the AMF core communities from different groups were not significantly phylogenetically different from each other (PERMANOVA R^2^ =`r round(permanova_uf_core$aov.tab$R2[1],2)`, *p* =`r  permanova_uf_core$aov.tab$"Pr(>F)"[1]`, and Table \@ref(tab:PERMANOVA-uf-core)), but differed in their multivariate dispersion (PERMDISP F = `r round (permdisp_uf_core$tab$F[1], 2)`, *p* = `r permdisp_uf_core$tab$"Pr(>F)"[1]`). Table \@ref(tab:PERMDISP-uf-core) shows the results of the pairwise comparison.

\newpage

## Discussion

The aim of this study was to describe and compare the plant-AMF interaction of common plant species from a former pasture grassland, especially regarding specificity in the partner selection of the plant-AMF symbiosis. I hypothesised that grasses which are known to be numeric AMF generalists would show less selective filtering of the AMF pool in comparison to the numeric specialist plants. Concentrating on different properties of the plant-AMF interaction niche, I found instead that the studied plants employed diverging interaction strategies. Depending on the tested property of the interaction niche, some plants acted as both generalist and specialist in their interaction with AMF at the same time. This indicated that filtering of the AMF pool might be independent of their identity as a numeric generalist or specialist plant, therefore I must reject my hypothesis 1.  Second, I studied the mechanisms underpinning the numeric plant specialists and hypothesised that they use a variety of advantageous filtering strategies for their AMF community. While the numeric specialists seemed to employ different interaction strategies, analogous to the numeric generalists, these did not seem to be based on complementary advantages of hosting diverging AMF communities. Instead, several plants of the numeric specialists - and also from the numeric generalists - had a significant phylogenetic signal regarding their AMF communities, some preferably hosting a phylogenetic over-dispersed AMF community while others interacted with clustered communities. Among the tested plants, the differences in the intraspecific variability of the AMF community composition were wide-ranging and independent of their degree of specificity. Taken together, this indicates that the mechanism underpinning the plant-AMF interaction rely on diverging interaction strategies. In what follows, I discuss these findings in detail and argue how the detected variety of species’ interaction strategies could benefit the ecosystem. 

### Numeric generalist and specialist plants: Plant family strongly affects AMF richness 

The interaction niches of the studied plant families differed significantly regarding their AMF richness. The plant members of the Asteraceae, Fabaceae and Cyperaceae interacted with a smaller number of AMF in comparison to the members of the Plantaginaceae and Poaceae. Observations from field studies on plant communities have found the differences in AMF richness between the Asteraceae and Fabaceae on one hand and Plantaginaceae and Poaceae on the other are much less pronounced than what I have reported here [@Geel2018; @Wehner2014; @Sepp2019]. One reason for that could be an introduced glasshouse bias. Šmilauer et al. [-@Smilauer2021a] found a significant decline in AMF richness in glasshouse-grown forbs compared to field collected forbs, whereas they did not find a similar pattern of AMF richness of the studied grasses when grown separately in a glasshouse or collected from plant communities under field conditions. Šmilauer et al. argued that a better nutrient availability was a significant contributor to this cultivation bias [@Smilauer2021]. Although this could be a contributing factor to the differences in α-diversity reported here, other plausible explanations also exist. The different plant families tested also have different adaptations for P acquisition: root architecture or the ability to exudate compounds that help liberate P from normally unavailable minerals are relevant traits for increasing P accessibility [@Lambers2006]. Some forbs, for example, mobilise P from mineral sources such as CaPO~4~ therefore remaining mostly independent from mycorrhizal symbiosis. Similarly, many sedges are typically non-mycorrhizal due to their ability to form dauciform roots of high root surface area under P deficiency [@Playsted2006] and they increase secretion of low molecular weight organic acids and chelators to aid in P acquisition. What was surprising was the small AMF diversity in the plant species from the Fabaceae (*T. pratense* and *M. sativa*), both root-nodule forming legumes which are highly mycorrhizal-dependent because the nodule formation and N fixation processes demand high P supply [@Scheublin2004; @Zhang2021]. One reason for this might be the origin of the seeds. Although they had been sourced from the field side, the original use of the site as a pasture could hint that these plants might be descendants from domesticated cultivars which often have lower symbiotic interaction diversity compared to their wild types [@Liu2020]. Both Asteracean species - *A. millefolium* and *C. vulgare* - had a low AMF α-diversity which could be explained by their mycorrhizal status described as facultative mycorrhizal [@Hempel2016]. It has been suggested that being facultative mycorrhizal increases plants’ realised niche compared to plants that are either obligate or non-mycorrhizal, because of their ability to change the occurrence of AMF in their roots in response to environmental factors [@Hempel2013; @Gerz2018]. In our glasshouse study, the plants were not affected by typical stressors like drought, and their supply with nutrients was comparably high. Taking energetic cost-benefit dynamics as underlying mechanisms into account, the association with mycorrhiza would not necessarily have been advantageous [@Johnson2013], and the facultative mycorrhizal plants therefore could have remained mostly uncolonised. However, the fact that some as facultative mycorrhizal described Poaceae had high AMF interaction diversity in this study indicates that, for the Poaceae at least, influences other than nutrient availability might have played a role in the AMF community assembly. The similarity in AMF richness among the plant species from the same family might be explained by the host plants’ physiologic variations in the mutualistic partner selectivity which in turn are different among plant phylogenies [@Gollotte2004; @Kiers2011]. These changes in AMF α-diversity could have been induced by regulative processes of the plant that differ among plant families. For example, the expression of a series of genes is necessary to enable the complex communication that is required for the interaction between arbuscular mycorrhiza and prospective host [@Lanfranco2018], and non-mycorrhizal plants were found to lack genes required for a functional interaction [@Cosme2018]. It has also been suggested that changes in that genetic equipment, such as abandonment of genes, might be the reason for facultative mycorrhizal associations which has been discussed as a transitional state from the mycorrhizal to the non-mycorrhizal state [@Maherali2016]. In diverse plant communities, these deficiencies in the plant-AMF-communication could be compensated by neighbouring plants with broad interaction niches because these generalist plants might be able to initiate steps of the AMF development, such as spore germination or hyphal development [@Akiyama2005; @Akiyama2006; @Steinkellner2007; @Lanfranco2017]. This neighbour effect might therefore enable the colonisation of co-occurring specialist plants, which in turn could broaden their interaction niche width.

### Some numeric generalist plants show specialism in their AMF selection 

Although the numeric interaction niche width revealed a phylogenetic signal among the plant families, the picture was decidedly mixed for other properties of the interaction niche. When considering host preferences, both the selectivity at the plant species level, for example favouring AMF taxa to maintain either a phylogenetic clustered or over-dispersed community, and the within-selectivity, i.e., to what degree the individuals of one species host the same AMF partners, are of interest. Taken together, all measured metrics fall into one of these three categories of specialism and are relevant aspects for the comprehensive description of the plant-AMF interaction niche. 
Under the conditions of the experiment, most plant species hosted random AMF communities: concentrating on the eleven species from the Poaceae, which tended to be numeric generalists, showed that some of them hosted phylogenetically over-dispersed AMF communities, whereas others seemed to have significantly phylogenetically clustered communities (MPD), indicating selective processes in the plant-AMF interaction. The AMF community of *P. pratense*, for example, was significantly phylogenetically over-dispersed, but at the same time also highly variable among its replicates (β~core~). It has been suggested that it might be beneficial for a plant to be colonised by taxonomically diverse AMF to take advantage of proposed niche complementary effects [@Batstone2018], but the effects of the co-occupation by different AMF are poorly understood [@Maherali2012; @Thonar2014]. *D. glomerata*, on the other hand, shows phylogenetic clustering but also high inter-replicate variability. While the plant-AMF interaction of *P. pratense* can be described as more of a generalist one, the phylogenetic interaction niche for *D. glomerata* seems to be smaller as expected by a random colonisation process, indicating some selectivity. 
The native grass *P. cita* acted as a specialist in the context of preferential interaction with AMF. Its replicates had a highly similar core AMF community, and furthermore *P. cita* had also a slightly different AMF community composition than the majority of the other Poaceae. *P. cita* was also the only grass exhibiting small intraspecific variability of the AMF community, a property that was more often found in the numeric specialists. However, as only three replicates of this plant species were analysed, this result might be an artefact of the analysis. Interestingly, other comparisons of native grasses to non-natives revealed differences in their AMF communities, with non-native grasses harbouring more diverse AMF communities and higher root colonisation rates than native grasses [@Wilson2012; @Duell2016; @Lekberg2013]. As the studied species originated from a pasture (semi-natural) grassland with significantly smaller abundances of native species, one reason for the reduced AMF community in *P. cita* might be that the soil AMF spore bank does not mirror the plant’s natural symbionts, as co-invasion by plants and their fungi has been reported for other mycorrhizae [@Dickie2010]. Alien fungi might outcompete native species in parallel with their host plant, displacing AMF species relevant for native plants [@Dickie2017]. However, as only one native plant has been studied here, there is only limited evidence if the origin of *P. cita* is linked to its notable specific AMF partner selection in comparison to the other tested Poaceae, as differences might be due to other characteristics of the plant AMF relationship, for example the mycorrhizal status or AMF dependency [@Lekberg2013; @Majewska2018]. Taken together, this study indicates that although all studied grasses might be inherent numeric generalists in their AMF-interaction, their propensity for specificity in AMF interactions depends on the plant species identity. 


### Numeric specialist plants employ different interaction strategies with AMF

Although the numeric specialist plants had AMF communities that were subsets of the generalist plant species, these communities differed somewhat in their taxonomic composition from the other specialists. However, further analysis revealed that - except for *A. millefolium* - the specialist plants’ AMF communities were most likely assembled without selection of specific AMF taxa. Unfortunately, a conclusive interpretation of the specificity of the plant-AMF interaction for most of the numeric specialist species remained elusive as they hosted such small numbers of AMF that statistical applications became unreliable. *A. millefolium*, however, hosted a significant phylogenetic clustered AMF community and showed a low intraspecific variability in the AMF community composition which both hint at selection of beneficial AMF partners. The same tendency could be detected for *C. intybus*, but it was not as pronounced as for *A. millefolium*. The interaction with specific AMF partners can be an advantageous strategy if these offer the most beneficial rewards under the present conditions. On the other hand, specialist plants could also benefit from the interaction with AMF taxa that differ from that from other plants in the ecosystem, but the data did not offer conclusive results on that strategy. 
That these plant species of low numeric AMF richness hosted highly variable AMF communities among their individuals indicates their potential ability to employ a multitude of different AMF. In turn, this might enable these plant species to expand their environmental niche spaces, potentially increasing the stability of the ecosystem against perturbation [@Bolnick2011]. Furthermore, the taxonomic compositions of the AMF communities among the 20 studied plants were very similar, which might also hint at the colonisation by the AMF that are most prevalent. AMF families of high read abundance, like the Claroideoglomeraceae, were found in almost all studied plant species. The Claroideoglomeraceae are known to be early colonisers [@Thonar2014] and might have had a competitive advantage in this glasshouse study system. Early colonisers could have dominated the AMF community for a while due to positive feedback effects [@Werner2015], outcompeting other AMF for space and resources inside the plant roots. For the ecosystem, the presence of a variety of interaction strategies, that were employed by the numeric specialists, like the selection of specific, beneficial AMF and the potential to interact with different AMF taxa – can help ameliorate competition [@Palmer2003] and increase ecosystem resilience [@Dell2019].


### Plants differ in their interaction strategies with AMF

In plant-pollinator systems, generalism has also been described as positively influencing plant productivity success [@Maldonado2013], and generalist mutualists as having a key role for the functionality and stability of their interaction network [@MartinGonzalez2010]. On the other hand, interaction specialists (e.g., in plant-pollinator systems) might enhance their coexistence via decrease of competition and increase of reciprocal benefits [@Bastolla2009]. By characterisation of the different interaction niche properties, I was able to identify interaction strategies of generalism and specialism, as inferred by the multidimensional volume of the interaction niche properties. The studied plant-AMF interactions of pasture plant species showed a high diversity of these interaction strategies. They ranged from plants acting as specialists in all interaction aspects to such acting as generalist, whereas some species were both generalist and specialists at the same time, depending on the respective interaction niche property. Furthermore, the characterisation of the different properties of the interaction niche width resulted in the segregation into dissimilar groups of plant species, depending on their positioning in the multidimensional space of the interaction niche properties. The detected diversity of the interaction strategies among the 20 studied plant species might offer flexible options for the plant community to react to environmental changes and increase the communities’ stability and productivity. Furthermore, the presence of diverse interaction strategies might diversity-dependant enhance ecosystem functioning by reducing competition [@Turnbull2016], potentially benefitting the pasture plant species. Besides the aspects of specificity studied here, other aspects relevant to the interaction niche space like net benefit of the symbiosis, nutrient allocation rates, ratios of root to soil hyphal colonisation et. could be included as additional niche axes describing the interaction niche of the plant-AMF symbiosis. Uniting these additional aspects with the ones of the interaction niche width used here could give a more detailed assessment of the functional role of each host plant and its effect on ecosystem stability and biodiversity [@Hale2020]. 


###	Possible caveats of the experiment

Complex interactions in the rhizosphere might interfere with the AMF-plant symbiosis. Both mycorrhizal fungi and plants interact with a plethora of microorganisms in the soil [@Bever2012]. Fine root endophytes (FRE) are a group of arbuscule-forming fungi of the sub-phylum Mycoromycotina that often co-occur with arbuscular mycorrhizal fungi in plant roots [@Orchard2017; @Orchard2017a]. In extreme environments, for example waterlogged or low-temperature conditions, FRE were found to outcompete AMF [@Orchard2016]. Furthermore, it has been suggested that the colonisation ratios of FRE to AMF are often significantly higher when the plants are grown in glasshouse experiments [@Orchard2017a], because the disruptive handling of the soil seems to favour FRE over AMF. These observations, however, were similar for all analysed plant families, whereas in our study only some plant families show significantly lower AMF α-diversity values in comparison to field studies [@Sepp2019]. Additionally, our sequencing results found Mycoromycotina only in one sample (in *S. aundinaceaus*). It is therefore unclear if FRE played a significant role in our study, also because the primer-pair used is not well-suited for the amplification of FRE genes [@Asemaninejad2016]. The application of diversity metrics and network theory to address the structure and assembly of the AM fungal community in plant roots is based on the assumption that each occurrence of AMF in the roots displays also an interaction. However, the interaction is most often defined as the occurrence of a functional symbiosis, which in turn is characterised as the formation of arbuscules [@Caruso2012a; @Veiga2013; @Lekberg2015]. Under the conditions of our study, it can be assumed that interaction between the AMF and the plant took place as the studied plants were grown separately in the pots, and infection by AMF on non-mycorrhizal plants is not possible without a host present to induce the fungal development [@Veiga2013]. Another difficulty of the analyses was to use the variability among replicates as measure of specificity. One has to be careful not to overinterpret this variability as it might have emerged by chance. This seems to be especially relevant in the case of the numeric specialists, for which it can be assumed that the number of replicates had not been sufficient to depict the possible width of AMF interaction partners. This bias seemed to have particularly impinged on the calculation of the $\beta$~core~ metric. Furthermore, each instance of null model selections to calculate metrics like nestedness or z-scores comes with the different probabilities of either type I or II errors [@Mariani2019]. This is especially the case here due to the large differences in richness, i.e., the degree distribution, which in turn strongly influences nestedness metrics. To account for that, the JDM-NODF metric was included [@Jonhson2013] which captures the nestedness that remains unexplained by the degree distribution.
As a consequence, some but not all here applied metrics of nestedness identified the bipartite plant-AMF network as significantly nested. It has been suggested that nestedness is often a result of the degree distribution and the selection of the null model [@Mariani2019; @Ulrich2009], and my results corroborate this view. 
However, as the calculated bipartite network is not a natural but a theoretical one constructed from the separately grown plants, these results are not easily transferable to networks under field conditions or even to mesocosms of the studied plants. 
Further care has to be taken when comparing the AMF community of the soil treatment to that of plant roots as this comparison is somewhat constrained by the inherently different properties of the samples from different origin. The mycorrhizal biomass in soil, for example, might have been much smaller than in root samples because AMF as obligate symbionts are usually not able to grow without their mutualistic partner, which might have decreased their detectability in the sequencing approach.

###	Conclusion
This is the first comprehensive study that elucidates the plant-AMF interaction niche properties of a diverse set of pasture plant species. It allowed the interpretation of the assembled AMF community without the need to disentangle the host effect from other biotic effects of a plant community. Several aspects were incorporated to describe the width of the plant-AMF interaction niche sufficiently, which resulted in the emergence of several groups of plants reflecting different interaction strategies. The utilisation of a range of interaction strategies by the plant species from the studied grassland could, in turn, become relevant in a plant-AMF community to buffer environmental changes. However, to comprehensively describe the plant-AMF interaction, further aspects of potential impacts on the interaction partner, like reciprocal benefits by nutrient allocation or facilitation ought to be included, and how the plant-AMF interaction is influenced by co-occurring plants demands further assessments.

\newpage

## Appendix for Chapter 2

```{r pd-table1, tab.cap = "Faith's phylogenetic diversity. Results for all replicates. Results significantly different from the null community are in bold.", label = "pd-table1" , tab.id = "pd-table1" }
stand_pd_Glo_all %>%  
  left_join(metaM0) %>%  
  select (!c(sortedPlantSpecies, PlaSpe, runs, filename, pd.obs.rank)) %>%  
  mutate(round (across (3:6),2)) %>% 
  mutate(pd.obs.p =round (pd.obs.p,4)) %>%  
  flextable ()  %>% 
  set_header_labels(sampleID = "sample ID", ntaxa ="n (AMF)", pd.obs = "PD", pd.rand.mean = "mean PD in null community", 
                    pd.rand.sd = "SD of PD in null community", pd.obs.z = "Standardised effect size of PD vs. null communities",
                    pd.obs.p =" p-value of observed PD vs. null communities" , PlantFamily = "Plant family", 
                    PlantSpeciesfull = "Plant species", PlantType = "Plant functional role") %>% 
  bold (i= ~pd.obs.p < 0.05 ) %>%  
    italic (j = "PlantSpeciesfull", part = "body")  %>% 
  bold (i= ~pd.obs.p > 0.95 ) 
```

```{r pd-table2, tab.cap = "Faith's phylogenetic diversity. Results for merged samples per plant species. Results significantly different from the null community are in bold.", label = "pd-table2" , tab.id = "pd-table2" }
stand_pd_Glo_PlaSpe %>%  
  select (!c( runs, pd.obs.rank)) %>%  
  mutate(round (across (3:6),2)) %>% 
  mutate(pd.obs.p =round (pd.obs.p,4)) %>%  
  flextable ()  %>% 
  set_header_labels(sampleID = "Plant species", ntaxa ="n (AMF)", pd.obs = "PD", pd.rand.mean = "mean PD in null community", 
                    pd.rand.sd = "SD of PD in null community", pd.obs.z = "Standardised effect size of PD vs. null communities",
                    pd.obs.p =" p-value of observed PD vs. null communities" ) %>% 
  bold (i= ~pd.obs.p < 0.05 ) %>%  
    italic (j = "sampleID", part = "body")  %>% 
  bold (i= ~pd.obs.p > 0.95 ) 
```

```{r mpd-table1, tab.cap = "Mean phylogenetic distance (MPD). Results for all replicates. Results significantly different from the null community are in bold.", label = "mpd-table1" , tab.id = "mpd-table1" }
ses.MPD_Glo %>%  
  left_join(metaM0) %>%  
  select (!c(sortedPlantSpecies, PlaSpe, runs, filename, mpd.obs.rank)) %>%  
  mutate(round (across (3:6),2)) %>% 
  mutate(mpd.obs.p =round (mpd.obs.p,4)) %>%  
  flextable ()  %>% 
  set_header_labels(sampleID = "sample ID", ntaxa ="n (AMF)", mpd.obs = "MPD", mpd.rand.mean = "mean MPD in null community", 
                    mpd.rand.sd = "SD of MPD in null community", mpd.obs.z = "Standardised effect size of MPD vs. null communities",
                    mpd.obs.p =" p-value of observed MPD vs. null communities" , PlantFamily = "Plant family", 
                    PlantSpeciesfull = "Plant species", PlantType = "Plant functional role") %>% 
  bold (i= ~mpd.obs.p < 0.05 ) %>%  
  italic (j = "PlantSpeciesfull", part = "body")  %>% 
  bold (i= ~mpd.obs.p > 0.95 ) 
```




```{r mpd-table2, tab.cap = "Mean phylogenetic distance (MPD). Results for merged samples per plant species. Results significantly different from the null community are in bold.", label = "mpd-table2" , tab.id = "mpd-table2"}
ses.MPD_Glo_PlaSpe %>%  
  select (!c( runs, mpd.obs.rank)) %>%  
  mutate(round (across (3:6),2)) %>% 
  mutate(mpd.obs.p =round (mpd.obs.p,4)) %>%  
  flextable ()  %>% 
  set_header_labels(sampleID = "Plant species", ntaxa ="n (AMF)", mpd.obs = "MPD", mpd.rand.mean = "mean MPD in null community", 
                    mpd.rand.sd = "SD of MPD in null community", mpd.obs.z = "Standardised effect size of MPD vs. null communities",
                    mpd.obs.p =" p-value of observed MPD vs. null communities" ) %>% 
  bold (i= ~mpd.obs.p < 0.05 ) %>%  
    italic (j = "sampleID", part = "body")  %>% 
  bold (i= ~mpd.obs.p > 0.95 ) 
```



```{r plot-bipartite, fig.cap = " Bipartite plant-AMF interaction network. Plant species are colored by plant family. Top squares denote plant species and below squares are AMF ASVs. Lines between squares are detected interactions, i.e., links.",  fig.height= 6, fig.width= 8 }


##plot network####
 
#define colors 
Genus  <- c( "Archaeospora", "Acaulospora", "Funneliformis" , "Cetraspora" , 
   "Claroideoglomus",    "Glomus" , "Paraglomus" , 
   "Scutellospora", "unidentified" )  
colFams  <- c(   "#287D8EFF" ,   "#C1A363" , 
"#F6E8C3", "#DFC27D",  "#20A386FF", "#80CDC1" ,  "#35978F", 
 "#01665E", "#C7EAE5" ) 
dfcol  <- data.frame(tax_table(ps_bin_PlaSpe_S)) %>% 
  as_tibble  (rownames ="ASV_ID")  %>% 
  select (ASV_ID, Genus)  %>%   
  replace_na(list (Genus="unidentified")) %>% 
   left_join (as_tibble (data.frame (Genus, colFams))) 

 
 #sort from generalist to specialist
 sorted  <- sortweb (t(data.frame (otu_table(ps_bin_PlaSpe_S)))) %>% 
                       as_tibble(rownames ="ASV_ID")  %>% 
                    left_join  (dfcol, by ="ASV_ID") %>% data.frame(row.names = "ASV_ID") %>% 
   select (!Soil)
 
 #plot bipartite network
 colorbi  <- sortweb (data.frame (otu_table(ps_bin_PlaSpe_S))) %>%  
   as_tibble(rownames = "PlantSpeciesfull") %>%  
   select (PlantSpeciesfull) %>%  left_join(metaM0) %>%  
   left_join(colplants) %>%  
   select (PlantSpeciesfull, color) %>%  unique () %>% 
   filter (PlantSpeciesfull != "Soil")
 # Customizing the output

 
 # png("plot1.png", width = 1080, height = 960,   # File name
 #    bg = "white")          # Paper size

#plot
plotweb (sorted[,1:19], method = "normal", col.high = as.character (colorbi$color), #col.high = colgrad2(20), 
         text.rot = 90, bor.col.high = as.character (colorbi$color), #bor.col.high = colgrad2(20),
         col.low = as.character (sorted$colFams), 
         bor.col.low = as.character(sorted$colFams), labsize=1.2,  low.lablength = 0, y.lim = c(0,3))
# Closing the graphical device

# dev.off() 
# 
# plot1 <- readPNG("plot1.png")
# 
# plot1



# ggarrange(rasterGrob(plot1),rasterGrob(plot2),ncol=1, nrow =2, labels= c("A)", "B)"), align = "v", heights = c(1.5, 1), hjust = -8 )
```

```{r plot-treemap,  fig.cap ="Distribution of ASVs across Glomeromycotinian genera and families and absolute links per AMF ASVs. Each box symbolises one AMF species and the size of the boxes is proportional to the number of root samples this AMF species occurred in, i.e., it is a proxy for the links per AMF ASV. Values reach from 1 (smallest box) to 59 (largest box) links per AMF species. Diversisporaceae are not labelled, with 2 ASV unidentified at genus level. AMF ASVs with a large number of links are interaction generalists. The absolute number of the ASVs per family of the Glomeromycotinia differs considerably. Each AMF family consists of a spectrum of generalist and specialist species, with the Gigasporaceae as notable exception containing in comparison a higher proportion of generalist AMF species than for example the Archaeosporaceae.", fig.height= 6, fig.width= 8}
#Which AMF taxa act as numeric generalists or specialists are illustrated in Fig. \@ref(fig:plot-treemap) and in Table \@ref(tab:AMFgeneralists). As already described above, tWhile the Archaeosporaceae comprise of the highest number of ASVs, other families, like the Gigasporaceae show a wide distribution in the root samples for a high proportion of their ASVs, i.e., more links per AMF ASV. Overall, all AMF families except Diversisporaceae are widely distributed in the 20 treatments.

# Plot Treemap ASVs Verteilung####
  
# das zeigt im prinzip welche ASVs genealisten 
#(große Kästchen und specialisten (kleine Kästchen sind) )
#Species degree in a treemap, bezogen aud die AMF genera
library(treemapify)
p   <-   ASV_table_Glo  %>%  
  select(-c(Kingdom, Phylum, Class, Order, Species))  %>% 
  gather (-c(ASV_ID, Genus, Family), key=sampleID, value = ASV_counts)  %>%
  left_join(metaM0, by="sampleID") %>% 
  filter (ASV_counts!=0)  %>% 
  group_by( Genus,ASV_ID )  %>% 
  tally ()  %>%  
left_join(tax_table (ps_Glo) %>%  
            data.frame () %>%  
            as_tibble(rownames = "ASV_ID") %>%  select (!Genus), by="ASV_ID")   %>% 
  replace_na(list (Genus="unidentified")) %>% 
  replace_na(list (Family="unidentified")) %>% 
  replace_na(list (Order="unidentifiedOrder")) %>% 
  ggplot (aes(area= n, label= n,fill = Genus , subgroup= Family, subgroup2=Order)) +
  geom_treemap () 
  
 p  +  geom_treemap_subgroup_text(fontface = "italic", 
                                   colour = "white", place = "centre",
                                     grow = TRUE) + 
    scale_fill_manual(values = c( "unidentified" ="#C7EAE5", 
                                  "Archaeospora"  = "#287D8EFF" , 
                                  "Acaulospora" =  "#C1A363" , 
                                  "Funneliformis" = "#F6E8C3", 
                                  "Cetraspora"  = "#DFC27D", 
                                  "Claroideoglomus" =  "#20A386FF", 
                                  "Glomus" = "#80CDC1" , "Paraglomus" = "#35978F", 
                                  "Scutellospora"= "#01665E" ), name = "AMF genus") +
   theme(legend.text=element_text(size=20, face = "italic"), legend.title = element_text(size =24))
  #   annotate(geom = "text", x = 0.25, y = 0.25, label = "The annotation") +
  # geom_treemap_text()  +
  # scale_x_continuous(breaks=NULL, limits = c(0, 1), labels=NULL, name=NULL) +
  # scale_y_continuous(breaks =NULL, limits = c(0, 1), labels=NULL, name=NULL)+
   # falls doch eine Beschrifung der boxen erfolgen soll. 
  #Außerdem habe ich die treemap mit ASvs und mit Anzahl der links abgespeichert für Vergleiche

  





  

```


```{r PERMANOVA-raup-distance , include =T,  tab.cap ="Pairwise permutational multivariate analysis of variance (PERMANOVA) of the Raup-Crick distances. PERMANOVA tests if the centroids and variances of the pairs are equivalent. In conjunction with a non-significant PERMDISP of the same pair, a significant PERMANOVA implies unequal centroids. f = F-statistic (pseudo-F, a measure of effect size), p = p-value.", tab.id = "PERMANOVA-raup-distance", label = "PERMANOVA-raup-distance" }
options(knitr.kable.NA = "")

pair.perm_raup_core <- read_rds ("resultsCh2/pair_perm_raup_core.rds")
##SEE PERMANOVA.R script 
#might help to run the permanova outside of the markdown - something interferes sometimes.
#it helped to re install vegan first, then phyloseq.

#
#R2 column: 
#we can/ can not reject the null hypothesis that our plant species AMF communities have the same centroid.
#for p< 0.05 reject
#for p>0.05 con not reject (centroids are the same)
#the global effect of plant species is significant
#  permanova_raup_core

#pairwise permanova
# pair.perm_raup_core
#
#R2 gives me the effect size.. x% of variation explained by grouping
#Only the pairs with p value <0.05
# sign_permdisp_all  <-   permdisp_pairs_raup_core$pairwise$permuted  %>% 
#   as_tibble(rownames = "pairs") %>% 
#   filter (value <0.05 ) %>%
#   mutate (value =round (value,3))  %>% 
#   dplyr::rename ("p-value" = "value") %>% 
#   data.frame()
table  <-
  pair.perm_raup_core$F.value  %>%  as_tibble(rownames = "Plant")  %>%
pivot_longer(names_to = "Pl1", values_to = "valueF", cols = c(-Plant)) %>% 
  drop_na(valueF) %>% left_join ((pair.perm_raup_core$p.value  %>% 
                                   as_tibble(rownames = "Plant")  %>%
    pivot_longer(names_to = "Pl1", values_to = "value", cols = c(-Plant)) %>% 
    drop_na(value)), by = c("Plant" = "Plant", "Pl1" ="Pl1")) %>% 
  mutate (valueF =round (valueF, 2), value= round (value,3))  %>%  
  add_column (F = "F = ")   %>% mutate (FValue= str_c(F,valueF))  %>% 
  add_column (p= "p = ") %>%  mutate (Pval = str_c(p, value))  %>% 
  add_column (sym = "; ")  %>%     
  mutate (Zelle = str_c(FValue, sym,Pval)) %>%   
  select (Plant, Pl1, Zelle) %>% 
  pivot_wider(names_from = "Pl1", values_from = "Zelle") %>% 
flextable  ()  %>% 
    fontsize(size =5, part = "body") %>% 
    fontsize(size =7, part = "header") %>% 
    fontsize(size =6, part = "footer") %>% 
    add_footer_lines (values  = "Significant pairs (p < 0.05) are in bold.") %>%  
  autofit()

table

```

```{r PERMDISP-raup  , include =T, tab.cap = "Pairwise permutational multivariate analysis of dispersion (PERMDISP) of Raup-Crick distances. PERMDISP tests the equality of the variances, significant (p<0.05, in bold) values indicate pairs that differ in variance. F = F-statistic, p = p-value.", tab.id = "PERMDISP-raup", label = "PERMDISP-raup" }
options(knitr.kable.NA = "")

permdisp_pairs_raup_core <- read_rds ("resultsCh2/permdisp_pairs_raup_core.rds")
#Check that variance homogeneity assumptions hold (to ensure the reliability of the results)
#Permuted multivariate analysis of beta-dispersion (PERMDISP)
# The vegan function betadisper() performs PERMDISP. This test is different from the others in that it specifically tests for differences in the spread (dispersion, variability) among groups.
# 
# Therefore, if you use this test in combination with one of the other three, you will be able to tease apart whether groups of communities are different because they have different centroids or different spreads. For example, if PERMANOVA yields a significant difference, but PERMDISP does not, you can safely say that the distinction between groups can be attributed to differences in their centroid. Ordinations are often a good way to visually support and summarize these findings.

#permdisp_raup_core
# 
# permdisp_pairs_raup_core$pairwise$permuted
#see extra document PEMANOVA!!!


Table_permdisp_raup_core  <-
unlist (round (permdisp_pairs_raup_core$pairwise$permuted, 2)) %>%  
  as_tibble(rownames = "pair")  %>%  
  separate(pair, c("Plant", "Pl1"),"-" ) %>% 
  left_join (
   ( unlist (round (permdisp_pairs_raup_core$statistic, 2)) %>%  
               as_tibble(rownames = "pair")  %>%  
               separate(pair, c("Plant", "Pl1"),"-" ) %>% 
              drop_na (Pl1) %>%  add_column (t = "F = ") %>% 
      mutate (tval = str_c(t, value)) %>%  
      mutate (Pl1 = str_replace (Pl1," \\(t\\)", ""))  %>% 
      select (Plant, Pl1, tval)), by = c( "Plant" = "Plant", "Pl1" = "Pl1")) %>% 
  add_column (sym = "; ")  %>% 
  add_column (p= "p = ") %>%  mutate (Pval = str_c(p, value))  %>% 
  mutate (Zelle = str_c(tval, sym,Pval)) %>%   
  select (Plant, Pl1, Zelle) %>% 
  pivot_wider(names_from = "Pl1", values_from = "Zelle") %>% 
  flextable ()  %>% 
  fontsize(size =5, part = "body") %>% 
    fontsize(size =7, part = "header") %>% 
    fontsize(size =6, part = "footer") %>% 
  add_footer_lines (values = "Significant pairs (p < 0.05) are in bold. t-test results and p-values are given per pair. ") %>%  
  autofit ()
 
Table_permdisp_raup_core

```


```{r PERMANOVA-uf-core, include =T, tab.cap ="Pairwise permutational multivariate analysis of variance (PERMANOVA) of the Unifrac distances. PERMANOVA tests if the centroids and variances of the pairs are equivalent. In conjunction with a non-significant PERMDISP of the same pair, a significant PERMANOVA implies unequal centroids. f = F-statistic (pseudo-F, a measure of effect size), p = p-value.", tab.id = "PERMANOVA-uf-core", label = "PERMANOVA-uf-core" }
options(knitr.kable.NA = "")

pair.perm_uf_core  <- read_rds ("resultsCh2/pair_perm_uf_core.rds")

#Table of PERMANOVA results
pair.perm_uf_core$F.value  %>%  as_tibble(rownames = "Plant")  %>%
pivot_longer(names_to = "Pl1", values_to = "valueF", cols = c(-Plant)) %>% 
  drop_na(valueF) %>% 
  left_join ((pair.perm_uf_core$p.value  %>%  
             as_tibble(rownames = "Plant")  %>%
            pivot_longer(names_to = "Pl1", values_to = "value", cols = c(-Plant)) %>% 
            drop_na(value)), by = c("Plant" = "Plant", "Pl1" ="Pl1")) %>% 
  mutate (valueF =round (valueF, 2), value= round (value,3))  %>%  
  add_column (F = "F = ")   %>% mutate (FValue= str_c(F,valueF))  %>% 
  add_column (p= "p = ") %>%  mutate (Pval = str_c(p, value))  %>% 
  add_column (sym = "; ")  %>%     
  mutate (Zelle = str_c(FValue, sym,Pval)) %>%   
  select (Plant, Pl1, Zelle) %>% 
  pivot_wider(names_from = "Pl1", values_from = "Zelle") %>% 
flextable  ()  %>% 
    fontsize(size =5, part = "body") %>% 
    fontsize(size =7, part = "header") %>% 
    fontsize(size =6, part = "footer") %>% 
    add_footer_lines (values  = "Significant pairs (p < 0.05) are in bold.") 


```

```{r PERMDISP-uf-core, include =T,  tab.cap = "Pairwise permutational multivariate analysis of dispersion (PERMDISP) of UniFrac distances. PERMDISP tests the equality of the variances, significant (p<0.05, in bold) values indicate pairs that differ in variance. F = F-statistic, p = p-value.", tab.id = "PERMDISP-uf-core", label = "PERMDISP-uf-core" }
options(knitr.kable.NA = "")

permdisp_pairs_uf_core  <- read_rds ("resultsCh2/permdisp_pairs_uf_core.rds")
#Table of PERMDISP results
Table_permdisp_uf_core  <-
  unlist (round (permdisp_pairs_uf_core$pairwise$permuted, 2)) %>%  
  as_tibble(rownames = "pair")  %>%  
  separate(pair, c("Plant", "Pl1"),"-" ) %>% 
  left_join (
    ( unlist (round (permdisp_pairs_uf_core$statistic, 2)) %>%  
        as_tibble(rownames = "pair")  %>%  
        separate(pair, c("Plant", "Pl1"),"-" ) %>% 
        drop_na (Pl1) %>%  add_column (t = "F = ") %>% 
        mutate (tval = str_c(t, value)) %>%  
        mutate (Pl1 = str_replace (Pl1," \\(t\\)", ""))  %>% 
        select (Plant, Pl1, tval)), by = c( "Plant" = "Plant", "Pl1" = "Pl1")) %>% 
  add_column (sym = "; ")  %>% 
  add_column (p= "p = ") %>%  mutate (Pval = str_c(p, value))  %>% 
  mutate (Zelle = str_c(tval, sym,Pval)) %>%   
  select (Plant, Pl1, Zelle) %>% 
  pivot_wider(names_from = "Pl1", values_from = "Zelle")  %>% 
  flextable ()  %>% 
    fontsize(size =5, part = "body") %>% 
    fontsize(size =7, part = "header") %>% 
    fontsize(size =6, part = "footer") %>% 
  add_footer_lines (values = "Significant pairs (p < 0.05) are in bold.") 

Table_permdisp_uf_core

```





