---
title: "AMF in root and soil of eight single grown plant species"
author: "Natascha Lewe"
date: "16/02/2022"
output:
  word_document2:
    number_sections: yes
    toc_depth: 3
    reference_docx: bib/Test.docx
bibliography: testbookdown/20220513bib.bibtex
csl: bib/apa.csl

---
# Chapter 3


```{r setup, echo=FALSE, message=FALSE}

knitr::opts_knit$set(root.dir = "C:/Users/Nat/OneDrive - Victoria University of Wellington - STAFF/PhD/Results/Marsden1/Marsden_1_2 roots/R/M1_2_NLFA")  # set root directory
knitr::opts_chunk$set(echo = FALSE, message=FALSE, warning=FALSE, dpi = 300)
```


```{r libraries, echo=FALSE, message=FALSE}
#library(kableExtra) #https://cran.r-project.org/web/packages/kableExtra/vignettes/awesome_table_in_html.html
library(tidyverse) # the most important package in the universe
library(readxl)
library(ggplot2)
library (flextable)
library(ggpubr)
library(phyloseq)
library(phylosmith)
library (broom) #needed for tidy linear modelling
library (lmerTest)
library (picante)
library(rstatix)
library(FactoMineR) # simple ordinations
library(factoextra)
library (rmarkdown)
library (ggsignif) # adds signif to ggplot
library (lme4)
library (nlme)
library (broom) # for pretty lm tables
library(broom.mixed) # for pretty mixed model tables
#library (wPerm) #permutation ANOVA etc
#library(paletteer) 
library (iNEXT)
library (metagMisc)
library(edgeR)
library (limma)
library (Glimma)
library (ape)
library (stats)
library (ggtree)
library(ggrepel)
library (car)


```


## Introduction 

The interplay between the plant and its microbial community plays a central role in ecosystem functioning through its effects on nutrient acquisition and carbon (C) cycling [@Kardol2010]. In Chapter 2 I provided evidence that pasture plant species occupy a continuum of interaction niche properties with AMF when grown in isolation of neighbour effects. 
Similar differences in plant interactions with AMF have been described under community or field conditions [@Gollotte2004; @Vandenkoornhuyse2003; @Hiiesalu2014]. 
These distinct plant-AMF interaction niche properties may have profound effects on ecosystem functions, such as carbon cycling, because exchange of carbon-rich photosynthates for water and compounds containing phosphorus (P) or nitrogen (N) is of main importance for the plant-AMF interaction. Overall, plants can lose up to 40% of their photosynthetically fixed carbon by different paths via the roots. Carbon is transferred to the fungal partner in the form of hexoses and fatty acids, which are essential for the growth of the AMF mycelium [@Luginbuehl2017; @Jiang2017]. 
AM plants enhance the carbon sequestration to the soil [@Wang2016a], they can act as significant sinks for atmospheric carbon dioxide [@Treseder2016; @Parihar2020; @Wang2016a]. 
However, the interplay between AMF and plant regarding carbon allocation is complex. The carbon-phosphorus exchange rates seem to be context-dependent and also vary among the species involved [@Charters2020]. 
Furthermore, the carbon transfer to an AMF symbiont decreases when a more beneficial AMF symbiont is present [@Kiers2011]. 
As a result, the AMF richness on a plant has been suggested to regulate the AMF mutualism via control of less cooperative fungal partners [@Argueello2016]. 
It is, however, seldom considered if plants differing in their interactions with AMF vary in their total carbon allocation to their AMF communities.

AMF taxa differ in their colonisation strategies and abilities in nutrient transfer [@Thonar2014; @Schuetz2022; @Lendenmann2011], which reflects a degree of functional complementarity in regard to their mutualistic role [@Jansa2008]. For example, while for some AMF families most of the fungal biomass is located in extraradical hyphae, others grow instead extensively inside the root spaces of the host plant [@Maherali2007]. 
For example, the mycelium from Gigasporaceaen species expands extensively into the soil, but they show little intraradical colonisation, whereas members of the Glomeraceae have the opposite growth strategy [@Hart2002]. In natural ecosystems, when the AMF communities potentially consist of a plethora of different taxa, these co-existing colonisation strategies of fungi might reflect an ecological strategy to avoid competition for space and resources, while offering complementary functional strategies of nutrient transfer to maximise host fitness [@Powell2018]. 
In fact, a study measuring the intra- and extraradical AMF biomass across a grassland plant community hosting a diverse AMF community reported a strong positive relationship between root and soil AMF biomass, indicating a proportional distribution of carbon-rich products between root and soil AMF biomass [@Barcelo2020]. However, it is unclear how the differences in interaction niche properties of the plant species influence the carbon allocation to the mycorrhizal fungi. Studies using simple AMF communities have shown that competition for space and nutrients is a major determinant for AMF abundance, and therefore biomass [@Engelmoer2014; @Padje2021]. While competition between AMF decreases their overall biomass, the species’ relative growth responses also depended on their identity [@Thonar2014] and relatedness [@Padje2021]. 

Carbon allocation to the fungal partner comes at the cost of plant biomass and the reason why the relationship is maintained even when the host does not seem to gain a mutualistic benefit is a topic of vigorous debate [@Johnson1997; @Grimoldi2006; @Verbruggen2012]. 
One possible mechanism is that plants benefit from fitness contribution by AMF fungi under conditions of low nutrients or stress [@Veresoglou2012a; @Lekberg2014], but is has also been suggested that the AMF mutualism could instead stabilise plant fitness under adverse conditions, resulting in a higher ecosystem resilience [@MartinezGarcia2015]. 
Parsimonious with this view is that AMF employ a variety of strategies of growth, P acquisition and biomass allocation. 
For example, the AMF *Glomus mosseae* stores large amounts of the acquired P in the plant root, but in a form that is inaccessible for the plant host [@Jansa2005]. 
This strategy might be advantageous under stress conditions for both partners because the AMF partner might be able to upkeep his side of the exchange deal. Plants select different strategies by retaining specific interaction niches with AMF. While some plant species might select benefitting AMF partners, others interact with a diverse AMF community, equipped with a variety of nutrient acquisition and colonisation strategies. Both strategies can induce significant advantages for the host plant. Plant communities consisting of plants with a mixture of interaction niche properties are therefore an important component underpinning the functional diversity in ecosystems [@Davison2020].

It is widely accepted that plants are structuring their soil microbial community [@Lundberg2012; @Bulgarelli2013]. 
Most known soil bacteria, for example, are organotrophs that feed on carbon-rich products from plant roots. Plants deposit large amounts of their photosynthetically fixed carbon directly into the soil via rhizodeposition and release of root exudates, which enrich soil bacterial taxa that are typically associated with plant roots [@Bulgarelli2013; @Eilers2010]. Some plant families have been found to differ in their exudation rates and composition of released sugars [@Okubo2016], therefore promoting distinct microbial communities, which in turn might explain the positive effects of plant diversity on bacterial abundances [@DeDeyn2011].
In addition to being a carbon-source for microbes, plants release a complex mix of secondary metabolites [@Mueller2019] containing compounds that attract or repel specific bacteria, regulate their growth, or influence bacterial metabolism, which, in turn, results in a fine-tuned beneficial bacterial community in the rhizosphere. In addition to the effect created by the plant, interactions between the members of the microbial community also influence their biomass and composition. Indeed, AM fungi create their own hyphosphere, increasing bacterial growth and influencing its community composition in the vicinity of the hyphae [@Marschner2003; @Emmett2021; @Singh2008; @Toljander2007]. Extraradical hyphae release a mixture of metabolites into the soil, analogously to the exudates released from plant roots [@Luthfiana2021; @Honvault2021]. 
Some of these hyphal metabolites induce growth and shifts in the bacterial community and indirectly change the nutrient availability in the soil, for example by influencing the ability of the bacterial community to mineralise P and N [@Zhang2020]. 
Induced changes in the microbial community of the soil can, in turn, have both positive and negative effects on the focal plant or heterospecifics, subsequently as plant-soil feedbacks play a key role in plant community assembly processes [@Crawford2019; @Bezemer2006]. However, knowledge of the relative impact of the plant-AMF interaction on bacterial communities is still sparse.


### Aims of this study

The objective of this chapter was to determine if the plants' interaction niche properties are related to their carbon allocation to their microbial community.
Specifically, I tested the following hypotheses: 
**Hyp 1**: The plant-AMF interaction properties influence the AMF biomass allocation in roots and soil. For example, AMF generalist hosts are expected to have an enhanced nutrient supply due to complementarity effects of their numerically and phylogenetically more diverse AMF partners. 
This could, in turn, result in an increased exchange of carbon to the AMF, resulting in positive effects on the AMF biomass. 
Increasing plant species richness increases the bacterial biomass and results in changes of the Gram-positive and Gram-negative bacteria biomass ratios [@Chen2019]. AMF can recruit distinct bacteria by releasing hyphal exudates as well but is still unresolved if AMF communities of different diversities recruit distinct bacterial communities. I will therefore test **Hyp 2**: The plant-AMF interaction property affects the bacterial biomass and composition of the bacterial community in soil. 
**Hyp 3**: Changes of the soil AMF community evoked by the host plant can be predicted by the plants’ interaction niche properties. For example, specialist plants with low AMF richness are expected to evoke greater change in their associated soil AMF community by interacting with and therefore enriching only a small number of selected AMF.

To test these hypotheses, I grew eight plants identified as having a diverse set of interaction niche properties (see chapter 2) separately in a glasshouse. I identified the AMF community in their roots and their associated soils by sequencing of the ITS2 region and determined the biomass of different microbial taxa by phospholipid fatty acid analysis (PLFA) and neutral lipid fatty acid analysis (NLFA). PLFA is a biochemical method that quantitatively delivers fatty acid profiles, with several fatty acids used as signature biomarkers for microbial groups, for example Gram-positive bacteria or AMF, therefore resulting in approximations of the biomass for microbial taxa. NLFA, in turn, is used to determine the biomass of AMF storage structures in roots and soil which contain large amount of neutral lipids (NL) [@JabajiHare1988]. 
The fatty acid (FA) 16:1$\omega$5 is the most abundant component of the NL fraction [@Graham1995] and correlates well with the amounts of AMF storage structures in roots and soil [@Sharma2015; @VanDiepen2007]. Because AMF are unable to synthesise FAs due to their lack of a *de novo* FA synthetase I, but they receive the FA C14:0 from their plant host [@Luginbuehl2017; @Sugiura2019] which can be modified by the AMF.
Therefore, the measurement of AMF derived NL is a particularly suitable proxy for the carbon allocation from the plants to their AMF.


```{r data-import-sequences, echo=FALSE, message=FALSE, warning=FALSE}


# meta 
meta_M1Wcontrol<- read_xlsx("data/meta_M1Wcontrol.xlsx")%>%  mutate (DW_roots = as.double(DW_roots), DW_above = as.double(DW_above))
meta_M2 <- read_xlsx ("data/M2_data/meta_table.xlsx") %>% 
  filter (AC.BC == "AC"| AC.BC == "BC") %>% 
  select (!c(Primer.comb, Lib))
meta_M1 <- 
  meta_M1Wcontrol %>%  
  filter (AC.BC == "M1")

# phyloseq objects##
ps_ALL <- readRDS ("data/M2_data/ps_ALL.rds")  ## inlcudes ALL samples, ALL ASVs
ps_M2 <- subset_samples(ps_ALL, AC.BC != "M1")
ps_M2 <- subset_samples(ps_M2, AC.BC != "control")

### ps for M1 including controlsoil samples , all sequences
ps_M1_allASVs <- subset_samples (ps_ALL, AC.BC == "M1"|AC.BC =="control")
sample_data (ps_M1_allASVs) <- sample_data (data.frame (meta_M1Wcontrol, row.names = "sampleID"))
ps_M1_allASVs  <- prune_samples (sample_sums(ps_M1_allASVs)>=1, ps_M1_allASVs)
ps_M1_allASVs  <- prune_taxa(taxa_sums(ps_M1_allASVs)>=1, ps_M1_allASVs)

### only Glo
ps_M1_Glo <- subset_taxa (ps_M1_allASVs, Phylum == "Glomeromycota")
ps_M1_Glo  <- prune_samples (sample_sums(ps_M1_Glo)>=1, ps_M1_Glo)
ps_M1_Glo  <- prune_taxa(taxa_sums(ps_M1_Glo)>=1, ps_M1_Glo)


# track reads####
track_reads_M2  <- read_csv("data/M2_data/track_reads_ALL.csv") 
track_reads_M1 <- track_reads_M2 %>%  left_join (meta_M1Wcontrol, by ="sampleID") %>%  filter (AC.BC =="M1"|AC.BC =="control")  

#Input reads Lib01 and Lib02, incl control samples
input_M1 <- track_reads_M1 %>% summarise(input_all =sum (input))
no_chim_M1 <-track_reads_M1 %>% summarise(input_all =sum (nonchim))
# Average 
ave_input_M1  <- track_reads_M1 %>%  summarise(ave_reads= mean(input))
ave_nochim_M1  <- track_reads_M1 %>%  summarise(ave_reads= mean(nonchim))

# number of ASVs before removal of non-Glomeromycota
No.ASVs_M1<- estimate_richness(ps_M1_allASVs,measures = "Observed") %>% 
  as_tibble (rownames = "sampleID")
# mean number of ASvs per sample before removal of non_glo
ave_ASV_M1 <- No.ASVs_M1 %>%  summarize (ave_ASV = mean(Observed), SD_ASV = sd(Observed))

#ASV table alle samples before rarefaction
ASV_table_Glo_M1 <- data.frame (otu_table(ps_M1_Glo))  %>%  
  as_tibble(rownames ="ASV_ID") %>% 
  pivot_longer(!ASV_ID, names_to = "sampleID", values_to = "ASV_count") %>% 
  left_join((tax_table(ps_M1_Glo) %>% 
               data.frame () %>%  
               as_tibble (rownames = "ASV_ID")), by= "ASV_ID")
# total count of reads for all Glo before rarefaction
total_reads_Glo_M1  <- ASV_table_Glo_M1 %>%  summarise(total =sum(ASV_count))
total_reads_Glo_M1_persample  <- ASV_table_Glo_M1 %>% left_join (meta_M1Wcontrol) %>%  group_by (roots_soil, sampleID) %>% summarise(totals= sum(ASV_count))
# Mean percentage of Glo families in root and soil samples before rarefaction####
Perc_reads_Glo_Fam_M1 <- 
  ASV_table_Glo_M1 %>% 
  group_by (sampleID, Family) %>%  
  summarize (Fam_sum = sum (ASV_count)) %>%  
  left_join(total_reads_Glo_M1_persample) %>% 
  mutate (perc_Fam = 100*Fam_sum/totals) %>% 
  group_by (roots_soil, Family) %>% 
  summarize (mean =mean (perc_Fam)) %>%  
  group_by (roots_soil) %>%  
  arrange (desc(mean))

Perc_Fam_soil  <- Perc_reads_Glo_Fam_M1 %>%  filter (roots_soil == "soil")
Perc_Fam_roots  <- Perc_reads_Glo_Fam_M1 %>%  filter (roots_soil == "roots")

# number of ASVs per Glo family before rarefaction
ASV_per_Fam_soil <- 
ASV_table_Glo_M1 %>% 
  select (sampleID, ASV_ID, Family, ASV_count) %>% 
  left_join (meta_M1Wcontrol) %>% 
  filter (roots_soil == "soil") %>% 
  filter (ASV_count > 0) %>%  
  select (ASV_ID, Family) %>%  
  unique () %>% 
  group_by (Family) %>%  
  tally() %>%  
  arrange (desc(n))

ASV_per_Fam_roots <- 
ASV_table_Glo_M1 %>% 
  select (sampleID, ASV_ID, Family, ASV_count) %>% 
  left_join (meta_M1Wcontrol) %>% 
  filter (roots_soil == "roots") %>% 
  filter (ASV_count > 0) %>%  
  select (ASV_ID, Family) %>%  
  unique () %>% 
  group_by (Family) %>%  
  tally()%>%  
  arrange (desc(n))


# later in the script, coverage based arfaction is run, resultin in a NEW otu-table for M1
# it is called ps_M1_Glo_rar


```



```{r data-import-NLFA-PLFA}

#PLFA NLPFA####
### read all excel sheets into one tibble
path<-"data/M1_NLRoots.xlsx" ###path of the file
path2 <- "data/M1_NLSoil.xlsx" ###path of the file
path3 <- "data/M1_PLSoil.xlsx"
Biomarker<- read_xlsx("data/FAME_RFF_biomarker.xlsx", sheet = "BM")


# import NLFA data - ROOTS ####
NL<- path%>%
  excel_sheets()%>%
  set_names() %>% 
  map_df(~read_excel(path=path, sheet = .x), .id= "sheet")  %>%  ###here, it would be possible to give a range for the sheets etc
  dplyr::rename("sampleID"="sheet")  #if using dplyr, add dplyr:: because of interference with plyr

#add Biomarker info to data
NL<-merge(NL, Biomarker, by="Name")

#add meta
NL<-merge(NL, meta_M1Wcontrol, by="sampleID") ###here, it would be possible to give a range for the sheets etc
 
#Removal of a sample R09 (no IS)
NL<- NL %>% filter (sampleID!="R09")

# import NLFA data - SOIL ####
NL_soil  <- path2%>%
  excel_sheets()%>%
  set_names() %>% 
  map_df(~read_excel(path=path2, sheet = .x), .id= "sheet")  %>%  ###here, it would be possible to give a range for the sheets etc
  dplyr::rename("sampleID"="sheet")  #if using dplyr, add dplyr:: because of interference with plyr

#add Biomarker info to data
NL_soil<-merge(NL_soil, Biomarker, by="Name")

#add meta
NL_soil<-merge(NL_soil, meta_M1Wcontrol, by="sampleID") ###here, it would be possible to give a range for the sheets etc
 

#PLFA SOil
# import PLFA data - SOIL ####
PL_soil  <- path3%>%
  excel_sheets()%>%
  set_names() %>% 
  map_df(~read_excel(path=path3, sheet = .x), .id= "sheet")  %>%  
  dplyr::rename("sampleID"="sheet")  #if using dplyr, add dplyr:: because of interference with plyr

#add Biomarker info to data
PL_soil<-merge(PL_soil, Biomarker, by="Name")

#add meta
PL_soil<-merge(PL_soil, meta_M1Wcontrol, by="sampleID") %>%  
select (-c(DW_above, DW_roots ))

```

```{r data-import-NLFA-PLFA-M2}
#PLFA NLPFA####
### read all excel sheets into one tibble
path<-"data/NLFA_soil_M2.xlsx" ###path of the file
path2 <- "data/PLFA_soil_M2.xlsx" ###path of the file
Biomarker<- read_xlsx("data/FAME_RFF_biomarker.xlsx", sheet = "BM")
meta_plants  <- read_xlsx("data/metadata_M2_soil.xlsx", sheet = "meta_plants") %>% unique()
meta_PLFA  <- read_xlsx("data/metadata_M2_soil.xlsx", sheet = "meta_M2_PLFA", range = "A1:G100")
# import NLFA data - M2 soil ####
NL_M2<- path%>%
  excel_sheets()%>%
  set_names() %>% 
  map_df(~read_excel(path=path, sheet = .x), .id= "sheet")  %>%  ###here, it would be possible to give a range for the sheets etc
  dplyr::rename("sampleID"="sheet") %>%   #if using dplyr, add dplyr:: because of interference with plyr
  mutate (sampleID = str_replace_all (sampleID, "NL_", "")) %>% 
  add_column (PL_NL= "NLFA") %>% 
  select(sampleID, Name, area, PL_NL)
  
#add Biomarker info to data
NL_M2<-merge(NL_M2, Biomarker %>%  select(-c(Biom1Soil, Soil_A1_kons, MW, RI)), by="Name")

#add meta
NL_M2<-left_join(NL_M2, meta_PLFA, by="sampleID") %>% left_join (meta_plants %>% unique(), by = "focal.sp")

# import PLFA data - M2 soil  ####
PL_M2<- path2%>%
  excel_sheets()%>%
  set_names() %>% 
  map_df(~read_excel(path=path2, sheet = .x), .id= "sheet")  %>%  ###here, it would be possible to give a range for the sheets etc
  dplyr::rename("sampleID"="sheet") %>%   #if using dplyr, add dplyr:: because of interference with plyr
  mutate (sampleID = str_replace_all (sampleID, "PL_", "")) %>% 
  add_column (PL_NL= "PLFA") %>% 
  select(sampleID, Name, area, PL_NL)

#add Biomarker info to data
PL_M2<-merge(PL_M2, Biomarker %>%  select(-c(Biom1Soil, Soil_A1_kons, MW, RI)), by="Name")

#add meta
PL_M2<-left_join(PL_M2, meta_PLFA, by="sampleID")%>% left_join (meta_plants %>% unique(), by = "focal.sp")

##Calculate the loss of sample and correct areas of all FAMEs for loss
#need: area of 19:0 (IS)
# need: Area of 19:0 at 100% (calculated from calibration runs), IS is at 20nmol per sample 
#  19:0 in Feb 2021
#conc= 20nmol/0.075 ml =2.6 exp-4 mol/l 
#'calculate the theoretical area as RF * conc
### theoretical area = 50,270,574 for IS - but I think there was a mistake somewhere - I assume 5Mio looks correct
#take into account split etc - it was split 10 - but the 190 was too, so no back and forth calculation needed
correction<- NL_M2 %>% select(c(sampleID,area, Name))%>% 
  filter(Name=="19:0")%>%
  mutate(IS_area=1*area) %>%  ###area as numerical
  select(c(sampleID,IS_area ))

# Full table for all NL fatty acids
NL_FA_conc_M2 <-  left_join(NL_M2, correction, by ="sampleID")%>% 
  mutate(conc_temp=area*(0.266667/IS_area)*(1/RFF)) %>%  ###the conc is now in e-03 mol/l
  filter(!is.na (conc_temp)) %>% as_tibble() %>% mutate (m_soil=as.numeric(m_soil)) %>% 
  mutate(conc_FA=75*conc_temp/m_soil)  
    ##conc is changed from e-03 mol/l per g soil to nmol per g soil (by multiplication with 75 µL)

## calculate conc of AMF biomarker per root sample only
AMF_conc_M2<-  NL_FA_conc_M2  %>% 
  group_by(sampleID, Name) %>%   #
  dplyr::summarize (sum_groups=sum(conc_FA))  %>% ##not relevant for NL because only one FA
  pivot_wider (names_from =  Name, values_from = sum_groups) %>% 
  mutate_all(~replace(., is.na(.), 0))  %>%
  select( '16:1w5', sampleID) %>% 
  dplyr::rename(AMF='16:1w5') %>%  #  easier accessible name
  left_join( meta_PLFA %>%  select (sampleID, focal.sp, treatment, AC.BC), by='sampleID') %>%  # add meta again
  left_join (meta_plants)




###Calculation for PLFA
correction<- PL_M2 %>% select(c(sampleID,area, Name))%>% 
  filter(Name=="19:0")%>%
  mutate(IS_area=area*1) %>%  ###to get the IS area for each sample
  select(c(sampleID,IS_area ))

# Full table for all PLFA fatty acids
PL_FA_conc_M2 <-  left_join(PL_M2, correction, by ="sampleID")%>% 
  mutate(conc_temp=area*(0.065/IS_area)*(1/RFF)) %>%  ###the conc is now in e-03 mol/l
  filter(!is.na (conc_temp)) %>% as_tibble() %>% mutate (m_soil=as.numeric(m_soil)) %>% 
  mutate(conc_FA=75*conc_temp/m_soil)  
##conc is changed from e-03 mol/l per g soil to nmol per g soil (by multiplication with 75 µL)

## get control samples 
PL_FA_conc_control <- 
  PL_FA_conc_M2 %>%  filter (AC.BC == "control")  # in nmol per g Soil

## full table pivotted, FA as columns
PL_conc_M2<-  PL_FA_conc_M2  %>% 
  group_by(sampleID, Name) %>%   #
  dplyr::summarize (sum_groups=sum(conc_FA))  %>% ##not relevant for NL because only one FA
  pivot_wider (names_from =  Name, values_from = sum_groups) %>% 
  mutate_all(~replace(., is.na(.), 0))  %>%
  left_join( meta_PLFA %>%  select (sampleID, focal.sp, treatment, AC.BC), by='sampleID') %>%  # add meta again
  left_join (meta_plants )   # in nmol per g Soil




```


```{r correction-IS, echo=FALSE, message=FALSE}
##Calculate the loss of sample and correct areas of all FAMEs for loss
#need: area of 19:0 (IS)
# need: Area of 19:0 at 100% (calculated from calibration runs), IS is at 20nmol per sample 
# RF for IS= 18842783498 Area per mol/l for 19:0 in sept 2019
#conc= 20nmol/0.075 ml =2.6 exp-4 mol/l 
#'calculate the theoretical area as RF * conc
### theoretical area = 4899124 IS
#take into account split etc
# correction<- NL %>% select(c(sampleID,Area, Name))%>% 
#this is splitless here

###test run relative to IS
correction<- NL %>% select(c(sampleID,Area, Name))%>% 
  filter(Name=="19:0")%>%
  mutate(IS_area=Area*1) %>%  
  select(c(sampleID,IS_area ))


# Full table for all NL fatty acids
NL_FA_conc <-  left_join(NL, correction, by ="sampleID")%>% 
  mutate(conc_temp=Area*(0.266667/IS_area)*(1/RFF))%>%  ###the conc is now in e-03 mol/l
  mutate(conc_FA=75*conc_temp/sampleMass)  %>%    ##conc is changed from e-03 mol/l per g soil to nmol per g soil (by multiplication with 75 µL)
  select (sampleID, ID, Name, conc_FA)

## calculate conc of AMF biomarker per root sample only in nmol per g root
AMF_conc<-  NL_FA_conc  %>% 
  group_by(sampleID, Name) %>%   #
  dplyr::summarise (sum_groups=sum(conc_FA))  %>% ##not relevant for NL because only one FA
  pivot_wider (names_from =  Name, values_from = sum_groups) %>% 
  mutate_all(~replace(., is.na(.), 0))  %>%
  select( '16:1w5', sampleID) %>% 
  add_column(type="root") %>%  ## needed for comparison soil vs root
  dplyr::rename(AMF='16:1w5') %>%  #  easier accessible name
  left_join( meta_M1Wcontrol, by='sampleID') %>%  # add meta again
  select(sampleID, ID, AMF, M1_M2_order, PlaSpe, type, PlantFamily, PlantSpeciesfull)

## Same calcualtions for soil####
correction<- NL_soil %>% select(c(sampleID,Area, Name))%>% 
  filter(Name=="19:0")%>%
  mutate(IS_area=1*Area) %>%  ###in den anderen ist das nur Area
  select(c(sampleID,IS_area ))

# Full table for all NL fatty acids
NL_FA_conc_Soil <-  NL_soil %>% left_join( correction, by ="sampleID")%>% 
  mutate(conc_temp=Area*(0.266667/IS_area)*(1/RFF))%>%  ###the conc is now in e-03 mol/l
  mutate(conc_FA=75*conc_temp/sampleMass)  %>%    ##conc is changed from e-03 mol/l per g soil to nmol per g soil (by multiplication with 75 µL)
  select (sampleID, ID, Name, conc_FA)

## calculate conc of AMF biomarker per soil sample only, in nmol per g soil
AMF_conc_Soil<-  NL_FA_conc_Soil  %>% 
  group_by(sampleID, Name) %>%   #
  dplyr::summarise (sum_groups=sum(conc_FA))  %>% ##not relevant for NL because only one FA
  pivot_wider (names_from =  Name, values_from = sum_groups) %>% 
  mutate_all(~replace(., is.na(.), 0))  %>%  # NA changes to 0
  select( '16:1w5', sampleID) %>% 
  add_column(type="soil") %>%  ## needed for comparison soil vs root
  dplyr::rename(AMF='16:1w5') %>%  #  easier accessible name
  left_join( meta_M1Wcontrol, by='sampleID') %>%  # add meta again
  select( sampleID, ID, AMF , M1_M2_order, PlaSpe,  type, PlantFamily, PlantSpeciesfull)

#Bind data
NL_all  <- AMF_conc %>% 
  bind_rows (AMF_conc_Soil)  # nmol per g soil or g roots

##PLFA###
##Calculate the loss of sample and correct areas of all FAMEs for loss
#need: area of 19:0 (IS)
# need: Area of 19:0 at 100% (calculated from calibration runs), IS is at 20nmol per sample 
# RF for IS= 18842783498 Area per mol/l for 19:0 in sept 2019
#conc= 20nmol/0.075 ml =2.6 exp-4 mol/l 
#'calculate the theoretical area as RF * conc
### theoretical area = 4899124 IS
## correction of PL for soil####
correction<- PL_soil %>% select(c(sampleID,Area, Name))%>% 
  filter(Name=="19:0")%>%
  mutate(IS_area=1*Area) %>%    
  select(c(sampleID,IS_area ))

# Full table for all PL fatty acids
PL_FA_conc_Soil <-  PL_soil %>% left_join( correction, by ="sampleID")%>% 
  mutate(conc_temp=Area*(0.065/IS_area)*(1/RFF))%>%  ###the conc is now in e-03 mol/l
  mutate(conc_FA=75*conc_temp/sampleMass)  %>%    ##conc is changed from e-03 mol/l per g soil to nmol per g soil (by multiplication with 75 µL)
  select (sampleID, ID, Name, conc_FA, Biom2Soil, FvsB )




```


## Method

### Glasshouse experiment and harvest
Using field-collected soil and one individual of each of eight plant species co-occurring within a pasture plant community, microcosms of each plant species were constructed in replicates of five and grown in a glasshouse. At harvest, soil and root samples were taken for detection of biomarker FAs and DNA extraction for subsequent metabarcoding of the ITS2 region.

### Study site and plant species
A detailed description of the study site can be found in Chapter 2.

```{r plant-table, tab.id = "plant-table", label = "plant-table", tab.cap = "Plant species used in this study."}
Ch3_plant_table <- read_excel("data/Ch3_plant_table.xlsx", 
    col_types = c("text", "text", "text", 
        "numeric", "text", "skip"))
Ch3_plant_table %>%  flextable %>% 
  italic (part = "body" , j = "Plant species") %>% 
  bold (part = "header")
```



### Soil collection
Within the study site, soil was collected from random locations along a 150 m transect. This was to ensure that the AMF pool represents the conditions the plants would encounter in the field. At each collection site, the vegetation was removed, and the upper 20 cm of soil were collected, which was then air dried in the glasshouse, sieved to 4 mm to remove plant roots and other coarse material and thoroughly mixed for 10 min using shovels to ensure a homogeneous mix. After homogenisation, the soil was mixed in a one to three ratio with a sand mix of coarse sand (Dalton’s washed sand No.2) and fine sand (Dalton’s washed sand No. 1) which had been mixed in a ratio of two to three. 

### Seedling growth and planting
To ensure seed viability, the seeds of the eight plant species that were shown to have a range of different interaction niche properties with the AMF communities were sourced from local suppliers (Table \@ref(tab:plant-table)). Before the start of the experiment, the germination rate and time were determined for each plant species to ensure that seedlings would be at the same growth stage at the time of the planting. Because these tests showed very variable germination times between 4 days and 13 days, and varying germination rates of 30 to 95 %, depending on the species, the seedling preparation for the experiments was set up staggered over a time frame of two weeks. For each species, multiple replicate Petri dishes were prepared on 4 consecutive days. Plant seeds were surface sterilised in 70 % ethanol for 5 min, thoroughly washed with distilled water, and then placed on moistened filter paper in Petri dishes. They were left in a temperature-controlled room at 18 °C with a 12 h day-and-night cycle and monitored daily and watered when needed. On the day of the planting, only seedlings that had a primary root developed were transplanted into 2 L pots containing 1.7 L soil-mix. Five replicate pots per plant species were established and grown in a glasshouse over the summer of 2018/2019. Plants were automatically watered, monitored daily and the position of the pots on the benches randomised biweekly. Seedlings that developed from the soil’s seed bank were promptly removed. After 113 days of growth all plants were destructively harvested.

### Root harvest, soil collection and biomass determination
At harvest, soil samples were obtained by mixing the total soil from each pot thoroughly after sampling of the majority of the root system. The soil samples were frozen to -20 °C, lyophilised overnight (FreeZone 2.5 L Benchtop Freeze Dryer, Labconco, US) and sieved to 2 mm to remove coarse sand particles, prior to storage at -20 °C. The remaining fine roots were separated from the remaining soil by thorough wet floating. Then, the total root system of each individual plant was dried, weighed and a subsample for DNA extraction and for lipid extraction taken which were kept frozen at -20 °C until further processing. 
Each remaining root system was weighed again, then oven dried at 60 °C and its dry weight determined to calculate the total dry weight of the root system before subsampling. The above ground biomass per sample was determined after oven drying at 60 °C. 





### Description of the AMF community in roots and soil by sequencing
#### DNA extraction
Roots were cut into small segments of approximately 0.5 to 1 cm length with a scalpel, and about 50 mg were subsampled randomly into a 1.5 mL Eppendorf tube filled with steel beads (Green Eppendorf Lysis Kit, Next Advance, inc., USA).
The root samples were then homogenised for 2 x 30 s at 8,000 rpm (~6,000 X g) (Precellys Evolution, Bertin instruments, FR), after which 600 µL of CTAB extraction buffer (2% Hexadecyl(trimethyl)ammonium bromide, 1% PVP40, 100 mM Tris-HCl, 1.4 M NaCl, 20 mM EDTA) [@Doyle1991] were added, and the root homogenate incubated at 60 °C for 1 hr while inverting the tube every 20 min to ensure uniform treatment.
After cooling down to room temperature, 600 µL Chloroform were added to the homogenate and the mixture was centrifuged at 10,000 rpm (~ 9,200 X g) for 2 s (Eppendorf centrifuge 5415R, Germany).
The upper aqueous layer was transferred carefully to a new 1.5 mL tube using a wide bore pipette tip. 
After gently inverting the sample with additional 800 µL cold isopropanol, it was put on ice for 10 min to precipitate the DNA. 
The DNA was further pelleted by centrifugation at 10,000 rpm for 1 min, and the supernatant then carefully discarded. The DNA pellet was washed twice with 800 µL of 80% ethanol by gently mixing followed by centrifugation at 10,000 rpm for 1 min. The supernatant was again discarded and the pellet air-dried upside down at room temperature. The DNA pellet was resuspended in 40 µL TE buffer (10 nM Tris-HCl, 1mM EDTA, pH = 7.4) and the quality and concentration of the DNA extract assessed by spectrophotometry (Nanodrop NP80, Implen, Germany) before it was frozen at -80 °C until further processing. If samples failed at spectrophotometric quality control, they were column cleaned using a shortened protocol of the GeneJet DNA extraction kit (Thermo Scientific, NZ), starting with the addition of gDNA-binding solution to the DNA extract and following the remaining steps of the protocol. 

For DNA extraction from soil, the NucleoSpin Soil kit (Macherey-Nagel, Germany) was applied as described in its protocol with the following modifications in the lysis step: after 300 mg of the lyophilised and sieved soil were weighed into bead tubes, lysis buffer 2 and 150 µL enhancer were added to adjust the conditions of the lysis. Quality and concentration of the DNA extract were assessed by spectrophotometry before freezing it at -80 °C until further processing.

#### PCR and sequencing
To identify the AMF community the internal transcribed spacer 2 (ITS2) region of the eukaryotic ribosomal DNA was amplified by polymerase chain reaction (PCR) with modified versions of the primers ITS3 (5’-CAHCGATGAAGAACGYRG-3’) and ITS4 (5’-TCCTSCGCTTATTGATATGC-3’) after Tedersoo *et al* [-@Tedersoo2015].
General fungal primers were used instead of primers specific to the Glomeromycotina to capture fungi from the Mucoromycotina as well. Some taxa from the Mucoromycotina have been described as arbuscular mycorrhizal, and their inclusion might be relevant if they show comparable high distribution as the Glomeromycotina. Primers were barcoded to discriminate between samples (Integrated DNA Technologies IDT, USA), which resulted in 38 possible primer combinations as ITS3 was available with two different barcodes and ITS4 had been obtained with 19 different barcodes (Table \@ref(tab:primer-table)). 
The 50 µL polymerase chain reaction (PCR) reaction mix contained the extracted DNA at a concentration of 2 ng µL^-1^, 25 µL of 2 x MyTaqTM Red Mix (Bioline, London, UK), 1 µL of each 10 µM primer and 8 µL 5 x TBT-Par (750 mM trehalose, 1 mg/mL nonacetylated bovine serum albumin (BSA), 1 % polysorbate-20 (Tween-20), and 8.5 mM Tris hydrochloride, pH 8.0) as amplification enhancer [@Samarakoon2013].
All components of the PCR reaction mix were kept on ice during the preparation. A touchdown protocol for the PCR was used to enhance product formation by preventing mispriming and the production of nonspecific PCR products. For that, the PCR included a “touchdown” phase, in which the annealing temperature is successively lowered by 1 °C per cycle to the melting temperature to ensure that in the first, more critical cycles of the PCR the probability of spurious primer hybridisation is significantly reduced [@Korbie2008]. Thermal cycling was performed under the following conditions on a GeneAmp PCR System 2700 (Applied Biosciences, USA): After denaturation of the DNA at 95 °C for 5 min, 13 cycles of the touchdown phase followed. They consisted of a denaturation step (95 °C, 30 s), followed by an annealing step (45 s, temperature touchdown at 50 °C), and ended with a 1 min elongation at 72 °C. Further 12 cycles at the condition of the last cycle were appended and the reaction terminated at 4 °C. Qualitative and quantitative evaluation of the PCR product was done by gel imagining under UV on an UVITEC Essential V6 (Cleaver Scientific, UK) after gel electrophoresis on a Agarose gel (2% Agarose in 1 x TAE buffer (40 mM 2-Amino-2-(hydroxymethyl)propane-1,3-diol, 20 mM acetic acid, and 1 mM EDTA)) which contained SybrSafe® DNA gel stain (Invitrogen/ThermoFisher, US) at 1 µL per 10 mL gel. 
Electrophoresis ran at 85 V for 30 min (Labnet Power Station 300, US). The DNA concentration was determined by camparing the intensity of the DNA bands to a quantitative DNA ladder (GeneRuler 1 kb Plus DNA Ladder SM1331, Thermo Fisher Scientific, US).

Pools of 38 samples that had been amplified using different barcode combinations were then prepared at equimolar concentrations, and the products cleaned by gel extraction following the protocol of the Monarch DNA gel extraction kit (New England Biolabs, UK). The sample pools were sent to sequencing on an Illumina sequencer (Macrogen, Australia) to generate paired end reads of 300 bp lengths each. The Illumina bcl2fastq 2 pipeline was used to infer the raw sequence data.

#### Processing of raw sequence reads
Each of the 19 sequencing libraries was demultiplexed using *cutadapt*  3.2 to obtain the paired-end sequencing reads per sample. For that, the barcode sequences were given as fasta file for *cutadapt* to identify and remove the barcodes from the sequences (overlap = 6, maximal error rate = 0.35). At demultiplexing, sequences that could not be assigned to a sample file were gathered in one file for the forward (file R1) and reverse reads (file R2) each. These two files containing the unassigned sequences were again demultiplexed with the forward file as reverse file and vice versa because the barcoded sequences had been sequenced in mixed orientation. As these two demultiplexing steps resulted in two R1 files and two R2 files per sample, each pair of files was concatenated into one file R1 or R2 respectively using linux bash commands. 
Then, the primer sequences were removed again using *cutadapt* with an overlap of 6 bases and a maximal error rate of 0.12. All steps of the demultiplexing, primer removal and concatenation of the sequences were computed on the Rāpoi High Performance Compute Cluster (HPC Cluster) from Victoria University of Wellington (New Zealand). 
hen, the paired-end sample files were imported into R 4.0.4. (R Core Team, 2016) to quality check, merge and filter the reads using the DADA2 ITS pipeline 1.18 (https://benjjneb.github.io/dada2/ITS_workflow.html).
Quality filtering was applied using the default parameters except for the maximum allowance of errors for the reads which was relaxed to an error of 4 to allow more reads to pass the filtering as the reads tended to be of low quality [@Callahan2016].
At the sample inference step, pseudo-pooling across samples was included in the DADA2 method to increase sensitivity for amplicon sequence variants (ASVs) that might be rare in some samples but not in others. 
The pseudo-pooling mode includes an additional step in which each sample is compared against the whole set of ASVs of the data to identify rare reads without the need to pool all samples together for sample inference. 
In the last step, the chimeric reads were discarded using default parameters.

#### Identification of fungal sequences and tree construction
After sample inference and disposal of chimeric reads, the fungal ITS classifiers were naïve Bayes trained on the dynamic UNITE 8.3 reference database [@Koljalg2020; @Nilsson2018a] to assign the fungal taxonomy.
After identifying the ASVs, a phylogenetic tree was built by aligning the sequences with *ClustalW* [@Thompson1994] in the R package *msa* `r packageVersion("msa")` [@Bodenhofer2015].
Using the R packages *seqinr* `r packageVersion("seqinr")` [@seqinr2007] and *ape* `r packageVersion("ape")` [@ape2019], the distance matrix of the alignment was built, and the tree constructed by neighbour joining. The construction was checked by plotting the tree distances against the original distances. 
The tree was then imported into *phyloseq* `r packageVersion("phyloseq")` [@phyloseq2013], where the data was agglomerated to the genus level and subset for the Glomeromycotina. The resulting tree was plotted with the R package *ggtree* `r packageVersion("ggtree")` [@Yu2020ggtree].

#### Rarefaction
To examine if the detected pool of AMF ASV represents the total possible pool, inter- and extrapolated species rarefaction curves and sampling completeness curves were computed per plant species and substrate (i.e., roots or soil) in R package *iNEXT* `r packageVersion("iNEXT")`. The samples were then rarefied to the smallest common sample coverage before measures of $\alpha$ -diversities were applied (see Chapter 2 for details on coverage-based rarefaction). Then, the data was filtered to contain only sequences assigned to the sub-phylum Glomeromycotina to obtain the AM fungi only. 

### Determination of the interaction niche property
The interaction niche properties of a plant species with AMF encompass several numerical and phylogenetic metrics of $\alpha$- and $\beta$-diversity of the plants’ AMF community. 
Therefore, it describes comprehensively the generalist or specialist status in regard to a plant species’ interaction with AMF. The numerical and phylogenetic α- and β-diversity metrics of the interaction niche (richness S, uniqueASVs, Shannon’s diversity H’, Chao1 diversity, Faith’s phylogenetic diversity (PD), mean phylogenetic diversity (MPD), percentage of core AMF species (β~core~) and compositional units (CU)) were computed as described in Chapter 2. Principal component analysis (PCA) was conducted to determine plant groups of similar interaction niche properties.

### Determination of the microbial biomass by phospholipid and neutral lipid fatty acid analyses

The identification of fatty acids (FAs) as biomarkers is a biochemical method that quantitatively delivers fatty acid profiles without the biases associated with culturing soil microbes or polymerase chain reaction (PCR) [@Kirk2004].

#### Sample preparation
For FA analysis, all samples were lyophilised as described before and a high-throughput phospho– and neutral lipid fatty acid (PLFA and NLFA) analysis was applied [@Bligh1959; modified by @Buyer2012] as described in detail in Lewe *et al* [-@Lewe2021]. 
In short, lipids were extracted from either about 100 mg root material or at least 600 mg soil with a one-phase chloroform:methanol:phosphate-buffer mixture spiked with the phospholipid 1,2-dinonadecanoyl-sn-glycero-3-phosphocholine (PL 19:0) and the neutral lipid 1,2,3- trinonadecanoyl-sn-glycerol (NL 19:0) as internal standards (each at 20 nmol per sample). 
The lipids were fractionated into neutral lipids (NLs), glycolipids and phospholipids (PLs) on a silica column. 
The PLs and NLs were derivatised by alkaline methanolysis to generate fatty acid methyl esters (FAMEs), which were concentrated under a nitrogen stream until dry and then dissolved in 75 µL hexane for further analysis on a Gas chromatography-mass spectrometry (GC-MS). 
The FAMEs were separated on a capillary column in a gas chromatograph equipped with a mass spectrometer [@Lewe2021].


#### Fatty acid methyl ester identification
For identification of the FAMEs, the mass spectra and retention times were compared with those of 31 commercially available standards (Matreya, USA; Nu-Chek, USA). 
The ω-reference convention X:YωZ was used for designation of the FAs [@Frostegaard1996; modified by @Lewe2021]. 
To determine the quantity of the analytes in response to the mass spectrometer used, the calculation of the response factor (RF) for each FAME is required [@Dodds2005]. 
Therefore, calibration curves were obtained for 25 FAME standards of the 31 selected FAMEs, from two replicates each at five different concentrations, and analysed to calculate the response factor (RF) for each FAME. 
These FAME standards consisted of representatives from each structural group of interest (saturated, iso-branched, anteiso-branched, cyclic, monounsaturated, polyunsaturated). For FAMEs that could not be commercially obtained, the RF from the structurally most related FAME standard was applied. 

Throughout the steps of the lipid analysis, lipid material is usually lost. To determine the amount of a FAME of interest in the original sample, the ration of the FAME's RF to the RF of the internal standard was calculated, resulting in the relative response factor of the FAME of interest.
After FAMEs were identified and their peak area determined using the Shimadzu GC-MS solution software 4.44, data was imported into R for all following calculations. 
Using the above equation, the amount of analyte per g dry weight (DW) sample was calculated, resulting in the molality b in mol·g^-1^ DW substrate. 
Although FAMEs are analysed with GC-MS, the substances of interests are FAs, derived either from extracted PLs or NLs as described before. 
Hereafter, I only use the terms PLFA or NLFA to describe these substances.

#### Biomarker Designation
A comprehensive set of 31 PLFA biomarkers was utilised, taking into consideration the large variability in the composition of microbial phospholipids (Table \@ref(tab:FA-table)). 
Because the soil samples were thoroughly freed from plant material, PLFAs that are common both in plants as well as in fungi were used as fungal biomarkers, e.g., 18:2$\omega$6. 


```{r FA-table, label = "FA-table", tab.id = "FA-table", tab.cap = "Phospholipid fatty acid (PLFA) and neutral lipid fatty acid (NLFA) biomarker designation. The nomenclature follows the omega-reference convention X:YomegaZ. X= number of the carbon atoms of the fatty acid, Y = number of double bonds, Z= position of the first double bond counted from the methyl end (omega-end) of the carbon chain. Prefixes: i= Iso branching, a= anteiso branching, cy = cyclopropane ring, 2OH = hydroxyl-group at position 2 counted from the alpha-carbon atom; 10Me = methyl group at position 10 counted from the a-carbon atom Suffixes: c= cis-configuartion, t = trans-configurations."}

Ch3_FAs <- read_excel("data/Ch3_plant_table.xlsx", 
    sheet = "Sheet2", col_types = c("text", 
        "text", "text", "text", "text", "text", 
        "text"))
Ch3_FAs %>%  flextable () %>%  
  bold (part = "header") %>% 
  bold (part = "body", j = "Microbial group") %>% 
  vline (j = "Microbial group")
# OLD text    Lipids are a major component of all organisms, especially the phospholipids (PL) in cellular membranes and neutral lipids (NL) as storage lipids. Their main structural component are fatty acids (FAs) , which can be used tentatively as biomarkers to approximate the biomass of microbial groups. For example, the FA 10Me16:0 is a signature biomarker for GP bacteria (see Table \@ref(tab:)). The PLFA 16:1w5 correlates well with AMF hyphal length within the soil and in roots, and has therefore been used to approximate the active biomass of the AMF [@Olsson1995]. However, it has also been detected in bacterial PL, somewhat confounding measurement of AMF hyphae in the soil [@Olsson1999]. In addition, the NLFA 16:1w5 is considered a suitable signature biomarker to measure the AMF lipid storage both in soil as in roots. It is often the most abundant FA is the NL fraction (graham 1995), with NL contributing up to 90% of the lipids in the extraradical mycelium where they are mainly found in storage tructures like spores, and vesicles. Fatty acids are furthermore a highly suitable proxy to measure the carbon allocation from plants to their mycorrhizal fungi. AMF do not contain a de novo FA synthetase I, which renders them unable to synthetise FAs. Instead, they are allocated certain FAs (, mostly myristates from their host plants, which they use to construct other FAs.

```


### Statistical analysis

#### Effects of host interaction niche property on AMF storage biomass

To examine if the generalism of the host plant influences the AMF storage biomass in roots and in soil, linear regression analyses were applied with AMF storage biomass per g dry weight (DW) roots and per g DW soil each as response variable and AMF richness, aboveground plant biomass, root biomass and plant taxonomic identity as predictor variables. To select the most parsimonious model, stepwise model selection was implemented. For that, all variables were iteratively removed, and the most parsimonious fit of the model selected by comparison using ANOVA. Normality was tested on the residuals of the
linear model by calculating the Shapiro-Wilk test [@Royston1982]. The homogeneity of the variance of the data was assessed using Bartlett’s test. Linear regression analysis and the additional tests were performed with the R package *stats* `r packageVersion ("stats")` if not stated otherwise [@R2021].
If no appropriate model solution was reached, or the residuals did not meet the assumptions of the regression analysis, the mean AMF biomass were statistically compared among the plants by computing the Kruskal-Wallis test and the Games-Howell *post-hoc* test in the R package *rstatix* `r packageVersion("rstatix")` [@rstatix2021]. In addition, non-parametric correlation analysis between variables of interest were applied using Spearman’s rank correlation. For visualisation of the results, the packages *ggplot2* `r packageVersion("ggplot2")` [@tidyverse2019]  and *ggsignif* `r packageVersion("ggsignif")` [@AhlmannEltze2021] were employed. 


Because host species differed in their root volume, the total AMF storage amount per microcosm (pot) was determined as well. 
For that, the total DW of the soil per pot was calculated and the result then multiplied with the determined amount of FA biomarker per g DW soil. Analogously, to obtain the total AMF biomass per root system, the total DW for each root sample was determined and multiplied with the amount of FA biomarker per g DW root as determined by NLFA analysis. 
The means of these values were statistically compared using the Kruskal-Wallis test followed by a *post-hoc* Games-Howell test.

The total AMF biomass of each pot, divided by the aboveground DW plant biomass was calculated to examine the carbon allocation of each plant species, accounting for possible differences in plant biomass among the species. Plants differing in their interaction niche properties with AMF might have distinct AMF communities that differ in their colonisation strategies.
Therefore, plants of different interaction niche properties with AMF might differ in their AMF soil to root biomass ratios. To test the influence of the interaction niche properties on the soil *versus* root AMF colonisation, the average soil:root AMF storage ratios were calculated based on the total values per microcosm and statistically compared as described above.



#### Plant host effect on the bacterial biomass

To examine the relationship between bacterial biomass, AMF richness and AMF biomass, linear regression modelling was employed. 
First, the bacterial biomass was calculated by summing the amounts of the bacterial PLFA biomarkers (PLFA for bacteria in general, Gram-positive bacteria, Gram-negative bacteria and Actinobacteria) and used as response variable in the model. 
To select the most parsimonious model, model selection and tests for homoscedasticity and normality were applied as described above with aboveground plant DW, root DW, AMF richness and AMF hyphal biomass in the soil as predictor variables. 
The AMF hyphal biomass was included because AMF hyphae have been described as influential on bacterial communities due to offering additional niches in the soil compartment. 
To approximate the hyphal growth in the soil, the AMF hyphal biomass was determined based on the PLFA AMF biomarker 16:1ω5 analogously to the calculation of the NLFA biomarker as described above. 

To compare the mean bacterial PLFA amounts among the eight plant species and the three plant families, analysis of variance (one-way ANOVA) and Tukey’s *post-hoc* test were calculated (R package *rstatix*) [@rstatix2021] after checking the normality and heteroscedasticity on the ANOVA model as described above. 

#### Description of the microbial community composition by PLFA
To examine how plant species identity influences the soil microbial communities, the microbial compositions per plant species were compared. Two different approaches were utilised for comparison. 
Firstly, the amounts of the relevant FA biomarkers per microbial group were summed and divided by the total PLFA to obtain the relative abundances of each microbial group. These proportions were graphically compared by plant species. 
Secondly, principal component analyses (PCA) were employed to compare PLFA profiles and the microbial community composition based on absolute PLFA values. 
PCA was calculated and a biplot visualised using the R package *FactoMineR* `r packageVersion("FactoMineR")` [@Factominer2008]. 

#### Differential abundance analysis
Metabarcoding data is compositional data, which does not infer any information about absolute abundances which in turn impedes the interpretation of changes in taxon abundances across treatments. While sequencing provides information on relative taxon abundances, each taxon is dependent on the relative abundances of all other taxa. 
Compositional data analysis therefore requires methods to account for the data compositionality. For that reason, differential abundance analysis (DAA) was used to identify significant changes in taxon abundance in the soil across the eight different plant species. For DAA, the R package *edgeR* [@Robinson2010] was used following the *edgeR* user's guide version May 2021 [@Chen2021edgeR].
First, ASVs with low counts were filtered from the raw data using standard parameters, with each group of replicates per plant species for the root or for the soil samples used as grouping factor. 
Next, a scaling factor was computed to account for the differences in sequencing read depth among the samples. The dispersion was then estimated by applying the empirical Bayes method for each ASV and negative binomial generalised linear models (GLM) were fitted. Differential abundant ASVs between plant-associated soils and the control soil were identified by quasi-likelihood F-tests, resulting in a *p*-value and associated log~2~-fold change (log2FC) per ASV for each comparison (i.e., soil from *P. cita* compared to control soil etc.). 




## Results

```{r plot-rarefaction, fig.cap = "Rarefaction and sampling completeness curves for fungal ASVs by plant species, for roots (A-C) and associated soils (D-F). Solid line segments are calculated rarefaction and completeness curves, dotted line segments are extrapolations. Shaded areas in each plot show the 95 % confidence interval of the extrapolation. A), D) rarefaction sampling curves of fungal ASV richness. B), E) Sample completeness curves by sequence reads. C), F) Sample coverage curves in percent. ", fig.height=8, fig.width=13, dpi = 350 }

#Similar numbers of ASVs were found for the `r ASV_per_Fam_roots_rar$Family[3]` (`r ASV_per_Fam_soil_rar$n[3]` ASVs), `r ASV_per_Fam_roots_rar$Family[4]` (`r ASV_per_Fam_soil_rar$n[4]` ASVs) and `r ASV_per_Fam_roots_rar$Family[6]` with `r ASV_per_Fam_soil_rar$n[6]` ASVs.

#agglomerate to Plant Species
ps_soil_allASVs <- subset_samples(ps_M1_allASVs, roots_soil == "soil")
ps_soil_allASVs <- prune_taxa (taxa_sums(ps_soil_allASVs) >=1, ps_soil_allASVs)
ps_soil_allASVs  <- prune_samples(sample_sums(ps_soil_allASVs)>=1, ps_soil_allASVs)

ps_roots_allASVs  <- subset_samples(ps_M1_allASVs, roots_soil == "roots")
ps_roots_allASVs<- prune_taxa(taxa_sums(ps_roots_allASVs)>=1, ps_roots_allASVs)
ps_roots_allASVs<- prune_samples(sample_sums(ps_roots_allASVs)>=1, ps_roots_allASVs)


ps_soil_M1_allASVs_perPlaSpe  <- merge_samples(ps_soil_allASVs, group = "PlantSpeciesfull")
ps_roots_M1_allASVs_perPlaSpe  <- merge_samples(ps_roots_allASVs, group = "PlantSpeciesfull")

### Rooots #### -----------------
#make df for Glo_PlaSpe data (empty samples samples pruned )
comm_df_M1_PlaSpe  <- data.frame (otu_table (ps_roots_M1_allASVs_perPlaSpe)) 

#rarefaction in iNEXT 
iNext_M1_PlaSpe_richness <- iNEXT (t (comm_df_M1_PlaSpe), 
                                q= 0, # richness
                                datatype =  "abundance", 
                                endpoint = NULL, 
                                knots = 400 )
# plot 
p1 <- ggiNEXT(iNext_M1_PlaSpe_richness, type = 1) +
  theme_classic() + 
  scale_shape_manual(values = c(0,1, 2,3,4,5,6,7)) + 
  scale_y_continuous (name = "ASV richness") +
  scale_x_continuous(name = "Number of sequence reads")+
  theme (axis.text = element_text(size =16), 
         axis.title = element_text(size =20),
         legend.text = element_text(face = "italic"))+
 guides(     colour=guide_legend(title="Plant species"), 
             fill=guide_legend(title="Plant species"), 
             shape=guide_legend(title="Plant species"))

p2 <- ggiNEXT(iNext_M1_PlaSpe_richness, type = 2) +
  theme_classic() + 
  scale_shape_manual(values = c(0,1, 2,3,4,5,6,7)) + 
  scale_x_continuous(name = "Number of sequence reads")+
  theme (axis.text = element_text(size =16), 
         axis.title = element_text(size =20),
         legend.text = element_text(face = "italic"))+
 guides(     colour=guide_legend(title="Plant species"), 
             fill=guide_legend(title="Plant species"), 
             shape=guide_legend(title="Plant species"))

p3 <- ggiNEXT(iNext_M1_PlaSpe_richness, type = 3) +
  theme_classic() + 
  scale_shape_manual(values = c(0,1, 2,3,4,5,6,7)) + 
  scale_y_continuous(name = "ASV richness")+
  scale_x_continuous (labels = scales::percent) + 
  theme (axis.text = element_text(size =16), 
         axis.title = element_text(size =20),
         legend.text = element_text(face = "italic"))+
 guides(     colour=guide_legend(title="Plant species"), 
             fill=guide_legend(title="Plant species"), 
             shape=guide_legend(title="Plant species"))

### Soil #### -----------------
#make df for Glo_PlaSpe data (empty samples samples pruned )
comm_df_M1_PlaSpe_soil  <- data.frame (otu_table (ps_soil_M1_allASVs_perPlaSpe)) 

#rarefaction in iNEXT 
iNext_M1_PlaSpe_richness_soil <- iNEXT (t (comm_df_M1_PlaSpe_soil), 
                                q= 0, # richness
                                datatype =  "abundance", 
                                endpoint = NULL, 
                                knots = 400 )
# plot 
p4 <- ggiNEXT(iNext_M1_PlaSpe_richness_soil, type = 1) +
  theme_classic() + 
  scale_shape_manual(values = c(0,1, 2,3,4,5,6,7,8)) + 
  scale_y_continuous (name = "ASV richness") +
  scale_x_continuous(name = "Number of sequence reads")+
  theme (axis.text = element_text(size =16), 
         axis.title = element_text(size =20), 
         legend.text = element_text(face = "italic")) +
 guides(     colour=guide_legend(title="Plant species"), 
             fill=guide_legend(title="Plant species"), 
             shape=guide_legend(title="Plant species"))


p5 <- ggiNEXT(iNext_M1_PlaSpe_richness_soil, type = 2) +
  theme_classic() + 
  scale_shape_manual(values = c(0,1, 2,3,4,5,6,7,8)) + 
  scale_x_continuous(name = "Number of sequence reads")+
  theme (axis.text = element_text(size =16), 
         axis.title = element_text(size =20),  
         legend.text = element_text(face = "italic"))+
 guides(     colour=guide_legend(title="Plant species"), 
             fill=guide_legend(title="Plant species"), 
             shape=guide_legend(title="Plant species"))

p6 <- ggiNEXT(iNext_M1_PlaSpe_richness_soil, type = 3) +
  theme_classic() + 
  scale_shape_manual(values = c(0,1, 2,3,4,5,6,7,8)) + 
  scale_y_continuous(name = "ASV richness")+
  scale_x_continuous (labels = scales::percent) + 
  theme (axis.text = element_text(size =16), 
         axis.title = element_text(size =20), 
         legend.text = element_text(face = "italic"))+
 guides(     colour=guide_legend(title="Plant species"), 
             fill=guide_legend(title="Plant species"), 
             shape=guide_legend(title="Plant species"))

ggarrange (ggarrange (p1,p2,p3, ncol = 3, common.legend = T, labels = c("A", "B", "C"), legend = "none"), 
           ggarrange (p4,p5,p6, ncol = 3, common.legend = T, labels = c("D", "E", "F"), legend = "bottom"), common.legend = T, nrow = 2)


```



```{r coverage-based-rarefaction}

#Use metamMisq for
 #ps_ALL_iter <-  phyloseq_coverage_raref(ps_M1_allASVs, iter =99) #produces 99 ps objects in list
#write_rds (ps_ALL_iter, "results/ps_M1_allASVs_iter99.rds")
ps_ALL_iter <- read_rds("results/ps_M1_allASVs_iter99.rds")
#get mean result of 99 iterations of resampling
ALL_coverage_tables <-
  map(ps_ALL_iter, otu_table) %>%
  map (data.frame) %>%   map (t) %>%
  map (function (df) as_tibble (df,rownames = "sampleID" )) %>%
  map_dfr (bind_rows, .id = "TableNumber") %>%
  pivot_longer(!c(TableNumber, sampleID), names_to =  "ASV_ID", values_to = "counts") %>%
  group_by (sampleID, ASV_ID) %>% select (!TableNumber) %>%
  summarize (mean_counts = round (mean(counts))) %>%
  unique () %>%
  pivot_wider(id_cols = , names_from = "sampleID", values_from = "mean_counts") %>%
  data.frame()
rownames (ALL_coverage_tables)<- ALL_coverage_tables$ASV_ID
ALL_coverage_tables <- ALL_coverage_tables[,-1]


# # # get back into ps_ALL as otu_table
ps_M1_allASVs_rarefied <- ps_M1_allASVs   # to get a ps_ALL object

# Exchange the otu table to new ones from rarefaction
otu_table (ps_M1_allASVs_rarefied) <- otu_table(ALL_coverage_tables, taxa_are_rows = T)

# get new 
ps_M1_Glo_rar <- subset_taxa (ps_M1_allASVs_rarefied, Phylum == "Glomeromycota")
sample_data(ps_M1_Glo_rar) <- sample_data (data.frame (meta_M1Wcontrol, row.names = "sampleID"))
ps_M1_Glo_rar <- prune_taxa(taxa_sums(ps_M1_Glo_rar)>=1, ps_M1_Glo_rar)
ps_M1_Glo_rar  <- prune_samples(sample_sums(ps_M1_Glo_rar)>=1, ps_M1_Glo_rar)

# Soil and roots
ps_M1_roots_Glo_rar  <- subset_samples (ps_M1_Glo_rar, roots_soil == "roots") 
ps_M1_roots_Glo_rar <- prune_taxa (taxa_sums (ps_M1_roots_Glo_rar)>=1,ps_M1_roots_Glo_rar )

ps_M1_soil_Glo_rar  <- subset_samples (ps_M1_Glo_rar, roots_soil == "soil") 
ps_M1_soil_Glo_rar <- prune_taxa (taxa_sums (ps_M1_soil_Glo_rar)>=1,ps_M1_soil_Glo_rar )

write_rds (ps_M1_soil_Glo_rar , "results/ps_M1_soil_Glo_rar.rds")

No.ASVs_M1_rar<- estimate_richness(ps_M1_Glo_rar,measures = "Observed") %>% 
  as_tibble (rownames = "sampleID")
# mean number of ASvs per sample after removal of non_glo and after arefacion
ave_ASV_M1_rar <- No.ASVs_M1_rar %>%  summarize (ave_ASV = mean(Observed), SD_ASV = sd(Observed))

# Read counts
ave_allASV_rar <- sample_sums (ps_M1_allASVs_rarefied) %>%  as_tibble (rownames = "sampleID") %>%  summarize (mean= mean (value))
sum_allASV_rar <- sample_sums (ps_M1_allASVs_rarefied) %>%  as_tibble (rownames = "sampleID") %>%  summarize (sum= sum (value))

sum_Glo_M1_rar  <- sample_sums (ps_M1_Glo_rar) %>%  as_tibble (rownames = "sampleID") %>%  summarize (sum= sum (value))
ave_Glo_M1_rar  <- sample_sums (ps_M1_Glo_rar) %>%  as_tibble (rownames = "sampleID") %>%  summarize (mean= mean (value))
```


```{r data-reads-perc-rarefied}


#ASV table all samples after rarefaction
ASV_table_Glo_M1_rar <- data.frame (otu_table(ps_M1_Glo_rar))  %>%  
  as_tibble(rownames ="ASV_ID") %>% 
  pivot_longer(!ASV_ID, names_to = "sampleID", values_to = "ASV_count") %>% 
  left_join((tax_table(ps_M1_Glo_rar) %>% 
               data.frame () %>%  
               as_tibble (rownames = "ASV_ID")), by= "ASV_ID")
# total count of reads for all Glo after rarefaction
total_reads_Glo_M1_rar  <- ASV_table_Glo_M1_rar %>%  summarise(total =sum(ASV_count))
total_reads_Glo_M1_persample_rar  <- ASV_table_Glo_M1_rar %>% left_join (meta_M1Wcontrol) %>%  group_by (roots_soil, sampleID) %>% summarise(totals= sum(ASV_count))
# Mean percentage of Glo families in root and soil samples after rarefaction
Perc_reads_Glo_Fam_M1_rar <- 
  ASV_table_Glo_M1_rar %>% 
  group_by (sampleID, Family) %>%  
  summarize (Fam_sum = sum (ASV_count)) %>%  
  left_join(total_reads_Glo_M1_persample_rar) %>% 
  mutate (perc_Fam = 100*Fam_sum/totals) %>% 
  group_by (roots_soil, Family) %>% 
  summarize (mean =mean (perc_Fam)) %>%  
  group_by (roots_soil) %>%  
  arrange (desc(mean))

Perc_Fam_soil_rar  <- Perc_reads_Glo_Fam_M1_rar %>%  filter (roots_soil == "soil")
Perc_Fam_roots_rar  <- Perc_reads_Glo_Fam_M1_rar %>%  filter (roots_soil == "roots")

# number of ASVs per Glo family before rarefaction
ASV_per_Fam_soil_rar <- 
ASV_table_Glo_M1_rar %>% 
  select (sampleID, ASV_ID, Family, ASV_count) %>% 
  left_join (meta_M1Wcontrol) %>% 
  filter (roots_soil == "soil") %>% 
  filter (ASV_count > 0) %>%  
  select (ASV_ID, Family) %>%  
  unique () %>% 
  group_by (Family) %>%  
  tally() %>%  
  arrange (desc(n))

ASV_per_Fam_roots_rar <- 
ASV_table_Glo_M1_rar %>% 
  select (sampleID, ASV_ID, Family, ASV_count) %>% 
  left_join (meta_M1Wcontrol) %>% 
  filter (roots_soil == "roots") %>% 
  filter (ASV_count > 0) %>%  
  select (ASV_ID, Family) %>%  
  unique () %>% 
  group_by (Family) %>%  
  tally()%>%  
  arrange (desc(n))
```


### Sequencing data and rarefaction
Eight plant species of distinct interaction niche properties (see Chapter 2) with AMF were chosen to study the effect of these properties on the AMF biomass allocation between plant roots and soil compartments, and the bacterial biomass of soils. 
To account for the changes in growth conditions between the two experiments, including newly sourced seeds, and a different soil mix, these plant species were grown again and metabarcoding of the plants' AMF community was applied. 
The resulting dataset included root and soil samples from each plant species in replicates of five, except for *A. millefolium*, for which four soil samples existed. Furthermore, three soil control samples were obtained. Sequencing results of the fungal communities for the 40 root samples and 42 soil samples were obtained with an average of `r  round (ave_input_M1$ave_reads)%>% prettyNum(big.mark = ",")` reads per sample at a total read count of `r (input_M1$input_all)%>% prettyNum(big.mark = ",")`. `r round (ave_nochim_M1$ave_reads) %>% prettyNum(big.mark = ",")` reads remained on average per sample after filtering. 
A total number of `r ntaxa (ps_M1_allASVs) %>% prettyNum(big.mark = ",")` ASVs resulted from the sample inference of the DADA2 pipeline with an average number of `r round(ave_ASV_M1$ave_ASV)` $\pm$ `r round(ave_ASV_M1$SD_ASV)` ASVs per sample. 
The rarefaction and sampling completeness curves (Figure \@ref(fig:plot-rarefaction)) showed that all plant species, their soils and the soil controls had been sufficiently sampled for fungal ASVs. However, the richness of fungal ASVs differed notably between the plant roots and their soils, with soils having about twice as diverse fungal communities as the plant roots. 
After coverage-based rarefaction of the samples, a total of `r sum_allASV_rar$sum %>% prettyNum(big.mark = ",")` reads with an average of `r round (ave_allASV_rar$mean) %>% prettyNum(big.mark = ",")` reads per sample remained. The removal of ASVs not belonging to the sub-phylum Glomeromycotina left `r ntaxa (ps_M1_Glo_rar)` ASVs with a total count of `r total_reads_Glo_M1_rar$total %>% prettyNum(big.mark = ",")` reads. Six samples were removed at that stage as they did not include any sequences after filtering for Glomeromycotina. 

The distribution of the Glomeromycotinan families varied greatly among the plant species' roots in also in their rhizosphere. In the roots, the `r Perc_Fam_roots_rar$Family[1]` vastly dominated with `r round (Perc_Fam_roots_rar$mean[1])`% of the reads. The other identified families accounted for `r round (Perc_Fam_roots_rar$mean[2])`% (`r Perc_Fam_roots_rar$Family[2]`), `r round (Perc_Fam_roots_rar$mean[3])`% (`r Perc_Fam_roots_rar$Family[3]`), `r round (Perc_Fam_roots_rar$mean[4])`% (`r Perc_Fam_roots_rar$Family[4]`), `r round (Perc_Fam_roots_rar$mean[5])`% (`r Perc_Fam_roots_rar$Family[5]`), `r round (Perc_Fam_roots_rar$mean[7],2)`% (`r Perc_Fam_roots_rar$Family[7]`) and `r round (Perc_Fam_roots_rar$mean[6],2)`% of the reads were unidentified at the family level. The proportions of the read counts were more evenly distributed in the soils, with `r round (Perc_Fam_soil_rar$mean[1])` %  of the reads originating from the `r Perc_Fam_roots_rar$Family[1]`, but a noticeable `r round (Perc_Fam_soil_rar$mean[2])`% of the reads were unidentified at the family level. This was followed by the `r Perc_Fam_soil_rar$Family[3]` with `r round (Perc_Fam_soil_rar$mean[3])` %, `r Perc_Fam_soil_rar$Family[4]` (`r round (Perc_Fam_soil_rar$mean[4])`%), `r Perc_Fam_soil_rar$Family[5]` (`r round (Perc_Fam_soil_rar$mean[5])`%), `r Perc_Fam_soil_rar$Family[6]` (`r round (Perc_Fam_soil_rar$mean[6])`%), `r Perc_Fam_soil_rar$Family[7]` (`r round (Perc_Fam_soil_rar$mean[7])`%) and `r Perc_Fam_soil_rar$Family[8]` with `r round (Perc_Fam_soil_rar$mean[8], 2)`% of the total read count. 
If considering the diversity of ASVs affiliated with these families, a different picture between roots and the soil emerges. In roots, the `r ASV_per_Fam_roots_rar$Family[1]` dominated with a total of `r ASV_per_Fam_roots_rar$n[1]` of `r ntaxa (ps_M1_roots_Glo_rar)` detected sequences, whereas in soil, the unidentified ASVs made up the largest proportion of the ASVs (`r ASV_per_Fam_soil_rar$n[1]` of `r ntaxa(ps_M1_soil_Glo_rar)`). In contrast, the `r ASV_per_Fam_soil_rar$Family[2]` were represented  by only `r ASV_per_Fam_soil_rar$n[2]` ASVs in soil.  



### The properties of the plant-AMF interaction niche 



```{r PCA-AMF-diversity}

## richness, chao, shannon
adiv_richness_M1  <- 
  estimate_richness(ps_M1_roots_Glo_rar, measures = c("Observed", "Chao1", "Shannon")) %>% 
  as_tibble (rownames = "sampleID") %>% 
  left_join (meta_M1 %>%  select (sampleID, PlaSpe, PlantSpeciesfull, PlantFamily), by = "sampleID")  


## richness per species -average
meannumberASVsper_species_M1  <-
  adiv_richness_M1 %>% 
  group_by (PlantSpeciesfull) %>% 
  summarize (meanASV = mean (Observed))


ASV_table_Glo_roots_M1 <- 
  otu_table (ps_M1_roots_Glo_rar) %>%  
  data.frame ()  %>%  t () %>%  
  as_tibble (rownames = "sampleID") 

## unique asvs
unique_M1  <- 
  ASV_table_Glo_roots_M1  %>% 
  pivot_longer(!sampleID, names_to = "ASV_ID", values_to = "ASV_count") %>% 
  filter (ASV_count != 0) %>%  
  left_join(meta_M1 %>%  select (sampleID, PlantSpeciesfull)) %>%  
  group_by (ASV_ID,PlantSpeciesfull ) %>%  
  tally () %>% 
  group_by(PlantSpeciesfull)  %>% 
  tally (name ="uniqueASVsperPlSpe")

## CU ###
comp_units   <- meannumberASVsper_species_M1  %>% 
  left_join(unique_M1, by = "PlantSpeciesfull")  %>% 
  mutate (CUnits = uniqueASVsperPlSpe/(meanASV*5))%>% # repl adjustment nor needed here, because all 5 replicates are available
  mutate ("1-CU" = 1 - CUnits)

### Cores####

ps_test <- ps_M1_roots_Glo_rar  # make a new ps to change its sample_data (core and conglomerte have problems when sample-data can not conglomerated)
sample_data (ps_test)  <- sample_data (ps_M1_roots_Glo_rar) %>% data.frame () %>%  select (PlantSpeciesfull, PlaSpe) %>%  sample_data()
ps_focal_roots_M1_core  <- phylosmith::taxa_core (ps_test, treatment =  "PlantSpeciesfull", frequency =0.6)#see above
ps_focal_roots_M1_core <- conglomerate_samples(ps_focal_roots_M1_core, treatment = "PlantSpeciesfull", merge_on = "PlantSpeciesfull")

#calculate beta diversity as percentage 
cores_roots_M1  <- 
  as.data.frame (otu_table (ps_focal_roots_M1_core )) %>%   as_tibble(rownames = "ASV_ID") %>% 
  gather (!ASV_ID, key=PlantSpeciesfull, value = ASV_counts)  %>%  
  group_by (PlantSpeciesfull) %>%  
  dplyr::filter (ASV_counts != "0")  %>% 
  dplyr::tally(name="core") %>% 
  left_join(unique_M1, by = "PlantSpeciesfull")  %>% 
  mutate (perc_core = 100* core/uniqueASVsperPlSpe) %>% 
  mutate (perc_core = round (perc_core,1) ) %>% 
  select (PlantSpeciesfull, perc_core) 

### PD####
comm_df_Glo_M1 <- data.frame (t(otu_table (ps_M1_roots_Glo_rar))) # vegan expects samples as rows and ASVs species as columns

#hier total randomised community data frame as null model
stand_pd_Glo_all_M1  <- as_tibble (ses.pd(comm_df_Glo_M1, phy_tree(ps_M1_roots_Glo_rar), include.root = FALSE, null.model = "independentswap", runs=100, iterations=999), rownames="sampleID")

stand_pd_Glo_all_M1  <- 
  stand_pd_Glo_all_M1 %>%  
  left_join (meta_M1 %>% select (sampleID, PlantSpeciesfull)) 

## MPD ###

dist_Glo_M1  <- cophenetic(phy_tree(ps_M1_roots_Glo_rar))
ses.MPD_Glo_M1  <-  as_tibble (ses.mpd (comm_df_Glo_M1, dist_Glo_M1, null.model = "independentswap" ), rownames = "sampleID")

# together PD, MPD
PD_MPD_M1 <-  
  stand_pd_Glo_all_M1 %>%  
  select (sampleID, pd.obs) %>% 
  left_join(ses.MPD_Glo_M1, by = "sampleID") %>%  
  left_join (meta_M1) %>% 
  select (sampleID, pd.obs, mpd.obs, PlaSpe, PlantSpeciesfull) %>% 
  drop_na() %>% 
  group_by (PlantSpeciesfull) %>% 
  mutate (mean_pd = mean (pd.obs), mean_mpd = mean (mpd.obs)) %>%  
  select (PlantSpeciesfull,  mean_pd, mean_mpd)   %>% 
  unique ()  





#Get values for all metrics in one tibble
All_metrics_M1  <-
  adiv_richness_M1 %>% 
  group_by(PlantFamily, PlantSpeciesfull)  %>% 
  mutate (Chao1 = mean (Chao1)) %>% 
  mutate (Shannon = mean (Shannon)) %>%  
  select (sampleID, Chao1, Shannon) %>% 
  select (!sampleID) %>% 
  unique()  %>%  
  left_join (comp_units %>%  select (!'1-CU'))    %>%   #includes CU and unique and richness
  left_join(cores_roots_M1) %>% 
  left_join (PD_MPD_M1) %>% 
  dplyr::rename("Richness"= "meanASV", "CU" = "CUnits", "β(core)"  = "perc_core",
                "PD" = "mean_pd", "MPD" = "mean_mpd" , "uniqueASV" = "uniqueASVsperPlSpe")  %>% 
  as.data.frame(row.names = NULL) %>% 
  relocate (where (is.numeric))
rownames(All_metrics_M1)  <- All_metrics_M1$PlantSpeciesfull

##calc of eigenvalues for PC1 for text
PCA_all_metrics_M1 <- PCA(All_metrics_M1, quali.sup = c(9,10),   scale.unit = T, graph = F)


```

The interaction niche properties of the eight plant species with AMF separated strongly along principal component 1 (PC1) (Fig. \@ref(fig:PCA-AMF-diversity-plot)). As with Chapter 2, the Asteraceae were numeric specialists while the Poaceae and Plantaginaceae had higher AMF $\alpha$-diversities (e.g. richness, PD, unique ASVs etc.). 
This gradient of $\alpha$-diversity was important in forming PC1 which explained about `r round (PCA_all_metrics_M1$eig [1,2])`% of the variances in plant generalism for AMF. 
In addition, the plant species also separated along PC2, which indicates differences in $\beta$-diversities. 
While the plant species differ in their interaction niche properties, they can be assigned to three groups based on their placement in the space of their interaction niche properties. 
The first group compromises the numeric specialists *A. millefolium* and *C. intybus*. The second group includes *P. lanceolata*, *P. cita* and *S. arundinaceus* as plant numeric generalists with a tendency to high within-species $\beta$-diversity. 
In contrast, the plant species from the third group (*B. willdenowii*, *A. capillaris* and *H. lanatus*) are numeric generalists but show low $\beta$-diversity, i.e., the AMF communities of each plant species' individuals are highly similar among each other. In what follows, I apply this grouping as sorting factor in figures for ease of interpretation. 


```{r PCA-AMF-diversity-plot, fig.cap= "Principal component analysis (PCA) of numerical and phylogenetic $\\alpha$ and $\\beta$-diversities per plant species A) PCA-biplot. Dots are plant species, arrows are metrics of diversity. contrib= contribution of variable to placement of treatment in biplot. B)-D) Contributions of metrics to the first three dimensions. PD = Faith's phylogenetic diversity, unique = number of unique ASVs per treatment, $\\beta$(core) = percentage of AMF species occurring in 60% of replicates per treatment, MPD = mean phylogenetic distance, CU = compositional units/true $\\beta$-diversity, Chao1 = Chao1 diversity, Shannon = Shannon's diversity H', Richness = species richness S.", fig.height=8, fig.width=10}

#PCA_all_metrics  <- vegan::rda (All_metrics, scale = T)
PCA_all_metrics_M1 <- PCA(All_metrics_M1, quali.sup = c(9,10),   scale.unit = T, graph = F)




# get_eigenvalue(res.pca): Extract the eigenvalues/variances of principal components
# fviz_eig(res.pca): Visualise the eigenvalues
# get_pca_ind(res.pca), get_pca_var(res.pca): Extract the results for individuals and variables, respectively.
# fviz_pca_ind(res.pca), fviz_pca_var(res.pca): Visualise the results individuals and variables, respectively.
# fviz_pca_biplot(res.pca): Make a biplot of individuals and variables.

#Plot 
#fviz_pca_var(PCA_all_metrics) # only the vatiables, no biplot

plotPca  <- fviz_pca_biplot(PCA_all_metrics_M1,   repel =T, addEllipses = F) +
  xlab (label = "PC1 (75.6%)") +
  ylab (label = "PC2 (16.2%)")

plotPca2  <- fviz_pca_biplot(PCA_all_metrics_M1,     axes=c (2,3),                        repel =T, addEllipses = F)

plot1 <- fviz_contrib(PCA_all_metrics_M1, choice = "var", axes = 1,  title= "Contribution of variables to PC1")
plot2 <- fviz_contrib(PCA_all_metrics_M1, choice = "var", axes = 2,  title= "Contribution of variables to PC2")
plot3 <- fviz_contrib(PCA_all_metrics_M1, choice = "var", axes = 3,  title= "Contribution of variables to PC3")

ggarrange (plotPca,
           ggarrange (plot1, plot2, plot3, nrow = 3, ncol= 1, labels= c("B", "C", "D")), 
           nrow=1, ncol=2, widths =  c(2, 1), labels= "A") 


### PC  data for LM 
PC_data_M1roots  <- 
  PCA_all_metrics_M1$ind$coord %>% 
  as_tibble  (rownames = "PlantSpeciesfull") %>% 
  arrange (Dim.1) %>% 
  add_column (PC_order= c(1:8))
```





```{r plant-biomass, include = T, tab.id = "plant-biomass", label = "plant-biomass", tab.cap = "Average root and aboveground biomass of all species. Values are in g dry weight (DW), after oven-drying of the plant material." }
 plantbiomass <- meta_M1 %>% 
  filter (roots_soil=="roots") %>% 
  group_by(PlantSpeciesfull) %>% 
  mutate (mean_dwroots = mean (DW_roots), mean_DW = mean (DW_above), ratio = mean_DW/mean_dwroots) %>% 
  select (PlantSpeciesfull, mean_dwroots, mean_DW, ratio) %>% 
  unique () 

plantbiomass %>% 
  flextable() %>%  set_header_labels(PlantSpeciesfull = "Plant species", mean_dwroots= "mean root biomass in g", mean_DW= "mean aboveground biomass in g", ratio = "plant biomass aboveground:root") %>% 
  colformat_double(digits = 2) %>% 
  italic (j= "PlantSpeciesfull", part = "body")



```

```{r lm-biomass-interaction-niche, include = F, label= "lm-biomass-interaction-niche", tab.id = "lm-biomass-interaction-niche"  , tab.cap = "Results of stepwise model selection for linear regression of plant interaction niche trait described by principal component 1 of the interaction niche PCA by above- and belowground biomass. The resulting model was: PC1 ~ plant abovegrund biomass + root biomass. Model selection with PC2 as xx variable did not result in any significant linear relationship." }
## Add biomass to PV data
dataNLrootstosoil  <- 
   AMF_conc %>%  # is in nmol
   select (ID, AMF)  %>%  #AMF is the biomarker in µmol per g substrate
   dplyr::rename ("AMFroot" = "AMF") %>% #renamed AMF for the root samples to calculate root to soil later
   left_join ((AMF_conc_Soil %>%  ungroup () %>% select (ID, AMF)), by = "ID") %>%
   #filter (!is.na (AMFroot)) %>% filter (!is.na (AMF)) %>%  
  mutate (AMFroot = AMFroot/1000) %>% ## now in µmol 
   mutate (AMF = AMF/1000) %>% ## now in µmol
   left_join (meta_M1Wcontrol) %>%  mutate (totalAMF = AMFroot * DW_roots) %>%  
  left_join(PC_data_M1roots)


## After model selection by Anova coparison and stepwise:
mdl1 <-   lm (Dim.1~  DW_roots + DW_above  , data = dataNLrootstosoil)
# Anova (mdl1,  type =3)
# summary (mdl1)

## nonetheless, some more tests for model
# #stepwise model selection
selectedmdl1  <- step (mdl1) 
# 
#summary (selectedmdl1)
## model explanins 0.7 % , nothing significant
mdl1 %>%  as_flextable () %>%  colformat_double(digits = 2) %>%  autofit()

# mdl2 <-   lm (Dim.1~  DW_roots + DW_above   , data = dataNLrootstosoil)
# mdl3 <-   lm (Dim.1~  DW_above   , data = dataNLrootstosoil)
# mdl4<-   lm (Dim.1~  DW_roots    , data = dataNLrootstosoil)



## Overall, same results FOR DIM2!! --- 


# 
# #summary(mdl1)
# 
# #check if residuals are okay
# #plot (mdl1, select = c(1))
# p_mdl1 <- round (glance (selectedmdl1)$p.value[[1]],3)
# 
# #Check assumptions
#   #normality
# resid_selectedmdl1  <-resid(mdl1)
# shapiro.resid_mdl1  <- shapiro.test(resid_selectedmdl1) # if p> 0.05, data is normal #  NORMAL ##qqnorm (resid_selectedmdl1)

```


### Assessment of the influences on the AMF biomass in roots and plant-associated soils

```{r NLFAs-distribution, echo=FALSE, include =F,  tab.cap = "Distribution of NLFA over all samples", tab.id = "NLFAs-distribution" }
NL %>% group_by(Name) %>% 
  tally() %>% 
  flextable ()  %>%  
  autofit()
```

```{r mean-sd-AMFbiomass-species, tab.id = "mean-sd-AMFbiomass-species", tab.cap = "Average amount of the neutral lipid fatty acid (NLFA) AMF biomarker biomarker in µmol per g dry weight roots. The NLFA AMF biomarker was used as proxy for the average biomass of AMF storage compartiments in roots. Plants are ordered along principal component 1.", include =F}

# This is calculated in nmol per g roots
 AMF_conc %>% ##in nmol per g roots
  filter (type == "root") %>% select(PlantSpeciesfull, AMF) %>% 
  group_by(PlantSpeciesfull) %>%  
  summarise("mean amount of AMF biomarker" = mean(AMF)/1000,  SD=sd(AMF)/1000, n = n()) %>% ### now it is in µmol per g 
  left_join (PC_data_M1roots) %>% arrange (Dim.1) %>% select (PlantSpeciesfull, "mean amount of AMF biomarker", SD, n) %>% 
  flextable ()  %>%  colformat_double(digits = 2)

# for results: mean and sd in µmol for roots
 Roots_mean_AMF <- 
   AMF_conc %>% filter (type == "root") %>% select(PlantSpeciesfull, AMF) %>% 
  group_by(PlantSpeciesfull) %>%  
  summarise(meanAMFbiomass = round (mean(AMF)/1000,2), n = n(), SD=round (sd(AMF)/1000,2)) %>%  arrange (meanAMFbiomass)
 
 # for results: mean and sd in nmol for soil
Soil_mean_AMF <- 
   AMF_conc_Soil %>% filter (type == "soil") %>% select(PlantSpeciesfull, AMF) %>% 
  group_by(PlantSpeciesfull) %>%  
  summarise(meanAMFbiomass = round (mean(AMF),2), n = n(), SD=round (sd(AMF),2)) %>%  arrange (meanAMFbiomass)


# NL AMF biomarker for controls in nol per g soil
Soil_NL_AMF_mean_control <- NL_FA_conc_M2 %>%  filter (AC.BC == "control")%>% filter (Biom2Soil =="AMF") %>%  
  select(PlantSpeciesfull, conc_FA) %>% 
  group_by(PlantSpeciesfull) %>%  
  summarise(meanAMFbiomass = round (mean(conc_FA),2), n = n(), SD=round (sd(conc_FA),2)) %>%  arrange (meanAMFbiomass)


```

```{r stats-plot-AMFbiomass}

### test assumptions first
Shapiro_NL_AMF <-AMF_conc_Soil %>%  
    group_by (PlantFamily, PlantSpeciesfull, M1_M2_order) %>%  shapiro_test(AMF) ### Normal

## bartletts 
Bartlett_NL_AMF <- bartlett.test(AMF_conc$AMF, AMF_conc$PlantSpeciesfull)### okay


### Therefore, AnOVA and Tukeys are fine for the roots, but not for the soil samples
# to not overcomplicate, I will use the non-para for these 4 connected measures (of the one figure)
 
# anova_NL_AMF <- aov(AMF ~ PlantSpeciesfull, data = AMF_conc_Soil)
# #summary (anova_NL_AMF)
#  
# #Post Hoc test
# # here Tukey is appropriate bec of normality etc
# TukeyHSD_NL_AMF <- aov(AMF ~ PlantSpeciesfull, data = AMF_conc_Soil) %>%  tukey_hsd()

## no sign pairs

#Calculate the root to soil ratios for each sample
# How to do that - shall we calcualte the AMF biomarker amount per POT or per g of the sample?
# Depends on the data we have

# kruskal wallis for no-parametric  - rank sum test
KW_AMF_roots <-
  AMF_conc %>%
select (sampleID, ID, AMF)  %>%  #AMF is the biomarker in µmol per g substrate
dplyr::rename ("AMFroot" = "AMF") %>%
left_join(meta_M1Wcontrol, by = "sampleID") %>%  select (sampleID, AMFroot, PlantSpeciesfull) %>%  ungroup () %>%
  kruskal_test(AMFroot~PlantSpeciesfull)

# kruskal wallis for no-parametric  - rank sum test SOIL
KW_AMF_soil <-
  AMF_conc_Soil %>%
select (sampleID, ID, AMF)  %>%  #AMF is the biomarker in µmol per g substrate
dplyr::rename ("AMFroot" = "AMF") %>%
left_join(meta_M1Wcontrol, by = "sampleID") %>%  select (sampleID, AMFroot, PlantSpeciesfull) %>%  ungroup () %>%
  kruskal_test(AMFroot~PlantSpeciesfull)
# 
#games howell  roots and soil
GH_AMF <-
  AMF_conc %>%
  bind_rows(AMF_conc_Soil) %>%
  select (sampleID, ID, AMF, type)  %>%
  left_join(meta_M1Wcontrol, by = "sampleID") %>%
  select (sampleID, AMF, PlantSpeciesfull, type) %>%
  group_by (type) %>%
 games_howell_test(AMF~PlantSpeciesfull) %>% filter (p.adj < 0.05)


#games howell  roots and soil TOTAL
GH_AMF_total <-
AMF_conc %>% left_join (meta_M1Wcontrol %>%  select (sampleID, DW_roots)) %>%
  mutate (AMFtotal = DW_roots * AMF ) %>%
  select (sampleID, type, AMFtotal) %>%
  bind_rows (AMF_conc_Soil %>%  mutate (AMFtotal = 1200 * AMF ) %>%  select (AMFtotal, sampleID, type)) %>%
    left_join(meta_M1Wcontrol, by = "sampleID") %>%
    select (sampleID, AMFtotal, PlantSpeciesfull, type) %>%
    group_by (type) %>%
    games_howell_test(AMFtotal~PlantSpeciesfull) %>% filter (p.adj < 0.05)

```


The signature AMF biomarker NL 16:1$\omega$5  was detected in 36 root samples (one replicate sample each was lost for *A. millefolium*, *P. lanceolata*, *P. cita* and *S. arundinaceus*) at mean molalities between `r Roots_mean_AMF$meanAMFbiomass[1]` ± `r Roots_mean_AMF$SD[1]` µmol $\cdot$ g^-1^ DW root in *`r Roots_mean_AMF$PlantSpeciesfull[1]`* and `r Roots_mean_AMF$meanAMFbiomass[8]` ± `r Roots_mean_AMF$SD[8]` µmol $\cdot$ g^-1^ DW root in *`r Roots_mean_AMF$PlantSpeciesfull[8]`*.
In the soil, the AMF biomarker molality reached from `r Soil_mean_AMF$meanAMFbiomass[1]` ± `r Soil_mean_AMF$SD[1]` to `r Soil_mean_AMF$meanAMFbiomass[8]` ± `r Soil_mean_AMF$SD[8]` nmol$\cdot$g^-1^ DW soil. In contrast, the average amount of the NL AMF biomarker in control soils was only `r round (Soil_NL_AMF_mean_control$meanAMFbiomass,2)` ± `r Soil_NL_AMF_mean_control$SD` nmol$\cdot$g^-1^ DW soil, indicating that all examined plants translocated large amounts of carbon into the AMF mycelium. 

#### AMF storage biomass in the roots
```{r lm-AMF-roots, include =FALSE}
#label= "lm-AMF-roots", tab.id = "lm-AMF-roots"  , tab.cap = "Results of stepwise model selection for linear regression of AMF storage biomass in roots by above- and belowground biomass and plant species. Amount of neutral lipid fatty acid AMF biomarker was used as proxy for AMF biomass. The estimates and the standard error of the NLFA biomarker are in µmol per g dry weight root. The resulting model was: AMF biomass ~ root biomass. The residuals of the model did not meet the assumption of normality."
#https://ourcodingclub.github.io/tutorials/modelling/
 
dataNLrootstosoil  <- 
   AMF_conc %>%  # is in nmol
   select (ID, AMF)  %>%  #AMF is the biomarker in µmol per g substrate
   dplyr:: rename ("AMFroot" = "AMF") %>% #renamed AMF for the root samples to calculate root to soil later
   left_join ((AMF_conc_Soil %>%  ungroup () %>% select (ID, AMF)), by = "ID") %>%
   #filter (!is.na (AMFroot)) %>% filter (!is.na (AMF)) %>%  
  mutate (AMFroot = AMFroot/1000) %>% ## now in µmol 
  mutate (AMF = AMF/1000) %>% ## now in µmol
  left_join (meta_M1Wcontrol) %>%  
  mutate (totalAMF = AMFroot * DW_roots) %>%  
  left_join(PC_data_M1roots)


# ## root AMF biomarker Full model 
#mdl0 <- lm (AMFroot ~ Observed + DW_roots * DW_above * PlaSpe, data = dataNLrootstosoil)
#mdl1 <-   lm (AMFroot~  DW_roots * DW_above *PlaSpe  , data = dataNLrootstosoil)
#anova (mdl0, mdl1)
 # #stepwise model selection
#selectedmdl1  <- step (mdl0) 
# 
# 
# #summary (selectedmdl1)
# selectedmdl1 %>%  as_flextable () %>%  colformat_double(digits = 2) %>%  autofit()
# 
# #summary(selectedmdl1)
# 
# #check if residuals are okay
# #plot (mdl1, select = c(1))
# p_mdl1 <- round (glance (selectedmdl1)$p.value[[1]],3)
# 
# #Check assumptions
#   #normality
# resid_selectedmdl1  <-resid(selectedmdl1)
# shapiro.resid_mdl1  <- shapiro.test(resid_selectedmdl1) # if p> 0.05, data is normal # NOT NORMAL ##qqnorm (resid_selectedmdl1)
# 
# #homoscedasticity
# bartlett.AMFroot <- bartlett.test(dataNLrootstosoil$AMFroot, dataNLrootstosoil$PlaSpe)

#### Second approach


### Slect a model - per hand, stepwise##
# ## Full model
 mdl_AMF_roots_0 <- lm (AMFroot ~   DW_above +  PlaSpe   , data = dataNLrootstosoil)
# 
# # plaSpe ?
# mdl_AMF_roots_1<- lm (totalAMF~  PlaSpe  , data = dataNLrootstosoil)
# # Pla Spe almost significant
# 
# #
mdl_AMF_roots_2 <- lm (AMFroot~  DW_above , data = dataNLrootstosoil)
# Anova (mdl_AMF_roots_7, type ="III")
# 
# 
# anova (mdl_AMF_roots_1, mdl_AMF_roots_2) # no sign diff

 mdl_AMF_roots_3 <- lm (AMFroot~  DW_roots   , data = dataNLrootstosoil)

# anova (mdl_AMF_roots_1, mdl_AMF_roots_3) # no sign diff
# Anova (mdl_AMF_roots_3, type ="III") ### Significant
# 
# mdl_AMF_roots_4 <- lm (AMFroot~  DW_roots + DW_above + PlaSpe  , data = dataNLrootstosoil)
# anova (mdl_AMF_roots_1, mdl_AMF_roots_4) # no sign diff
# 
# 
# mdl_AMF_roots_5 <- lm (AMFroot~  PlaSpe + DW_roots *DW_above   , data = dataNLrootstosoil)
# anova (mdl_AMF_roots_1, mdl_AMF_roots_5) # no sign diff
# 
# mdl_AMF_roots_6<- lm (AMFroot~  1  , data = dataNLrootstosoil)
# anova (mdl_AMF_roots_3, mdl_AMF_roots_6) # no sign diff
# 
# 
# mdl_AMF_roots_7<- lm (AMFroot~  PlaSpe*DW_roots  , data = dataNLrootstosoil)
# anova (mdl_AMF_roots_7, mdl_AMF_roots_6) # no sign diff

# best model was only intercept and DW_roots
#summary (mdl_AMF_roots_3)

dl3_table <- mdl_AMF_roots_3 %>%  as_flextable () %>%  
  colformat_double(digits = 2) %>%  
  add_header_lines ("Model: AMF storage biomass in roots ~ root biomass") %>% 
  autofit() %>%  theme_booktabs()

#summary(selectedmdl1)

#check if residuals are okay
#plot (mdl_AMF_roots_3, select = c(1))
p_mdl_AMF_roots <- round (glance (mdl_AMF_roots_3)$p.value[[1]],3)

#Check assumptions
  #normality
resid_selectedmdl_AMF_roots  <-resid(mdl_AMF_roots_3)
shapiro.resid_mdl_AMF_roots  <- shapiro.test(resid_selectedmdl_AMF_roots) # if p> 0.05, data is normal # NOT NORMAL ##qqnorm (resid_selectedmdl1)

#homoscedasticity
#
bartlett.AMFroot <- bartlett.test(dataNLrootstosoil$AMFroot, dataNLrootstosoil$PlaSpe)


## Because normality not reached, non parametric test used:
# Also, transformations would not be the correct way for a better normality of residuals
AMFroot_cortest <- 
  cor.test (dataNLrootstosoil$AMFroot, dataNLrootstosoil$DW_roots, method = "spearman") %>%  
  tidy ()

# no sign, low correlation
```

```{r plot-mean-AMFbiomass, fig.cap = "AMF storage biomass in roots and soil by plant species. A) Mean AMF biomarker amount in nmol per g root and per g soil. Results are mean amounts  with standard deviation (SD). B) Mean AMF biomarker amount in µmol per total root system or per total soil volume. The calculations were based on the amount of the AMF biomarker per mesocosm (pot). Significance levels (Games-Howell test): * at p<0.05, ** at p<0.005, *** at p<0.0005, **** at p<0.0001.", dpi = 300, fig.width= 8, fig.height=6}

# get significant  results  ROOTS and soil NL 
df <- GH_AMF %>%   filter (!p.adj.signif == "ns")  %>%   filter (!p.adj.signif == "")  %>%  
  dplyr::mutate(groups = purrr::pmap(.l = list(group1, group2), .f = c)) %>%
  dplyr::arrange(group1)

#boxplot of NL AMFbiomarker per g in roots and soil
plot1 <- 
  NL_all  %>%  
    group_by (type,PlantSpeciesfull) %>%  
    summarize (mean_AMF = mean (AMF), SD =sd(AMF)) %>% 
    left_join (meta_plants) %>% 
  left_join (PC_data_M1roots) %>% 
    ggplot( aes(x=reorder (PlantSpeciesfull, M1_M2_order), y=mean_AMF, fill=PlantFamily)) + # reorder to get plants in preferred order
  facet_grid( scales = "free", rows = vars (type)) + 
    geom_col (stat = "identity", colour = "black") +
    geom_errorbar(aes (ymin = mean_AMF, ymax =mean_AMF+SD)) + 
    xlab("Plant species") +
    ylab(expression("NL 16:1"*omega*"5 in nmol/g substrate"))+
    theme_light () +
    theme (axis.text.x=element_text(family= "sans", size = 9, angle=45, hjust =1, color = "black" ))  +
    theme (axis.text.y=element_text(family= "sans", size = 9, color = "black"))  +
    theme (axis.title = element_text(family = "sans", size = 11 ),   
           legend.title=element_text(size=12),
           legend.text=element_text(size=11)
           ) + 
    theme (strip.text = element_text (family = "sans", size = 11, color = "black"))  +
    scale_fill_manual(values=c("Asteraceae"= "#edc948" ,
                               "Plantaginaceae"= "#4e79a7",
                               "Poaceae" = "#7F9A65" ), name = "Plant family") +
  ggsignif::geom_signif (      comparisons = df$groups,
    annotation = c (df %>% pull (p.adj.signif)),
    step_increase = 0.06, vjust = 0.5) +
   theme (legend.position = "none")

# get rid of wrong annotation

pg<-ggplot_build(plot1) #disassemble plot and obtain information


new<-pg$data[[3]] %>%  filter (PANEL =="2")
pg$data[[3]]<-new #swap out the original annotation

q<-ggplot_gtable(pg) #reassemble the plot

plot1 <- ggplotify::as.ggplot(q) #generate new plot
  

# plot 2 total 

# Get GH for plot annotation
df2 <- GH_AMF_total  %>%  filter (!p.adj.signif == "ns")  %>%   filter (!p.adj.signif == "")  %>%  
  dplyr::mutate(groups = purrr::pmap(.l = list(group1, group2), .f = c)) %>%
  dplyr::arrange(group1)

plot2 <- 
  AMF_conc  %>%  left_join(meta_M1Wcontrol) %>% 
    mutate (fullAMF = AMF* DW_roots) %>% #AMF biomarker amount for total root system
    bind_rows(AMF_conc_Soil %>% 
    left_join (meta_M1Wcontrol) %>% 
    mutate (fullAMF = AMF * 1200))  %>% # AMF biomarker amount for total pot volume
    group_by (type,PlantSpeciesfull) %>%  
    summarize (mean_fullAMF = mean (fullAMF)/1000, SD =sd(fullAMF)/1000) %>%  
  left_join((meta_M1Wcontrol %>% select (PlantSpeciesfull, M1_M2_order, PlantFamily)))  %>%
    unique() %>% 
  left_join (PC_data_M1roots) %>% 
ggplot( aes(x=reorder (PlantSpeciesfull, M1_M2_order), y=mean_fullAMF, fill=PlantFamily)) + # reorder to get plants in preferred order
  geom_bar(stat = "identity", color ="black")+
   geom_errorbar(aes (ymin = mean_fullAMF, ymax =mean_fullAMF+SD))+ 
  xlab("Plant species")+
  ylab(expression("NL 16:1"*omega*"5 in"~mu*"mol per microcosm"))+
  theme_light () +
  facet_grid("type", scales = "free" ) + 
  theme (axis.text.x=element_text(family= "sans", size = 9, angle=45, hjust =1, color = "black"))  +
  theme (axis.text.y=element_text(family= "sans", size = 9, color = "black"))  +
  theme (axis.title = element_text(family = "sans", size = 11 ),  
         legend.title=element_text(size=12), 
         legend.text=element_text(size=11)) + 
  theme (strip.text = element_text (family = "sans", size = 11, color = "black"))  +
  scale_fill_manual(values=c("Asteraceae"= "#edc948" ,
                             "Plantaginaceae"= "#4e79a7",
                             "Poaceae" = "#7F9A65" ), name = "Plant family") +
    ggsignif::geom_signif (      comparisons = df2$groups,
    annotation = c (df2 %>% pull (p.adj.signif)),
    step_increase = 0.06, vjust = 0.5)

  plot_legend <- get_legend(plot2)
  
plot2 <- plot2 +theme (legend.position = "none")
# get rid of wrong signif annotatin 

pg2<-ggplot_build(plot2) #disassemble plot and obtain information


new2<-pg2$data[[3]] %>%  filter (PANEL =="2")
pg2$data[[3]]<-new2 #swap out the original annotation

q2<-ggplot_gtable(pg2) #reassemble the plot

plot2 <- ggplotify::as.ggplot(q2) #generate new plot

ggarrange (plot1, plot2, ncol =2, widths = c(1,1),legend = "right", labels = "AUTO", legend.grob =plot_legend)
```



Model selection for linear regression models was applied to examine the relative impact of AMF richness, plant species identity, root and shoot biomass on the carbon allocation to the AMF storage biomass in the plant roots. 
The model selection revealed that the plant species identity, the plant shoot and root biomass influenced the AMF biomass in the roots, but the best model only included the root biomass as significant factor. 
However, this model did not meet the assumptions of normality of the residuals (Shapiro-Wilk normality test W = `r round ( shapiro.resid_mdl_AMF_roots$statistic[[1]],2)`, *p* = `r round (shapiro.resid_mdl_AMF_roots$p[[1]],3)`) and homoscedasticity of the variances (Bartlett's test K^2^ = `r round (bartlett.AMFroot$statistic, 1)`, *p* = `r round (bartlett.AMFroot$p.value, 3)`), and therefore variance partitioning was not possible.
Comparison of the mean AMF storage biomass revealed no significant differences among the tested plant species (Kruskal-Wallis test $\chi$^2^ = `r round (KW_AMF_roots$statistic[[1]],2)`, df = `r KW_AMF_roots$df[[1]]`, p = `r round (KW_AMF_roots$p,2)`) (Fig. \@ref(fig:plot-mean-AMFbiomass)).

To account for the differences in plant biomass among the plant species at the time of sampling, the total AMF biomass per microcosm was calculated. This resulted in a slightly different pattern of the plant species' AMF biomass in comparison to their AMF biomass per g DW soil, with the most notable change happening for *P. cita* and *C. intybus*. 
While the AMF biomarker amount per g in *P. cita* roots was high, the total AMF biomass per microcosm of *P. cita* was comparably low due to the small root biomass of the *P. cita* plants. 
For *C. intybus*, this change happened in the opposite direction, with having a low AMF biomass per g DW root material but a high total AMF biomass. 
As this this could have stemmed from the difference in root biomass, non-parametric Spearman correlation was tested, which showed a significant negative correlation (Spearman's $\rho$ = `r AMFroot_cortest$estimate`, *p* = `r AMFroot_cortest$p.value`) between AMF storage biomass per g DW root biomass and plant root biomass, indicating that smaller root systems were higher colonised than larger ones in this experiment. 
*P. cita* had the smallest root biomass (Table \@ref(tab:plant-biomass)) but its AMF biomass per total root system was not overly different from that of the other plant species (\@ref(fig:plot-mean-AMFbiomass) B), indicating a high percentage root colonisation in *P. cita*. 




#### AMF storage biomass in the soil

```{r lm-AMF-soil, include = F, label= "lm-AMF-soil", tab.id = "lm-AMF-soil"  , tab.cap = "Result of stepwise model selection for the linear regression of AMF storage biomass in soil. The amount of neutral lipid fatty acid (NLFA) AMF biomarker was used as proxy for the AMF biomass. The estimates and the standard error of the NLFA biomarker are in µmol per g dry weight soil." }
# 
# ## root AMF biomarker Full model 
# mdl2 <-   lm (AMF ~  PlaSpe*DW_roots * DW_above  , data = dataNLrootstosoil) # tested log (AMF) and sqrt(), both the same- no normality
# #stepwise model selection
# selectedmdl2  <- step (mdl2)
# 
# #summary (selectedmdl1)
# selectedmdl2 %>%  as_flextable () %>%  colformat_double(digits = 2) %>%  autofit () 
# 
# #summary(selectedmdl1)
# 
# #check if residuals are okay
# #plot (mdl1, select = c(1))
# p_mdl2 <- round (glance (selectedmdl2)$p.value[[1]],3)
# 
# #Check assumptions
#   #normality
# resid_selectedmdl2  <-resid(selectedmdl2)
# shapiro.resid_mdl2  <- shapiro.test(resid_selectedmdl2) # if p> 0.05, data is normal # NOT NORMAL
# qqnorm (resid_selectedmdl2)
# #homoscedasticity
 bartlett.AMFsoil <- bartlett.test(dataNLrootstosoil$AMF, dataNLrootstosoil$PlaSpe)


#library (car)
### Slect a model - per hand, stepwise##
# ## Full model
#mdl_AMF_Soil_0 <- lm (AMF~  Observed + DW_roots * DW_above *PlaSpe  , data = dataNLrootstosoil)
#Anova (mdl_AMF_Soil_0, type ="III")
# #Check assumptions
#   #normality

### Tested ALL combination step wise , comparison with anova and BIC - chose 2 models:
# plaSpe 
mdl_AMF_Soil_1<- lm (AMF~  DW_roots  , data = dataNLrootstosoil)
# mdl_AMF_Soil_1 %>%  as_flextable () %>%  colformat_double(digits = 2) %>%  
#  add_header_lines ("Model 1: AMF biomass in soil ~ root biomass") %>%  autofit ()
  
# Pla Spe  significant
# Anova (mdl_AMF_Soil_1, type ="III")
# resid_mdl_AMF_Soil_1  <-resid(mdl_AMF_Soil_1)
# shapiro.resid_mdl_AMF_Soil_1  <- shapiro.test(resid_mdl_AMF_Soil_1) # if p> 0.05, data is not normal #   NORMAL!!!
# qqnorm (resid_mdl_AMF_Soil_1)
# #homoscedasticity
# bartlett.AMFsoil_1 <- bartlett.test(dataNLrootstosoil$AMF, dataNLrootstosoil$PlaSpe)

## DW also significant
mdl_AMF_Soil_8<- lm (AMF~  PlaSpe  , data = dataNLrootstosoil)  # this one chose because of normality and higher R2
# anova (mdl_AMF_Soil_1, mdl_AMF_Soil_8) # no sign improvement
# Anova (mdl_AMF_Soil_8, type ="III")

 mdl_AMF_Soil_8 %>%  as_flextable () %>%  colformat_double(digits = 2) %>%  
  add_header_lines ("Model: AMF biomass in soil ~ plant species") %>%  autofit () %>%  theme_booktabs()

 resid_mdl_AMF_Soil_8  <-resid(mdl_AMF_Soil_8)
shapiro.resid_mdl_AMF_Soil_8  <- shapiro.test(resid_mdl_AMF_Soil_8) # if p> 0.05, data is normal #  NORMAL!!!
p_mdl_AMF_Soil_8   <- round (glance (mdl_AMF_Soil_8)$p.value[[1]],3)
# qqnorm (resid_mdl_AMF_Soil_1)
# #homoscedasticity
bartlett.AMFsoil_1 <- bartlett.test(dataNLrootstosoil$AMF, dataNLrootstosoil$PlaSpe)

# mdl_AMF_Soil_9<- lm (AMF~  DW_above  , data = dataNLrootstosoil)
# anova (mdl_AMF_Soil_1, mdl_AMF_Soil_9) # DW above better than PlaSpe 
# 
# Anova (mdl_AMF_Soil_9, type ="III")

#
# mdl_AMF_Soil_2 <- lm (AMF~  PlaSpe+ DW_roots , data = dataNLrootstosoil)
# anova (mdl_AMF_Soil_1, mdl_AMF_Soil_2) # no sign improvement

# mdl_AMF_Soil_3 <- lm (AMF~  PlaSpe + DW_above   , data = dataNLrootstosoil)
# 
# anova (mdl_AMF_Soil_1, mdl_AMF_Soil_3) # no sign improvement
# Anova (mdl_AMF_Soil_3, type ="III") ### Significant
# 
# mdl_AMF_Soil_4 <- lm (AMF~  DW_above * PlaSpe  , data = dataNLrootstosoil)
# anova (mdl_AMF_Soil_1, mdl_AMF_Soil_4) # no sign improvement
# 
# mdl_AMF_Soil_5 <- lm (AMF~  PlaSpe + DW_roots *DW_above   , data = dataNLrootstosoil)
# anova (mdl_AMF_Soil_1, mdl_AMF_Soil_5) # no sign improvement
# 
# mdl_AMF_Soil_6<- lm (AMF~  1  , data = dataNLrootstosoil)
# anova (mdl_AMF_Soil_1, mdl_AMF_Soil_6) # no sign improvement
# 
# 
# mdl_AMF_Soil_7<- lm (AMF~  PlaSpe*DW_roots  , data = dataNLrootstosoil)
# anova (mdl_AMF_Soil_7, mdl_AMF_Soil_1) # no sign improvement

Anova_AMF_soil  <- aov (AMF ~PlaSpe, data = dataNLrootstosoil)

```

For the soil AMF storage biomass, the best linear regression model only included the plant species identity as significant effect on AMF storage biomass but did not meet the assumptions of homogeneity of variances (Bartlett's test K^2^ = `r round (bartlett.AMFsoil$statistic, 1)`, *p* = `r round (bartlett.AMFsoil$p.value, 3)`), and therefore variance partitioning was not possible.
However, the AMF biomass per g DW soil differed significantly (Kruskal-Wallis test $\chi$^2^ = `r round (KW_AMF_soil$statistic[[1]],2)`, df = `r KW_AMF_soil$df[[1]]`, p = `r round (KW_AMF_soil$p,2)`) among at *P. cita* and the two Asteraceaen plant species.

Interestingly, the pattern of the mean AMF biomass in roots differed from its pattern in soil, with the most notable change in *P. cita* and *C. intybus*. These species showed opposing trends of the AMF biomass in roots and soils, for example, *P. cita* had a comparably high AMF biomass in roots but the lowest AMF biomarker amount of all plant species in the soil. 



#### Carbon allocation per plant biomass
```{r plot-total-AMF-per-DW, include=T, fig.cap = "Total amount of AMF biomarker per microcosm and per g aboveground plant dry weight. Horizontal lines show median values, boxes denote the interquartile range (IQR), while whiskers show 1.5$*$IQR ranges. Outliers are indicated by dots. Significance levels between plant species are based on Games-Howell test, with * at p<0.05, ** at p<0.005, *** at p<0.0005 and **** at p<0.0001.", fig.height=6, fig.width=6}

#total AMF NL biomarker per DW aboveground plant
total_AMF_perDW <- 
  AMF_conc  %>%  
  left_join(meta_M1) %>% 
  mutate (fullAMF = AMF* DW_roots) %>% 
  select (sampleID, ID, fullAMF, PlantSpeciesfull, PlantFamily, DW_above, DW_roots, M1_M2_order)  %>% 
  left_join(    AMF_conc_Soil %>%  mutate (fullAMFSoil = AMF * 1200) %>%  select (sampleID, ID, fullAMFSoil), by = "ID")  %>% 
  mutate (AMFsum = fullAMFSoil+fullAMF)  %>%  
  filter (!is.na (AMFsum))    %>% 
  mutate (AMFperDW = AMFsum/DW_above) %>% 
  left_join (PC_data_M1roots)

# STATS ###
KW_tAMFDW <- total_AMF_perDW %>% kruskal_test(AMFperDW ~PlantSpeciesfull)

GH_tAMFDW <- total_AMF_perDW %>% games_howell_test(AMFperDW ~PlantSpeciesfull) %>%  filter (p.adj < 0.05)

# For plot 
df <- GH_tAMFDW  %>%  filter (!p.adj.signif == "ns")  %>%   filter (!p.adj.signif == "")  %>%  
  dplyr::mutate(groups = purrr::pmap(.l = list(group1, group2), .f = c)) %>%
  dplyr::arrange(group1)


#Plot boxplot
total_AMF_perDW%>% 
  ggplot( aes(x=reorder (PlantSpeciesfull, M1_M2_order), y=AMFperDW, fill=PlantFamily)) + # reorder to get plants in preferred order
  geom_boxplot()+
  xlab("Plant species")+
  ylab(expression("Total AMF-biomarker NL 16:1"*omega*"5 in"~mu*"mol per DW aboveground"))+
  theme_light () +
  theme (axis.text.x=element_text(family= "sans", size = 10, angle=45, hjust =1, color = "black" ))  +
  theme (axis.text.y=element_text(family= "sans", size = 10, color = "black"))  +
  theme (axis.title = element_text(family = "sans", size = 12 ),
         legend.title=element_text(size=12),
         legend.text=element_text(size=10)) +
  theme (strip.text = element_text (family = "sans", size = 12, color = "black"))  +
  scale_fill_manual(values=c("Asteraceae"= "#edc948" ,
                             "Plantaginaceae"= "#4e79a7",
                             "Poaceae" = "#7F9A65" ), name = "Plant family") +
    ggsignif::geom_signif (      comparisons = df$groups,
    annotation = c (df %>% pull (p.adj.signif)),
    step_increase = 0.06, vjust = 0.5)
## 

```


Relative to their photosynthetic active biomass, plants did not allocate the same carbon amounts to their AMF partners (Kruskal-Wallis test $\chi$^2^ = `r round (KW_tAMFDW$statistic[[1]],2)`, df = `r KW_tAMFDW$df[[1]]`, p = `r round (KW_tAMFDW$p,2)`, Fig. \@ref(fig:plot-total-AMF-per-DW)). 
The numerical specialists, *A. millefolium* and *C. intybus* (both Asteraceae) showed significantly higher total AMF biomass accumulation per g DW aboveground biomass than the other six plant species from the Poaceae and Plantaginaceae which associated with high values of $\alpha$-diversity.

#### Soil to root ratios of the AMF storage biomass

```{r stats-AMF-ratios}
#
##table with the ratios per g substrate 
NLratio_root_soil_table  <- 
  AMF_conc %>% 
    select (ID, AMF)  %>%  #AMF is the biomarker in µmol per g substrate
    dplyr:: rename ("AMFroot" = "AMF") %>% #renamed AMF for the root samples to calculate root to soil later
    left_join (AMF_conc_Soil, by = "ID") %>%  
    filter (!is.na (AMF)) %>% 
    mutate (ratio = AMFroot/AMF) %>%  
  group_by(PlantSpeciesfull) %>%  
  summarise (mean_ratio = mean (ratio), SD =sd(ratio)) %>% 
  mutate (mean_ratio = round(mean_ratio,2)) %>%  
  mutate (SD = round (SD,2))

# Stats for the ratios#####


# ##GH and Kruskal wallis for the ratios per g -------------
# AMF_ratio_gram <-
# AMF_conc %>% ungroup () %>% select (ID, AMF) %>%  dplyr::rename ("AMFroot" = "AMF" ) %>% 
#   left_join(AMF_conc_Soil %>% mutate (AMFtotal = )) %>% mutate (ratio = AMFroot/AMF) %>% 
#   select (sampleID, ID, ratio, type)  %>% 
#   left_join(meta_M1Wcontrol, by = "sampleID") %>%  
#   select (sampleID, ratio, PlantSpeciesfull, type) 
# 
# 
# 
# GH_AMF_ratios_gram <- AMF_ratio_gram %>%  
#   games_howell_test(ratio~PlantSpeciesfull) %>% filter (p.adj < 0.05)
# 
# KW_AMF_ratios_gram <-AMF_ratio_gram %>% 
#   kruskal_test(ratio ~PlantSpeciesfull)

## GH and KW for the ratios from total per pot------------
AMF_ratio_pot <-
  AMF_conc  %>%  left_join(meta_M1Wcontrol) %>% 
  mutate (fullAMF = AMF* DW_roots) %>% 
  select (sampleID, ID, fullAMF, PlantSpeciesfull, PlantFamily)  %>% 
left_join(    AMF_conc_Soil %>%  mutate (fullAMFSoil = AMF * 1200) %>%  
                select (sampleID, ID, fullAMFSoil), by = "ID") %>% #full pot weight
  mutate (ratio = fullAMFSoil/fullAMF)  %>%  filter (!is.na (ratio)) 
# mean ratio per pot
mean_AMF_ratio_pot <- 
  AMF_ratio_pot %>% 
  group_by (PlantSpeciesfull) %>%  
  summarize ("AMF biomass soil:root" = mean (ratio), SD = sd (ratio)) %>% 
  left_join(meta_plants) %>% 
  arrange(M1_M2_order) %>% 
  select (PlantSpeciesfull, "AMF biomass soil:root", SD) %>% 
  flextable () %>%  set_header_labels(PlantSpeciesfull = "Plant species") %>% 
  colformat_double(digits = 1) %>% 
  hline(i=c(2,5), part = "body") %>%  
  italic (part = "body", j = "PlantSpeciesfull")

### Test for table in ggarrange - could not get it fitted to figure 
# ratio_table <-  
#   AMF_ratio_pot %>% 
#     group_by (PlantSpeciesfull) %>%  
#     summarize ("mean ratios" = round (mean (ratio),1), SD = round (sd (ratio),1))  %>% 
#   dplyr::rename ("Plant species" = "PlantSpeciesfull")  %>% 
#   ggtexttable(theme = ttheme("light") , rows = NULL) %>% 
#   tab_add_title("Average soil to root ratios")
#GH
GH_AMF_ratio_pot <- AMF_ratio_pot %>%  games_howell_test(ratio ~PlantSpeciesfull) %>%  filter (p.adj <0.05)

#KW test
KW_AMF_ratios_pot <- AMF_ratio_pot %>%  kruskal_test(ratio ~ PlantSpeciesfull)

```



Calculating the ratios between the total AMF biomass in the soil and AMF biomass in the roots showed that most of the AMF carbon is allocated into the soil with amounts between 10 to 90 x higher than the amount found in the root systems.
With the exception of *P. lanceolata* and *P. cita*, the mean ratios were in similar ranges (Table \@ref(tab:AMF-table-mean-ratios)), for each microcosm and did not show significant differences among the plant species (Kruskal-Wallis test $\chi$^2^ = `r round (KW_AMF_ratios_pot$statistic[[1]],2)`, df = `r KW_AMF_ratios_pot$df[[1]]`, p = `r round (KW_AMF_ratios_pot$p,2)`). This indicates a relatively stable relationship between the colonisation of the roots and the soil regardless of the plant species and their interaction niche properties with AMF. Furthermore, the mean AMF biomass did not indicate any pattern following the three groups of differing interaction properties with AMF.

```{r AMF-table-mean-ratios,tab.id = "AMF-table-mean-ratios", label = "AMF-table-mean-ratios", tab.cap = "The mean ratios and standard deviations (SD) of AMF storage biomass in soil and plant roots by plant species. The calculation was based on the total AMF biomass per microcosm with the neutral lipid fatty acid (NLFA) AMF biomarker as proxy for AMF biomass. No significant differences could be detected among the mean ratios of the plant species. The plant species are sorted by the three groups of similar interaction niche properties." , include=T }



# show the table
mean_AMF_ratio_pot
```


```{r plot-AMF-ratios, include=F, fig.cap= "Ratios of AMF biomarker amounts between roots and soil by plant species. A) Root to soil ratio of the neutral lipid fatty acid (NLFA) AMF biomarker based on the amount of biomarker per g DW of each substrate. B) Root to soil ratio of the NLFA AMF biomarker based on the the total amount of biomarker per microcosm (plant pot). Horizontal lines show median values, boxes denote the interquartile range (IQR), while whiskers show 1.5$*$IQR ranges. Outliers are indicated by dots. No significant differences of the mean ratios could be detected among the plant species. ", dpi = 300   }
# Plot Ratios per g 
# no significance included heree, as nothing was significant
plot3<-
AMF_conc %>% 
  select (ID, AMF)  %>%  #AMF is the biomarker in µmol per g substrate
  dplyr:: rename ("AMFroot" = "AMF") %>% #renamed AMF for the root samples to calculate root to soil later
  left_join (AMF_conc_Soil, by = "ID") %>%  
  filter (!is.na (AMF)) %>% 
  mutate (ratio = AMFroot/AMF)  %>% ## RATIO rot to soil, each per g
  ggplot (aes (x = reorder (PlantSpeciesfull, M1_M2_order), y = ratio, fill = PlantFamily)) +
  geom_boxplot() + 
  xlab("Plant Species") +
  ylab(expression("Root to soil ratio of NL 16:1"*omega*"5 per g subtrate"))+
   theme_light () +
  theme (axis.text.x=element_text(family= "sans", size = 8, angle=45, hjust =1, color = "black" ))  +
    theme (axis.text.y=element_text(family= "sans", size = 9, color = "black"))  +
  theme (axis.title = element_text(family = "sans", size = 11 ),  
         legend.title=element_text(size=12), 
         legend.text=element_text(size=11)) + 
  scale_fill_manual(values=c("Asteraceae"= "#edc948" ,
                             "Plantaginaceae"= "#4e79a7",
                             "Poaceae" = "#7F9A65" ),name = "Plant Family") 

#RATIOS of total AMF biomass per pot
# nothing was significant (see stats with GH and KW)

plot4 <-
AMF_conc  %>%  left_join(meta_M1Wcontrol) %>% 
  mutate (fullAMF = AMF* DW_roots) %>% 
  select (sampleID, ID, fullAMF, PlantSpeciesfull, PlantFamily, M1_M2_order)  %>% 
left_join(    AMF_conc_Soil %>%  mutate (fullAMFSoil = AMF * 1200) %>%  
                select (sampleID, ID, fullAMFSoil), by = "ID")  %>% #full pot weight
  mutate (ratio = fullAMF/fullAMFSoil)  %>%  filter (!is.na (ratio)) %>% #ratio
    ggplot( aes(x=reorder (PlantSpeciesfull, M1_M2_order), y=ratio, fill=PlantFamily)) + # reorder to get plants in preferred order
  geom_boxplot()+
  xlab("Plant Species")+
  ylab(expression("Root to soil ratio of NL 16:1"*omega*"5 per microcosm"))+
  theme_light () +
  theme (axis.text.x=element_text(family= "sans", size = 8, angle=45, hjust =1, color = "black" ))  +
  theme (axis.text.y=element_text(family= "sans", size = 9, color = "black"))  +
  theme (axis.title = element_text(family = "sans", size = 11 ),
         legend.title=element_text(size=12),
         legend.text=element_text(size=11)) +
  theme (strip.text = element_text (family = "sans", size = 13, color = "black"))  +
  scale_fill_manual(values=c("Asteraceae"= "#edc948" ,
                             "Plantaginaceae"= "#4e79a7",
                             "Poaceae" = "#7F9A65" ), name = "Plant Family")

ggarrange (NULL, plot3,NULL ,plot4, labels  = c("A","", "B", ""), ncol =4, widths = c(0.1, 1,0.1, 1), legend = "right", nrow =1, common.legend = T)


## If ratios are similar - check Pearson?



```


#### Plant biomass determines hyphal biomass
For the AMF hyphal biomass, the best linear regression model included the aboveground plant biomass and the root biomass (Table \@ref(tab:lm-AMF-PLFA-soil)), indicating that an increase in either of them resulted in an increase in hyphal biomass. The rate of the increase was, however, independent from the plant species identity, and also from the AMF storage biomass. 

```{r mean-hyphal}

# Mean Hyphal biomass
mean_hyphal_FA <-  PL_FA_conc_Soil %>% 
    select (!ID) %>%  select (!FvsB) %>%  
    bind_rows(PL_FA_conc_control %>%  select (sampleID, Name, conc_FA, Biom2Soil )) %>% 
    filter (Biom2Soil !="IS") %>%
    filter (Biom2Soil != "NAFA") %>%
    #filter (Biom2Soil!="bacteria") %>%  ## oder nur 18:0 raus?
    select  (sampleID, conc_FA, Name) %>%  
  filter (Name == "16:1w5") %>%  
  left_join(meta_M1Wcontrol) %>% 
  group_by(PlantSpeciesfull) %>% 
  mutate (meanFA = mean (conc_FA), SD = sd (conc_FA)) %>% 
  select (PlantSpeciesfull, meanFA, SD) %>% 
  unique () %>%  arrange (meanFA)
```


The smallest amount of the PLFA biomarker 16:1$\omega$5 was detected in the control soil (`r round (mean_hyphal_FA$meanFA[1],2)` $\pm$ `r round (mean_hyphal_FA$SD[1],2)` nmol$\cdot$g^-1^ DW soil), followed closely by *`r mean_hyphal_FA$PlantSpeciesfull[2]`* (`r round (mean_hyphal_FA$meanFA[2],2)` $\pm$ `r round (mean_hyphal_FA$SD[2],2)` nmol$\cdot$g^-1^ DW soil), with the highest values found in *`r mean_hyphal_FA$PlantSpeciesfull[9]`* (`r round (mean_hyphal_FA$meanFA[9],2)` $\pm$ `r round (mean_hyphal_FA$SD[9],2)` nmol$\cdot$g^-1^ DW soil). The comparably high amount of the PLFA biomarker in the control soil could indicate that a high amount of the biomarker might have originated from bacterial PLFAs.  

```{r lm-AMF-PLFA-soil, label= "lm-AMF-PLFA-soil", tab.id = "lm-AMF-PLFA-soil"  , tab.cap = "Results of linear regression of AMF hyphal biomass in soil by above- and belowground biomass. Amount of phospholipid fatty acid (PLFA) AMF biomarker was used as proxy for AMF hyphal biomass. The estimates and the standard error of the PLFA biomarker are in nmol per g dry weight soil " }

#https://ourcodingclub.github.io/tutorials/modelling/
PL_NL_AMF_soil <-  
PL_FA_conc_Soil %>%  ## this is the PLFA data
  filter (Biom2Soil=="AMF") %>%  ## now only for PL AMF biomarker
  left_join(meta_M1Wcontrol %>%  filter (roots_soil == "roots") %>%  select (ID, PlaSpe, DW_roots, DW_above, PlantSpeciesfull, PlantFamily)) %>% ## this adds root weight and other meta info
  left_join (AMF_conc_Soil %>%  ungroup () %>%  select (ID, AMF)) %>%  ## this adds the AMF storage biomass in the soil 
 left_join(PC_data_M1roots)
Mdl_PL_AMF_Soil_4 <- lm (conc_FA~  DW_roots + DW_above  , data = PL_NL_AMF_soil)### Best model
#Anova (Mdl_PL_AMF_Soil_4, type ="III") ### Significant

# tested Dim.1 too

#summary (selectedmdl1)
Mdl_PL_AMF_Soil_4 %>%  as_flextable () %>%  
  colformat_double(digits = 2) %>%  
  add_header_lines ("Model: AMF hyphal biomass ~ DW root biomass + DW aboveground biomass") %>% 
  autofit ()
#summary(Mdl_PL_AMF_Soil_4)

#check if residuals are okay
#plot (mdl1, select = c(1))
p_Mdl_PL_AMF_Soil_4 <- round (glance (Mdl_PL_AMF_Soil_4)$p.value[[1]],3)

#Check assumptions
  #normality
resid_Mdl_PL_AMF_Soil_4   <-resid(Mdl_PL_AMF_Soil_4)
shapiro.resid_Mdl_PL_AMF_Soil_4  <- shapiro.test(resid_Mdl_PL_AMF_Soil_4) # if p> 0.05, data is normal # NORMAL



```


```{r plot-hyphae-AMF,include =F, fig.cap = "AMF hyphal biomass by plant species.  No sign differences means" }
## shapiro test normality 
Shapiro_PL_AMF <-PL_FA_conc_Soil %>%  
    filter (Biom2Soil =="AMF") %>%  
    left_join(meta_M1Wcontrol) %>%
    group_by (PlantFamily, PlantSpeciesfull, M1_M2_order) %>%  shapiro_test(conc_FA) ### Normal

## bartletts 
Bartlett_PL_AMF <- PL_FA_conc_Soil %>%  
    filter (Biom2Soil =="AMF") %>%  
    left_join(meta_M1Wcontrol) 
Bartlett_PL_AMF1 <- bartlett.test(Bartlett_PL_AMF$conc_FA, Bartlett_PL_AMF$PlantSpeciesfull)### okay


### Therefore, AnOVA and Tukeys are fine
 
anova_PL_AMF <- aov(conc_FA ~ PlantSpeciesfull, data = Bartlett_PL_AMF)
#summary (anova_PL_AMF)
 
# ANOVA partitioned the variance into  component can be explained by the predictor variable
# and A component that cannot be explained by the predictor variable (variance within levels, the residual variance)
# F, is the ratio of these two sources of variation

#Post Hoc test
# here Tukey is appropriate bec of normality etc
TukeyHSD_PL_AMF <- aov(conc_FA ~ PlantSpeciesfull, data = Bartlett_PL_AMF) %>%  tukey_hsd()

df4 <- TukeyHSD_PL_AMF  %>%  filter (!p.adj.signif == "ns")  %>%   filter (!p.adj.signif == "")  %>%  
  dplyr::mutate(groups = purrr::pmap(.l = list(group1, group2), .f = c)) %>%
  dplyr::arrange(group1)
## no sign pairs


# Plot 
PL_FA_conc_Soil %>%  
  filter (Biom2Soil =="AMF") %>%  
  left_join(meta_M1Wcontrol) %>%
  group_by (PlantFamily, PlantSpeciesfull, M1_M2_order) %>%  
  summarise (mean = mean (conc_FA), SD = sd (conc_FA))  %>% 
  ggplot (aes (x= reorder (PlantSpeciesfull, M1_M2_order), y = mean, fill = PlantFamily)) + 
  geom_col (stat = "identity", colour = "black") +
    geom_errorbar(aes (ymin = mean, ymax =mean+SD)) + 
    xlab("Plant Species") +
    ylab(expression("PL 16:1"*omega*"5 in nmol/g substrate"))+
    theme_light () +
    theme (axis.text.x=element_text(family= "sans", size = 9, angle=45, hjust =1, color = "black" ))  +
    theme (axis.text.y=element_text(family= "sans", size = 9, color = "black"))  +
    theme (axis.title = element_text(family = "sans", size = 11 ),   
           legend.title=element_text(size=12),
           legend.text=element_text(size=11)
           ) + 
    theme (strip.text = element_text (family = "sans", size = 11, color = "black"))  +
    scale_fill_manual(values=c("Asteraceae"= "#edc948" ,
                               "Plantaginaceae"= "#4e79a7",
                               "Poaceae" = "#7F9A65" ), name = "Plant Family") 
 
   #  ggsignif::geom_signif (      comparisons = df$groups,
   #  annotation = c (df4 %>% pull (p.adj.signif)),
   #  step_increase = 0.06, vjust = 0.5)  +
   # theme (legend.position = "none")
 
# ### Old: Ratios NL to PL
# PL_FA_conc_Soil  %>% 
#     group_by(sampleID,  Biom2Soil) %>%   #
#     dplyr::summarise (sum_groups=sum(conc_FA))  %>% dplyr::filter(Biom2Soil == "AMF")  %>%  
#   left_join (NL_all %>%  dplyr::filter(type =="soil")) %>% 
#   mutate (AMF_NLtoPL = AMF /sum_groups) %>%  
#   filter (!is.na (AMF_NLtoPL))    %>% 
#   ggplot (aes (y= AMF_NLtoPL, x= PlaSpe, fill = PlantFamily))+ 
#   geom_boxplot() +
#      
# # Table mean FA
# PL_FA_conc_Soil  %>% 
#     group_by(sampleID,  Biom2Soil) %>%   #
#     dplyr::summarise (sum_groups=sum(conc_FA))  %>% dplyr::filter(Biom2Soil == "AMF")  %>%  
#   left_join (NL_all %>%  dplyr::filter(type =="soil")) %>% 
#   mutate (AMF_NLtoPL = AMF /sum_groups) %>%  
#   filter (!is.na (AMF_NLtoPL)) %>% 
#   group_by (PlantSpeciesfull) %>% summarize (meanratioNLPL = round (mean (AMF_NLtoPL),1)) %>% flextable()
# 
# # Statistic wilcox - alle means untereinander gleich
# wilcox_PLFA_mean_soil  <- PL_FA_conc_Soil  %>% 
#     group_by(sampleID,  Biom2Soil) %>%   #
#     dplyr::summarise (sum_groups=sum(conc_FA))  %>% dplyr::filter(Biom2Soil == "AMF")  %>%  
#   left_join (NL_all %>%  dplyr::filter(type =="soil")) %>% 
#   mutate (AMF_NLtoPL = AMF /sum_groups) %>%  
#   filter (!is.na (AMF_NLtoPL)) %>% ungroup () %>% wilcox_test(AMF_NLtoPL~PlantSpeciesfull) 


# besser means berechnen
```



###  Assessment of the soil microbial biomass by phospholipid fatty acid (PLFA) analysis


```{r detected-PLFA, include = F, tab.id = "detected-PLFA", label = "detected-PLFA", tab.cap = "Detected phospholipid fatty acids (PLFA) and their distribution in soil samples. For 37 of 40 possible samples, PLFA data could be obtained. n = number of samples the PLFA could be detected in."}
PL_soil %>% group_by(Name) %>% 
  tally() %>% 
  filter (Name != "19:0") %>% 
  filter (Name != "20:0") %>% 
  flextable ()  %>%  
  autofit()
```


```{r bact-PLFA-PlaSpe-PlaFam}
## alter text für total PLFA - kann einfach zurückgeändert werden
# These values varied significantly among plant species  (ANOVA F (`r summary(anova_PlaSpe)[[1]][["Df"]][1]`, `r summary(anova_PlaSpe)[[1]][["Df"]][2]`) = `r round ((summary (anova_PlaSpe))[[1]][["F value"]][1],2)`, *p*-value = `r prettyNum(round ((summary (anova_PlaSpe))[[1]][["Pr(>F)"]][1],4))`), ranging from `r round (Table_meanPLFA$totalPLFA[1],1)` ± `r round (Table_meanPLFA$SD[1],1)` nmol$\cdot$g^-1^ DW soil for *`r Table_meanPLFA$PlantSpeciesfull[1]`* to `r round (Table_meanPLFA$totalPLFA[8],1)` ± `r round (Table_meanPLFA$SD[8],1)` nmol$\cdot$g^-1^ DW soil for *`r Table_meanPLFA$PlantSpeciesfull[8]`* (Fig. \@ref(fig:plot-total-PLFA-PlaSpe-PlaFam)). 
# 
# A notable difference in microbial biomass (total PLFA) in the soil could also be detected at the plant family level (Kruskal-Wallis H (`r KW_PLFA_PlaFam$df`) = `r round (KW_PLFA_PlaFam$statistic,1)`, *p*-value = `r round (KW_PLFA_PlaFam$p, 4)`), with soils from the Poaceae containing intermediate levels of PLFA (mean PLFA = `r round (Table_meanPLFA_PlaFam$totalPLFA[2],1)` ± `r round (Table_meanPLFA_PlaFam$SD[2],1)` nmol$\cdot$g^-1^ DW soil), while the Asteraceae had the highest PLFA content with `r round (Table_meanPLFA_PlaFam$totalPLFA[3],1)` ± `r round (Table_meanPLFA_PlaFam$SD[3],1)` nmol$\cdot$g^-1^ DW soil and Plantaginaceae soil contained only about half of that amount with a mean PLFA of `r round (Table_meanPLFA_PlaFam$totalPLFA[1],1)` ± `r round (Table_meanPLFA_PlaFam$SD[1],1)` nmol$\cdot$g^-1^ DW soil.

#get data
  bactPLFA <-  PL_FA_conc_Soil  %>% 
  group_by(sampleID,  Biom2Soil) %>%   #
  dplyr::summarise (sum_groups=sum(conc_FA))  %>% #result in microbial groups
  dplyr::filter (Biom2Soil != "IS" & Biom2Soil != "NAFA" & Biom2Soil != "remove" &Biom2Soil != "Fungi" & Biom2Soil != "AMF")  %>%  #removes all FA that are of unknown origin or IS, or of fungal origin
  group_by (sampleID) %>% 
  dplyr::summarise(PLFA = sum (sum_groups)) %>% 
  left_join(meta_M1, by = "sampleID") %>% 
    select (-c(DW_above, DW_roots)) %>%  
    left_join(meta_M1 %>% filter (roots_soil == "roots") %>%   select (pot, DW_above, DW_roots), by = "pot") %>% 
  left_join (dataNLrootstosoil %>%  select (pot, AMF, AMFroot), by = "pot")  %>% #####ACHTUNG! AMFroot and AMF are in µmol NOW!!!!! CHANGE OF PLFA?????? PLFA is in nmol
left_join (PC_data_M1roots, by = "PlantSpeciesfull") %>% 
    mutate (PLFAperDWabove = PLFA/(DW_above)) %>% 
  mutate (PLFAperDWabovefull = PLFAperDWabove *1.2)    # times 1200 for the soil amount per pot, here from nmol to µmol because times 1.2 used


## STATS ##
#test normality
PLFA_shapiro_bact <- bactPLFA %>%  group_by (PlaSpe) %>% shapiro_test(PLFA )
  #p-value > 0.05 implying that the distribution of the data are not significantly different from normal distribution.
  #In other words, we can assume the normality for all Pla Spe

PLFA_bartlett_bact <- bartlett.test(bactPLFA$PLFA ~ bactPLFA$PlaSpe ) # p > 0.05 variances are the same (less than 0.05 = variances not equal).
#ANOVA for the comparison of PlaSpe to add to the plot text
 anova_PlaSpe_bact<- aov(PLFA ~ PlaSpe, data = bactPLFA)
#summary (anova_PlaSpe_bact)
 
# ANOVA partitioned the variance into  component can be explained by the predictor variable
# and A component that cannot be explained by the predictor variable (variance within levels, the residual variance)
# F, is the ratio of these two sources of variation

#Post Hoc test
# here Tukey is appropriate bec of normality etc
TukeyHSD_PlaSpe_bact <- aov(PLFA ~ PlantSpeciesfull, data = bactPLFA) %>%  tukey_hsd()

df3 <- TukeyHSD_PlaSpe_bact  %>%  filter (!p.adj.signif == "ns")  %>%   filter (!p.adj.signif == "")  %>%  
  dplyr::mutate(groups = purrr::pmap(.l = list(group1, group2), .f = c)) %>%
  dplyr::arrange(group1)

  
## Plot total PLFA by Plant species#####
p1 <- 
  bactPLFA %>%  
  group_by(PlantSpeciesfull) %>% 
  summarise(meanPLFA = mean (PLFA), SD = sd (PLFA)) %>% 
  left_join (meta_plants %>%  select (PlantSpeciesfull, PlantFamily, M1_M2_order) ) %>% unique () %>% 
  left_join (PC_data_M1roots) %>% 
  ggplot (aes (x= reorder (PlantSpeciesfull, M1_M2_order), y = meanPLFA, fill = PlantFamily)) + 
  geom_bar (stat = "identity", color = "black") +  
  geom_errorbar(aes (ymin = meanPLFA, ymax =meanPLFA+SD))+     
  scale_fill_manual(values=c("Asteraceae"= "#edc948" ,  "Plantaginaceae"= "#4e79a7","Poaceae" = "#7F9A65" ), name = "Plant family") +
  ylab("Bacterial PLFA in nmol/g DW soil") +
  xlab ("Plant species") +
  theme_bw () +
  theme(axis.text.x= element_text(family= "sans", face= "italic", size = 11, angle = 45, hjust = 1 )) +
  theme (axis.text.y=element_text(family= "sans", size = 11))  +
  theme (axis.title = element_text(family = "sans", size = 12 ),  
         legend.title=element_text(size=12), 
         legend.text=element_text(size=11)) + 
  ggsignif::geom_signif (      comparisons = df3$groups,
    annotation = c (df3 %>% pull (p.adj.signif)),  step_increase = 0.06, vjust = 0.5)

# table for mean  SD PLFA 
Table_meanPLFA_bact <- bactPLFA %>% 
  group_by(PlantSpeciesfull) %>%  
  summarize (totalPLFA = mean (PLFA), SD = sd (PLFA)) %>%  
  arrange (totalPLFA)  # sort ascending


## Plant Family ####
#STATS ### Pla Fam
#test normalizy
PLFA_shapiro_PlaFam_bact <- bactPLFA %>%  group_by (PlantFamily) %>% shapiro_test(PLFA )
  #p-value > 0.05 implying that the distribution of the data are not significantly different from normal distribution.
  #In other words, we can assume the normality for all PlaFam

PLFA_bartlett_PlaFam_bact <- bartlett.test(bactPLFA$PLFA ~ bactPLFA$PlantFamily ) # okay
## anova okay

#ANOVA for the comparison of PlaSpe to add to the plot text
 anova_PlaFam_bact<- aov(PLFA ~ PlantFamily, data = bactPLFA)
#summary (anova_PlaFam_bact)

TukeyHSD_PlaFam_bact <- aov(PLFA ~ PlantFamily, data = bactPLFA) %>%  tukey_hsd()


df4 <- TukeyHSD_PlaFam_bact  %>%  filter (!p.adj.signif == "ns")  %>%   filter (!p.adj.signif == "")  %>%  
  dplyr::mutate(groups = purrr::pmap(.l = list(group1, group2), .f = c)) %>%
  dplyr::arrange(group1)

  
## Plot total PLFA by Plant Family#####
p2 <- 
  bactPLFA %>%  
  group_by(PlantFamily) %>% 
  summarise(meanPLFA = mean (PLFA), SD = sd (PLFA)) %>% 
  ggplot (aes (x= PlantFamily, y = meanPLFA, fill = PlantFamily)) + 
  geom_bar (stat = "identity", color = "black") +  
  geom_errorbar(aes (ymin = meanPLFA, ymax =meanPLFA+SD))+     
  scale_fill_manual(values=c("Asteraceae"= "#edc948" ,  "Plantaginaceae"= "#4e79a7","Poaceae" = "#7F9A65" ), name = "Plant family") +
  ylab("Bacterial PLFA in nmol/g DW soil") +
  xlab ("Plant family") +
  theme_bw () +
  theme(axis.text.x= element_text(family= "sans", size = 11, angle = 45, hjust = 1 )) +
  theme (axis.text.y=element_text(family= "sans", size = 11))  +
  theme (axis.title = element_text(family = "sans", size = 12 ),  
         legend.title=element_text(size=12), 
         legend.text=element_text(size=11)) + 
  ggsignif::geom_signif (      comparisons = df4$groups,
    annotation = c (df4 %>% pull (p.adj.signif)),  step_increase = 0.06, vjust = 0.5)


# Table Pla Fam
Table_meanPLFA_PlaFam_bact <- bactPLFA %>% 
  group_by(PlantFamily) %>%  
  summarize (totalPLFA = mean (PLFA), SD = sd (PLFA)) %>%  
  arrange (totalPLFA)  # sort ascending

## mean LFA bacteria for soil control
Table_meanPLFA_soilControl_bact <- 
  PL_FA_conc_control %>% group_by(sampleID,  Biom2Soil) %>%   #
    dplyr::summarise (sum_groups=sum(conc_FA))  %>% #result in microbial groups
    dplyr::filter (Biom2Soil != "IS" & Biom2Soil != "NAFA" & Biom2Soil != "remove" &Biom2Soil != "Fungi" & Biom2Soil != "AMF")  %>%  #removes all FA that are of unknown origin or IS, or of fungal origin
    group_by (sampleID) %>% 
    dplyr::summarise(PLFA = sum (sum_groups)) %>% 
     summarize (meanPLFASC = mean (PLFA), sdPLFASC = sd (PLFA))

# Full plot is below

##ggarrange (NULL,p1,NULL, p2, ncol = 4, widths = c(0.1,2,0.1,1), legend = "bottom", common.legend = T, labels  = c("A", "", "B", ""), align = "h")
```


```{r plot-bact-PLFA-perDW, include = F, fig.cap = "Total amount of bacterial PLFA in µmol per microcosm and per g aboveground plant dry weight. Horizontal lines show median values, boxes denote the interquartile range (IQR), while whiskers show 1.5$*$IQR ranges. Outliers are indicated by dots. Significance levels between plant species are based on Games-Howell test, with * at p<0.05, ** at p<0.005, *** at p<0.0005 and **** at p<0.0001.", fig.height=6, fig.width=6}

# old text 
# To account for the influence of the photosynthetic active aboveground biomass on the carbon allocation to the bacterial community , the total bacterial PLFA of each microcosm (pot) was calculated per aboveground biomass. 
#The resulting values varied significantly among plant species  (ANOVA F (`r summary(anova_PlaSpe_DW_bact)[[1]][["Df"]][1]`, `r summary(anova_PlaSpe_DW_bact)[[1]][["Df"]][2]`) = `r round ((summary (anova_PlaSpe_DW_bact))[[1]][["F value"]][1],2)`, *p*-value = `r prettyNum(round ((summary (anova_PlaSpe_DW_bact))[[1]][["Pr(>F)"]][1],4))`), ranging from `r round (Table_meanPLFA_DW_bact$totalPLFA[1],1)` ± `r round (Table_meanPLFA_DW_bact$SD[1],1)` µmol$\cdot$g^-1^ DW aboveground plant biomass for *`r Table_meanPLFA_DW_bact$PlantSpeciesfull[1]`* to `r round (Table_meanPLFA_DW_bact$totalPLFA[8],1)` ± `r round (Table_meanPLFA_DW_bact$SD[8],1)` µmol$\cdot$g^-1^ DW for *`r Table_meanPLFA_DW_bact$PlantSpeciesfull[8]`* (Fig. \@ref(fig:plot-total-PLFA-perDW)). In the soil of both Asteraceaen species, similar high amounts of PLFAs were detected, indicating that they allocate larger amount of carbon-rich products per photosynthetic biomass to the soil microbial community. In contrast, the total PLFA amount for the plant species that are AMF numerical generalist differed significant from the Asteraceae, with about double the amount of total PLFA detected in the soil of the latter plant species.

## STATS ##
#test normality
PLFA_shapiro <- bactPLFA %>%  group_by (PlaSpe) %>% shapiro_test(PLFAperDWabovefull )
#p-value > 0.05 implying that the distribution of the data are not significantly different from normal distribution.
#In other words, we can assume the normality for all Pla Spe

PLFA_bartlett_DW <- bartlett.test(bactPLFA$PLFAperDWabovefull~bactPLFA$PlaSpe ) # p > 0.05 variances are the same (less than 0.05 = variances not equal).
#ANOVA for the comparison of PlaSpe to add to the plot text
anova_PlaSpe_DW_bact <- aov(PLFAperDWabovefull ~ PlaSpe, data = bactPLFA)
#summary (anova_PlaSpe_DW_bact)

# ANOVA partitioned the variance into  component can be explained by the predictor variable
# and A component that cannot be explained by the predictor variable (variance within levels, the residual variance)
# F, is the ratio of these two sources of variation

#Post Hoc test
# here Tukey is appropriate bec of normality etc
TukeyHSD_PlaSpeDW_bact <- aov(PLFAperDWabovefull ~ PlantSpeciesfull, data = bactPLFA) %>%  tukey_hsd()

df5 <- TukeyHSD_PlaSpeDW_bact  %>%  filter (!p.adj.signif == "ns")  %>%   filter (!p.adj.signif == "")  %>%  
  dplyr::mutate(groups = purrr::pmap(.l = list(group1, group2), .f = c)) %>%
  dplyr::arrange(group1)


## Plot total PLFA by Plant species and DW above#####
  bactPLFA %>%  
  group_by(PlantSpeciesfull) %>% 
  left_join (meta_plants %>%  select (PlantSpeciesfull, PlantFamily, M1_M2_order) ) %>% unique () %>% 
  ggplot (aes (x= reorder (PlantSpeciesfull, M1_M2_order), y = PLFAperDWabovefull, fill = PlantFamily)) + 
  geom_boxplot () +  
  scale_fill_manual(values=c("Asteraceae"= "#edc948" ,  "Plantaginaceae"= "#4e79a7","Poaceae" = "#7F9A65" )) +
  ylab("Bacterial PLFA in µmol per g DW aboveground plant biomass") +
  xlab ("Plant species") +
  theme_bw () +
  theme(axis.text.x= element_text(family= "sans", face= "italic", size = 11, angle = 45, hjust = 1 )) +
  theme (axis.text.y=element_text(family= "sans", size = 11))  +
  theme (axis.title = element_text(family = "sans", size = 12 ),  
         legend.title=element_text(size=12), 
         legend.text=element_text(size=11)) + 
  ggsignif::geom_signif (      comparisons = df5$groups,
                               annotation = c (df5 %>% pull (p.adj.signif)),  step_increase = 0.06, vjust = 0.5)

# table for mean  SD PLFA 
Table_meanPLFA_DW_bact <- bactPLFA %>% 
  group_by(PlantSpeciesfull) %>%  
  summarize (totalPLFA = mean (PLFAperDWabovefull), SD = sd (PLFAperDWabovefull)) %>%  
  arrange (totalPLFA)  # sort ascending


```

Of the 31 PLFA selected for assessing the biomass of the microbial community, 17 could be detected (see Fig. \@ref(fig:PLFA-PCA-biplot-FA)), of which 15 are of bacterial origin. 
The average bacterial biomass in the soil was estimated as the averages of the total bacterial PLFAs in the soil. These values varied among plant species  (ANOVA F (`r summary(anova_PlaSpe_bact)[[1]][["Df"]][1]`, `r summary(anova_PlaSpe_bact)[[1]][["Df"]][2]`) = `r round ((summary (anova_PlaSpe_bact))[[1]][["F value"]][1],2)`, *p*-value = `r prettyNum( ((summary (anova_PlaSpe_bact))[[1]][["Pr(>F)"]][1]))`), ranging from `r round (Table_meanPLFA_bact$totalPLFA[1],1)` ± `r round (Table_meanPLFA_bact$SD[1],1)` nmol$\cdot$g^-1^ DW soil for *`r Table_meanPLFA_bact$PlantSpeciesfull[1]`* to `r round (Table_meanPLFA_bact$totalPLFA[8],1)` ± `r round (Table_meanPLFA_bact$SD[8],1)` nmol$\cdot$g^-1^ DW soil for *`r Table_meanPLFA_bact$PlantSpeciesfull[8]`* (Fig. \@ref(fig:plot-bact-PLFA-PlaSpe-PlaFam)). The control soil contained `r round (Table_meanPLFA_soilControl_bact$meanPLFASC[1],1)` ± `r round (Table_meanPLFA_soilControl_bact$sdPLFASC[1],1)` nmol$\cdot$g^-1^ DW soil bacterial PLFAs.

A significant difference in bacterial biomass in the soil was also detected at the plant family level (ANOVA F (`r summary(anova_PlaFam_bact)[[1]][["Df"]][1]`, `r summary(anova_PlaFam_bact)[[1]][["Df"]][2]`) = `r round ((summary (anova_PlaFam_bact))[[1]][["F value"]][1],2)`, *p*-value = `r prettyNum((summary (anova_PlaFam_bact))[[1]][["Pr(>F)"]][1])`), with soils from the Poaceae containing intermediate levels of PLFA (mean PLFA = `r round (Table_meanPLFA_PlaFam_bact$totalPLFA[2],1)` ± `r round (Table_meanPLFA_PlaFam_bact$SD[2],1)` nmol$\cdot$g^-1^ DW soil), while the Asteraceae had the highest PLFA content with `r round (Table_meanPLFA_PlaFam_bact$totalPLFA[3],1)` ± `r round (Table_meanPLFA_PlaFam_bact$SD[3],1)` nmol$\cdot$g^-1^ DW soil and Plantaginaceae soil contained only about half of that amount with a mean PLFA of `r round (Table_meanPLFA_PlaFam_bact$totalPLFA[1],1)` ± `r round (Table_meanPLFA_PlaFam_bact$SD[1],1)` nmol$\cdot$g^-1^ DW soil.

```{r plot-bact-PLFA-PlaSpe-PlaFam, include = T, fig.cap= " Mean bacterial PLFA in nmol FA per g dry weight soil by plant species (A) and by plant family (B). Whiskers denote standard deviation (SD) of the mean bacterial PLFA. Significance levels of Tukey's *post-hoc* test: * at p<0.05, ** at p<0.005, *** at p<0.0005, **** at p<0.0001",fig.height= 6, fig.width=8,  dpi =300}
# Full plot

ggarrange (NULL,p1,NULL, p2, ncol = 4, widths = c(0.1,2,0.1,1), legend = "bottom", common.legend = T, labels  = c("A", "", "B", ""), align = "h")

```




Because the amount of carbon allocated to the soil bacteria differed among the plant families, linear mixed-effect modelling, eliminating the effect of the plant identity was applied to assess the relative effect of the root mediated versus the AMF mediated carbon translocation to the soil bacterial community. 
After removal of the effects of the plant family, a highly significant positive relationship between bacterial biomass and AMF hyphal biomass remained, whereas neither plant biomass not AMF richness were significant predictors of bacterial biomass.


```{r lm-PLFA-PlaSpe, include= F, tab.id = "lm-PLFA-PlaSpe",  label = "lm-PLFA-PlaSpe", tab.cap = "Results of linear regression, testing response of total microbial biomass to plant species, aboveground dry weight and root dry weight of the plant. Microbial biomass was measured by proxy using total phospholipid fatty acid (PLFA) amount in the soil. Stepwise selection resulted in the following model structure: PLFA ~ plant species + aboveground biomass * root biomass. The estimates and standard errors for the amount of PLFA are given in nmol per g dry weight soil." }

#response variable PLFA 
#explanatory variable PlaSpe
#explains effect of Pla Spe on PLFA
# this is continuous data, therefore family= Gaussian (default) used
# model selection 
 totalPLFA <-  PL_FA_conc_Soil  %>% 
  group_by(sampleID,  Biom2Soil) %>%   #
  dplyr::summarise (sum_groups=sum(conc_FA))  %>% #result in microbial groups
  dplyr::filter (Biom2Soil != "IS" & Biom2Soil != "NAFA" & Biom2Soil != "remove" )  %>%  #removes all FA that are of unknown origin or IS, or of fungal origin
  group_by (sampleID) %>% 
  dplyr::summarise(PLFA = sum (sum_groups)) %>% 
  left_join(meta_M1, by = "sampleID") %>% 
    select (-c(DW_above, DW_roots)) %>%  
    left_join(meta_M1 %>% filter (roots_soil == "roots") %>%   select (pot, DW_above, DW_roots), by = "pot") %>% 
  left_join (dataNLrootstosoil %>%  select (pot, AMF, AMFroot), by = "pot")  %>% #####ACHTUNG! AMFroot and AMF are in µmol NOW!!!!! CHANGE OF PLFA?????? PLFA is in nmol
left_join (PC_data_M1roots, by = "PlantSpeciesfull") %>% 
    mutate (PLFAperDWabove = PLFA/(DW_above)) %>% 
  mutate (PLFAperDWabovefull = PLFAperDWabove *1.2)    

#https://ourcodingclub.github.io/tutorials/modelling/
#LM to test normality and heterosc on the residuals
lm_PLFA_PlaSpe <- lm (PLFA~ PlaSpe + DW_above*DW_roots , data = totalPLFA)
#intercept is the mean for the first factor level AchMil
#Adjusted R2 = variance explained by predictor =  %

#selectedmdl_PlaSpe_PLFA  <- step (lm_PLFA_PlaSpe) ### I did also the other way, comparing all models stepwise to each other by anova and testing with Anova type 3 - same resulting model!!
selectedmdl_PlaSpe_PLFA  <- lm (PLFA~ PlaSpe+DW_above +DW_roots, data = totalPLFA)
selectedmdl_PlaSpe_PLFA %>%  as_flextable() %>%    colformat_double(digits = 2) %>%  
add_header_lines("Model: total bacterial biomass (PLFA) ~ plant species + plant aboveground biomass * root biomass") %>%  
  autofit () %>%  theme_booktabs()
#Check assumptions
  #normality
lm_resid  <-resid(selectedmdl_PlaSpe_PLFA)
shapiro.resid  <- shapiro.test(lm_resid) # if p> 0.05, data is normal
#normalityplots <-plot(lm_PLFA_PlaSpe) # check visually plots

#homoscedasticity
bartlettPLFA <- bartlett.test(totalPLFA$PLFA, totalPLFA$PlaSpe)


#ANOVA for the comparison of PlaSpe to add to the plot text
 anova_PlaSpe_PLFA <- aov(PLFA ~ PlaSpe, data = totalPLFA)  ## habe ich oben schon berechnet asl anova_PlaSpe
#summary (anova_PlaSpe)
 
# ANOVA partitioned the variance into  component can be explained by the predictor variable
# and A component that cannot be explained by the predictor variable (variance within levels, the residual variance)
# F, is the ratio of these two sources of variation

```


```{r PLFAs, include =F}
# Table PLFA in 
Table_mean_singlePLFAs <- 
PL_FA_conc_Soil %>%  
  filter (Biom2Soil != "IS" ) %>%  
  filter (Biom2Soil != "NAFA") %>% 
  left_join(meta_M1Wcontrol) %>% 
  group_by(PlantSpeciesfull, Name) %>%
  summarize (concFA= mean (conc_FA), SD = sd(conc_FA)) %>%  
    mutate_if(is.numeric, round, 2) %>% 
    unite("mean ± SD", concFA:SD, sep=" ± ")  %>%  
  pivot_wider(Name, values_from = "mean ± SD",names_from= PlantSpeciesfull) %>%  
  dplyr::rename (PLFA = Name) %>%   
  flextable ()


# quick plot percentage PLFA per pla spe   --- not included (PCA instead)
FA_PLFA_perc_PlaSpe  <- 
  PL_FA_conc_Soil %>%     
  filter (Biom2Soil != "IS")  %>%  #removes all FA that are of unknown origin or IS
  filter (Biom2Soil != "NAFA") %>% 
  left_join(meta_M1) %>% 
    group_by(PlantSpeciesfull, Name) %>% 
    summarize (concFA= mean (conc_FA), SD = sd(conc_FA))  %>%
    ggplot (aes(x=PlantSpeciesfull, y= concFA, fill = Name ))+  
    geom_bar(stat="identity", position = "fill" )
```


```{r Bact-AMF-LM, label = "Bact-AMF-LMM2"  , tab.id = "Bact-AMF-LM", tab.cap= "Results of linear mixed-effects model selection of the effect of AMF hyphal biomass, AMF richness, plant identity and plant biomass on soil bacterial biomass. The resulting model included AMF hyphal biomass as fixed effect and plant family as random effect."}

### only bacteria instead of all PLFA
### HYPHAL -- AMF biomarker from PLFA only!!
Bact_to_AMF_PLFA <-   PL_FA_conc_Soil  %>%
  group_by(sampleID, ID,  FvsB) %>%   
  filter(Biom2Soil != "Fungi") %>% #this removes the other Fungi from the PLFA dataset  - leaves only AMF PLFA behind
  dplyr::summarise (sum_groups=sum(conc_FA))  %>% #result in microbial groups
  ungroup () %>%
  dplyr::filter (FvsB != "IS")  %>%  #removes  IS
  dplyr::filter (FvsB != "NAFA")  %>%  #removes all NAFA that are of unknown origin or IS
  dplyr::filter (FvsB != "remove") %>%  
  pivot_wider(names_from = "FvsB", values_from = "sum_groups") %>% 
  left_join(meta_M1 %>% filter (roots_soil =="roots") %>%  select (!sampleID), by = "ID") %>% ### only the meta for the roots are useable here,as they contain info on DW!!
  filter (!is.na(Fungi))  %>% 
  dplyr::rename(AMF= Fungi) %>%  
  left_join((adiv_richness_M1))

## the AMF column was  called Fungi, but it contains only the AMF biomarker results, nenamed as AMF (hyphae!!!!!

## LM instead of LMM??? not helpful result of model selection
# model selection with comparison by AIC and anova.
# lm_Bact_AMF1  <- lm (Bacteria ~ AMF + DW_roots + DW_above + PlaSpe + Observed, data = Bact_to_AMF_PLFA)
# # 
# lm_Bact_AMF2  <- lm (Bacteria ~ AMF , data = Bact_to_AMF_PLFA)
# 
# #lm_Bact_AMF3  <- lm (Bacteria ~ AMF + PlaSpe, data = Bact_to_AMF_PLFA)
# lm_Bact_AMF6  <- lm (Bacteria ~ AMF + DW_roots* DW_above + PlaSpe , data = Bact_to_AMF_PLFA)
# 
# lm_Bact_AMF6 %>% 
#   as_flextable () %>%  
#   colformat_double(digits = 2) %>%  
#   set_formatter(p.value = function(x) {
#     formatC(x, format = "e", digits = 3)
#   }) %>% 
#   add_header_lines ("Model: Soil bacterial biomass ~ soil AMF hyphal biomass + plant species + root biomass * aboveground biomass") %>% 
#   autofit ()
# lm_Bact_AMF4  <- lm (Bacteria ~ AMF + DW_roots  , data = Bact_to_AMF_PLFA)
# lm_Bact_AMF5  <- lm (Bacteria ~ AMF + DW_roots +  PlaSpe, data = Bact_to_AMF_PLFA)



### random model selection
# lmerB1 <- lmer (Bacteria ~AMF +  DW_roots * DW_above+(1|PlantFamily/PlaSpe),  data = Bact_to_AMF_PLFA , REML = T)
# lmerB2<-  lmer (Bacteria ~AMF  + DW_roots *DW_above + (1|PlaSpe),  data = Bact_to_AMF_PLFA  , REML = T)
# lmerB3 <-  lmerTest::lmer (Bacteria ~AMF   + DW_roots + DW_above +(1|PlantFamily),  data = Bact_to_AMF_PLFA )
# lmLmerB1 <-  gls  (Bacteria ~AMF   + DW_roots * DW_above ,  data = Bact_to_AMF_PLFA , method = "REML")
# lmerB4 <-  lmer (Bacteria ~  AMF +  DW_above+ (DW_roots|PlaSpe),  data = Bact_to_AMF_PLFA )
# lmerB5 <-  lmer  (Bacteria ~AMF   +DW_roots  +  (DW_above|PlantFamily/PlaSpe),  data = Bact_to_AMF_PLFA )
# lmerB6 <-  lmer (Bacteria ~ AMF +  DW_above+ (DW_roots|PlantFamily/PlaSpe),  data = Bact_to_AMF_PLFA )
# lmerB7 <- lmer  (Bacteria ~AMF   + DW_roots * DW_above + (DW_roots * DW_above|PlantFamily),  data = Bact_to_AMF_PLFA , REML = T)


#anova (lmerB1, lmerB2, lmerB3, lmLmerB1, lmerB4, lmerB5, lmerB6)

# b3 is best random
# best model: random effect 1|PlantFamily

# #Results for all fixed effects
# tidy (lmerBtoAMF2) %>%  flextable()

# stepwise approach for fixed effects

#selectedmdlPLFA_hyphalAMF  <- step (lmerB3)
LMM_PLFA_hyphalAMF <-  lmerTest::lmer  (Bacteria ~AMF   + (1|PlantFamily),  data = Bact_to_AMF_PLFA )

tidy (LMM_PLFA_hyphalAMF) %>%  flextable () %>%
  add_header_lines("Model: Bacterial biomass ~ AMF hyphal biomass (soil), random effect = Plant family") %>%
  colformat_double(j = c(4:7), digits =2) %>%
  set_formatter(p.value = function(x) {
    formatC(x, format = "e", digits = 3)
  }) %>%
  autofit ()
# 
# 
# VarianceLMM_hyphal <- as.data.frame (VarCorr (LMM_PLFA_hyphalAMF))
# varianceRandomFamily_hyphal <- round (100 * VarianceLMM_hyphal$vcov[1]/(VarianceLMM_hyphal$vcov[1] + VarianceLMM_hyphal$vcov[2]))

```






#### Examination of the impact of the plants' interaction strategy on the soil microbial community composition

Based on PLFA, the relative abundances of the soil microbial community did not show strong variability among the plant species or based on their interaction niche properties with AMF (Fig. \@ref(fig:plot-perc-microbial-groups)). 
However, metabarcoding revealed that the soil AMF communities varied strongly among the plants  without any notable influence by the plant-AMF interaction niche properties (Fig. \@ref(fig:plot-rel-abu) B). 
In contrast, the analysis of absolute abundances also found notable differences among plants of different interaction niche properties (Fig. \@ref(fig:PLFA-PCA-biplot-groups) and Fig. \@ref(fig:plot-DAA-soil)). In what follows, I describe these results in detail.


PLFA analysis as a proxy for microbial biomass revealed only small variability in the relative abundances of the microbial groups  among the plant species (Fig. \@ref(fig:plot-perc-microbial-groups)). The general biomarkers for bacteria were found to have the highest relative abundances in all soils, whereas AMF constitute the smallest amounts of the microbial biomass.


```{r plot-perc-microbial-groups, fig.cap= "Percentage composition of microbial communities in soil. Determined by phospholipid fatty acid analysis (PLFA). For the bacterial fraction, both unspecific fatty acid biomarkers for bacteria and specific biomarkers for Gram-positive bacteria (=GP), Gram-negative bacteria (GN) and Actinobacteria (order of GP) are depicted.  AMF = arbuscular mycorrhizal fungi.", fig.height=6, fig.width=8, dpi = 300}


# Plot percentage microbial groups per pla spe
p <- PL_FA_conc_Soil %>%  
    filter (Biom2Soil != "IS" ) %>%  
    filter (Biom2Soil != "NAFA") %>% 
    left_join(meta_M1) %>% 
   # group_by( PlantSpeciesfull, Biom2Soil) %>%
    #summarize (concFA= mean (conc_FA), SD = sd(conc_FA)) %>%  
  mutate (Biom2Soil = fct_relevel(Biom2Soil, "AMF" , "Fungi", "Actinobacteria","gram.pos","gram.neg","bacteria")) %>% 
    ggplot (aes(x=reorder (PlantSpeciesfull,M1_M2_order),  y=  conc_FA, fill = as.factor(Biom2Soil )))+  
    geom_bar(stat="identity", position = "fill" )
 
p + scale_fill_manual(values =c("AMF" = "#83A552FF" ,"Fungi" = "#3D4928FF",  "Actinobacteria"= "#7B201FFF",   "gram.pos" ="#B19F8EFF","gram.neg"= "#8C5228FF", "bacteria"= "#4E0B0CFF" ), labels = c("AMF", "Fungi", "Actinobacteria",  "GP", "GN", "bacteria")) +
  scale_y_continuous(labels = scales::percent) +
  labs(x= "Plant species", y = "Relative abundance of soil microbial groups in %" , fill = " Microbial group") +
   theme_bw () +
  theme(axis.text.x= element_text(family= "sans", face= "italic", size = 11, angle = 45, hjust = 1 )) +
  theme (axis.text.y=element_text(family= "sans", size = 11))  +
  theme (axis.title = element_text(family = "sans", size = 12 ),  
         legend.title=element_text(size=11), 
         legend.text=element_text(size=10))

## Table microbial groups####
Table_micr_groups <- PL_FA_conc_Soil %>%  
  filter (Biom2Soil != "IS" ) %>%  
  filter (Biom2Soil != "NAFA") %>% 
  left_join(meta_M1) %>% 
  group_by(PlantSpeciesfull, Biom2Soil) %>%
  summarize (concFA= mean (conc_FA), SD = sd(conc_FA)) %>%  
    mutate_if(is.numeric, round, 2) %>% 
    unite("mean ± SD", concFA:SD, sep=" ± ")  %>%  
  pivot_wider(Biom2Soil, values_from = "mean ± SD",names_from= PlantSpeciesfull) %>%  
  dplyr::rename ("Microbial group" = Biom2Soil) %>%   
  flextable ()

```

To include information of absolute values for the PLFAs (Fig. \@ref(fig:PLFA-PCA-biplot-FA)) and for the microbial groups (Fig. \@ref(fig:PLFA-PCA-biplot-groups)), PCA was conducted based on the 17 PLFAs detected in the soil samples. 



```{r PLFA-PCA-biplot-FA, fig.cap= "Principal component analysis (PCA) of phospholipid fatty acid (PLFA) data per soil sample. A) PCA-biplot of PLFAs. Arrows show the loadings for the fatty acids. Dots are samples color coded by the grouping of similar interaction niche properties. Large dots are the centroids of each group. Group A = plants of low alpha-diversities (A. millefolium, C. intybus), Group B = plant species of high alpha- and beta-diversities (P. lanceolata, S. arundinaceus, P. cita), Group C = plant species of high alpha-diversities and low beta-diversities (B. willdenowii, A. capillaris, H. lanatus), SC = soil control. Shapes indicates the plant families. Ellipses are 95% confidence ellipses. The first two dimensions (PC 1 and 2) explain about 77% of the variation. B) - D) Contributions in percentage of the variables to the first three principal components. The dotted red line indicates the expected average contribution, fatty acids with a larger than expected contribution can be interpreted as important contributions.", include =T, fig.height= 8, fig.width=10, dpi= 300}
## Use PCA for biplot
#http://www.sthda.com/english/articles/31-principal-component-methods-in-r-practical-guide/112-pca-principal-component-analysis-essentials/
 
#Variante 1 FAMEs
PLFA_PCA1  <- PL_FA_conc_Soil %>% 
   select (!ID) %>%  select (!FvsB) %>%  
  bind_rows(PL_FA_conc_control %>%  select (sampleID, Name, conc_FA, Biom2Soil )) %>% 
  filter (Biom2Soil !="IS") %>%
  filter (Biom2Soil != "NAFA") %>%
  #filter (Biom2Soil!="bacteria") %>%  ## oder nur 18:0 raus?
  select  (sampleID, conc_FA, Name) %>%
  pivot_wider(names_from = Name, values_from = conc_FA, values_fn = sum) %>%
   left_join(meta_M1Wcontrol %>%  select (sampleID, PlantSpeciesfull, PlaSpe, PlantFamily))  %>% 
  mutate (group = case_when (PlantFamily == "Asteraceae" ~ "A", PlaSpe == "PlaLan" ~ "B" ,
                             PlaSpe ==  "PoaCit" ~ "B" ,PlaSpe ==  "SchAru" ~ "B" ,PlaSpe == "BroWil"~ "C",
                             PlaSpe ==  "AgrCap"~ "C", PlaSpe ==  "HolLan"~ "C", PlaSpe =="SoiCon"~ "SC"))


#calculate PCA
  PLFA_PCA.res1 <- PCA (PLFA_PCA1[,2:22], quali.sup = c(18,19,20,21), graph = F, scale.unit =T)

PCA_arrows1<-
  PLFA_PCA.res1$var$coord %>%  as_tibble (rownames = "FAs") %>% 
  dplyr::rename("D1end" = "Dim.1", "D2end"= "Dim.2")


PCA_PLFA_data1   <- PLFA_PCA.res1$ind$coord  %>%  as_tibble() %>% cbind (PLFA_PCA1)
## get spiderplot data #
center <- 
  aggregate(cbind (Dim.1, Dim.2) ~ group, data = PCA_PLFA_data1, FUN= mean) %>% 
  dplyr::rename("cDim.1" = "Dim.1", "cDim.2" = "Dim.2") 
PCA_PLFA_data1 <- PCA_PLFA_data1 %>% left_join (center)

PCA_eig_Dim1 <- round (PLFA_PCA.res1$eig[1,2],1)

PCA_eig_Dim2 <- round (PLFA_PCA.res1$eig[2,2],1)

plot1  <-
  PCA_PLFA_data1 %>% 
  ggplot (aes (x = Dim.1, y = Dim.2, color = group, shape = PlantFamily)) + 
  geom_point (size =3) +
  geom_segment( mapping = aes (xend = Dim.1, yend = Dim.2, color = group)) + 
  geom_point ( aes (x= cDim.1, y = cDim.2), size = 7, shape = 18) +
  geom_hline(yintercept = 0, lty = 2, color = "grey", alpha = 0.9) + 
  geom_vline(xintercept = 0, lty = 2, color = "grey", alpha = 0.9) + 
  stat_ellipse(aes (x= Dim.1, y = Dim.2, color = group, type = "norm"))  +   # this is 95%confidence
  geom_segment(data = PCA_arrows1, aes (x=0, xend= D1end*5, y = 0, yend = D2end*5), 
               arrow = arrow(length = unit(0.3, "picas")), color = "grey", inherit.aes = F)  +
  geom_text_repel (data = PCA_arrows1, aes (x  = D1end*6, y = D2end*6, label = FAs), 
                   color = "black", inherit.aes = F , force = 0.6) + 
  theme_minimal() + 
  xlab(label = "PC1 (67.0 %)") +
  ylab ("PC2 (10.5 %)") + 
  guides (color= guide_legend( "Grouping by interaction niche properties"), shape= guide_legend(NULL)) +
  theme (legend.position = "bottom")
  
  
 #lots- contribution to dimension 
  plotdim1 <- fviz_contrib(PLFA_PCA.res1, choice = "var", axes = 1, font.main = c(size =12 ), title= "Contribution of variables to PC1")
plotdim2 <- fviz_contrib(PLFA_PCA.res1, choice = "var", axes = 2, font.main = c(size =12 ), title= "Contribution of variables to PC2")
plotdim3 <- fviz_contrib(PLFA_PCA.res1, choice = "var", axes = 3, font.main = c(size =12 ), title= "Contribution of variables to PC3")

# Plot PCA
 # plot1 <-  fviz_pca_biplot (PLFA_PCA.res1, col.ind = PLFA_PCA1$PlantFamily, legend.title = "Plant families", addEllipses = T,  
 #                            #ellipse.type = "confidence", 
 #                            repel=T, palette = c("Asteraceae"= "#edc948" ,  "Plantaginaceae"= "#4e79a7",
 #                                                                              "Poaceae" = "#7F9A65", "Soil Control" ="black" ), 
 #                            col.var = "darkslategrey", legend = "bottom" )
 
 #arrage plots into one fig
 ggarrange (plot1,
           ggarrange (plotdim1, plotdim2, plotdim3, nrow = 3, ncol= 1, labels= c("B", "C", "D")), 
           nrow=1, ncol=2, widths =  c(2, 1), labels= "A") 
```

The first two principal components explained `r round(PLFA_PCA.res1$eig[2,3],1)` % of the total variance of the soil microbes' FA composition. Several biomarker PLFAs from the Gram-positive and Gram-negative bacteria contributed heavily to PC1 (Fig. \@ref(fig:PLFA-PCA-biplot-FA)), and the saturated PLFA 16:0, a general biomarker for bacteria, also weighted heavily on PC1. 
PC2 was dominated by the saturated PLFA biomarkers 17:0 and 18:0, and both the iso- and anteiso-15:0 PLFA. Grouping the samples by their interaction niche properties with AMF revealed that group A (*A. millefolium*, *C. intybus*) separated somewhat from the other plant species along PC1, suggesting differences in the GP to GN ratios between these groups. 
Groups B and C overlapped but a slight separation of the Plantaginaceae from most other samples along PC2 was detected.



```{r PLFA-PCA-biplot-groups, fig.cap= "Principal component analysis (PCA) of microbial groups as determined by phospholipid fatty acid (PLFA) analysis. A) PCA-biplot of the microbial groups bacteria, fungi, arbuscular mycorrhizal fungi (AMF), Gram-positive bacteria (GP), Actinobacteria and Gram-negative bacteria (GN). Arrows show the loadings for each group. Dots are samples colour coded by the grouping of similar interaction niche properties. Large dots are the centroids of each group. Group A = plants of low alpha-diversities (A. millefolum, C. intybus), Group B = plant species of high alpha- and beta-diversities (P. lanceolata, S. arundinaceus, P. cita), Group C = plant species of high alpha-diversities and low beta-diversities (B. willdenowii, A. capillaris, H. lanatus), SC = soil control. Shapes indicates the plant families. Ellipses are 95% confidence intervals. The first two dimensions (PC1 and 2) explain about 87% of the variation. B) - D) Contributions in percentage of the variables to the first three principal components (PC). The dotted red line indicates the expected average contribution. Microbial groups with a larger than expected contribution can be interpreted as important contributions.", fig.height = 8, fig.width=10, dpi= 300}

# Variante 2 biomarker groups 
PLFA_PCA2 <-   
    PL_FA_conc_Soil %>%
    filter (Biom2Soil !="IS") %>%
    filter (Biom2Soil != "NAFA") %>%  
  group_by(sampleID, Biom2Soil) %>%  
  summarise (group = sum (conc_FA)) %>%  
  bind_rows(PL_FA_conc_control %>%  group_by(Biom2Soil, sampleID) %>%  summarise (group = sum (conc_FA)) %>%  filter (Biom2Soil !="NAFA" & Biom2Soil != "IS")) %>% 
  left_join (meta_M1Wcontrol %>%  select (sampleID, PlantFamily, PlaSpe, PlantSpeciesfull)) %>%  
  pivot_wider(names_from = Biom2Soil, values_from = group)  %>% 
  dplyr::rename (GP = gram.pos, GN = gram.neg) %>% 
  mutate (group = case_when (PlantFamily == "Asteraceae" ~ "A", PlaSpe == "PlaLan" ~ "B" , PlaSpe ==  "PoaCit" ~ "B" ,PlaSpe ==  "SchAru" ~ "B" ,PlaSpe == "BroWil"~ "C", PlaSpe ==  "AgrCap"~ "C", PlaSpe ==  "HolLan"~ "C", PlaSpe =="SoiCon"~ "SC"))

PLFA_PCA.res2 <- PCA (PLFA_PCA2[,5:11], graph = F, quali.sup = 7, scale.unit =T) # oder hier FALSE for scale unit?


PCA_arrows<-
  PLFA_PCA.res2$var$coord %>%  as_tibble (rownames = "microbes") %>% 
  dplyr::rename("D1end" = "Dim.1", "D2end"= "Dim.2")


PCA_PLFA_data   <- PLFA_PCA.res2$ind$coord  %>%  as_tibble() %>% cbind (PLFA_PCA2)


# get centroids
center1 <- 
  aggregate(cbind (Dim.1, Dim.2) ~ group, data = PCA_PLFA_data, FUN= mean) %>% 
  dplyr::rename("cDim.1" = "Dim.1", "cDim.2" = "Dim.2") 
PCA_PLFA_data <- PCA_PLFA_data %>%  left_join (center1)
  
PCA_eig_Dim1 <- round (PLFA_PCA.res2$eig[1,2],1)

PCA_eig_Dim2 <- round (PLFA_PCA.res2$eig[2,2],1)

plot2  <-
  PCA_PLFA_data %>%  ggplot (aes (x = Dim.1, y = Dim.2, color = group, shape = PlantFamily)) + 
  geom_point (size =3) +
  geom_point (aes (x = cDim.1, y = cDim.2),  shape = 18, size = 7) +
  geom_hline(yintercept = 0, lty = 2, color = "grey", alpha = 0.9) + 
  geom_vline(xintercept = 0, lty = 2, color = "grey", alpha = 0.9) + 
  stat_ellipse(aes (x= Dim.1, y = Dim.2, color = group), type = "norm")  +   # this is 95%confidence
  geom_segment(data = PCA_arrows, aes (x=0, xend= D1end*3, y = 0, yend = D2end*3), 
               arrow = arrow(length = unit(0.3, "picas")), color = "grey", inherit.aes = F)  +
  geom_text (data = PCA_arrows, aes (x  = D1end*3.3, y = D2end*3.3, label = microbes), color = "black", inherit.aes = F ) + 
  theme_minimal() + 
  xlab(label = "PC1 (78.3 %)") +
  ylab ("PC2 (8.7 %)") + 
  guides (color= guide_legend( "Grouping by interaction niche properties"), shape = guide_legend(NULL)) +
  theme (legend.position = "bottom")

  
  ##wäre ggf noch schön die Plant speices zu unterscheiden muss noc in anderen ggplot system gemacht werden
  
plotdim21 <- fviz_contrib(PLFA_PCA.res2, choice = "var", axes = 1, font.main = c(size =12 ), title= "Contribution of variables to PC1")
plotdim22 <- fviz_contrib(PLFA_PCA.res2, choice = "var", axes = 2, font.main = c(size =12 ), title= "Contribution of variables to PC2")
plotdim23 <- fviz_contrib(PLFA_PCA.res2, choice = "var", axes = 3, font.main = c(size =12 ), title= "Contribution of variables to PC3")

# Plot PCA
 # plot2 <-  fviz_pca_biplot (PLFA_PCA.res2, col.ind = PLFA_PCA2$PlantFamily, legend.title = "Plant families", 
 #                            addEllipses = T,     #ellipse.level = 0.95, ellipse.type = "confidence", 
 #                            repel=T, palette = c("Asteraceae"= "#edc948" ,  "Plantaginaceae"= "#4e79a7","Poaceae" = "#7F9A65", "Soil Control" = "black" ), col.var = "darkslategrey", legend = "bottom" )
 
 #arrange plots into one fig
 ggarrange (plot2,
           ggarrange (plotdim21, plotdim22, plotdim23, nrow = 3, ncol= 1, labels= c("B", "C", "D")), 
           nrow=1, ncol=2, widths =  c(2, 1), labels= "A") 
```

Conducting PCA per biomarker group showed a separation of the microbial community composition on the plant family level (Fig. \@ref(fig:PLFA-PCA-biplot-groups)). PC1 and PC2 explained `r round(PLFA_PCA.res2$eig[2,3],1)` % of the total variation. The contributions of all groups to PC1 were similarly strong, with a slightly stronger contribution of the biomarkers for Gram-negative bacteria and for bacteria in general. In PC2, however, the AMF biomarker showed a strong weight, followed by the Actinomycetales biomarker, which are an order of GP bacteria, and other GP bacterial biomarkers. 
In both PCAs, group A and group C separated along PC1 and PC2, demonstrating the differences of their soil bacterial composition, especially regarding GP and GN bacteria. 


### AMF richness and community composition in plant roots and their associated soils
Comparisons of AMF diversities in plant roots *versus* their associated soil indicate that the richness, which is one of the main determinants of the interaction niche properties, does not determine the composition of the AMF community in the soil (Fig. \@ref(fig:plot-rel-abu)). 
In agreement with the results from Chapter 2, the Asteraceae had a smaller AMF richness than the other tested plants, whereas the Poaceae and the Plantaginaceae hosted similar numbers of AMF in their roots. However, only *H. lanatus*, having the highest mean AMF richness in roots, was significantly different from both Asteraceaen species.

In comparison to the roots, the AMF richnesses in the soil of different plant species were overall lower and more similar to each other. One notable exception was *S. arundinaceus*, which had a high average AMF richness in its associated soil, similar to its root AMF richness. 
The high $\alpha$-diversity of AMF in soil samples of *S. arundinaceus* was surprising, but a closer look at each of its replicates could not identify outliers, nor a mistake in sample preparation. 
Taking the AMF identity into account (Fig. \@ref(fig:plot-rel-abu) B), a different picture emerges when comparing the soils to their associated root samples. In the roots, the *Acaulospora* clearly dominated the AMF communities. 
In contrast, the soil contains more evenly mixed AMF communities including several genera that were not detected in the root samples, like the genus *Dominikia.* Other genera were only found in some plant species, for example *Rhizophagus* ASVs were only detected in *S. arundinaceus* and *A. capillaris*. There was no clear relationship between the plant root AMF community and that of their associated soils. For example, in six of the eight plant species *Funelliformis* ASVs were detected but only in two of their associated soils (*B. willdenowii*, *C. intybus*), indicating that the detected soil AMF community does not necessarily mirror the AMF community actively interacting with the plant roots. 
However, the overall smaller number of AMF detected in the soil could also indicate a difficulty of detecting the Glomeromycotina within the much larger fungal community, as they account for only a small fraction of the total fungal community in the soils. 
Interestingly, the proportions of the AMF genera in the soil differed from that in the roots. In the soil, the relative abundance of the root-dominating *Acaulospora* decreased while other taxa, for example, the *Cetraspora* and *Chlaroideoglomus* increased in their relative abundance. The comparison of the control soil with the plant-associated soils shows a similar trend, with a stronger increase of taxa other than the *Acaulospora.* 

```{r plot-diversity-uniqueASVs, include = F, fig.cap= "Measures of alpha-diversity. A) ASV richness per plant species. Horizontal lines show median values, boxes denote the interquartile range (IQR), while whiskers show 1.5$*$IQR ranges. Outliers are indicated by dots. Significance levels (are)Games-Howell test): * at p<0.05, ** at p<0.005, *** at p<0.0005, **** at p<0.0001.  B) Unique AMF ASVs per plant species. ASVs are colored by AMF genus. C) Relative read abundances of the AMF genera by plant species.Note that the number of replicates influences the number of unique ASVs per treatment: ", fig.height= 8, fig.width= 6}

# diversity calculated 
adiv_richness  <- estimate_richness(ps_M1_Glo_rar, measures = c("Observed")) %>% 
  as_tibble (rownames = "sampleID")

## signif different?
KW_richness_roots <- adiv_richness %>%  left_join(meta_M1Wcontrol) %>% filter (roots_soil =="roots") %>%  kruskal_test(Observed ~ PlaSpe )
KW_richness_soil <- adiv_richness %>%  left_join(meta_M1Wcontrol) %>% filter (roots_soil =="soil") %>%  kruskal_test(Observed ~ PlaSpe )

#Games Howell
GW_richness<- adiv_richness %>%  left_join(meta_M1Wcontrol)  %>% group_by (roots_soil) %>%  games_howell_test(Observed ~ PlantSpeciesfull) 
df0 <- GW_richness %>%  filter (!p.adj.signif == "ns")  %>%   filter (!p.adj.signif == "")  %>%  
  dplyr::mutate(groups = purrr::pmap(.l = list(group1, group2), .f = c)) %>%
  dplyr::arrange(group1) 


GW_richness_roots <- adiv_richness %>%  left_join(meta_M1Wcontrol)  %>% filter (roots_soil =="roots")%>%  games_howell_test(Observed ~ PlantSpeciesfull) 
df <- GW_richness_roots %>%  filter (!p.adj.signif == "ns")  %>%   filter (!p.adj.signif == "")  %>%  
  dplyr::mutate(groups = purrr::pmap(.l = list(group1, group2), .f = c)) %>%
  dplyr::arrange(group1) 
                 
GW_richness_soil <- adiv_richness %>%  left_join(meta_M1Wcontrol)  %>% filter (roots_soil =="soil")%>%  games_howell_test(Observed ~ PlantSpeciesfull)
df2 <- GW_richness_soil  %>%  filter (!p.adj.signif == "ns")  %>%   filter (!p.adj.signif == "")  %>%  
  dplyr::mutate(groups = purrr::pmap(.l = list(group1, group2), .f = c)) %>%
  dplyr::arrange(group1) 
                                                                                                               

#adiv_richness Plot### only for the roots - includes the significant levels
p01 <- adiv_richness %>% 
  left_join(meta_M1Wcontrol, by="sampleID") %>% 
  filter (Observed!=0) %>% 
  group_by (roots_soil,PlantSpeciesfull) %>% 
  mutate(meannumberASVs=mean(Observed)) %>% 
  select (sampleID, Observed, PlantSpeciesfull, meannumberASVs, PlantFamily, M1_M2_order)  %>%   ## ggf sollte man den alten myorder hier rauswerfen
  pivot_longer(cols=c(Observed), names_to ="diversity_measure", values_to = "div", values_drop_na = TRUE) %>% 
  arrange(meannumberASVs)   %>% 
  ggplot(aes(y=reorder(PlantSpeciesfull,-M1_M2_order), x=div, fill=PlantFamily))  +
  geom_boxplot()  +
  facet_wrap(vars(roots_soil)) +
  theme_light () +
  ylab ("Plant species")+
    xlab ("ASV richness") +
  theme(axis.text.y= element_text(family= "sans", face= "italic", size = 9)) +
  theme (axis.text.x=element_text(family= "sans", size = 9))  +
  theme (axis.title = element_text(family = "sans", size = 11 ),  
         legend.title=element_text(size=11), 
         legend.text=element_text(size=9), 
         legend.position = "right") +
    theme (strip.text = element_text (family = "sans", size = 13, color = "black"))+
    scale_fill_manual(values=c("Asteraceae"= "#edc948" ,
                             "Plantaginaceae"= "#4e79a7",
                             "Poaceae" = "#7F9A65" ), name = "Plant family") +
    ggsignif::geom_signif (    comparisons = df0$groups, 
    annotation = c (df0 %>% pull (p.adj.signif)),
    step_increase = 0.06, vjust = 0.5)

# get rid of wrong annotation

pg01<-ggplot_build(p01) #disassemble plot and obtain information


new<-pg01$data[[2]] %>%  filter (PANEL =="1")
pg01$data[[2]]<-new #swap out the original annotation

q<-ggplot_gtable(pg01) #reassemble the plot

p0 <- ggplotify::as.ggplot(q) #generate new plot
  



# Plot unique number of ASvs 
p1  <- data.frame (otu_table(ps_M1_Glo_rar))  %>%  as_tibble(rownames ="ASV_ID") %>%
  pivot_longer(!ASV_ID, names_to = "sampleID", values_to = "ASV_counts") %>% 
  left_join((tax_table(ps_M1_Glo_rar) %>% data.frame () %>%  as_tibble (rownames = "ASV_ID")), by= "ASV_ID")  %>% 
  filter (ASV_counts!=0)  %>% 
  left_join (meta_M1Wcontrol) %>% 
  group_by(roots_soil,PlantSpeciesfull, M1_M2_order, ASV_ID, Genus)  %>% 
  tally ()  %>% 
  group_by(roots_soil,PlantSpeciesfull,M1_M2_order,  Genus)  %>% 
  tally ()  %>% 
    replace_na(list (Genus="unidentified"))   %>% 
  ggplot (aes(y=n, x= reorder (PlantSpeciesfull, -M1_M2_order), fill = Genus)) +
geom_bar(stat = "identity", position = "stack") + 
  facet_wrap(~roots_soil)

p1 <- p1 + theme_light() + 
  coord_flip () +
  ylab("Unique AMF ASVs per plant species") +
  xlab ("Plant species") +
  theme_light () +
    theme(axis.text.y= element_text(family= "sans", face= "italic", size = 9),  #
          axis.title.y = element_text(family= "sans", size = 11),
          axis.text.x = element_text(family= "sans", size = 9),
          axis.title.x  = element_text(family = "sans", size = 11 ),  
           legend.title =element_text(size=11), 
           legend.text =element_text(size=9)) +
    theme (strip.text = element_text (family = "sans", size = 13, color = "black"))  +
    scale_fill_manual(values = c( "unidentified" ="#C7EAE5", "Archaeospora"  = "#287D8EFF" , 
                                  "Acaulospora" =  "#C1A363" , "Diversispora" = "#993333",
                                  "Funneliformis" = "#F6E8C3", "Cetraspora"  = "#DFC27D", 
                                  "Claroideoglomus" =  "#20A386FF", 
                                  "Glomus" = "#80CDC1" , "Paraglomus" = "#35978F", 
                                  "Scutellospora"= "#01665E", "Dominikia" = "#FF9933", "Rhizophagus" = "black")    )  

# plot relative abundances of AMF genera in percent
p2 <- data.frame (otu_table(ps_M1_Glo_rar))  %>%  as_tibble(rownames ="ASV_ID") %>%
    pivot_longer(!ASV_ID, names_to = "sampleID", values_to = "ASV_counts") %>% 
    left_join((tax_table(ps_M1_Glo_rar) %>% data.frame () %>% as_tibble (rownames = "ASV_ID")), by= "ASV_ID")  %>% 
    filter (ASV_counts!=0)  %>% 
    left_join (meta_M1Wcontrol) %>% 
    group_by(roots_soil,sampleID, Genus) %>%  
  summarise (count_genus_per_sample =sum(ASV_counts)) %>%  # count per sample the number of reads per genus
  left_join(meta_M1Wcontrol %>% select (sampleID, PlantSpeciesfull, M1_M2_order)) %>% 
  group_by(roots_soil,PlantSpeciesfull,M1_M2_order, Genus) %>% 
  summarise (mean_genus = mean (count_genus_per_sample)) %>% # take the mean per PlaSpe of the genera
      replace_na(list (Genus="unidentified"))   %>% 
  ggplot (aes(x= reorder (PlantSpeciesfull, -M1_M2_order), y = mean_genus, fill = Genus)) + 
  geom_bar(position = "fill", stat = "identity") + 
  facet_wrap(~roots_soil) 
p2  <-  p2 + 
  theme_light()+ 
    coord_flip () +
    ylab("Relative abundances of AMF genera in %") +
    xlab ("Plant species") +
    theme(axis.text.y= element_text(family= "sans", face= "italic", size = 9),  #
          axis.title.y = element_text(family= "sans", size = 11),
          axis.text.x = element_text(family= "sans", size = 9),
          axis.title.x  = element_text(family = "sans", size = 11 ),  
           legend.title =element_text(size=11), 
           legend.text =element_text(size=9, face = "italic")) +
    theme (strip.text = element_text (family = "sans", size = 13, color = "black"))  +
    scale_fill_manual(values = c( "unidentified" ="#C7EAE5", "Archaeospora"  = "#287D8EFF" , 
                                  "Acaulospora" =  "#C1A363" , "Diversispora" = "#993333",
                                  "Funneliformis" = "#F6E8C3", "Cetraspora"  = "#DFC27D", 
                                  "Claroideoglomus" =  "#20A386FF", 
                                  "Glomus" = "#80CDC1" , "Paraglomus" = "#35978F", 
                                  "Scutellospora"= "#01665E", "Dominikia" = "#FF9933", "Rhizophagus" = "black"), name = "AMF genus"    )  +
    scale_y_continuous(labels = scales::percent) #+
  #scale_x_discrete(labels = NULL)
pBC <- ggarrange (p1, p2, labels = c("B","C"), nrow = 2, ncol =1, widths =   2, heights = 1.5, legend = "right", common.legend = T)
ggarrange (p0,pBC, labels = c("A", ""), nrow =2, ncol=1, heights = c(2,4), widths = 2 ) 


```

```{r plot-rel-abu, fig.cap= "Measures of alpha-diversity. A) ASV richness per plant species. Horizontal lines show median values, boxes denote the interquartile range (IQR), while whiskers show 1.5$*$IQR ranges. Outliers are indicated by dots. Significance levels are (Games-Howell test): * at p<0.05, ** at p<0.005, *** at p<0.0005, **** at p<0.0001.  B) Relative read abundances of the AMF genera by plant species. ", fig.height= 8, fig.width= 6}

# diversity calculated 
adiv_richness  <- estimate_richness(ps_M1_Glo_rar, measures = c("Observed")) %>% 
  as_tibble (rownames = "sampleID")

## signif different?
KW_richness_roots <- adiv_richness %>%  left_join(meta_M1Wcontrol) %>% filter (roots_soil =="roots") %>%  kruskal_test(Observed ~ PlaSpe )
KW_richness_soil <- adiv_richness %>%  left_join(meta_M1Wcontrol) %>% filter (roots_soil =="soil") %>%  kruskal_test(Observed ~ PlaSpe )

#Games Howell
GW_richness<- adiv_richness %>%  left_join(meta_M1Wcontrol)  %>% group_by (roots_soil) %>%  games_howell_test(Observed ~ PlantSpeciesfull) 
df0 <- GW_richness %>%  filter (!p.adj.signif == "ns")  %>%   filter (!p.adj.signif == "")  %>%  
  dplyr::mutate(groups = purrr::pmap(.l = list(group1, group2), .f = c)) %>%
  dplyr::arrange(group1) 


GW_richness_roots <- adiv_richness %>%  left_join(meta_M1Wcontrol)  %>% filter (roots_soil =="roots")%>%  games_howell_test(Observed ~ PlantSpeciesfull) 
df <- GW_richness_roots %>%  filter (!p.adj.signif == "ns")  %>%   filter (!p.adj.signif == "")  %>%  
  dplyr::mutate(groups = purrr::pmap(.l = list(group1, group2), .f = c)) %>%
  dplyr::arrange(group1) 
                 
GW_richness_soil <- adiv_richness %>%  left_join(meta_M1Wcontrol)  %>% filter (roots_soil =="soil")%>%  games_howell_test(Observed ~ PlantSpeciesfull)
df2 <- GW_richness_soil  %>%  filter (!p.adj.signif == "ns")  %>%   filter (!p.adj.signif == "")  %>%  
  dplyr::mutate(groups = purrr::pmap(.l = list(group1, group2), .f = c)) %>%
  dplyr::arrange(group1) 
                                                                                                               

#adiv_richness Plot### only for the roots - includes the significant levels
p01 <- adiv_richness %>% 
  left_join(meta_M1Wcontrol, by="sampleID") %>% 
  filter (Observed!=0) %>% 
  group_by (roots_soil,PlantSpeciesfull) %>% 
  mutate(meannumberASVs=mean(Observed)) %>% 
  select (sampleID, Observed, PlantSpeciesfull, meannumberASVs, PlantFamily, M1_M2_order)  %>%   ## ggf sollte man den alten myorder hier rauswerfen
  pivot_longer(cols=c(Observed), names_to ="diversity_measure", values_to = "div", values_drop_na = TRUE) %>% 
  arrange(meannumberASVs)   %>% 
  ggplot(aes(y=reorder(PlantSpeciesfull,-M1_M2_order), x=div, fill=PlantFamily))  +
  geom_boxplot()  +
  facet_wrap(vars(roots_soil)) +
  theme_light () +
  ylab ("Plant species")+
    xlab ("ASV richness") +
  theme(axis.text.y= element_text(family= "sans", face= "italic", size = 9)) +
  theme (axis.text.x=element_text(family= "sans", size = 9))  +
  theme (axis.title = element_text(family = "sans", size = 11 ),  
         legend.title=element_text(size=11), 
         legend.text=element_text(size=9), 
         legend.position = "right") +
    theme (strip.text = element_text (family = "sans", size = 13, color = "black"))+
    scale_fill_manual(values=c("Asteraceae"= "#edc948" ,
                             "Plantaginaceae"= "#4e79a7",
                             "Poaceae" = "#7F9A65" ), name = "Plant family") +
    ggsignif::geom_signif (    comparisons = df0$groups, 
    annotation = c (df0 %>% pull (p.adj.signif)),
    step_increase = 0.06, vjust = 0.5)

# get rid of wrong annotation

pg01<-ggplot_build(p01) #disassemble plot and obtain information


new<-pg01$data[[2]] %>%  filter (PANEL =="1")
pg01$data[[2]]<-new #swap out the original annotation

q<-ggplot_gtable(pg01) #reassemble the plot

p0 <- ggplotify::as.ggplot(q) #generate new plot
  



# plot relative abundances of AMF genera in percent
p2 <- data.frame (otu_table(ps_M1_Glo_rar))  %>%  
  as_tibble(rownames ="ASV_ID") %>%
    pivot_longer(!ASV_ID, names_to = "sampleID", values_to = "ASV_counts") %>% 
    left_join((tax_table(ps_M1_Glo_rar) %>% data.frame () %>% as_tibble (rownames = "ASV_ID")), by= "ASV_ID")  %>% 
    filter (ASV_counts!=0)  %>% 
    left_join (meta_M1Wcontrol) %>% 
    group_by(roots_soil,sampleID, Genus) %>%  
  summarise (count_genus_per_sample =sum(ASV_counts)) %>%  # count per sample the number of reads per genus
  left_join(meta_M1Wcontrol %>% select (sampleID, PlantSpeciesfull, M1_M2_order)) %>% 
  group_by(roots_soil,PlantSpeciesfull,M1_M2_order, Genus) %>% 
  summarise (mean_genus = mean (count_genus_per_sample)) %>% # take the mean per PlaSpe of the genera
  replace_na(list (Genus="unidentified"))   %>% 
  ggplot (aes(x= reorder (PlantSpeciesfull, -M1_M2_order), y = mean_genus, fill = Genus)) + 
  geom_bar(position = "fill", stat = "identity") + 
  facet_wrap(~roots_soil) 
p2  <-  p2 + 
  theme_light()+ 
    coord_flip () +
    ylab("Relative abundances of AMF genera in %") +
    xlab ("Plant species") +
    theme(axis.text.y= element_text(family= "sans", face= "italic", size = 9),  #
          axis.title.y = element_text(family= "sans", size = 11),
          axis.text.x = element_text(family= "sans", size = 9),
          axis.title.x  = element_text(family = "sans", size = 11 ),  
           legend.title =element_text(size=11), 
           legend.text =element_text(size=9, face = "italic")) +
    theme (strip.text = element_text (family = "sans", size = 13, color = "black"))  +
    scale_fill_manual(values = c( "unidentified" ="#C7EAE5", "Archaeospora"  = "#287D8EFF" , 
                                  "Acaulospora" =  "#C1A363" , "Diversispora" = "#993333",
                                  "Funneliformis" = "#F6E8C3", "Cetraspora"  = "#DFC27D", 
                                  "Claroideoglomus" =  "#20A386FF", 
                                  "Glomus" = "#80CDC1" , "Paraglomus" = "#35978F", 
                                  "Scutellospora"= "#01665E", "Dominikia" = "#FF9933", "Rhizophagus" = "black"), name = "AMF genus"    )  +
    scale_y_continuous(labels = scales::percent) #+
  #scale_x_discrete(labels = NULL)

ggarrange (p0, p2, labels = "AUTO", nrow=2, ncol =1)

```


```{r plot-DAA-soil, include = T, fig.cap = "Alteration of the soil AMF community by eight different plant species. The plot panels show the results from the differential abundance analysis (DAA) between soils associated with eight different plant species and control soil. Each point represents one ASV, with the color indicating the AMF order, and the size the mean relative abundance of the ASV in regards to all AMF ASVs of the respective plant. The AMF taxa are organised by the phylogenetic tree of the identified AMF genera (left). The log2-fold change between the soils describes the log-ratio of the respective ASV between the two soils. Significant log2-fold changes are indicated by non-transparent coloring (p-value < 0.05).", fig.height= 8, fig.width=13}


### Add a new meta column to the data
# combination of PlaSpe and root/soil, e.g. AchMil-soil, AchMil-roots
# for use as grouping factor in edgeR!
ps_edgeR <- ps_M1_allASVs

new_sampledata <- 
  sample_data (ps_edgeR) %>% 
  data.frame () %>% 
  unite (col =PlaSpe_RS, c(PlaSpe, roots_soil), sep= "_" ) %>% 
  sample_data()
sample_data(ps_edgeR)  <- new_sampledata


# at genus level - needed for Tree building and taxon names
ps_edgeR_genus <- tax_glom(ps_edgeR, taxrank = "Genus", NArm = F)

#Object for edgeR ##
# rows OTUs, columns = samples
# get the count table as  df
df <- otu_table(ps_edgeR) %>%  
  data.frame ()

#Get taxon table for the ASVs - needed as meta table for the DAA results
taxa_edgeR <- 
  tax_table (ps_edgeR) %>% 
  data.frame () %>%  
  mutate(GenusLabel = ifelse(!is.na(Genus), paste(Genus), 
                             ifelse(!is.na(Family), paste('Unid. ', Family, sep = ""), 
                                    ifelse(!is.na(Order), paste('Unid. ', Order, sep = ""),
                                           ifelse(!is.na(Class), paste('Unid. ', Class, sep = ""), paste("Unid. ", Phylum, sep = "")))))) %>% 
  as_tibble (rownames = "ASV_ID" ) %>%  
  select (ASV_ID, GenusLabel,Phylum, Class, Family, Genus)

## Grouping : PlaSpe in roots and PlaSpe in soil = 16 treatments
group = factor (sample_data (ps_edgeR)$PlaSpe_RS)

## prepare edgeR object y
y <- DGEList(counts = df, group = group)
DAA_dim_before_filter <- dim(y)
# Filtering ####
#filter small OTUs- that removes about 2000 OTUs
keep <- filterByExpr(y)
# filter the dataset for the ASVs to keep
y <- y[keep, , keep.lib.sizes=FALSE]
DAA_dim_after_filter <-  dim (y)

#Normalisation####
#The calcNormFactors function normalises the library sizes by finding a set of scaling factors
#for the library sizes that minimizes the log-fold changes between the samples for most genes.
y<- calcNormFactors(y)
#y$samples
#A normalization factor below one indicates that a small number of high count genes
#are monopolizing the sequencing, causing the counts for other genes to be lower than would
#be usual given the library size.
y$samples$group  <- relevel (y$samples$group, ref = "SoiCon_soil")

design <- model.matrix (~ group , data = y$samples)

#Dispersions - GLM####
#For general experiments (with multiple factors), edgeR uses the Cox-Reid profile-adjusted
#likelihood (CR) method in estimating dispersions [25].
y <- estimateDisp(y, design)

#add sample information
y$samples$PlaSpe <- factor (sample_data (ps_edgeR)$PlaSpe)
y$samples$PlantSpeciesfull <- factor (sample_data (ps_edgeR)$PlantSpeciesfull)
y$samples$PlantFamily <- factor (sample_data (ps_edgeR)$PlantFamily)


# relevel reference level

# DESIGN: Decide on grouping ####
# see Law 2020, A guide to creating design matrices ...
## Model design 

colnames(design) <- levels(y$samples$group)
#check the names of the columns with 
# colnames(design)
# to get the needed comparisons (either by writing contrasts or by direct use of the coefficient's results)

# testing for differentially abundant (DA) OTUs
#While the likelihood ratio test is a more obvious choice for inferences with GLMs, the QL
#F-test is preferred as it reflects the uncertainty in estimating the dispersion for each gene. It
#provides more robust and reliable error rate control when the number of replicates is small.
#The QL dispersion estimation and hypothesis testing can be done by using the functions
#glmQLFit() and glmQLFTest().


#GLM ####
fit <- glmQLFit(y, design)

#GLM F-test #####
#compare the coefficients of the glm 
# i.e. compare treatments
AchMil_soil <- glmQLFTest(fit, coef=3) 
#Soil PlaSpe results####
# Ach Mil data soil ##
DAA_AchMil_soil <-
  AchMil_soil$table %>%  
  as_tibble (rownames = "ASV_ID") %>% 
  left_join (taxa_edgeR) %>% 
  filter (Phylum == "Glomeromycota" ) %>% 
  add_column (PlaSpe = "AchMil", roots_soil = "soil") %>% ### change here 
  mutate (Sign = case_when(PValue<=0.05 ~ "sign.", PValue > 0.05 ~ "ns")) 

# Agr Cap soil ##
# i.e. compare treatments
AgrCap_soil <- glmQLFTest(fit, coef=5) 

# Ach Mil data soil
DAA_AgrCap_soil <-
  AgrCap_soil$table %>%  
  as_tibble (rownames = "ASV_ID") %>% 
  left_join (taxa_edgeR) %>% 
  filter (Phylum == "Glomeromycota" ) %>% 
  add_column (PlaSpe = "AgrCap", roots_soil = "soil") %>% ### change here 
  mutate (Sign = case_when(PValue<=0.05 ~ "sign.", PValue > 0.05 ~ "ns")) 

## BroWIl soil
BroWil_soil <- glmQLFTest(fit, coef=7) 

DAA_BroWil_soil <-
  BroWil_soil$table %>%  
  as_tibble (rownames = "ASV_ID") %>% 
  left_join (taxa_edgeR) %>% 
  filter (Phylum == "Glomeromycota" ) %>% 
  add_column (PlaSpe = "BroWil", roots_soil = "soil") %>% ### change here 
  mutate (Sign = case_when(PValue<=0.05 ~ "sign.", PValue > 0.05 ~ "ns"))

# CicInt data soil


CicInt_soil <- glmQLFTest(fit, coef=9) 

DAA_CicInt_soil <-
  CicInt_soil$table %>%  
  as_tibble (rownames = "ASV_ID") %>% 
  left_join (taxa_edgeR) %>% 
  filter (Phylum == "Glomeromycota" ) %>% 
  add_column (PlaSpe = "CicInt", roots_soil = "soil") %>% ### change here 
  mutate (Sign = case_when(PValue<=0.05 ~ "sign.", PValue > 0.05 ~ "ns")) 

# HolLan data soil


HolLan_soil <- glmQLFTest(fit, coef=11) 

DAA_HolLan_soil <-
  HolLan_soil$table %>%  
  as_tibble (rownames = "ASV_ID") %>% 
  left_join (taxa_edgeR) %>% 
  filter (Phylum == "Glomeromycota" ) %>% 
  add_column (PlaSpe = "HolLan", roots_soil = "soil") %>% ### change here 
  mutate (Sign = case_when(PValue<=0.05 ~ "sign.", PValue > 0.05 ~ "ns")) 

# PlaLan data soil

PlaLan_soil <- glmQLFTest(fit, coef=13) 

DAA_PlaLan_soil <-
  PlaLan_soil$table %>%  
  as_tibble (rownames = "ASV_ID") %>% 
  left_join (taxa_edgeR) %>% 
  filter (Phylum == "Glomeromycota" ) %>% 
  add_column (PlaSpe = "PlaLan", roots_soil = "soil") %>% ### change here 
  mutate (Sign = case_when(PValue<=0.05 ~ "sign.", PValue > 0.05 ~ "ns")) 

# PoaCit data soil


PoaCit_soil <- glmQLFTest(fit, coef=15) 

DAA_PoaCit_soil <-
  PoaCit_soil$table %>%  
  as_tibble (rownames = "ASV_ID") %>% 
  left_join (taxa_edgeR) %>% 
  filter (Phylum == "Glomeromycota" ) %>% 
  add_column (PlaSpe = "PoaCit", roots_soil = "soil") %>% ### change here 
  mutate (Sign = case_when(PValue<=0.05 ~ "sign.", PValue > 0.05 ~ "ns")) 


# SchAru data soil

SchAru_soil <- glmQLFTest(fit, coef=17) 

DAA_SchAru_soil <-
  SchAru_soil$table %>%  
  as_tibble (rownames = "ASV_ID") %>% 
  left_join (taxa_edgeR) %>% 
  filter (Phylum == "Glomeromycota" ) %>% 
  add_column (PlaSpe = "SchAru", roots_soil = "soil") %>% ### change here 
  mutate (Sign = case_when(PValue<=0.05 ~ "sign.", PValue > 0.05 ~ "ns")) 

## Subset the genus level data  for Glo
ps_edgeR_glo_genus <- subset_taxa(ps_edgeR_genus, Phylum == "Glomeromycota")
ps_edgeR_glo_genus <- prune_taxa (taxa_sums(ps_edgeR_glo_genus)>=1, ps_edgeR_glo_genus)

# get the tree data
MyTree <-    ps_edgeR_glo_genus %>% phy_tree

# Save names of taxa in tree
TreeTax <-  taxa_names(ps_edgeR_glo_genus)

# create new taxonomy labels
df.tax <-  ps_edgeR_glo_genus %>% tax_table %>% as.data.frame
df.tax$ASV_ID   <-  df.tax %>% row.names # add column with the ASV iD 

df.tax <-  df.tax  %>% mutate( TaxLabel = paste(Family, Genus, sep = "_")) %>%
            select(ASV_ID, TaxLabel, Phylum, Class, Order, Family, Genus)

# Change the NA in the taxon table to the nearest identified taxon
df.tax = df.tax %>%
  mutate(GenusLabel = ifelse(!is.na(Genus), paste(Genus), 
                             ifelse(!is.na(Family), paste('Unid. ', Family, sep = ""), 
                                    ifelse(!is.na(Order), paste('Unid. ', Order, sep = ""),
                                          ifelse(!is.na(Class), paste('Unid. ', Class, sep = ""), paste("Unid. ", Phylum, sep = "")))))) 

# get a tibble of the whole taxa table incl new GenusLabel
taxa_names <- df.tax %>% as_tibble ()


# Tree plot ####
test_tree <- rotateConstr(MyTree, constraint = c("ASV_3904" ,"ASV_1540", "ASV_1320",   "ASV_1856", 
                                                 "ASV_2852","ASV_1705","ASV_1334", "ASV_550" ,
                                                 "ASV_270","ASV_386",  "ASV_662" , "ASV_988" , 
                                                 "ASV_247" , "ASV_713"  , "ASV_357", "ASV_253" ,  "ASV_14" ,   "ASV_3003"))

p  = ggtree(test_tree, ladderize = F) %<+% df.tax

p_tree <- 
  p +  geom_tiplab(aes(label = GenusLabel, color = Order), align = TRUE, size = 4) +
 # geom_tippoint(aes(color= Order), size=3, show.legend = F) +
  theme_tree2() +
  theme (plot.margin = unit (c(8,5,6.5,5), "mm")) + 
  xlim(NA, 0.7) +
  theme (legend.position = "none")

#  get the order of the tree tips for use in the DAA plot
order_taxa <- get_taxa_name(p) %>%  
  as_tibble () %>% 
  dplyr::rename ("ASV_ID" = "value") %>%  
  left_join (taxa_names) %>% 
  add_column (order = c(1:18))

# get relative abundances for the ASVs for later use in the DAA plot
ps_edgeR_glo_ASV <- subset_taxa(ps_edgeR, Phylum == "Glomeromycota")
ps_edgeR_glo_ASV_rel <- relative_abundance(ps_edgeR_glo_ASV)

rel_ASV_abundance_soil <- 
  data.frame (otu_table(ps_edgeR_glo_ASV_rel))  %>%  
  as_tibble(rownames ="ASV_ID") %>%
  pivot_longer(!ASV_ID, names_to = "sampleID", values_to = "rel_ASV_per_sample") %>% 
  left_join((tax_table(ps_edgeR_glo_ASV_rel) %>% data.frame () %>% as_tibble (rownames = "ASV_ID")), by= "ASV_ID")  %>% 
  #filter (rel_ASV_per_sample!=0)  %>% 
  left_join (meta_M1Wcontrol) %>% ## these are the relative abundances per sample
  group_by(roots_soil,PlaSpe, ASV_ID) %>%  # group to seperate soil from roots for each  PlaSpe , to get rel abu of ASVs per PlaSpe
  summarise (mean_rel_ASV_per_PlaSpe =mean(rel_ASV_per_sample)) %>% 
  filter (roots_soil =="soil") # for Soil data only


#DAA ALL ####
#get data  for all DAA results into one tibble
# add information on relative abundance of the ASVs per Glo community!
# add genus label
DAA <- bind_rows(DAA_AchMil_soil, DAA_AgrCap_soil, DAA_BroWil_soil, DAA_CicInt_soil, 
                 DAA_HolLan_soil, DAA_PlaLan_soil, DAA_PoaCit_soil,  DAA_SchAru_soil) %>%  
  left_join (order_taxa %>% select (Order, GenusLabel, order), by = "GenusLabel") %>% 
  left_join (rel_ASV_abundance_soil)  %>% 
  filter (!is.na (mean_rel_ASV_per_PlaSpe))  # remove all rel abundances and logFC that have 0 abundance in the respective PlaSpe

## make a dummy df to add a blank geom to the DAA plot (without it, some of the 
# AMF genera are mssing in the DAA plot
# and it cannot be aligned with tree)
DAA_empty_fields <- order_taxa %>%  add_column(logFC = 0, Sign = "ns", mean_rel_ASV_per_PlaSpe = 0) 




# Plot DAA ####
plotDAA <-
DAA %>%   mutate (across (PlaSpe, factor, levels = c("AchMil", "CicInt", "PlaLan","PoaCit", "SchAru", "BroWil", "AgrCap", "HolLan" ))) %>% 
  ggplot (aes (x= logFC,  y= reorder (GenusLabel, -order), color =Order, alpha = Sign, size = mean_rel_ASV_per_PlaSpe)) + 
  geom_blank (data =DAA_empty_fields, mapping = aes (x= logFC, y = reorder (GenusLabel, -order), size =mean_rel_ASV_per_PlaSpe)) +  ##this adds blank data - to include All AMF that are in the tree!!
  geom_jitter(width =0.4) +  # geom_jitter
  facet_wrap(~PlaSpe, nrow =1) +
  theme_classic() + 
  theme (axis.title.y = element_blank(),
         axis.ticks.y = element_blank(), 
         axis.line.y = element_blank(),
         axis.text.y = element_blank()) +
  geom_vline(xintercept = 0, linetype = "dashed", color ="darkgrey") + 
  scale_alpha_discrete(range = c(0.3, 1)) +
  theme (legend.position = "right" ) +
  xlim(-9,13) +
  labs (size = "Mean relative abundance of ASV ", color = "AMF order", alpha = "Significance")


# Plot tree and DAA together #####
ggarrange (p_tree, plotDAA, nrow = 1, widths = c(0.4,1))
```

Differential abundance analysis (DAA) was applied as a statistical tool for detecting ASVs with significantly different abundances between treatments.
Here, the AMF community of each plant-associated soils was compared to that of the control soil to assess how each plant species altered the AMF community of its associated soil (Fig. \@ref(fig:plot-DAA-soil)) by determining the significant log~2~-fold changes between the ASV read abundances calculated for each plant species' soil in comparison to the control soil. 
All plants enriched the soil significantly with AMF taxa, especially with *Acaulospora.* Overall, two different enrichment patterns can be identified. In the soils of *A. capillaris*, *B. willdenowii* and *H. lanatus*, the same three AMF genera were significantly enriched, with ASVs from the *Acaulospora* and other Diversisporales also dominating the relative abundances. 
Although these three plant species showed some enrichment of the Gigasporales in their soils, none of these log~2~-fold changes were significant. In contrast, the ASV enrichments in the soils of *A. millefolium*, *C. intybus*, *P. lanceolata*, *S. arundinaceus* and *P. cita* were more evenly distributed among the AMF genera, especially in *S. arundinaceus* and *C. intybus*. These two different patterns separate the three plant species from group C (*B. willdenowii*, *A. capillaris*, *H. lanatus*) that were described as having AMF communities of high $\alpha$-diversities but low $\beta$-diversities in their roots from the other two groups described afore (see PCA diversities). 
This indicates that differences in the interaction niche properties determined from the plant roots can result in different patterns of AMF enrichment in the plant associated soils. 

```{r plot-DAA-roots,include = F,  fig.cap = "Difference in abundance of AMF taxa between root and associated soil by plant species. The plot panels show the results from the differential abundance analysis (DAA) between the plant roots and their associated soil. Each point represents one ASVs, with the color indicating the AMF order, and the size the mean relative abundance of the ASV in regards to all AMF ASVs. The AMF taxa are organised by the phylogenetic tree of the identified AMF genera (left). The log2-fold change between the sampling groups describes the log-ratio of the respective ASV between the root and the soil samples for each plant species. Significant log2-fold changes are indicated by non-transparent coloring (edgeR p-value < 0.05).", fig.height=8, fig.width= 13}


### Comparison plant roots to control soil
#colnames (design)
#GLM F-test #####
#compare the coefficients of the glm 
# i.e. compare treatments
AchMil_roots<- glmQLFTest(fit, coef=2) 
#rootsPlaSpe results####
# Ach Mil data roots##
DAA_AchMil_roots<-
  AchMil_soil$table %>%  
  as_tibble (rownames = "ASV_ID") %>% 
  left_join (taxa_edgeR) %>% 
  filter (Phylum == "Glomeromycota" ) %>% 
  add_column (PlaSpe = "AchMil", roots_soil = "roots") %>% ### change here 
  mutate (Sign = case_when(PValue<=0.05 ~ "sign.", PValue > 0.05 ~ "ns")) 

# Agr Cap roots ##
# i.e. compare treatments
AgrCap_roots <- glmQLFTest(fit, coef=4) 

# Ach Mil data roots
DAA_AgrCap_roots <-
  AgrCap_roots$table %>%  
  as_tibble (rownames = "ASV_ID") %>% 
  left_join (taxa_edgeR) %>% 
  filter (Phylum == "Glomeromycota" ) %>% 
  add_column (PlaSpe = "AgrCap", roots_soil = "roots") %>% ### change here 
  mutate (Sign = case_when(PValue<=0.05 ~ "sign.", PValue > 0.05 ~ "ns")) 

## BroWIl roots
BroWil_roots <- glmQLFTest(fit, coef=6) 

DAA_BroWil_roots <-
  BroWil_roots$table %>%  
  as_tibble (rownames = "ASV_ID") %>% 
  left_join (taxa_edgeR) %>% 
  filter (Phylum == "Glomeromycota" ) %>% 
  add_column (PlaSpe = "BroWil", roots_soil = "roots") %>% ### change here 
  mutate (Sign = case_when(PValue<=0.05 ~ "sign.", PValue > 0.05 ~ "ns"))

# CicInt data roots


CicInt_roots <- glmQLFTest(fit, coef=8) 

DAA_CicInt_roots <-
  CicInt_roots$table %>%  
  as_tibble (rownames = "ASV_ID") %>% 
  left_join (taxa_edgeR) %>% 
  filter (Phylum == "Glomeromycota" ) %>% 
  add_column (PlaSpe = "CicInt", roots_soil = "roots") %>% ### change here 
  mutate (Sign = case_when(PValue<=0.05 ~ "sign.", PValue > 0.05 ~ "ns")) 

# HolLan data roots


HolLan_roots <- glmQLFTest(fit, coef=10)

DAA_HolLan_roots <-
  HolLan_roots$table %>%  
  as_tibble (rownames = "ASV_ID") %>% 
  left_join (taxa_edgeR) %>% 
  filter (Phylum == "Glomeromycota" ) %>% 
  add_column (PlaSpe = "HolLan", roots_soil = "roots") %>% ### change here 
  mutate (Sign = case_when(PValue<=0.05 ~ "sign.", PValue > 0.05 ~ "ns")) 

# PlaLan data roots

PlaLan_roots <- glmQLFTest(fit, coef=12) 

DAA_PlaLan_roots <-
  PlaLan_roots$table %>%  
  as_tibble (rownames = "ASV_ID") %>% 
  left_join (taxa_edgeR) %>% 
  filter (Phylum == "Glomeromycota" ) %>% 
  add_column (PlaSpe = "PlaLan", roots_soil = "roots") %>% ### change here 
  mutate (Sign = case_when(PValue<=0.05 ~ "sign.", PValue > 0.05 ~ "ns")) 

# PoaCit data roots


PoaCit_roots <- glmQLFTest(fit, coef=14) 

DAA_PoaCit_roots <-
  PoaCit_roots$table %>%  
  as_tibble (rownames = "ASV_ID") %>% 
  left_join (taxa_edgeR) %>% 
  filter (Phylum == "Glomeromycota" ) %>% 
  add_column (PlaSpe = "PoaCit", roots_soil = "roots") %>% ### change here 
  mutate (Sign = case_when(PValue<=0.05 ~ "sign.", PValue > 0.05 ~ "ns")) 


# SchAru data roots

SchAru_roots <- glmQLFTest(fit, coef=16) 

DAA_SchAru_roots <-
  SchAru_roots$table %>%  
  as_tibble (rownames = "ASV_ID") %>% 
  left_join (taxa_edgeR) %>% 
  filter (Phylum == "Glomeromycota" ) %>% 
  add_column (PlaSpe = "SchAru", roots_soil = "roots") %>% ### change here 
  mutate (Sign = case_when(PValue<=0.05 ~ "sign.", PValue > 0.05 ~ "ns")) 

# ## Subset the genus level data  for Glo
# ps_edgeR_glo_genus <- subset_taxa(ps_edgeR_genus, Phylum == "Glomeromycota")
# ps_edgeR_glo_genus <- prune_taxa (taxa_sums(ps_edgeR_glo_genus)>=1, ps_edgeR_glo_genus)
#
# # get the tree data
# MyTree <-    ps_edgeR_glo_genus %>% phy_tree
#
# # Save names of taxa in tree
# TreeTax <-  taxa_names(ps_edgeR_glo_genus)
#
# # create new taxonomy labels
# df.tax <-  ps_edgeR_glo_genus %>% tax_table %>% as.data.frame
# df.tax$ASV_ID   <-  df.tax %>% row.names # add column with the ASV iD
#
# df.tax <-  df.tax  %>% mutate( TaxLabel = paste(Family, Genus, sep = "_")) %>%
#             select(ASV_ID, TaxLabel, Phylum, Class, Order, Family, Genus)
#
# # Change the NA in the taxon table to the nearest identified taxon
# df.tax = df.tax %>%
#   mutate(GenusLabel = ifelse(!is.na(Genus), paste(Genus),
#                              ifelse(!is.na(Family), paste('Unid. ', Family, sep = ""),
#                                     ifelse(!is.na(Order), paste('Unid. ', Order, sep = ""),
#                                           ifelse(!is.na(Class), paste('Unid. ', Class, sep = ""), paste("Unid. ", Phylum, sep = ""))))))
#
# # get a tibble of the whole taxa table incl new GenusLabel
# taxa_names <- df.tax %>% as_tibble ()
#
#
# # Tree plot ####
# test_tree <- rotateConstr(MyTree, constraint = c("ASV_3904" ,"ASV_1540", "ASV_1320",   "ASV_1856",
#                                                  "ASV_2852","ASV_1705","ASV_1334", "ASV_550" ,
#                                                  "ASV_270","ASV_386",  "ASV_662" , "ASV_988" ,
#                                                  "ASV_247" , "ASV_713"  , "ASV_357", "ASV_253" ,  "ASV_14" ,   "ASV_3003"))
# 
# p  = ggtree(test_tree, ladderize = F) %<+% df.tax
# 
# p_tree <-
#   p +  geom_tiplab(aes(label = GenusLabel, color = Order), align = TRUE, size = 4) +
#  # geom_tippoint(aes(color= Order), size=3, show.legend = F) +
#   theme_tree2() +
#   theme (plot.margin = unit (c(8,5,6.5,5), "mm")) +
#   xlim(NA, 0.7) +
#   theme (legend.position = "none")
# 
# #  get the order of the tree tips for use in the DAA plot
# order_taxa <- get_taxa_name(p) %>%
#   as_tibble () %>%
#   dplyr::rename ("ASV_ID" = "value") %>%
#   left_join (taxa_names) %>%
#   add_column (order = c(1:18))

# get relative abundances for the ASVs for later use in the DAA plot
# ps_edgeR_glo_ASV <- subset_taxa(ps_edgeR, Phylum == "Glomeromycota")
# ps_edgeR_glo_ASV_rel <- relative_abundance(ps_edgeR_glo_ASV)

rel_ASV_abundance_roots<- 
  data.frame (otu_table(ps_edgeR_glo_ASV_rel))  %>%  
  as_tibble(rownames ="ASV_ID") %>%
  pivot_longer(!ASV_ID, names_to = "sampleID", values_to = "rel_ASV_per_sample") %>% 
  left_join((tax_table(ps_edgeR_glo_ASV_rel) %>% data.frame () %>% as_tibble (rownames = "ASV_ID")), by= "ASV_ID")  %>% 
  #filter (rel_ASV_per_sample!=0)  %>% 
  left_join (meta_M1Wcontrol) %>% ## these are the relative abundances per sample
  group_by(roots_soil,PlaSpe, ASV_ID) %>%  # group to seperate rootsfrom roots for each  PlaSpe , to get rel abu of ASVs per PlaSpe
  summarise (mean_rel_ASV_per_PlaSpe =mean(rel_ASV_per_sample)) %>% 
  filter (roots_soil =="roots") # for roots data only


#DAA ALL ####
#get data  for all DAA results into one tibble
# add information on relative abundance of the ASVs per Glo community!
# add genus label
DAA_roots <- bind_rows(DAA_AchMil_roots, DAA_AgrCap_roots, DAA_BroWil_roots, DAA_CicInt_roots, 
                 DAA_HolLan_roots, DAA_PlaLan_roots, DAA_PoaCit_roots,  DAA_SchAru_roots) %>%  
  left_join (order_taxa %>% select (Order, GenusLabel, order), by = "GenusLabel") %>% 
  left_join (rel_ASV_abundance_roots)  %>% 
  filter (!is.na (mean_rel_ASV_per_PlaSpe))  # remove all rel abundances and logFC that have 0 abundance in the respective PlaSpe

## make a dummy df to add a blank geom to the DAA plot (without it, some of the 
# AMF genera are mssing in the DAA plot
# and it cannot be aligned with tree)
DAA_empty_fields <- order_taxa %>%  add_column(logFC = 0, Sign = "ns", mean_rel_ASV_per_PlaSpe = 0) 




# Plot DAA ####
plotDAA_roots <-
DAA_roots %>%   mutate (across (PlaSpe, factor, levels = c("AchMil", "CicInt", "PlaLan","PoaCit", "SchAru", "BroWil", "AgrCap", "HolLan" ))) %>% 
  ggplot (aes (x= logFC,  y= reorder (GenusLabel, -order), color =Order, alpha = Sign, size = mean_rel_ASV_per_PlaSpe)) + 
  geom_blank (data =DAA_empty_fields, mapping = aes (x= logFC, y = reorder (GenusLabel, -order), size =mean_rel_ASV_per_PlaSpe)) +  ##this adds blank data - to include All AMF that are in the tree!!
  geom_jitter() +  # geom_jitter
  facet_wrap(~PlaSpe, nrow =1) +
  theme_classic() + 
  theme (axis.title.y = element_blank(),
         axis.ticks.y = element_blank(), 
         axis.line.y = element_blank(),
         axis.text.y = element_blank()) +
  geom_vline(xintercept = 0, linetype = "dashed", color ="darkgrey") + 
  scale_alpha_discrete(range = c(0.3, 1)) +
  theme (legend.position = "right" ) +
    xlim(-9,13) +
  labs (size = "Mean relative abundance of ASV ", color = "AMF order", alpha = "Significance")


# Plot tree and DAA together #####
ggarrange (p_tree, plotDAA_roots, nrow = 1, widths = c(0.4,1))
```



## Discussion 


The aim of this experiment was to study the effect of different plant-AMF interaction niche properties on the soil microbial community. 
The determination of the plant-AMF interaction properties, based on root AMF diversities, placed the eight studied plant species into three groups of distinct α- and β-diversities. 
These results correlated well with the description of the plant-AMF interaction niche properties from Chapter 2, showing that these properties are inherent plant traits, even under variable growth conditions.
I found that the arrival of a plant host altered the soil community significantly, altering both microbial community structure and biomass. With some notable exceptions, the AMF biomass and bacterial biomass did not differ among the plant species and was also unrelated to the plant-AMF interaction niche properties. Instead, host plant family determined microbial biomass and the bacterial community composition. By contrast, the changes of the soil AMF community showed a mixed picture. While the AMF richness and relative abundances remained relatively unaffected by the plants’ interaction niche properties, the absolute abundances of the AMF showed a relationship with these interaction traits. In what follows, I discuss these findings in detail.


### Plants promote the growth of the soil AMF 

I found that all plant species transported large amounts of carbon into the mycelium if compared to the plant-free control soil. 
Overall, the absolute carbon allocation into the symbiotic AMF was similar among the plant species, suggesting that my hypothesis, that the properties of the plants’ interaction niche with AMF would influence the carbon transfers to AMF could not be supported. 
In addition to obtaining similar absolute carbon allocations, the root to soil ratio of the carbon allocation also did not differ between the plants. 
That these ratios were relatively constant suggests that even at high AMF richness the co-occurring AMF of the AMF community utilise complimentary colonisation strategies, therefore avoiding competition for space [@Barcelo2020]. 
However, when the plant biomass is taken into account, a more nuanced picture emerges which shows that the species of Asteraceae transferred more carbon per unit photosynthetic plant biomass than any other species, including those with higher AMF richness. 
Studies testing the effect of AMF richness on plant productivity described that competition between AMF resulted in adverse effects for the host plants [@Malicka2021] and decreased fungal abundance [@Engelmoer2014] which could explain why the numeric AMF generalist plants allocate less carbon to their AMF relative to their biomass. 
However, other morphological or physiological traits of the plant families tested might have evoked the differences in carbon allocation to the AMF community [@Ladygina2010]. 

Plant biomass had no linear relationship with the AMF storage biomass, which agrees with other findings that the number of spores and vesicles is not necessarily influenced by the plant biomass or photosynthetic activity, but also dependent on the maturity of the plant and the number and identity of the AMF species [@Smilauer2021; @Padje2021; @Maherali2007]. 
Notably, the NZ native grass *P. cita* did allocate only small amounts of carbon into the AMF soil community. 
While the exact reasons for that are unclear, *P. cita* had a comparably small biomass and might not have reached a state of maturity needed for larger carbon transfers to the microbial community at the time of the harvest [@Smilauer2021]. 
On the other hand, the total bacterial PLFA amount as a measure of bacterial biomass of the *P. cita* associated soil was in the same range as most of the other plant species, indicating that *P. cita* seemed to be able to allocate equally large amounts of carbon-rich exudates into the soil. 
One reason for that contradiction could lie in the type of interaction between the AMF and *P. cita*. 
While the plant-AMF relationship is mostly described as mutualistic, it is well known that - depending on the involved partner and the conditions - the interaction can be also become parasitic [@Johnson1997; @Carey2004; @Friede2016]. 
If the interaction between *P. cita* and its AMF tended to mostly benefit the fungal partner with only small reciprocal nutrient exchange, it could explain both the small *P. cita* biomass and the small AMF growth in *P. cita* microcosms. 

The measurement of the hyphal biomass revealed a positive linear relationship between the plant biomass and the biomass of the soil hyphae, indicating the positive mutualistic effects of the plant-AMF interaction. However, care must be taken in the application of the phospholipid fatty acid 16:1ω5 as an AMF biomarker because it is unclear to which proportion the detected levels of the FA might have originated from bacteria instead [@Olsson1995]. In fact, I detected high levels of the PLFA in the plant-free control soil as well, which were not statistically different from the FA values of the plant-associated soils. Taking this caveat into account, using the PL biomarker to determine the hyphal biomass might not have been as suitable here to examine the carbon allocation to the soil as it was for the NL biomarker for the determination of the AMF storage biomass. 


### Plant identity, not interaction niche property shapes the soil bacterial community 

Contrary to my hypothesis 2, the total bacterial biomass differed not among the groups of different plant-AMF interaction niche properties and was unrelated to the AMF richness. Instead, the plant species and the plant family identity determined, the bacterial biomass in the soil. 
The community composition of the soil bacteria varied among the plant families and was mostly unrelated to the properties of the interaction niche with AMF.
Any effect that AMF diversity might have had on the bacterial community was also confounded by the link between AMF richness and plant family. 
Both root and hyphal exudates influence microbial growth via trophic impacts and drive bacterial community assembly patterns due to distinct exudate compositions [@Zhalnina2018; @Zhang2021]. While AMF can form distinct bacterial communities [@Luthfiana2021; @Zhou2020], none of my results describing the differences in AMF community composition among plants relate to the differences in bacterial composition. 
Distinct root exudation profiles have been described for plant species and also among forbs and grasses [@Dietz2020; @Herz2018; @Steinauer2016; @Lange2014] which suggests that the changes in the bacterial community I observed may have resulted directly from root-mediated processes. The most notable difference among the soil bacterial community of the plant families concerned the ratios of Gram-positive to Gram-negative bacteria, a finding that was reported from plant communities under field conditions as well. Zhong *et al* [-@Zhong2020] showed that the composition of the plant communities significantly affected the ratio of Gram-positive to Gram-negative bacteria in the soil depending on the dominant plant family.
They found that under Asteraceae dominated conditions, the Actinobacteria (GP) dominated, while the Gram-negative Proteobacteria increased with the increase in Poaceae [@Zhong2020], a finding that corroborates with my results of single grown plants under glasshouse conditions. 


### The plant-AMF interaction niche property influences the soil AMF community 

All plant species inferred significant changes in the AMF community composition in their associated soils. However, neither the AMF richness nor the relative abundances of AMF taxa of the soil mirrored that of their respective plant species. 
While the taxa placed within the family Acaulosporaceae (order Diversisporales) dominated the root AMF communities and that of the control soil, other AMF families showed increased relative abundances in the plant-associated soils. 
One possible explanation for the obtained taxon-dependent difference in abundances between the two substrates could be based on distinct colonisation strategies with respect to root *versus* soil biomass allocation. 
Variations among the ratios of intraradical to extraradical fungal biomass have been described for some AMF families [@Hart2002; @Maherali2007]. For example, in comparison to the Acaulosporaceae, the Gigasporaceae and Archaesporaceae have been described to preferably colonise the soil [@Maherali2007; @Hart2002; @Barcelo2020]. 
These differences in AMF community composition between roots and soil also implicate that AMF metabarcoding data from root samples does not necessarily mirror the situation of the AMF composition in the soil.

Differential abundance analysis (DAA) revealed that the plant AMF interaction niches properties are related to the pattern of the AMF taxon enrichment in plant-associated soils. 
Plant species that are characterised by high AMF $\alpha$-diversities and low within β-diversities tended to enrich distinct taxonomic groups, while the other plants obtained more taxonomically mixed AMF communities with equally low relative abundant taxa. 
While the AMF composition and richness in the soil did not mirror that of the roots, the root AMF β-diversity appears to explain the patterns of soil AMF community enrichment by the plants. 
For example, plant individuals of a plant species described as hosting AMF communities of low within β-diversity concentrate their interaction efforts on specific AMF taxa, resulting in the pattern revealed by the DAA. In contrast, plants of both high α- and β-diversity impacted the abundances of the AMF taxa in a more even way by significantly enriching a larger number of phylogenetic different AMF taxa. 
Both strategies can be of advantage for the plants in a community, depending on its requirements, the environment and the services provided by the hosted AMF species [@Wagg2011; @Tsiknia2021]. 
For example, an evenly distributed enrichment of diverse AMF might offer more variability in nutrient uptake strategies for the plant host [@Jansa2008].
AMF species differ in their ability and efficiency to transfer nutrients like N [@Schuetz2022] and P [@Thonar2014], and a plant community equipped with a diverse AMF community of more evenly abundant AMF communities therefore might be better equipped for a range of environmental conditions due to complementarity effects [@Jansa2008]. 
On the other hand, plants can also benefit from preferably enriching AMF taxa that are better adapted to the present conditions [@Malicka2021]. Studies found plants preferable allocating carbon to the more cooperative mycorrhizal partner [@Kiers2011], a mechanism which could have played a role in the selection of specific taxa offering higher nutrient transfer services under the conditions of this glasshouse experiment.

An unusual dominance of the Acaulosporaceae in the soil and roots indicates that, besides the plants’ influence, other factors had an impact on the AMF community composition. 
Dominance by a minority of AMF taxa, most often from the Glomeraceae, had been previously reported in AM dominated systems [@Oepik2009; @Johansen2015]. Usually, Gigasporaceae and Acaulosporaceae isolates are relatively less common in AM fungal communities [@Davison2015; @Oepik2006]. 
However, AMF taxa have distinct niche-based ecological ranges. Acaulosporaceae, for example, preferably occur in soils with higher bulk density and lower pH [@Veresoglou2013]. 
Although the soil properties were not measured here, the employed soil mixture included a high proportion of medium and fine sand in relation to the small proportion of the field-sampled soil, which indicates a comparably high bulk density [@Ghezzehei2011]. 
Therefore, this soil condition might have benefitted the growth of the acaulosporoid spores. 
Furthermore, comparisons of Gigasporaceae and Glomeraceae species have shown that Gigasporaceae dominate in sandy soils over Glomeraceae, which in turn favor clay rich soils [@Lekberg2007]. It seems that the soil mixture used here had a significant effect on the relative abundances of the AMF taxa, which also differed from the obtained communities in the first experiment (see Chapter 2), in which the soil mixture included notably less sand. These specific soil properties might have given the Acaulosporaceae a distinct advantage in becoming the most competitive partner for the plant hosts. 

### Conclusion

This chapter’s aim was to assess the ability of plant species of different interaction niche properties to influence the microbial community of soil. All plants induced significant changes in the composition and biomass of the soil microbial community. The carbon allocation to the AMF was high, corroborating with studies reporting that AMF act as large carbon sinks in ecosystems. 
However, the plants' interaction niche properties with AMF did not influence either the AMF biomass or the soil bacterial biomass. Instead, all grassland species studied, except for *P. cita*, were equally able to increase the AMF abundance in the soil which, in turn, is a requirement for successful initiation of further plant-AMF interactions. 
The soil bacterial community composition depended only on the plant identity and was unrelated to the plants’ interaction with AMF as well, which indicates at a higher relevance of the root-mediated prozesses in comparison to possible AMF hyphae mediated processes influencing the soil bacteria. 
However, the differences in $\beta$-diversity were a good predictor for the changes in AMF absolute abundance in the soil. 
The characteristic AMF interaction strategies slightly determine the plants’ relative impact on the absolute abundances of AMF taxa in their associated soil. 
These reported impacts on the absolute AMF abundances, in turn, might determine the strength and direction of feedback mechanism by the soil community on a plant community.

\newpage

## Appendix for Chapter 3

```{r table-roots-properties, tab.id = "table-roots-properties", label = "table-roots-properties", tab.cap = "Mean interaction niche properties by plant species."  }

All_metrics_M1  %>%  
  mutate (CU = CU*5) %>% 
  select (- c( PlantFamily, PlantSpeciesfull))  %>% 
  as_tibble(rownames = "Plant species") %>% 
  mutate (round (across(where (is.numeric)),2))  %>% 
  flextable () %>% 
  italic (part = "body", j = "Plant species" )


```

\newpage

```{r table-soil-properties, tab.id = "table-soil-properties", label = "table-soil-properties", tab.cap = "Mean AMF interaction niche properties for the soils associated with each plant species."}

## richness, chao, shannon
adiv_richness_M1_soil  <- 
  estimate_richness(ps_M1_soil_Glo_rar, measures = c("Observed", "Chao1", "Shannon")) %>% 
  as_tibble (rownames = "sampleID") %>% 
  left_join (meta_M1Wcontrol %>%  select (sampleID, PlaSpe, PlantSpeciesfull, PlantFamily), by = "sampleID")  


## richness per species -average
meannumberASVsper_species_M1_soil  <-
  adiv_richness_M1_soil %>% 
  group_by (PlantSpeciesfull) %>% 
  summarize (meanASV = mean (Observed))


ASV_table_Glo_M1_soil <- 
  otu_table (ps_M1_soil_Glo_rar) %>%  
  data.frame ()  %>%  t () %>%  
  as_tibble (rownames = "sampleID") 

## unique asvs
unique_M1_soil  <- 
  ASV_table_Glo_M1_soil  %>% 
  pivot_longer(!sampleID, names_to = "ASV_ID", values_to = "ASV_count") %>% 
  filter (ASV_count != 0) %>%  
  left_join(meta_M1Wcontrol %>%  select (sampleID, PlantSpeciesfull)) %>%  
  group_by (ASV_ID,PlantSpeciesfull ) %>%  
  tally () %>% 
  group_by(PlantSpeciesfull)  %>% 
  tally (name ="uniqueASVsperPlSpe")

## CU ###
comp_units   <- meannumberASVsper_species_M1_soil  %>% 
  left_join(unique_M1_soil, by = "PlantSpeciesfull")  %>% 
  mutate (CUnits = uniqueASVsperPlSpe/(meanASV*5))%>% # repl adjustment nor needed here, because all 5 replicates are available
  mutate ("1-CU" = 1 - CUnits)

### Cores####

ps_test_soil <- ps_M1_soil_Glo_rar  # make a new ps to change its sample_data (core and conglomerte have problems when sample-data can not conglomerated)
sample_data (ps_test_soil)  <- sample_data (ps_M1_soil_Glo_rar) %>% data.frame () %>%  select (PlantSpeciesfull, PlaSpe) %>%  sample_data()
ps_focal_soil_M1_core  <- phylosmith::taxa_core (ps_test_soil, treatment =  "PlantSpeciesfull", frequency =0.6)#see above
ps_focal_soil_M1_core <- conglomerate_samples(ps_focal_soil_M1_core, treatment = "PlantSpeciesfull", merge_on = "PlantSpeciesfull")

#calculate beta diversity as percentage 
cores_soil_M1  <- 
  as.data.frame (otu_table (ps_focal_soil_M1_core )) %>%   as_tibble(rownames = "ASV_ID") %>% 
  gather (!ASV_ID, key=PlantSpeciesfull, value = ASV_counts)  %>%  
  group_by (PlantSpeciesfull) %>%  
  dplyr::filter (ASV_counts != "0")  %>% 
  dplyr::tally(name="core") %>% 
  left_join(unique_M1_soil, by = "PlantSpeciesfull")  %>% 
  mutate (perc_core = 100* core/uniqueASVsperPlSpe) %>% 
  mutate (perc_core = round (perc_core,1) ) %>% 
  select (PlantSpeciesfull, perc_core) 

### PD####
comm_df_Glo_M1_soil <- data.frame (t(otu_table (ps_M1_soil_Glo_rar))) # vegan expects samples as rows and ASVs species as columns

#hier total randomised community data frame as null model
stand_pd_Glo_all_M1_soil  <- as_tibble (ses.pd(comm_df_Glo_M1_soil, phy_tree(ps_M1_soil_Glo_rar), include.root = FALSE, null.model = "independentswap", runs=100, iterations=999), rownames="sampleID")

stand_pd_Glo_all_M1_soil  <- 
  stand_pd_Glo_all_M1_soil %>%  
  left_join (meta_M1Wcontrol %>% select (sampleID, PlantSpeciesfull)) 

## MPD ###

dist_Glo_M1_soil  <- cophenetic(phy_tree(ps_M1_soil_Glo_rar))
ses.MPD_Glo_M1_soil  <-  as_tibble (ses.mpd (comm_df_Glo_M1_soil, dist_Glo_M1_soil, null.model = "independentswap" ), rownames = "sampleID")

# together PD, MPD
PD_MPD_M1_soil <-  
  stand_pd_Glo_all_M1_soil %>%  
  select (sampleID, pd.obs) %>% 
  left_join(ses.MPD_Glo_M1_soil, by = "sampleID") %>%  
  left_join (meta_M1Wcontrol) %>% 
  select (sampleID, pd.obs, mpd.obs, PlaSpe, PlantSpeciesfull) %>% 
  drop_na() %>% 
  group_by (PlantSpeciesfull) %>% 
  mutate (mean_pd = mean (pd.obs), mean_mpd = mean (mpd.obs)) %>%  
  select (PlantSpeciesfull,  mean_pd, mean_mpd)   %>% 
  unique ()  





#Get values for all metrics in one tibble
All_metrics_M1_soil  <-
  adiv_richness_M1_soil %>% 
  group_by(PlantFamily, PlantSpeciesfull)  %>% 
  mutate (Chao1 = mean (Chao1)) %>% 
  mutate (Shannon = mean (Shannon)) %>%  
  select (sampleID, Chao1, Shannon) %>% 
  select (!sampleID) %>% 
  unique()  %>%  
  left_join (comp_units %>%  select (!'1-CU'))    %>%   #includes CU and unique and richness
  left_join(cores_soil_M1) %>% 
  left_join (PD_MPD_M1_soil) %>% 
  dplyr::rename("Richness"= "meanASV", "CU" = "CUnits", "β(core)"  = "perc_core",
                "PD" = "mean_pd", "MPD" = "mean_mpd" , "uniqueASV" = "uniqueASVsperPlSpe")  %>% 
  as.data.frame(row.names = NULL) %>% 
  relocate (where (is.numeric))
rownames(All_metrics_M1_soil)  <- All_metrics_M1_soil$PlantSpeciesfull

All_metrics_M1_soil  %>%  
  mutate (CU = CU*5) %>% 
  select (- c( PlantFamily, PlantSpeciesfull))  %>% 
  as_tibble(rownames = "Plant species") %>% 
  mutate (round (across(where (is.numeric)),2))  %>% 
  flextable () %>% 
  italic (part = "body", j = "Plant species" )

```

\newpage

```{r primer-table, tab.id = "primer-table", label = "primer-table", tab.cap = "Primer for metabarcoding of ITS2 region."}
read_excel ("data/PrimerTable.xlsx") %>%  flextable()
```

\newpage



